{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "# How to extend a LinkML modeling language (LinkML) model\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "pycharm": {
     "is_executing": false,
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "!pip install -q --disable-pip-version-check yamlmagic\n",
    "!pip install -q --disable-pip-version-check .."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext yamlmagic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "pycharm": {
     "is_executing": false,
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "data": {
      "application/javascript": "\n            require(\n                [\n                    \"notebook/js/codecell\",\n                    \"codemirror/mode/yaml/yaml\"\n                ],\n                function(cc){\n                    cc.CodeCell.options_default.highlight_modes.magic_yaml = {\n                        reg: [\"^%%yaml\"]\n                    }\n                }\n            );\n            ",
      "text/plain": [
       "<IPython.core.display.Javascript object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%yaml model\n",
    "\n",
    "# Every model must have a globally unique URI. This is the external name of the model\n",
    "id: http://example.org/examples/distributeExample\n",
    "\n",
    "# Every model should also have a globally unique name (well, global within the context of the particular modeling environment)\n",
    "name: dist1\n",
    "\n",
    "# Descriptions are always useful, but not required\n",
    "description: A toy extension to the base biolink model\n",
    "\n",
    "# Versions are recommended but not required.  The version is copied into the output artifacts.  An error will be raised\n",
    "# if two different versions of the same model are imported\n",
    "version: 0.0.1\n",
    "\n",
    "# A license is not required at this point -- should it be?\n",
    "license: https://creativecommons.org/publicdomain/zero/1.0/\n",
    "\n",
    "# Prefixes can be assigned specifically.  We define two below:\n",
    "#   biolink -- the prefix used by the biolink-model\n",
    "#   dist1 -- the URI prefix used by this example.  Note that the dist1 prefix may or may not align with the model id\n",
    "prefixes:\n",
    "  biolink: https://w3id.org/biolink/vocab/\n",
    "  linkml: https://w3id.org/linkml/\n",
    "  dist: http://example.org/examples/dist1#\n",
    "\n",
    "# Prefixes can also be pulled from a prefixcommons compliant site. The map below uses the definitions found in\n",
    "#     https://github.com/prefixcommons/biocontext/blob/master/registry/semweb_context.yaml.\n",
    "default_curi_maps:\n",
    "  - semweb_context\n",
    "\n",
    "# The default prefix is what is used in the subsets, types, slots, classes sections below if not otherwise supplied\n",
    "default_prefix:  dist\n",
    "default_range: string\n",
    "\n",
    "# The list of prefixes to emit target files.  Note that all prefixes that are used elsewhere in the model are automatically\n",
    "# emitted, with the exception of\n",
    "emit_prefixes:\n",
    "    - skos\n",
    "    - rdf\n",
    "    - dist\n",
    "\n",
    "# List of models to import.  Note that import specifications can (currently) be URI's, absolute (file://...file), curies\n",
    "# (biolink:model), or relative (includes/myfile) file names.  Note, however, that this latter form is being deprecated.\n",
    "# The location of imported files can now be specified in an accompanying mapping file.  The imports below reference:\n",
    "#   https://w3id.org/biolink/biolink-model    -- the biolink model\n",
    "#   https://w3id.org/linkml/types  -- the biolink modeling language types definitions\n",
    "imports:\n",
    "  - https://w3id.org/biolink/biolink-model\n",
    "  - linkml:types\n",
    "\n",
    "# Subsets that are defined in this model extension\n",
    "subsets:\n",
    "    experimental:\n",
    "        # A subset should have a description\n",
    "        description: model elements that have not yet been tested\n",
    "\n",
    "# Types that are defined in this model extension\n",
    "types:\n",
    "  gene sequence:\n",
    "    uri: dist:seq\n",
    "    typeof: string\n",
    "    description: A gene sequence\n",
    "\n",
    "# Slots that are defined in this model extension\n",
    "slots:\n",
    "    gene has sequence:\n",
    "        description: A gene pattern\n",
    "        domain: gene\n",
    "        range: gene sequence\n",
    "        slot_uri: dist:hasSeq\n",
    "        required: true\n",
    "\n",
    "# Classes that are defined in this model extension\n",
    "classes:\n",
    "    # The class name.  For most generators, this will be transformed to CamelCase (MyGene)\n",
    "    my gene:\n",
    "        description: This is an example extension.  Doesn't do a lot\n",
    "        is_a: gene\n",
    "        slots:\n",
    "            - gene has sequence\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Auto generated from None by pythongen.py version: 0.0.1\n",
      "# Generation date: 2025-01-09T17:06:28\n",
      "# Schema: dist1\n",
      "#\n",
      "# id: http://example.org/examples/distributeExample\n",
      "# description: A toy extension to the base biolink model\n",
      "# license: https://creativecommons.org/publicdomain/zero/1.0/\n",
      "\n",
      "import dataclasses\n",
      "import re\n",
      "from dataclasses import dataclass\n",
      "from datetime import (\n",
      "    date,\n",
      "    datetime,\n",
      "    time\n",
      ")\n",
      "from typing import (\n",
      "    Any,\n",
      "    ClassVar,\n",
      "    Dict,\n",
      "    List,\n",
      "    Optional,\n",
      "    Union\n",
      ")\n",
      "\n",
      "from jsonasobj2 import (\n",
      "    JsonObj,\n",
      "    as_dict\n",
      ")\n",
      "from linkml_runtime.linkml_model.meta import (\n",
      "    EnumDefinition,\n",
      "    PermissibleValue,\n",
      "    PvFormulaOptions\n",
      ")\n",
      "from linkml_runtime.utils.curienamespace import CurieNamespace\n",
      "from linkml_runtime.utils.dataclass_extensions_376 import dataclasses_init_fn_with_kwargs\n",
      "from linkml_runtime.utils.enumerations import EnumDefinitionImpl\n",
      "from linkml_runtime.utils.formatutils import (\n",
      "    camelcase,\n",
      "    sfx,\n",
      "    underscore\n",
      ")\n",
      "from linkml_runtime.utils.metamodelcore import (\n",
      "    bnode,\n",
      "    empty_dict,\n",
      "    empty_list\n",
      ")\n",
      "from linkml_runtime.utils.slot import Slot\n",
      "from linkml_runtime.utils.yamlutils import (\n",
      "    YAMLRoot,\n",
      "    extended_float,\n",
      "    extended_int,\n",
      "    extended_str\n",
      ")\n",
      "from rdflib import (\n",
      "    Namespace,\n",
      "    URIRef\n",
      ")\n",
      "\n",
      "from linkml_runtime.linkml_model.types import Boolean, Date, Double, Float, Integer, String, Time, Uriorcurie\n",
      "from linkml_runtime.utils.metamodelcore import Bool, URIorCURIE, XSDDate, XSDTime\n",
      "\n",
      "metamodel_version = \"1.7.0\"\n",
      "version = \"0.0.1\"\n",
      "\n",
      "# Overwrite dataclasses _init_fn to add **kwargs in __init__\n",
      "dataclasses._init_fn = dataclasses_init_fn_with_kwargs\n",
      "\n",
      "# Namespaces\n",
      "AGRKB = CurieNamespace('AGRKB', 'https://www.alliancegenome.org/')\n",
      "APO = CurieNamespace('APO', 'http://purl.obolibrary.org/obo/APO_')\n",
      "ASPGD = CurieNamespace('AspGD', 'http://www.aspergillusgenome.org/cgi-bin/locus.pl?dbid=')\n",
      "BFO = CurieNamespace('BFO', 'http://purl.obolibrary.org/obo/BFO_')\n",
      "BIGG_METABOLITE = CurieNamespace('BIGG_METABOLITE', 'http://identifiers.org/bigg.metabolite/')\n",
      "BIGG_REACTION = CurieNamespace('BIGG_REACTION', 'http://identifiers.org/bigg.reaction/')\n",
      "BIOGRID = CurieNamespace('BIOGRID', 'http://identifiers.org/biogrid/')\n",
      "BIOSAMPLE = CurieNamespace('BIOSAMPLE', 'http://identifiers.org/biosample/')\n",
      "BSPO = CurieNamespace('BSPO', 'http://purl.obolibrary.org/obo/BSPO_')\n",
      "BTO = CurieNamespace('BTO', 'http://purl.obolibrary.org/obo/BTO_')\n",
      "CAID = CurieNamespace('CAID', 'http://reg.clinicalgenome.org/redmine/projects/registry/genboree_registry/by_caid?caid=')\n",
      "CARO = CurieNamespace('CARO', 'http://purl.obolibrary.org/obo/CARO_')\n",
      "CAS = CurieNamespace('CAS', 'http://identifiers.org/cas/')\n",
      "CATH = CurieNamespace('CATH', 'http://identifiers.org/cath/')\n",
      "CATH_SUPERFAMILY = CurieNamespace('CATH_SUPERFAMILY', 'http://identifiers.org/cath.superfamily/')\n",
      "CDD = CurieNamespace('CDD', 'http://identifiers.org/cdd/')\n",
      "CHADO = CurieNamespace('CHADO', 'http://gmod.org/wiki/Chado/')\n",
      "CHEBI = CurieNamespace('CHEBI', 'http://purl.obolibrary.org/obo/CHEBI_')\n",
      "CHEMBL_COMPOUND = CurieNamespace('CHEMBL_COMPOUND', 'http://identifiers.org/chembl.compound/')\n",
      "CHEMBL_MECHANISM = CurieNamespace('CHEMBL_MECHANISM', 'https://www.ebi.ac.uk/chembl/mechanism/inspect/')\n",
      "CHEMBL_TARGET = CurieNamespace('CHEMBL_TARGET', 'http://identifiers.org/chembl.target/')\n",
      "CIO = CurieNamespace('CIO', 'http://purl.obolibrary.org/obo/CIO_')\n",
      "CL = CurieNamespace('CL', 'http://purl.obolibrary.org/obo/CL_')\n",
      "CLINVAR = CurieNamespace('CLINVAR', 'http://identifiers.org/clinvar')\n",
      "CLO = CurieNamespace('CLO', 'http://purl.obolibrary.org/obo/CLO_')\n",
      "COG = CurieNamespace('COG', 'https://www.ncbi.nlm.nih.gov/research/cog-project/')\n",
      "CPT = CurieNamespace('CPT', 'https://www.ama-assn.org/practice-management/cpt/')\n",
      "CTD = CurieNamespace('CTD', 'http://ctdbase.org/')\n",
      "CHEMBANK = CurieNamespace('ChemBank', 'http://chembank.broadinstitute.org/chemistry/viewMolecule.htm?cbid=')\n",
      "COMPLEXPORTAL = CurieNamespace('ComplexPortal', 'https://www.ebi.ac.uk/complexportal/complex/')\n",
      "DBSNP = CurieNamespace('DBSNP', 'http://identifiers.org/dbsnp/')\n",
      "DDANAT = CurieNamespace('DDANAT', 'http://purl.obolibrary.org/obo/DDANAT_')\n",
      "DGIDB = CurieNamespace('DGIdb', 'https://www.dgidb.org/interaction_types')\n",
      "DOID = CurieNamespace('DOID', 'http://purl.obolibrary.org/obo/DOID_')\n",
      "DOID_PROPERTY = CurieNamespace('DOID-PROPERTY', 'http://purl.obolibrary.org/obo/doid#')\n",
      "DRUGBANK = CurieNamespace('DRUGBANK', 'http://identifiers.org/drugbank/')\n",
      "DRUGCENTRAL = CurieNamespace('DrugCentral', 'http://drugcentral.org/drugcard/')\n",
      "EC = CurieNamespace('EC', 'http://www.enzyme-database.org/query.php?ec=')\n",
      "ECO = CurieNamespace('ECO', 'http://purl.obolibrary.org/obo/ECO_')\n",
      "ECTO = CurieNamespace('ECTO', 'http://purl.obolibrary.org/obo/ECTO_')\n",
      "EDAM_DATA = CurieNamespace('EDAM-DATA', 'http://edamontology.org/data_')\n",
      "EDAM_FORMAT = CurieNamespace('EDAM-FORMAT', 'http://edamontology.org/format_')\n",
      "EDAM_OPERATION = CurieNamespace('EDAM-OPERATION', 'http://edamontology.org/operation_')\n",
      "EDAM_TOPIC = CurieNamespace('EDAM-TOPIC', 'http://edamontology.org/topic_')\n",
      "EFO = CurieNamespace('EFO', 'http://www.ebi.ac.uk/efo/EFO_')\n",
      "EGGNOG = CurieNamespace('EGGNOG', 'http://identifiers.org/eggnog/')\n",
      "EMAPA = CurieNamespace('EMAPA', 'http://purl.obolibrary.org/obo/EMAPA_')\n",
      "ENSEMBL = CurieNamespace('ENSEMBL', 'http://identifiers.org/ensembl/')\n",
      "ENVO = CurieNamespace('ENVO', 'http://purl.obolibrary.org/obo/ENVO_')\n",
      "EXO = CurieNamespace('ExO', 'http://purl.obolibrary.org/obo/ExO_')\n",
      "FAO = CurieNamespace('FAO', 'http://purl.obolibrary.org/obo/FAO_')\n",
      "FB = CurieNamespace('FB', 'http://identifiers.org/fb/')\n",
      "FBBT = CurieNamespace('FBbt', 'http://purl.obolibrary.org/obo/FBbt_')\n",
      "FBCV = CurieNamespace('FBcv', 'http://purl.obolibrary.org/obo/FBcv_')\n",
      "FBDV = CurieNamespace('FBdv', 'http://purl.obolibrary.org/obo/FBdv_')\n",
      "FMA = CurieNamespace('FMA', 'http://purl.obolibrary.org/obo/FMA_')\n",
      "FOODON = CurieNamespace('FOODON', 'http://purl.obolibrary.org/obo/FOODON_')\n",
      "FYPO = CurieNamespace('FYPO', 'http://purl.obolibrary.org/obo/FYPO_')\n",
      "GENEPIO = CurieNamespace('GENEPIO', 'http://purl.obolibrary.org/obo/GENEPIO_')\n",
      "GENO = CurieNamespace('GENO', 'http://purl.obolibrary.org/obo/GENO_')\n",
      "GO = CurieNamespace('GO', 'http://purl.obolibrary.org/obo/GO_')\n",
      "GOLD_META = CurieNamespace('GOLD_META', 'http://identifiers.org/gold.meta/')\n",
      "GOP = CurieNamespace('GOP', 'http://purl.obolibrary.org/obo/go#')\n",
      "GOREL = CurieNamespace('GOREL', 'http://purl.obolibrary.org/obo/GOREL_')\n",
      "GSID = CurieNamespace('GSID', 'https://scholar.google.com/citations?user=')\n",
      "GTOPDB = CurieNamespace('GTOPDB', 'https://www.guidetopharmacology.org/GRAC/LigandDisplayForward?ligandId=')\n",
      "HAMAP = CurieNamespace('HAMAP', 'http://identifiers.org/hamap/')\n",
      "HANCESTRO = CurieNamespace('HANCESTRO', 'http://www.ebi.ac.uk/ancestro/ancestro_')\n",
      "HCPCS = CurieNamespace('HCPCS', 'http://purl.bioontology.org/ontology/HCPCS/')\n",
      "HGNC = CurieNamespace('HGNC', 'http://identifiers.org/hgnc/')\n",
      "HGNC_FAMILY = CurieNamespace('HGNC_FAMILY', 'http://identifiers.org/hgnc.family/')\n",
      "HMDB = CurieNamespace('HMDB', 'http://identifiers.org/hmdb/')\n",
      "HP = CurieNamespace('HP', 'http://purl.obolibrary.org/obo/HP_')\n",
      "HSAPDV = CurieNamespace('HsapDv', 'http://purl.obolibrary.org/obo/HsapDv_')\n",
      "IAO = CurieNamespace('IAO', 'http://purl.obolibrary.org/obo/IAO_')\n",
      "ICD10 = CurieNamespace('ICD10', 'https://icd.who.int/browse10/2016/en#/')\n",
      "ICD11 = CurieNamespace('ICD11', 'http://id.who.int/icd/entity/')\n",
      "ICD9 = CurieNamespace('ICD9', 'http://translator.ncats.nih.gov/ICD9_')\n",
      "IDO = CurieNamespace('IDO', 'http://purl.obolibrary.org/obo/IDO_')\n",
      "INCHI = CurieNamespace('INCHI', 'http://identifiers.org/inchi/')\n",
      "INCHIKEY = CurieNamespace('INCHIKEY', 'http://identifiers.org/inchikey/')\n",
      "INO = CurieNamespace('INO', 'http://purl.obolibrary.org/obo/INO_')\n",
      "INTACT = CurieNamespace('INTACT', 'http://identifiers.org/intact/')\n",
      "IUPHAR_FAMILY = CurieNamespace('IUPHAR_FAMILY', 'http://identifiers.org/iuphar.family/')\n",
      "KEGG = CurieNamespace('KEGG', 'http://www.kegg.jp/entry/')\n",
      "KEGG_BRITE = CurieNamespace('KEGG_BRITE', 'https://bioregistry.io/kegg.brite:')\n",
      "KEGG_COMPOUND = CurieNamespace('KEGG_COMPOUND', 'http://identifiers.org/kegg.compound/')\n",
      "KEGG_DISEASE = CurieNamespace('KEGG_DISEASE', 'http://identifiers.org/kegg.disease/')\n",
      "KEGG_DRUG = CurieNamespace('KEGG_DRUG', 'http://identifiers.org/kegg.drug/')\n",
      "KEGG_ENVIRON = CurieNamespace('KEGG_ENVIRON', 'http://identifiers.org/kegg.environ/')\n",
      "KEGG_GENES = CurieNamespace('KEGG_GENES', 'https://bioregistry.io/kegg.genes:bsu:')\n",
      "KEGG_GLYCAN = CurieNamespace('KEGG_GLYCAN', 'http://identifiers.org/kegg.glycan/')\n",
      "KEGG_MODULE = CurieNamespace('KEGG_MODULE', 'http://identifiers.org/kegg.module/')\n",
      "KEGG_ORTHOLOGY = CurieNamespace('KEGG_ORTHOLOGY', 'http://identifiers.org/kegg.orthology/')\n",
      "KEGG_PATHWAY = CurieNamespace('KEGG_PATHWAY', 'https://bioregistry.io/kegg.pathway:')\n",
      "KEGG_REACTION = CurieNamespace('KEGG_REACTION', 'http://identifiers.org/kegg.reaction/')\n",
      "LOINC = CurieNamespace('LOINC', 'http://loinc.org/rdf/')\n",
      "MA = CurieNamespace('MA', 'http://purl.obolibrary.org/obo/MA_')\n",
      "MAXO = CurieNamespace('MAXO', 'http://purl.obolibrary.org/obo/MAXO_')\n",
      "MEDDRA = CurieNamespace('MEDDRA', 'http://identifiers.org/meddra/')\n",
      "MESH = CurieNamespace('MESH', 'http://id.nlm.nih.gov/mesh/')\n",
      "METANETX_REACTION = CurieNamespace('METANETX_REACTION', 'https://www.metanetx.org/equa_info/')\n",
      "MGI = CurieNamespace('MGI', 'http://identifiers.org/mgi/')\n",
      "MI = CurieNamespace('MI', 'http://purl.obolibrary.org/obo/MI_')\n",
      "MIR = CurieNamespace('MIR', 'http://identifiers.org/mir/')\n",
      "MONDO = CurieNamespace('MONDO', 'http://purl.obolibrary.org/obo/MONDO_')\n",
      "MP = CurieNamespace('MP', 'http://purl.obolibrary.org/obo/MP_')\n",
      "MPATH = CurieNamespace('MPATH', 'http://purl.obolibrary.org/obo/MPATH_')\n",
      "MSIGDB = CurieNamespace('MSigDB', 'https://www.gsea-msigdb.org/gsea/msigdb/')\n",
      "MMUSDV = CurieNamespace('MmusDv', 'http://purl.obolibrary.org/obo/MMUSDV_')\n",
      "NBO = CurieNamespace('NBO', 'http://purl.obolibrary.org/obo/NBO_')\n",
      "NBO_PROPERTY = CurieNamespace('NBO-PROPERTY', 'http://purl.obolibrary.org/obo/nbo#')\n",
      "NCBIGENE = CurieNamespace('NCBIGene', 'http://identifiers.org/ncbigene/')\n",
      "NCBITAXON = CurieNamespace('NCBITaxon', 'http://purl.obolibrary.org/obo/NCBITaxon_')\n",
      "NCIT = CurieNamespace('NCIT', 'http://purl.obolibrary.org/obo/NCIT_')\n",
      "NCIT_OBO = CurieNamespace('NCIT-OBO', 'http://purl.obolibrary.org/obo/ncit#')\n",
      "NDC = CurieNamespace('NDC', 'http://identifiers.org/ndc/')\n",
      "NDDF = CurieNamespace('NDDF', 'http://purl.bioontology.org/ontology/NDDF/')\n",
      "NLMID = CurieNamespace('NLMID', 'https://www.ncbi.nlm.nih.gov/nlmcatalog/?term=')\n",
      "OBAN = CurieNamespace('OBAN', 'http://purl.org/oban/')\n",
      "OBI = CurieNamespace('OBI', 'http://purl.obolibrary.org/obo/OBI_')\n",
      "OGMS = CurieNamespace('OGMS', 'http://purl.obolibrary.org/obo/OGMS_')\n",
      "OMIM = CurieNamespace('OMIM', 'http://purl.obolibrary.org/obo/OMIM_')\n",
      "OMIM_PS = CurieNamespace('OMIM_PS', 'https://www.omim.org/phenotypicSeries/')\n",
      "OMIT = CurieNamespace('OMIT', 'http://purl.obolibrary.org/obo/OMIT_')\n",
      "ORCID = CurieNamespace('ORCID', 'https://orcid.org/')\n",
      "PANTHER_FAMILY = CurieNamespace('PANTHER_FAMILY', 'http://www.pantherdb.org/panther/family.do?clsAccession=')\n",
      "PANTHER_PATHWAY = CurieNamespace('PANTHER_PATHWAY', 'http://identifiers.org/panther.pathway/')\n",
      "PATO = CurieNamespace('PATO', 'http://purl.obolibrary.org/obo/PATO_')\n",
      "PCO = CurieNamespace('PCO', 'http://purl.obolibrary.org/obo/PCO_')\n",
      "PFAM = CurieNamespace('PFAM', 'http://identifiers.org/pfam/')\n",
      "PHARMGKB_DISEASE = CurieNamespace('PHARMGKB_DISEASE', 'https://www.pharmgkb.org/disease/')\n",
      "PHARMGKB_DRUG = CurieNamespace('PHARMGKB_DRUG', 'https://www.pharmgkb.org/chemical/')\n",
      "PHARMGKB_GENE = CurieNamespace('PHARMGKB_GENE', 'https://www.pharmgkb.org/gene/')\n",
      "PHARMGKB_PATHWAYS = CurieNamespace('PHARMGKB_PATHWAYS', 'https://www.pharmgkb.org/pathway/')\n",
      "PHARMGKB_VARIANT = CurieNamespace('PHARMGKB_VARIANT', 'https://www.pharmgkb.org/variant/')\n",
      "PHAROS = CurieNamespace('PHAROS', 'http://pharos.nih.gov')\n",
      "PIRSF = CurieNamespace('PIRSF', 'http://identifiers.org/pirsf/')\n",
      "PMC = CurieNamespace('PMC', 'http://europepmc.org/articles/PMC')\n",
      "PMCID = CurieNamespace('PMCID', 'http://www.ncbi.nlm.nih.gov/pmc/')\n",
      "PMID = CurieNamespace('PMID', 'http://www.ncbi.nlm.nih.gov/pubmed/')\n",
      "PO = CurieNamespace('PO', 'http://purl.obolibrary.org/obo/PO_')\n",
      "PR = CurieNamespace('PR', 'http://purl.obolibrary.org/obo/PR_')\n",
      "PRINTS = CurieNamespace('PRINTS', 'http://identifiers.org/prints/')\n",
      "PRODOM = CurieNamespace('PRODOM', 'http://identifiers.org/prodom/')\n",
      "PROSITE = CurieNamespace('PROSITE', 'http://identifiers.org/prosite/')\n",
      "PUBCHEM_COMPOUND = CurieNamespace('PUBCHEM_COMPOUND', 'http://identifiers.org/pubchem.compound/')\n",
      "PUBCHEM_SUBSTANCE = CurieNamespace('PUBCHEM_SUBSTANCE', 'http://identifiers.org/pubchem.substance/')\n",
      "PW = CurieNamespace('PW', 'http://purl.obolibrary.org/obo/PW_')\n",
      "PATHWHIZ = CurieNamespace('PathWhiz', 'http://smpdb.ca/pathways/#')\n",
      "POMBASE = CurieNamespace('PomBase', 'https://www.pombase.org/gene/')\n",
      "REACT = CurieNamespace('REACT', 'http://www.reactome.org/PathwayBrowser/#/')\n",
      "REPODB = CurieNamespace('REPODB', 'http://apps.chiragjpgroup.org/repoDB/')\n",
      "RFAM = CurieNamespace('RFAM', 'http://identifiers.org/rfam/')\n",
      "RGD = CurieNamespace('RGD', 'http://identifiers.org/rgd/')\n",
      "RHEA = CurieNamespace('RHEA', 'http://identifiers.org/rhea/')\n",
      "RNACENTRAL = CurieNamespace('RNACENTRAL', 'http://identifiers.org/rnacentral/')\n",
      "RO = CurieNamespace('RO', 'http://purl.obolibrary.org/obo/RO_')\n",
      "RXCUI = CurieNamespace('RXCUI', 'https://mor.nlm.nih.gov/RxNav/search?searchBy=RXCUI&searchTerm=')\n",
      "RXNORM = CurieNamespace('RXNORM', 'http://purl.bioontology.org/ontology/RXNORM/')\n",
      "RESEARCHID = CurieNamespace('ResearchID', 'https://publons.com/researcher/')\n",
      "SEED_REACTION = CurieNamespace('SEED_REACTION', 'https://modelseed.org/biochem/reactions/')\n",
      "SEMMEDDB = CurieNamespace('SEMMEDDB', 'https://skr3.nlm.nih.gov/SemMedDB')\n",
      "SEPIO = CurieNamespace('SEPIO', 'http://purl.obolibrary.org/obo/SEPIO_')\n",
      "SGD = CurieNamespace('SGD', 'http://identifiers.org/sgd/')\n",
      "SIDER_DRUG = CurieNamespace('SIDER_DRUG', 'http://identifiers.org/sider.drug/')\n",
      "SIO = CurieNamespace('SIO', 'http://semanticscience.org/resource/SIO_')\n",
      "SMART = CurieNamespace('SMART', 'http://identifiers.org/smart/')\n",
      "SMPDB = CurieNamespace('SMPDB', 'http://identifiers.org/smpdb/')\n",
      "SNOMED = CurieNamespace('SNOMED', 'http://purl.obolibrary.org/obo/SNOMED_')\n",
      "SNOMEDCT = CurieNamespace('SNOMEDCT', 'http://snomed.info/id/')\n",
      "SO = CurieNamespace('SO', 'http://purl.obolibrary.org/obo/SO_')\n",
      "SPDI = CurieNamespace('SPDI', 'https://api.ncbi.nlm.nih.gov/variation/v0/spdi/')\n",
      "STATO = CurieNamespace('STATO', 'http://purl.obolibrary.org/obo/STATO_')\n",
      "STY = CurieNamespace('STY', 'http://purl.bioontology.org/ontology/STY/')\n",
      "SUPFAM = CurieNamespace('SUPFAM', 'http://identifiers.org/supfam/')\n",
      "SCOPUSID = CurieNamespace('ScopusID', 'https://www.scopus.com/authid/detail.uri?authorId=')\n",
      "TAXRANK = CurieNamespace('TAXRANK', 'http://purl.obolibrary.org/obo/TAXRANK_')\n",
      "TCDB = CurieNamespace('TCDB', 'http://identifiers.org/tcdb/')\n",
      "TIGRFAM = CurieNamespace('TIGRFAM', 'http://identifiers.org/tigrfam/')\n",
      "TO = CurieNamespace('TO', 'http://purl.obolibrary.org/obo/TO_')\n",
      "UBERGRAPH = CurieNamespace('UBERGRAPH', 'http://translator.renci.org/ubergraph-axioms.ofn#')\n",
      "UBERON = CurieNamespace('UBERON', 'http://purl.obolibrary.org/obo/UBERON_')\n",
      "UBERON_CORE = CurieNamespace('UBERON_CORE', 'http://purl.obolibrary.org/obo/uberon/core#')\n",
      "UBERON_NONAMESPACE = CurieNamespace('UBERON_NONAMESPACE', 'http://purl.obolibrary.org/obo/core#')\n",
      "UMLS = CurieNamespace('UMLS', 'http://identifiers.org/umls/')\n",
      "UMLSSG = CurieNamespace('UMLSSG', 'https://lhncbc.nlm.nih.gov/semanticnetwork/download/sg_archive/SemGroups-v04.txt')\n",
      "UNII = CurieNamespace('UNII', 'http://identifiers.org/unii/')\n",
      "UNIPROT_ISOFORM = CurieNamespace('UNIPROT_ISOFORM', 'http://purl.uniprot.org/isoforms/')\n",
      "UO = CurieNamespace('UO', 'http://purl.obolibrary.org/obo/UO_')\n",
      "UO_PROPERTY = CurieNamespace('UO-PROPERTY', 'http://purl.obolibrary.org/obo/uo#')\n",
      "UPHENO = CurieNamespace('UPHENO', 'http://purl.obolibrary.org/obo/UPHENO_')\n",
      "UNIPROTKB = CurieNamespace('UniProtKB', 'http://purl.uniprot.org/uniprot/')\n",
      "VANDF = CurieNamespace('VANDF', 'https://www.nlm.nih.gov/research/umls/sourcereleasedocs/current/VANDF/')\n",
      "VMC = CurieNamespace('VMC', 'https://github.com/ga4gh/vr-spec/')\n",
      "WB = CurieNamespace('WB', 'http://identifiers.org/wb/')\n",
      "WBPHENOTYPE = CurieNamespace('WBPhenotype', 'http://purl.obolibrary.org/obo/WBPhenotype_')\n",
      "WBVOCAB = CurieNamespace('WBVocab', 'http://bio2rdf.org/wormbase_vocabulary')\n",
      "WBBT = CurieNamespace('WBbt', 'http://purl.obolibrary.org/obo/WBbt_')\n",
      "WBLS = CurieNamespace('WBls', 'http://purl.obolibrary.org/obo/WBls_')\n",
      "WIKIDATA = CurieNamespace('WIKIDATA', 'https://www.wikidata.org/entity/')\n",
      "WIKIDATA_PROPERTY = CurieNamespace('WIKIDATA_PROPERTY', 'https://www.wikidata.org/prop/')\n",
      "WIKIPATHWAYS = CurieNamespace('WIKIPATHWAYS', 'http://identifiers.org/wikipathways/')\n",
      "WORMBASE = CurieNamespace('WormBase', 'https://www.wormbase.org/get?name=')\n",
      "XAO = CurieNamespace('XAO', 'http://purl.obolibrary.org/obo/XAO_')\n",
      "XCO = CurieNamespace('XCO', 'http://purl.obolibrary.org/obo/XCO_')\n",
      "XPO = CurieNamespace('XPO', 'http://purl.obolibrary.org/obo/XPO_')\n",
      "XENBASE = CurieNamespace('Xenbase', 'http://www.xenbase.org/gene/showgene.do?method=display&geneId=')\n",
      "ZFA = CurieNamespace('ZFA', 'http://purl.obolibrary.org/obo/ZFA_')\n",
      "ZFIN = CurieNamespace('ZFIN', 'http://identifiers.org/zfin/')\n",
      "ZFS = CurieNamespace('ZFS', 'http://purl.obolibrary.org/obo/ZFS_')\n",
      "ZP = CurieNamespace('ZP', 'http://purl.obolibrary.org/obo/ZP_')\n",
      "BIOLINK = CurieNamespace('biolink', 'https://w3id.org/biolink/vocab/')\n",
      "BIOSCHEMAS = CurieNamespace('bioschemas', 'https://bioschemas.org/')\n",
      "DCAT = CurieNamespace('dcat', 'http://www.w3.org/ns/dcat#')\n",
      "DCID = CurieNamespace('dcid', 'https://datacommons.org/browser/')\n",
      "DCT = CurieNamespace('dct', 'http://purl.org/dc/terms/')\n",
      "DCTYPES = CurieNamespace('dctypes', 'http://purl.org/dc/dcmitype/')\n",
      "DICTYBASE = CurieNamespace('dictyBase', 'http://dictybase.org/gene/')\n",
      "DIST = CurieNamespace('dist', 'http://example.org/examples/dist1#')\n",
      "DOI = CurieNamespace('doi', 'https://doi.org/')\n",
      "FABIO = CurieNamespace('fabio', 'http://purl.org/spar/fabio/')\n",
      "FALDO = CurieNamespace('faldo', 'http://biohackathon.org/resource/faldo#')\n",
      "FOAF = CurieNamespace('foaf', 'http://xmlns.com/foaf/0.1/')\n",
      "FOODB_COMPOUND = CurieNamespace('foodb_compound', 'http://foodb.ca/compounds/')\n",
      "FOODB_FOOD = CurieNamespace('foodb_food', 'http://foodb.ca/foods/')\n",
      "GFF3 = CurieNamespace('gff3', 'https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md#')\n",
      "GPI = CurieNamespace('gpi', 'https://github.com/geneontology/go-annotation/blob/master/specs/gpad-gpi-2-0.md#')\n",
      "GTPO = CurieNamespace('gtpo', 'https://rdf.guidetopharmacology.org/ns/gtpo#')\n",
      "ICD11_FOUNDATION = CurieNamespace('icd11_foundation', 'http://id.who.int/icd/entity/')\n",
      "INTERPRO = CurieNamespace('interpro', 'https://www.ebi.ac.uk/interpro/entry/')\n",
      "ISBN = CurieNamespace('isbn', 'https://www.isbn-international.org/identifier/')\n",
      "ISNI = CurieNamespace('isni', 'https://isni.org/isni/')\n",
      "ISSN = CurieNamespace('issn', 'https://portal.issn.org/resource/ISSN/')\n",
      "LINKML = CurieNamespace('linkml', 'https://w3id.org/linkml/')\n",
      "MEDGEN = CurieNamespace('medgen', 'https://www.ncbi.nlm.nih.gov/medgen/')\n",
      "METACYC_REACTION = CurieNamespace('metacyc_reaction', 'http://identifiers.org/metacyc.reaction:')\n",
      "NCATS_BIOPLANET = CurieNamespace('ncats_bioplanet', 'https://tripod.nih.gov/bioplanet/detail.jsp?pid=')\n",
      "NCATS_DRUG = CurieNamespace('ncats_drug', 'https://drugs.ncats.io/drug/')\n",
      "OBOINOWL = CurieNamespace('oboInOwl', 'http://www.geneontology.org/formats/oboInOwl#')\n",
      "ORPHANET = CurieNamespace('orphanet', 'http://www.orpha.net/ORDO/Orphanet_')\n",
      "OS = CurieNamespace('os', 'https://github.com/cmungall/owlstar/blob/master/owlstar.ttl')\n",
      "OWL = CurieNamespace('owl', 'http://www.w3.org/2002/07/owl#')\n",
      "PAV = CurieNamespace('pav', 'http://purl.org/pav/')\n",
      "PROV = CurieNamespace('prov', 'http://www.w3.org/ns/prov#')\n",
      "QUD = CurieNamespace('qud', 'http://qudt.org/1.1/schema/qudt#')\n",
      "RDF = CurieNamespace('rdf', 'http://www.w3.org/1999/02/22-rdf-syntax-ns#')\n",
      "RDFS = CurieNamespace('rdfs', 'http://www.w3.org/2000/01/rdf-schema#')\n",
      "REGULATES = CurieNamespace('regulates', 'http://example.org/UNKNOWN/regulates/')\n",
      "SCHEMA = CurieNamespace('schema', 'http://schema.org/')\n",
      "SKOS = CurieNamespace('skos', 'http://www.w3.org/2004/02/skos/core#')\n",
      "WGS = CurieNamespace('wgs', 'http://www.w3.org/2003/01/geo/wgs84_pos')\n",
      "XSD = CurieNamespace('xsd', 'http://www.w3.org/2001/XMLSchema#')\n",
      "DEFAULT_ = DIST\n",
      "\n",
      "\n",
      "# Types\n",
      "class ChemicalFormulaValue(str):\n",
      "    \"\"\" A chemical formula \"\"\"\n",
      "    type_class_uri = XSD[\"string\"]\n",
      "    type_class_curie = \"xsd:string\"\n",
      "    type_name = \"chemical formula value\"\n",
      "    type_model_uri = DIST.ChemicalFormulaValue\n",
      "\n",
      "\n",
      "class GeneSequence(String):\n",
      "    \"\"\" A gene sequence \"\"\"\n",
      "    type_class_uri = DIST[\"seq\"]\n",
      "    type_class_curie = \"dist:seq\"\n",
      "    type_name = \"gene sequence\"\n",
      "    type_model_uri = DIST.GeneSequence\n",
      "\n",
      "\n",
      "class IriType(Uriorcurie):\n",
      "    \"\"\" An IRI \"\"\"\n",
      "    type_class_uri = XSD[\"string\"]\n",
      "    type_class_curie = \"xsd:string\"\n",
      "    type_name = \"iri type\"\n",
      "    type_model_uri = DIST.IriType\n",
      "\n",
      "\n",
      "class LabelType(String):\n",
      "    \"\"\" A string that provides a human-readable name for an entity \"\"\"\n",
      "    type_class_uri = XSD[\"string\"]\n",
      "    type_class_curie = \"xsd:string\"\n",
      "    type_name = \"label type\"\n",
      "    type_model_uri = DIST.LabelType\n",
      "\n",
      "\n",
      "class PredicateType(Uriorcurie):\n",
      "    \"\"\" A CURIE from the biolink related_to hierarchy. For example, biolink:related_to, biolink:causes, biolink:treats. \"\"\"\n",
      "    type_class_uri = XSD[\"string\"]\n",
      "    type_class_curie = \"xsd:string\"\n",
      "    type_name = \"predicate type\"\n",
      "    type_model_uri = DIST.PredicateType\n",
      "\n",
      "\n",
      "class NarrativeText(String):\n",
      "    \"\"\" A string that provides a human-readable description of something \"\"\"\n",
      "    type_class_uri = XSD[\"string\"]\n",
      "    type_class_curie = \"xsd:string\"\n",
      "    type_name = \"narrative text\"\n",
      "    type_model_uri = DIST.NarrativeText\n",
      "\n",
      "\n",
      "class SymbolType(String):\n",
      "    type_class_uri = XSD[\"string\"]\n",
      "    type_class_curie = \"xsd:string\"\n",
      "    type_name = \"symbol type\"\n",
      "    type_model_uri = DIST.SymbolType\n",
      "\n",
      "\n",
      "class FrequencyValue(String):\n",
      "    type_class_uri = UO[\"0000105\"]\n",
      "    type_class_curie = \"UO:0000105\"\n",
      "    type_name = \"frequency value\"\n",
      "    type_model_uri = DIST.FrequencyValue\n",
      "\n",
      "\n",
      "class PercentageFrequencyValue(Double):\n",
      "    type_class_uri = UO[\"0000187\"]\n",
      "    type_class_curie = \"UO:0000187\"\n",
      "    type_name = \"percentage frequency value\"\n",
      "    type_model_uri = DIST.PercentageFrequencyValue\n",
      "\n",
      "\n",
      "class Quotient(Double):\n",
      "    type_class_uri = UO[\"0010006\"]\n",
      "    type_class_curie = \"UO:0010006\"\n",
      "    type_name = \"quotient\"\n",
      "    type_model_uri = DIST.Quotient\n",
      "\n",
      "\n",
      "class Unit(String):\n",
      "    type_class_uri = UO[\"0000000\"]\n",
      "    type_class_curie = \"UO:0000000\"\n",
      "    type_name = \"unit\"\n",
      "    type_model_uri = DIST.Unit\n",
      "\n",
      "\n",
      "class TimeType(Time):\n",
      "    type_class_uri = XSD[\"time\"]\n",
      "    type_class_curie = \"xsd:time\"\n",
      "    type_name = \"time type\"\n",
      "    type_model_uri = DIST.TimeType\n",
      "\n",
      "\n",
      "class BiologicalSequence(String):\n",
      "    type_class_uri = XSD[\"string\"]\n",
      "    type_class_curie = \"xsd:string\"\n",
      "    type_name = \"biological sequence\"\n",
      "    type_model_uri = DIST.BiologicalSequence\n",
      "\n",
      "\n",
      "# Class references\n",
      "class OntologyClassId(extended_str):\n",
      "    pass\n",
      "\n",
      "\n",
      "class EntityId(extended_str):\n",
      "    pass\n",
      "\n",
      "\n",
      "class NamedThingId(EntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class AttributeId(NamedThingId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ChemicalRoleId(AttributeId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class BiologicalSexId(AttributeId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class PhenotypicSexId(BiologicalSexId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GenotypicSexId(BiologicalSexId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class SeverityValueId(AttributeId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class RelationshipTypeId(OntologyClassId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class TaxonomicRankId(OntologyClassId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class OrganismTaxonId(NamedThingId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class EventId(NamedThingId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class AdministrativeEntityId(NamedThingId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class AgentId(AdministrativeEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class InformationContentEntityId(NamedThingId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class StudyResultId(InformationContentEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class StudyVariableId(InformationContentEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class CommonDataElementId(InformationContentEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ConceptCountAnalysisResultId(StudyResultId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ObservedExpectedFrequencyAnalysisResultId(StudyResultId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class RelativeFrequencyAnalysisResultId(StudyResultId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class TextMiningResultId(StudyResultId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ChiSquaredAnalysisResultId(StudyResultId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class LogOddsAnalysisResultId(StudyResultId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class DatasetId(InformationContentEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class DatasetDistributionId(InformationContentEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class DatasetVersionId(InformationContentEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class DatasetSummaryId(InformationContentEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ConfidenceLevelId(InformationContentEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class EvidenceTypeId(InformationContentEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class PublicationId(InformationContentEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class BookId(PublicationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class BookChapterId(PublicationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class SerialId(PublicationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ArticleId(PublicationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class JournalArticleId(ArticleId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class PatentId(PublicationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class WebPageId(PublicationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class PreprintPublicationId(PublicationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class DrugLabelId(PublicationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class RetrievalSourceId(InformationContentEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class PhysicalEntityId(NamedThingId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ActivityId(NamedThingId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class StudyId(ActivityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ProcedureId(NamedThingId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class PhenomenonId(NamedThingId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class DeviceId(NamedThingId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class DiagnosticAidId(NamedThingId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class MaterialSampleId(PhysicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class PlanetaryEntityId(NamedThingId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class EnvironmentalProcessId(PlanetaryEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class EnvironmentalFeatureId(PlanetaryEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GeographicLocationId(PlanetaryEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GeographicLocationAtTimeId(GeographicLocationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class BiologicalEntityId(NamedThingId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ChemicalEntityId(NamedThingId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class MolecularEntityId(ChemicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class SmallMoleculeId(MolecularEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ChemicalMixtureId(ChemicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class NucleicAcidEntityId(MolecularEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class RegulatoryRegionId(BiologicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class AccessibleDnaRegionId(RegulatoryRegionId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class TranscriptionFactorBindingSiteId(RegulatoryRegionId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class MolecularMixtureId(ChemicalMixtureId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ComplexMolecularMixtureId(ChemicalMixtureId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class BiologicalProcessOrActivityId(BiologicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class MolecularActivityId(BiologicalProcessOrActivityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class BiologicalProcessId(BiologicalProcessOrActivityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class PathwayId(BiologicalProcessId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class PhysiologicalProcessId(BiologicalProcessId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class BehaviorId(BiologicalProcessId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ProcessedMaterialId(ChemicalMixtureId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class DrugId(MolecularMixtureId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class EnvironmentalFoodContaminantId(ChemicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class FoodAdditiveId(ChemicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class FoodId(ChemicalMixtureId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class OrganismAttributeId(AttributeId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class PhenotypicQualityId(OrganismAttributeId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GeneticInheritanceId(BiologicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class OrganismalEntityId(BiologicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class BacteriumId(OrganismalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class VirusId(OrganismalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class CellularOrganismId(OrganismalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class MammalId(CellularOrganismId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class HumanId(MammalId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class PlantId(CellularOrganismId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class InvertebrateId(CellularOrganismId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class VertebrateId(CellularOrganismId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class FungusId(CellularOrganismId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class LifeStageId(OrganismalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class IndividualOrganismId(OrganismalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class PopulationOfIndividualOrganismsId(OrganismalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class StudyPopulationId(PopulationOfIndividualOrganismsId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class DiseaseOrPhenotypicFeatureId(BiologicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class DiseaseId(DiseaseOrPhenotypicFeatureId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class PhenotypicFeatureId(DiseaseOrPhenotypicFeatureId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class BehavioralFeatureId(PhenotypicFeatureId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class AnatomicalEntityId(OrganismalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class CellularComponentId(AnatomicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class CellId(AnatomicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class CellLineId(OrganismalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GrossAnatomicalStructureId(AnatomicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GeneId(BiologicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class MyGeneId(GeneId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class MacromolecularComplexId(BiologicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class NucleosomeModificationId(BiologicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GenomeId(BiologicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ExonId(BiologicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class TranscriptId(BiologicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class CodingSequenceId(BiologicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class PolypeptideId(BiologicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ProteinId(PolypeptideId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ProteinIsoformId(ProteinId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ProteinDomainId(BiologicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class PosttranslationalModificationId(BiologicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ProteinFamilyId(BiologicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class NucleicAcidSequenceMotifId(BiologicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class RNAProductId(TranscriptId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class RNAProductIsoformId(RNAProductId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class NoncodingRNAProductId(RNAProductId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class MicroRNAId(NoncodingRNAProductId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class SiRNAId(NoncodingRNAProductId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GeneFamilyId(BiologicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ZygosityId(AttributeId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GenotypeId(BiologicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class HaplotypeId(BiologicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class SequenceVariantId(BiologicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class SnvId(SequenceVariantId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ReagentTargetedGeneId(BiologicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ClinicalAttributeId(AttributeId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ClinicalMeasurementId(ClinicalAttributeId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ClinicalModifierId(ClinicalAttributeId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ClinicalCourseId(ClinicalAttributeId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class OnsetId(ClinicalCourseId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ClinicalEntityId(NamedThingId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ClinicalTrialId(ClinicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ClinicalInterventionId(ClinicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ClinicalFindingId(PhenotypicFeatureId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class HospitalizationId(ClinicalInterventionId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class SocioeconomicAttributeId(AttributeId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class CaseId(IndividualOrganismId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class CohortId(StudyPopulationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ExposureEventId(OntologyClassId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GenomicBackgroundExposureId(AttributeId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class PathologicalProcessId(BiologicalProcessId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class PathologicalProcessExposureId(AttributeId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class PathologicalAnatomicalStructureId(AnatomicalEntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class PathologicalAnatomicalExposureId(AttributeId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class DiseaseOrPhenotypicFeatureExposureId(AttributeId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ChemicalExposureId(AttributeId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ComplexChemicalExposureId(AttributeId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class DrugExposureId(ChemicalExposureId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class DrugToGeneInteractionExposureId(DrugExposureId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class TreatmentId(NamedThingId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class BioticExposureId(AttributeId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class EnvironmentalExposureId(AttributeId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GeographicExposureId(EnvironmentalExposureId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class BehavioralExposureId(AttributeId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class SocioeconomicExposureId(AttributeId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class AssociationId(EntityId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ChemicalEntityAssessesNamedThingAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ContributorAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GenotypeToGenotypePartAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GenotypeToGeneAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GenotypeToVariantAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GeneToGeneAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GeneToGeneHomologyAssociationId(GeneToGeneAssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GeneToGeneFamilyAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GeneToGeneCoexpressionAssociationId(GeneToGeneAssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class PairwiseGeneToGeneInteractionId(GeneToGeneAssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class PairwiseMolecularInteractionId(PairwiseGeneToGeneInteractionId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class CellLineToDiseaseOrPhenotypicFeatureAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ChemicalToChemicalAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ReactionToParticipantAssociationId(ChemicalToChemicalAssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ReactionToCatalystAssociationId(ReactionToParticipantAssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ChemicalToChemicalDerivationAssociationId(ChemicalToChemicalAssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ChemicalToDiseaseOrPhenotypicFeatureAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ChemicalOrDrugOrTreatmentToDiseaseOrPhenotypicFeatureAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ChemicalOrDrugOrTreatmentSideEffectDiseaseOrPhenotypicFeatureAssociationId(ChemicalOrDrugOrTreatmentToDiseaseOrPhenotypicFeatureAssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GeneToPathwayAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class MolecularActivityToPathwayAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ChemicalToPathwayAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class NamedThingAssociatedWithLikelihoodOfNamedThingAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ChemicalGeneInteractionAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GeneRegulatesGeneAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ProcessRegulatesProcessAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ChemicalAffectsGeneAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GeneAffectsChemicalAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class DrugToGeneAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class MaterialSampleDerivationAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class MaterialSampleToDiseaseOrPhenotypicFeatureAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class DiseaseToExposureEventAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ExposureEventToOutcomeAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class PhenotypicFeatureToPhenotypicFeatureAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class InformationContentEntityToNamedThingAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class DiseaseOrPhenotypicFeatureToLocationAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class DiseaseOrPhenotypicFeatureToGeneticInheritanceAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GenotypeToPhenotypicFeatureAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ExposureEventToPhenotypicFeatureAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class DiseaseToPhenotypicFeatureAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class CaseToPhenotypicFeatureAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class BehaviorToBehavioralFeatureAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GeneToDiseaseOrPhenotypicFeatureAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GeneToPhenotypicFeatureAssociationId(GeneToDiseaseOrPhenotypicFeatureAssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GeneToDiseaseAssociationId(GeneToDiseaseOrPhenotypicFeatureAssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class CausalGeneToDiseaseAssociationId(GeneToDiseaseAssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class CorrelatedGeneToDiseaseAssociationId(GeneToDiseaseAssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class DruggableGeneToDiseaseAssociationId(GeneToDiseaseAssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class PhenotypicFeatureToDiseaseAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class VariantToGeneAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class VariantToGeneExpressionAssociationId(VariantToGeneAssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class VariantToPopulationAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class PopulationToPopulationAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class VariantToPhenotypicFeatureAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class VariantToDiseaseAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GenotypeToDiseaseAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GeneAsAModelOfDiseaseAssociationId(GeneToDiseaseAssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class VariantAsAModelOfDiseaseAssociationId(VariantToDiseaseAssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GenotypeAsAModelOfDiseaseAssociationId(GenotypeToDiseaseAssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class CellLineAsAModelOfDiseaseAssociationId(CellLineToDiseaseOrPhenotypicFeatureAssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class OrganismalEntityAsAModelOfDiseaseAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class OrganismToOrganismAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class TaxonToTaxonAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GeneHasVariantThatContributesToDiseaseAssociationId(GeneToDiseaseAssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GeneToExpressionSiteAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class SequenceVariantModulatesTreatmentAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class FunctionalAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class MacromolecularMachineToMolecularActivityAssociationId(FunctionalAssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class MacromolecularMachineToBiologicalProcessAssociationId(FunctionalAssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class MacromolecularMachineToCellularComponentAssociationId(FunctionalAssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class MolecularActivityToChemicalEntityAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class MolecularActivityToMolecularActivityAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GeneToGoTermAssociationId(FunctionalAssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class EntityToDiseaseAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class EntityToPhenotypicFeatureAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class SequenceAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GenomicSequenceLocalizationId(SequenceAssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class SequenceFeatureRelationshipId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class TranscriptToGeneRelationshipId(SequenceFeatureRelationshipId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class GeneToGeneProductRelationshipId(SequenceFeatureRelationshipId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ExonToTranscriptRelationshipId(SequenceFeatureRelationshipId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class ChemicalEntityOrGeneOrGeneProductRegulatesGeneAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class AnatomicalEntityToAnatomicalEntityAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class AnatomicalEntityToAnatomicalEntityPartOfAssociationId(AnatomicalEntityToAnatomicalEntityAssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class AnatomicalEntityToAnatomicalEntityOntogenicAssociationId(AnatomicalEntityToAnatomicalEntityAssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class OrganismTaxonToOrganismTaxonAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class OrganismTaxonToOrganismTaxonSpecializationId(OrganismTaxonToOrganismTaxonAssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class OrganismTaxonToOrganismTaxonInteractionId(OrganismTaxonToOrganismTaxonAssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "class OrganismTaxonToEnvironmentAssociationId(AssociationId):\n",
      "    pass\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class MappingCollection(YAMLRoot):\n",
      "    \"\"\"\n",
      "    A collection of deprecated mappings.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"MappingCollection\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:MappingCollection\"\n",
      "    class_name: ClassVar[str] = \"mapping collection\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.MappingCollection\n",
      "\n",
      "    predicate_mappings: Optional[Union[Union[dict, \"PredicateMapping\"], List[Union[dict, \"PredicateMapping\"]]]] = empty_list()\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if not isinstance(self.predicate_mappings, list):\n",
      "            self.predicate_mappings = [self.predicate_mappings] if self.predicate_mappings is not None else []\n",
      "        self.predicate_mappings = [v if isinstance(v, PredicateMapping) else PredicateMapping(**as_dict(v)) for v in self.predicate_mappings]\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class PredicateMapping(YAMLRoot):\n",
      "    \"\"\"\n",
      "    A deprecated predicate mapping object contains the deprecated predicate and an example of the rewiring that should\n",
      "    be done to use a qualified statement in its place.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"exact_match\", \"narrow_match\", \"broad_match\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"PredicateMapping\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:PredicateMapping\"\n",
      "    class_name: ClassVar[str] = \"predicate mapping\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.PredicateMapping\n",
      "\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    mapped_predicate: Optional[str] = None\n",
      "    subject_aspect_qualifier: Optional[str] = None\n",
      "    subject_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    subject_form_or_variant_qualifier: Optional[str] = None\n",
      "    subject_part_qualifier: Optional[str] = None\n",
      "    subject_derivative_qualifier: Optional[str] = None\n",
      "    subject_context_qualifier: Optional[str] = None\n",
      "    qualified_predicate: Optional[str] = None\n",
      "    object_aspect_qualifier: Optional[str] = None\n",
      "    object_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    object_form_or_variant_qualifier: Optional[str] = None\n",
      "    object_part_qualifier: Optional[str] = None\n",
      "    object_derivative_qualifier: Optional[str] = None\n",
      "    object_context_qualifier: Optional[str] = None\n",
      "    causal_mechanism_qualifier: Optional[Union[str, \"CausalMechanismQualifierEnum\"]] = None\n",
      "    anatomical_context_qualifier: Optional[str] = None\n",
      "    species_context_qualifier: Optional[Union[str, OrganismTaxonId]] = None\n",
      "    exact_match: Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]] = empty_list()\n",
      "    narrow_match: Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]] = empty_list()\n",
      "    broad_match: Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]] = empty_list()\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self.mapped_predicate is not None and not isinstance(self.mapped_predicate, str):\n",
      "            self.mapped_predicate = str(self.mapped_predicate)\n",
      "\n",
      "        if self.subject_aspect_qualifier is not None and not isinstance(self.subject_aspect_qualifier, str):\n",
      "            self.subject_aspect_qualifier = str(self.subject_aspect_qualifier)\n",
      "\n",
      "        if self.subject_direction_qualifier is not None and not isinstance(self.subject_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.subject_direction_qualifier = DirectionQualifierEnum(self.subject_direction_qualifier)\n",
      "\n",
      "        if self.subject_form_or_variant_qualifier is not None and not isinstance(self.subject_form_or_variant_qualifier, str):\n",
      "            self.subject_form_or_variant_qualifier = str(self.subject_form_or_variant_qualifier)\n",
      "\n",
      "        if self.subject_part_qualifier is not None and not isinstance(self.subject_part_qualifier, str):\n",
      "            self.subject_part_qualifier = str(self.subject_part_qualifier)\n",
      "\n",
      "        if self.subject_derivative_qualifier is not None and not isinstance(self.subject_derivative_qualifier, str):\n",
      "            self.subject_derivative_qualifier = str(self.subject_derivative_qualifier)\n",
      "\n",
      "        if self.subject_context_qualifier is not None and not isinstance(self.subject_context_qualifier, str):\n",
      "            self.subject_context_qualifier = str(self.subject_context_qualifier)\n",
      "\n",
      "        if self.qualified_predicate is not None and not isinstance(self.qualified_predicate, str):\n",
      "            self.qualified_predicate = str(self.qualified_predicate)\n",
      "\n",
      "        if self.object_aspect_qualifier is not None and not isinstance(self.object_aspect_qualifier, str):\n",
      "            self.object_aspect_qualifier = str(self.object_aspect_qualifier)\n",
      "\n",
      "        if self.object_direction_qualifier is not None and not isinstance(self.object_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.object_direction_qualifier = DirectionQualifierEnum(self.object_direction_qualifier)\n",
      "\n",
      "        if self.object_form_or_variant_qualifier is not None and not isinstance(self.object_form_or_variant_qualifier, str):\n",
      "            self.object_form_or_variant_qualifier = str(self.object_form_or_variant_qualifier)\n",
      "\n",
      "        if self.object_part_qualifier is not None and not isinstance(self.object_part_qualifier, str):\n",
      "            self.object_part_qualifier = str(self.object_part_qualifier)\n",
      "\n",
      "        if self.object_derivative_qualifier is not None and not isinstance(self.object_derivative_qualifier, str):\n",
      "            self.object_derivative_qualifier = str(self.object_derivative_qualifier)\n",
      "\n",
      "        if self.object_context_qualifier is not None and not isinstance(self.object_context_qualifier, str):\n",
      "            self.object_context_qualifier = str(self.object_context_qualifier)\n",
      "\n",
      "        if self.causal_mechanism_qualifier is not None and not isinstance(self.causal_mechanism_qualifier, CausalMechanismQualifierEnum):\n",
      "            self.causal_mechanism_qualifier = CausalMechanismQualifierEnum(self.causal_mechanism_qualifier)\n",
      "\n",
      "        if self.anatomical_context_qualifier is not None and not isinstance(self.anatomical_context_qualifier, str):\n",
      "            self.anatomical_context_qualifier = str(self.anatomical_context_qualifier)\n",
      "\n",
      "        if self.species_context_qualifier is not None and not isinstance(self.species_context_qualifier, OrganismTaxonId):\n",
      "            self.species_context_qualifier = OrganismTaxonId(self.species_context_qualifier)\n",
      "\n",
      "        if not isinstance(self.exact_match, list):\n",
      "            self.exact_match = [self.exact_match] if self.exact_match is not None else []\n",
      "        self.exact_match = [v if isinstance(v, NamedThingId) else NamedThingId(v) for v in self.exact_match]\n",
      "\n",
      "        if not isinstance(self.narrow_match, list):\n",
      "            self.narrow_match = [self.narrow_match] if self.narrow_match is not None else []\n",
      "        self.narrow_match = [v if isinstance(v, NamedThingId) else NamedThingId(v) for v in self.narrow_match]\n",
      "\n",
      "        if not isinstance(self.broad_match, list):\n",
      "            self.broad_match = [self.broad_match] if self.broad_match is not None else []\n",
      "        self.broad_match = [v if isinstance(v, NamedThingId) else NamedThingId(v) for v in self.broad_match]\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class OntologyClass(YAMLRoot):\n",
      "    \"\"\"\n",
      "    a concept or class in an ontology, vocabulary or thesaurus. Note that nodes in a biolink compatible KG can be\n",
      "    considered both instances of biolink classes, and OWL classes in their own right. In general you should not need\n",
      "    to use this class directly. Instead, use the appropriate biolink class. For example, for the GO concept of\n",
      "    endocytosis (GO:0006897), use bl:BiologicalProcess as the type.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"OntologyClass\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:OntologyClass\"\n",
      "    class_name: ClassVar[str] = \"ontology class\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.OntologyClass\n",
      "\n",
      "    id: Union[str, OntologyClassId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, OntologyClassId):\n",
      "            self.id = OntologyClassId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "class Annotation(YAMLRoot):\n",
      "    \"\"\"\n",
      "    Biolink Model root class for entity annotations.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Annotation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Annotation\"\n",
      "    class_name: ClassVar[str] = \"annotation\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Annotation\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class QuantityValue(Annotation):\n",
      "    \"\"\"\n",
      "    A value of an attribute that is quantitative and measurable, expressed as a combination of a unit and a numeric\n",
      "    value\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"QuantityValue\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:QuantityValue\"\n",
      "    class_name: ClassVar[str] = \"quantity value\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.QuantityValue\n",
      "\n",
      "    has_unit: Optional[Union[str, Unit]] = None\n",
      "    has_numeric_value: Optional[float] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self.has_unit is not None and not isinstance(self.has_unit, Unit):\n",
      "            self.has_unit = Unit(self.has_unit)\n",
      "\n",
      "        if self.has_numeric_value is not None and not isinstance(self.has_numeric_value, float):\n",
      "            self.has_numeric_value = float(self.has_numeric_value)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "class RelationshipQuantifier(YAMLRoot):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"RelationshipQuantifier\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:RelationshipQuantifier\"\n",
      "    class_name: ClassVar[str] = \"relationship quantifier\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.RelationshipQuantifier\n",
      "\n",
      "\n",
      "class SensitivityQuantifier(RelationshipQuantifier):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"SensitivityQuantifier\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:SensitivityQuantifier\"\n",
      "    class_name: ClassVar[str] = \"sensitivity quantifier\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.SensitivityQuantifier\n",
      "\n",
      "\n",
      "class SpecificityQuantifier(RelationshipQuantifier):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"SpecificityQuantifier\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:SpecificityQuantifier\"\n",
      "    class_name: ClassVar[str] = \"specificity quantifier\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.SpecificityQuantifier\n",
      "\n",
      "\n",
      "class PathognomonicityQuantifier(SpecificityQuantifier):\n",
      "    \"\"\"\n",
      "    A relationship quantifier between a variant or symptom and a disease, which is high when the presence of the\n",
      "    feature implies the existence of the disease\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"PathognomonicityQuantifier\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:PathognomonicityQuantifier\"\n",
      "    class_name: ClassVar[str] = \"pathognomonicity quantifier\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.PathognomonicityQuantifier\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class FrequencyQuantifier(RelationshipQuantifier):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"FrequencyQuantifier\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:FrequencyQuantifier\"\n",
      "    class_name: ClassVar[str] = \"frequency quantifier\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.FrequencyQuantifier\n",
      "\n",
      "    has_count: Optional[int] = None\n",
      "    has_total: Optional[int] = None\n",
      "    has_quotient: Optional[float] = None\n",
      "    has_percentage: Optional[float] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self.has_count is not None and not isinstance(self.has_count, int):\n",
      "            self.has_count = int(self.has_count)\n",
      "\n",
      "        if self.has_total is not None and not isinstance(self.has_total, int):\n",
      "            self.has_total = int(self.has_total)\n",
      "\n",
      "        if self.has_quotient is not None and not isinstance(self.has_quotient, float):\n",
      "            self.has_quotient = float(self.has_quotient)\n",
      "\n",
      "        if self.has_percentage is not None and not isinstance(self.has_percentage, float):\n",
      "            self.has_percentage = float(self.has_percentage)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "class ChemicalOrDrugOrTreatment(YAMLRoot):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ChemicalOrDrugOrTreatment\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ChemicalOrDrugOrTreatment\"\n",
      "    class_name: ClassVar[str] = \"chemical or drug or treatment\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ChemicalOrDrugOrTreatment\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Entity(YAMLRoot):\n",
      "    \"\"\"\n",
      "    Root Biolink Model class for all things and informational relationships, real or imagined.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Entity\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Entity\"\n",
      "    class_name: ClassVar[str] = \"entity\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Entity\n",
      "\n",
      "    id: Union[str, EntityId] = None\n",
      "    iri: Optional[Union[str, IriType]] = None\n",
      "    category: Optional[Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]]] = empty_list()\n",
      "    type: Optional[Union[str, List[str]]] = empty_list()\n",
      "    name: Optional[Union[str, LabelType]] = None\n",
      "    description: Optional[Union[str, NarrativeText]] = None\n",
      "    has_attribute: Optional[Union[Union[str, AttributeId], List[Union[str, AttributeId]]]] = empty_list()\n",
      "    deprecated: Optional[Union[bool, Bool]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, EntityId):\n",
      "            self.id = EntityId(self.id)\n",
      "\n",
      "        if self.iri is not None and not isinstance(self.iri, IriType):\n",
      "            self.iri = IriType(self.iri)\n",
      "\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "        if not isinstance(self.type, list):\n",
      "            self.type = [self.type] if self.type is not None else []\n",
      "        self.type = [v if isinstance(v, str) else str(v) for v in self.type]\n",
      "\n",
      "        if self.name is not None and not isinstance(self.name, LabelType):\n",
      "            self.name = LabelType(self.name)\n",
      "\n",
      "        if self.description is not None and not isinstance(self.description, NarrativeText):\n",
      "            self.description = NarrativeText(self.description)\n",
      "\n",
      "        if not isinstance(self.has_attribute, list):\n",
      "            self.has_attribute = [self.has_attribute] if self.has_attribute is not None else []\n",
      "        self.has_attribute = [v if isinstance(v, AttributeId) else AttributeId(v) for v in self.has_attribute]\n",
      "\n",
      "        if self.deprecated is not None and not isinstance(self.deprecated, Bool):\n",
      "            self.deprecated = Bool(self.deprecated)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "    def __new__(cls, *args, **kwargs):\n",
      "\n",
      "        type_designator = \"category\"\n",
      "        if not type_designator in kwargs:\n",
      "            return super().__new__(cls,*args,**kwargs)\n",
      "        else:\n",
      "            type_designator_value = kwargs[type_designator]\n",
      "            target_cls = cls._class_for(\"class_class_curie\", type_designator_value)\n",
      "\n",
      "\n",
      "            if target_cls is None:\n",
      "                target_cls = cls._class_for(\"class_class_uri\", type_designator_value)\n",
      "\n",
      "\n",
      "            if target_cls is None:\n",
      "                target_cls = cls._class_for(\"class_model_uri\", type_designator_value)\n",
      "\n",
      "\n",
      "            if target_cls is None:\n",
      "                raise ValueError(f\"Wrong type designator value: class {cls.__name__} \"\n",
      "                                 f\"has no subclass with ['class_class_curie', 'class_class_uri', 'class_model_uri']='{kwargs[type_designator]}'\")\n",
      "            return super().__new__(target_cls,*args,**kwargs)\n",
      "\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class NamedThing(Entity):\n",
      "    \"\"\"\n",
      "    a databased entity or concept/class\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"NamedThing\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:NamedThing\"\n",
      "    class_name: ClassVar[str] = \"named thing\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.NamedThing\n",
      "\n",
      "    id: Union[str, NamedThingId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    provided_by: Optional[Union[str, List[str]]] = empty_list()\n",
      "    xref: Optional[Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]]] = empty_list()\n",
      "    full_name: Optional[Union[str, LabelType]] = None\n",
      "    synonym: Optional[Union[Union[str, LabelType], List[Union[str, LabelType]]]] = empty_list()\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, NamedThingId):\n",
      "            self.id = NamedThingId(self.id)\n",
      "\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "        if not isinstance(self.provided_by, list):\n",
      "            self.provided_by = [self.provided_by] if self.provided_by is not None else []\n",
      "        self.provided_by = [v if isinstance(v, str) else str(v) for v in self.provided_by]\n",
      "\n",
      "        if not isinstance(self.xref, list):\n",
      "            self.xref = [self.xref] if self.xref is not None else []\n",
      "        self.xref = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.xref]\n",
      "\n",
      "        if self.full_name is not None and not isinstance(self.full_name, LabelType):\n",
      "            self.full_name = LabelType(self.full_name)\n",
      "\n",
      "        if not isinstance(self.synonym, list):\n",
      "            self.synonym = [self.synonym] if self.synonym is not None else []\n",
      "        self.synonym = [v if isinstance(v, LabelType) else LabelType(v) for v in self.synonym]\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "    def __new__(cls, *args, **kwargs):\n",
      "\n",
      "        type_designator = \"category\"\n",
      "        if not type_designator in kwargs:\n",
      "            return super().__new__(cls,*args,**kwargs)\n",
      "        else:\n",
      "            type_designator_value = kwargs[type_designator]\n",
      "            target_cls = cls._class_for(\"class_class_curie\", type_designator_value)\n",
      "\n",
      "\n",
      "            if target_cls is None:\n",
      "                target_cls = cls._class_for(\"class_class_uri\", type_designator_value)\n",
      "\n",
      "\n",
      "            if target_cls is None:\n",
      "                target_cls = cls._class_for(\"class_model_uri\", type_designator_value)\n",
      "\n",
      "\n",
      "            if target_cls is None:\n",
      "                raise ValueError(f\"Wrong type designator value: class {cls.__name__} \"\n",
      "                                 f\"has no subclass with ['class_class_curie', 'class_class_uri', 'class_model_uri']='{kwargs[type_designator]}'\")\n",
      "            return super().__new__(target_cls,*args,**kwargs)\n",
      "\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Attribute(NamedThing):\n",
      "    \"\"\"\n",
      "    A property or characteristic of an entity. For example, an apple may have properties such as color, shape, age,\n",
      "    crispiness. An environmental sample may have attributes such as depth, lat, long, material.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Attribute\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Attribute\"\n",
      "    class_name: ClassVar[str] = \"attribute\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Attribute\n",
      "\n",
      "    id: Union[str, AttributeId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_attribute_type: Union[str, OntologyClassId] = None\n",
      "    name: Optional[Union[str, LabelType]] = None\n",
      "    has_quantitative_value: Optional[Union[Union[dict, QuantityValue], List[Union[dict, QuantityValue]]]] = empty_list()\n",
      "    has_qualitative_value: Optional[Union[str, NamedThingId]] = None\n",
      "    iri: Optional[Union[str, IriType]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, AttributeId):\n",
      "            self.id = AttributeId(self.id)\n",
      "\n",
      "        if self._is_empty(self.has_attribute_type):\n",
      "            self.MissingRequiredField(\"has_attribute_type\")\n",
      "        if not isinstance(self.has_attribute_type, OntologyClassId):\n",
      "            self.has_attribute_type = OntologyClassId(self.has_attribute_type)\n",
      "\n",
      "        if self.name is not None and not isinstance(self.name, LabelType):\n",
      "            self.name = LabelType(self.name)\n",
      "\n",
      "        if not isinstance(self.has_quantitative_value, list):\n",
      "            self.has_quantitative_value = [self.has_quantitative_value] if self.has_quantitative_value is not None else []\n",
      "        self.has_quantitative_value = [v if isinstance(v, QuantityValue) else QuantityValue(**as_dict(v)) for v in self.has_quantitative_value]\n",
      "\n",
      "        if self.has_qualitative_value is not None and not isinstance(self.has_qualitative_value, NamedThingId):\n",
      "            self.has_qualitative_value = NamedThingId(self.has_qualitative_value)\n",
      "\n",
      "        if self.iri is not None and not isinstance(self.iri, IriType):\n",
      "            self.iri = IriType(self.iri)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ChemicalRole(Attribute):\n",
      "    \"\"\"\n",
      "    A role played by the molecular entity or part thereof within a chemical context.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ChemicalRole\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ChemicalRole\"\n",
      "    class_name: ClassVar[str] = \"chemical role\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ChemicalRole\n",
      "\n",
      "    id: Union[str, ChemicalRoleId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_attribute_type: Union[str, OntologyClassId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ChemicalRoleId):\n",
      "            self.id = ChemicalRoleId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class BiologicalSex(Attribute):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"BiologicalSex\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:BiologicalSex\"\n",
      "    class_name: ClassVar[str] = \"biological sex\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.BiologicalSex\n",
      "\n",
      "    id: Union[str, BiologicalSexId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_attribute_type: Union[str, OntologyClassId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, BiologicalSexId):\n",
      "            self.id = BiologicalSexId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class PhenotypicSex(BiologicalSex):\n",
      "    \"\"\"\n",
      "    An attribute corresponding to the phenotypic sex of the individual, based upon the reproductive organs present.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"PhenotypicSex\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:PhenotypicSex\"\n",
      "    class_name: ClassVar[str] = \"phenotypic sex\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.PhenotypicSex\n",
      "\n",
      "    id: Union[str, PhenotypicSexId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_attribute_type: Union[str, OntologyClassId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, PhenotypicSexId):\n",
      "            self.id = PhenotypicSexId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GenotypicSex(BiologicalSex):\n",
      "    \"\"\"\n",
      "    An attribute corresponding to the genotypic sex of the individual, based upon genotypic composition of sex\n",
      "    chromosomes.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GenotypicSex\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GenotypicSex\"\n",
      "    class_name: ClassVar[str] = \"genotypic sex\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GenotypicSex\n",
      "\n",
      "    id: Union[str, GenotypicSexId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_attribute_type: Union[str, OntologyClassId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GenotypicSexId):\n",
      "            self.id = GenotypicSexId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class SeverityValue(Attribute):\n",
      "    \"\"\"\n",
      "    describes the severity of a phenotypic feature or disease\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"SeverityValue\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:SeverityValue\"\n",
      "    class_name: ClassVar[str] = \"severity value\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.SeverityValue\n",
      "\n",
      "    id: Union[str, SeverityValueId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_attribute_type: Union[str, OntologyClassId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, SeverityValueId):\n",
      "            self.id = SeverityValueId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class RelationshipType(OntologyClass):\n",
      "    \"\"\"\n",
      "    An OWL property used as an edge label\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"RelationshipType\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:RelationshipType\"\n",
      "    class_name: ClassVar[str] = \"relationship type\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.RelationshipType\n",
      "\n",
      "    id: Union[str, RelationshipTypeId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, RelationshipTypeId):\n",
      "            self.id = RelationshipTypeId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class TaxonomicRank(OntologyClass):\n",
      "    \"\"\"\n",
      "    A descriptor for the rank within a taxonomic classification. Example instance: TAXRANK:0000017 (kingdom)\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"TaxonomicRank\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:TaxonomicRank\"\n",
      "    class_name: ClassVar[str] = \"taxonomic rank\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.TaxonomicRank\n",
      "\n",
      "    id: Union[str, TaxonomicRankId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, TaxonomicRankId):\n",
      "            self.id = TaxonomicRankId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class OrganismTaxon(NamedThing):\n",
      "    \"\"\"\n",
      "    A classification of a set of organisms. Example instances: NCBITaxon:9606 (Homo sapiens), NCBITaxon:2 (Bacteria).\n",
      "    Can also be used to represent strains or subspecies.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"OrganismTaxon\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:OrganismTaxon\"\n",
      "    class_name: ClassVar[str] = \"organism taxon\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.OrganismTaxon\n",
      "\n",
      "    id: Union[str, OrganismTaxonId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_taxonomic_rank: Optional[Union[str, TaxonomicRankId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, OrganismTaxonId):\n",
      "            self.id = OrganismTaxonId(self.id)\n",
      "\n",
      "        if self.has_taxonomic_rank is not None and not isinstance(self.has_taxonomic_rank, TaxonomicRankId):\n",
      "            self.has_taxonomic_rank = TaxonomicRankId(self.has_taxonomic_rank)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Event(NamedThing):\n",
      "    \"\"\"\n",
      "    Something that happens at a given place and time.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Event\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Event\"\n",
      "    class_name: ClassVar[str] = \"event\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Event\n",
      "\n",
      "    id: Union[str, EventId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, EventId):\n",
      "            self.id = EventId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class AdministrativeEntity(NamedThing):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"AdministrativeEntity\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:AdministrativeEntity\"\n",
      "    class_name: ClassVar[str] = \"administrative entity\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.AdministrativeEntity\n",
      "\n",
      "    id: Union[str, AdministrativeEntityId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Agent(AdministrativeEntity):\n",
      "    \"\"\"\n",
      "    person, group, organization or project that provides a piece of information (i.e. a knowledge association)\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Agent\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Agent\"\n",
      "    class_name: ClassVar[str] = \"agent\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Agent\n",
      "\n",
      "    id: Union[str, AgentId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    affiliation: Optional[Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]]] = empty_list()\n",
      "    address: Optional[str] = None\n",
      "    name: Optional[Union[str, LabelType]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, AgentId):\n",
      "            self.id = AgentId(self.id)\n",
      "\n",
      "        if not isinstance(self.affiliation, list):\n",
      "            self.affiliation = [self.affiliation] if self.affiliation is not None else []\n",
      "        self.affiliation = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.affiliation]\n",
      "\n",
      "        if self.address is not None and not isinstance(self.address, str):\n",
      "            self.address = str(self.address)\n",
      "\n",
      "        if self.name is not None and not isinstance(self.name, LabelType):\n",
      "            self.name = LabelType(self.name)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class InformationContentEntity(NamedThing):\n",
      "    \"\"\"\n",
      "    a piece of information that typically describes some topic of discourse or is used as support.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"InformationContentEntity\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:InformationContentEntity\"\n",
      "    class_name: ClassVar[str] = \"information content entity\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.InformationContentEntity\n",
      "\n",
      "    id: Union[str, InformationContentEntityId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    license: Optional[str] = None\n",
      "    rights: Optional[str] = None\n",
      "    format: Optional[str] = None\n",
      "    creation_date: Optional[Union[str, XSDDate]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self.license is not None and not isinstance(self.license, str):\n",
      "            self.license = str(self.license)\n",
      "\n",
      "        if self.rights is not None and not isinstance(self.rights, str):\n",
      "            self.rights = str(self.rights)\n",
      "\n",
      "        if self.format is not None and not isinstance(self.format, str):\n",
      "            self.format = str(self.format)\n",
      "\n",
      "        if self.creation_date is not None and not isinstance(self.creation_date, XSDDate):\n",
      "            self.creation_date = XSDDate(self.creation_date)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class StudyResult(InformationContentEntity):\n",
      "    \"\"\"\n",
      "    A collection of data items from a study that are about a particular study subject or experimental unit (the\n",
      "    'focus' of the Result) - optionally with context/provenance metadata that may be relevant to the interpretation of\n",
      "    this data as evidence.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"StudyResult\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:StudyResult\"\n",
      "    class_name: ClassVar[str] = \"study result\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.StudyResult\n",
      "\n",
      "    id: Union[str, StudyResultId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class StudyVariable(InformationContentEntity):\n",
      "    \"\"\"\n",
      "    a variable that is used as a measure in the investigation of a study\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"StudyVariable\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:StudyVariable\"\n",
      "    class_name: ClassVar[str] = \"study variable\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.StudyVariable\n",
      "\n",
      "    id: Union[str, StudyVariableId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, StudyVariableId):\n",
      "            self.id = StudyVariableId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class CommonDataElement(InformationContentEntity):\n",
      "    \"\"\"\n",
      "    A Common Data Element (CDE) is a standardized, precisely defined question, paired with a set of allowable\n",
      "    responses, used systematically across different sites, studies, or clinical trials to ensure consistent data\n",
      "    collection. Multiple CDEs (from one or more Collections) can be curated into Forms. (https://cde.nlm.nih.gov/home)\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"CommonDataElement\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:CommonDataElement\"\n",
      "    class_name: ClassVar[str] = \"common data element\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.CommonDataElement\n",
      "\n",
      "    id: Union[str, CommonDataElementId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, CommonDataElementId):\n",
      "            self.id = CommonDataElementId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ConceptCountAnalysisResult(StudyResult):\n",
      "    \"\"\"\n",
      "    A result of a concept count analysis.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ConceptCountAnalysisResult\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ConceptCountAnalysisResult\"\n",
      "    class_name: ClassVar[str] = \"concept count analysis result\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ConceptCountAnalysisResult\n",
      "\n",
      "    id: Union[str, ConceptCountAnalysisResultId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ConceptCountAnalysisResultId):\n",
      "            self.id = ConceptCountAnalysisResultId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ObservedExpectedFrequencyAnalysisResult(StudyResult):\n",
      "    \"\"\"\n",
      "    A result of a observed expected frequency analysis.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ObservedExpectedFrequencyAnalysisResult\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ObservedExpectedFrequencyAnalysisResult\"\n",
      "    class_name: ClassVar[str] = \"observed expected frequency analysis result\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ObservedExpectedFrequencyAnalysisResult\n",
      "\n",
      "    id: Union[str, ObservedExpectedFrequencyAnalysisResultId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ObservedExpectedFrequencyAnalysisResultId):\n",
      "            self.id = ObservedExpectedFrequencyAnalysisResultId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class RelativeFrequencyAnalysisResult(StudyResult):\n",
      "    \"\"\"\n",
      "    A result of a relative frequency analysis.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"RelativeFrequencyAnalysisResult\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:RelativeFrequencyAnalysisResult\"\n",
      "    class_name: ClassVar[str] = \"relative frequency analysis result\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.RelativeFrequencyAnalysisResult\n",
      "\n",
      "    id: Union[str, RelativeFrequencyAnalysisResultId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, RelativeFrequencyAnalysisResultId):\n",
      "            self.id = RelativeFrequencyAnalysisResultId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class TextMiningResult(StudyResult):\n",
      "    \"\"\"\n",
      "    A result of text mining.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"TextMiningResult\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:TextMiningResult\"\n",
      "    class_name: ClassVar[str] = \"text mining result\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.TextMiningResult\n",
      "\n",
      "    id: Union[str, TextMiningResultId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, TextMiningResultId):\n",
      "            self.id = TextMiningResultId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ChiSquaredAnalysisResult(StudyResult):\n",
      "    \"\"\"\n",
      "    A result of a chi squared analysis.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ChiSquaredAnalysisResult\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ChiSquaredAnalysisResult\"\n",
      "    class_name: ClassVar[str] = \"chi squared analysis result\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ChiSquaredAnalysisResult\n",
      "\n",
      "    id: Union[str, ChiSquaredAnalysisResultId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ChiSquaredAnalysisResultId):\n",
      "            self.id = ChiSquaredAnalysisResultId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class LogOddsAnalysisResult(StudyResult):\n",
      "    \"\"\"\n",
      "    A result of a log odds ratio analysis.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"LogOddsAnalysisResult\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:LogOddsAnalysisResult\"\n",
      "    class_name: ClassVar[str] = \"log odds analysis result\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.LogOddsAnalysisResult\n",
      "\n",
      "    id: Union[str, LogOddsAnalysisResultId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, LogOddsAnalysisResultId):\n",
      "            self.id = LogOddsAnalysisResultId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Dataset(InformationContentEntity):\n",
      "    \"\"\"\n",
      "    an item that refers to a collection of data from a data source.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Dataset\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Dataset\"\n",
      "    class_name: ClassVar[str] = \"dataset\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Dataset\n",
      "\n",
      "    id: Union[str, DatasetId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, DatasetId):\n",
      "            self.id = DatasetId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class DatasetDistribution(InformationContentEntity):\n",
      "    \"\"\"\n",
      "    an item that holds distribution level information about a dataset.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"DatasetDistribution\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:DatasetDistribution\"\n",
      "    class_name: ClassVar[str] = \"dataset distribution\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.DatasetDistribution\n",
      "\n",
      "    id: Union[str, DatasetDistributionId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    distribution_download_url: Optional[str] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, DatasetDistributionId):\n",
      "            self.id = DatasetDistributionId(self.id)\n",
      "\n",
      "        if self.distribution_download_url is not None and not isinstance(self.distribution_download_url, str):\n",
      "            self.distribution_download_url = str(self.distribution_download_url)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class DatasetVersion(InformationContentEntity):\n",
      "    \"\"\"\n",
      "    an item that holds version level information about a dataset.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"DatasetVersion\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:DatasetVersion\"\n",
      "    class_name: ClassVar[str] = \"dataset version\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.DatasetVersion\n",
      "\n",
      "    id: Union[str, DatasetVersionId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_dataset: Optional[Union[str, DatasetId]] = None\n",
      "    ingest_date: Optional[str] = None\n",
      "    has_distribution: Optional[Union[str, DatasetDistributionId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, DatasetVersionId):\n",
      "            self.id = DatasetVersionId(self.id)\n",
      "\n",
      "        if self.has_dataset is not None and not isinstance(self.has_dataset, DatasetId):\n",
      "            self.has_dataset = DatasetId(self.has_dataset)\n",
      "\n",
      "        if self.ingest_date is not None and not isinstance(self.ingest_date, str):\n",
      "            self.ingest_date = str(self.ingest_date)\n",
      "\n",
      "        if self.has_distribution is not None and not isinstance(self.has_distribution, DatasetDistributionId):\n",
      "            self.has_distribution = DatasetDistributionId(self.has_distribution)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class DatasetSummary(InformationContentEntity):\n",
      "    \"\"\"\n",
      "    an item that holds summary level information about a dataset.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"DatasetSummary\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:DatasetSummary\"\n",
      "    class_name: ClassVar[str] = \"dataset summary\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.DatasetSummary\n",
      "\n",
      "    id: Union[str, DatasetSummaryId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    source_web_page: Optional[str] = None\n",
      "    source_logo: Optional[str] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, DatasetSummaryId):\n",
      "            self.id = DatasetSummaryId(self.id)\n",
      "\n",
      "        if self.source_web_page is not None and not isinstance(self.source_web_page, str):\n",
      "            self.source_web_page = str(self.source_web_page)\n",
      "\n",
      "        if self.source_logo is not None and not isinstance(self.source_logo, str):\n",
      "            self.source_logo = str(self.source_logo)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ConfidenceLevel(InformationContentEntity):\n",
      "    \"\"\"\n",
      "    Level of confidence in a statement\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ConfidenceLevel\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ConfidenceLevel\"\n",
      "    class_name: ClassVar[str] = \"confidence level\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ConfidenceLevel\n",
      "\n",
      "    id: Union[str, ConfidenceLevelId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ConfidenceLevelId):\n",
      "            self.id = ConfidenceLevelId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class EvidenceType(InformationContentEntity):\n",
      "    \"\"\"\n",
      "    Class of evidence that supports an association\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"EvidenceType\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:EvidenceType\"\n",
      "    class_name: ClassVar[str] = \"evidence type\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.EvidenceType\n",
      "\n",
      "    id: Union[str, EvidenceTypeId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, EvidenceTypeId):\n",
      "            self.id = EvidenceTypeId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Publication(InformationContentEntity):\n",
      "    \"\"\"\n",
      "    Any published piece of information. Publications are considered broadly to include any document or document part\n",
      "    made available in print or on the web - which may include scientific journal issues, individual articles, and\n",
      "    books - as well as things like pre-prints, white papers, patents, drug labels, web pages, protocol documents, and\n",
      "    even a part of a publication if of significant knowledge scope (e.g. a figure, figure legend, or section\n",
      "    highlighted by NLP).\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Publication\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Publication\"\n",
      "    class_name: ClassVar[str] = \"publication\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Publication\n",
      "\n",
      "    id: Union[str, PublicationId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    publication_type: Union[str, List[str]] = None\n",
      "    authors: Optional[Union[Union[str, AgentId], List[Union[str, AgentId]]]] = empty_list()\n",
      "    pages: Optional[Union[str, List[str]]] = empty_list()\n",
      "    summary: Optional[str] = None\n",
      "    keywords: Optional[Union[str, List[str]]] = empty_list()\n",
      "    mesh_terms: Optional[Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]]] = empty_list()\n",
      "    xref: Optional[Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]]] = empty_list()\n",
      "    name: Optional[Union[str, LabelType]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, PublicationId):\n",
      "            self.id = PublicationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.publication_type):\n",
      "            self.MissingRequiredField(\"publication_type\")\n",
      "        if not isinstance(self.publication_type, list):\n",
      "            self.publication_type = [self.publication_type] if self.publication_type is not None else []\n",
      "        self.publication_type = [v if isinstance(v, str) else str(v) for v in self.publication_type]\n",
      "\n",
      "        if not isinstance(self.authors, list):\n",
      "            self.authors = [self.authors] if self.authors is not None else []\n",
      "        self.authors = [v if isinstance(v, AgentId) else AgentId(v) for v in self.authors]\n",
      "\n",
      "        if not isinstance(self.pages, list):\n",
      "            self.pages = [self.pages] if self.pages is not None else []\n",
      "        self.pages = [v if isinstance(v, str) else str(v) for v in self.pages]\n",
      "\n",
      "        if self.summary is not None and not isinstance(self.summary, str):\n",
      "            self.summary = str(self.summary)\n",
      "\n",
      "        if not isinstance(self.keywords, list):\n",
      "            self.keywords = [self.keywords] if self.keywords is not None else []\n",
      "        self.keywords = [v if isinstance(v, str) else str(v) for v in self.keywords]\n",
      "\n",
      "        if not isinstance(self.mesh_terms, list):\n",
      "            self.mesh_terms = [self.mesh_terms] if self.mesh_terms is not None else []\n",
      "        self.mesh_terms = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.mesh_terms]\n",
      "\n",
      "        if not isinstance(self.xref, list):\n",
      "            self.xref = [self.xref] if self.xref is not None else []\n",
      "        self.xref = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.xref]\n",
      "\n",
      "        if self.name is not None and not isinstance(self.name, LabelType):\n",
      "            self.name = LabelType(self.name)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Book(Publication):\n",
      "    \"\"\"\n",
      "    This class may rarely be instantiated except if use cases of a given knowledge graph support its utility.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Book\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Book\"\n",
      "    class_name: ClassVar[str] = \"book\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Book\n",
      "\n",
      "    id: Union[str, BookId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    publication_type: Union[str, List[str]] = None\n",
      "    type: Optional[Union[str, List[str]]] = empty_list()\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, BookId):\n",
      "            self.id = BookId(self.id)\n",
      "\n",
      "        if not isinstance(self.type, list):\n",
      "            self.type = [self.type] if self.type is not None else []\n",
      "        self.type = [v if isinstance(v, str) else str(v) for v in self.type]\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class BookChapter(Publication):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"BookChapter\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:BookChapter\"\n",
      "    class_name: ClassVar[str] = \"book chapter\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.BookChapter\n",
      "\n",
      "    id: Union[str, BookChapterId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    publication_type: Union[str, List[str]] = None\n",
      "    published_in: Union[str, URIorCURIE] = None\n",
      "    volume: Optional[str] = None\n",
      "    chapter: Optional[str] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, BookChapterId):\n",
      "            self.id = BookChapterId(self.id)\n",
      "\n",
      "        if self._is_empty(self.published_in):\n",
      "            self.MissingRequiredField(\"published_in\")\n",
      "        if not isinstance(self.published_in, URIorCURIE):\n",
      "            self.published_in = URIorCURIE(self.published_in)\n",
      "\n",
      "        if self.volume is not None and not isinstance(self.volume, str):\n",
      "            self.volume = str(self.volume)\n",
      "\n",
      "        if self.chapter is not None and not isinstance(self.chapter, str):\n",
      "            self.chapter = str(self.chapter)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Serial(Publication):\n",
      "    \"\"\"\n",
      "    This class may rarely be instantiated except if use cases of a given knowledge graph support its utility.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Serial\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Serial\"\n",
      "    class_name: ClassVar[str] = \"serial\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Serial\n",
      "\n",
      "    id: Union[str, SerialId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    publication_type: Union[str, List[str]] = None\n",
      "    iso_abbreviation: Optional[str] = None\n",
      "    volume: Optional[str] = None\n",
      "    issue: Optional[str] = None\n",
      "    type: Optional[Union[str, List[str]]] = empty_list()\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, SerialId):\n",
      "            self.id = SerialId(self.id)\n",
      "\n",
      "        if self.iso_abbreviation is not None and not isinstance(self.iso_abbreviation, str):\n",
      "            self.iso_abbreviation = str(self.iso_abbreviation)\n",
      "\n",
      "        if self.volume is not None and not isinstance(self.volume, str):\n",
      "            self.volume = str(self.volume)\n",
      "\n",
      "        if self.issue is not None and not isinstance(self.issue, str):\n",
      "            self.issue = str(self.issue)\n",
      "\n",
      "        if not isinstance(self.type, list):\n",
      "            self.type = [self.type] if self.type is not None else []\n",
      "        self.type = [v if isinstance(v, str) else str(v) for v in self.type]\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Article(Publication):\n",
      "    \"\"\"\n",
      "    a piece of writing on a particular topic presented as a stand-alone section of a larger publication\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Article\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Article\"\n",
      "    class_name: ClassVar[str] = \"article\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Article\n",
      "\n",
      "    id: Union[str, ArticleId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    publication_type: Union[str, List[str]] = None\n",
      "    published_in: Union[str, URIorCURIE] = None\n",
      "    iso_abbreviation: Optional[str] = None\n",
      "    volume: Optional[str] = None\n",
      "    issue: Optional[str] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ArticleId):\n",
      "            self.id = ArticleId(self.id)\n",
      "\n",
      "        if self._is_empty(self.published_in):\n",
      "            self.MissingRequiredField(\"published_in\")\n",
      "        if not isinstance(self.published_in, URIorCURIE):\n",
      "            self.published_in = URIorCURIE(self.published_in)\n",
      "\n",
      "        if self.iso_abbreviation is not None and not isinstance(self.iso_abbreviation, str):\n",
      "            self.iso_abbreviation = str(self.iso_abbreviation)\n",
      "\n",
      "        if self.volume is not None and not isinstance(self.volume, str):\n",
      "            self.volume = str(self.volume)\n",
      "\n",
      "        if self.issue is not None and not isinstance(self.issue, str):\n",
      "            self.issue = str(self.issue)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class JournalArticle(Article):\n",
      "    \"\"\"\n",
      "    an article, typically presenting results of research, that is published in an issue of a scientific journal.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"JournalArticle\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:JournalArticle\"\n",
      "    class_name: ClassVar[str] = \"journal article\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.JournalArticle\n",
      "\n",
      "    id: Union[str, JournalArticleId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    publication_type: Union[str, List[str]] = None\n",
      "    published_in: Union[str, URIorCURIE] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, JournalArticleId):\n",
      "            self.id = JournalArticleId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Patent(Publication):\n",
      "    \"\"\"\n",
      "    a legal document granted by a patent issuing authority which confers upon the patenter the sole right to make, use\n",
      "    and sell an invention for a set period of time.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Patent\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Patent\"\n",
      "    class_name: ClassVar[str] = \"patent\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Patent\n",
      "\n",
      "    id: Union[str, PatentId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    publication_type: Union[str, List[str]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, PatentId):\n",
      "            self.id = PatentId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class WebPage(Publication):\n",
      "    \"\"\"\n",
      "    a document that is published according to World Wide Web standards, which may incorporate text, graphics, sound,\n",
      "    and/or other features.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"WebPage\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:WebPage\"\n",
      "    class_name: ClassVar[str] = \"web page\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.WebPage\n",
      "\n",
      "    id: Union[str, WebPageId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    publication_type: Union[str, List[str]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, WebPageId):\n",
      "            self.id = WebPageId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class PreprintPublication(Publication):\n",
      "    \"\"\"\n",
      "    a document reresenting an early version of an author's original scholarly work, such as a research paper or a\n",
      "    review, prior to formal peer review and publication in a peer-reviewed scholarly or scientific journal.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"PreprintPublication\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:PreprintPublication\"\n",
      "    class_name: ClassVar[str] = \"preprint publication\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.PreprintPublication\n",
      "\n",
      "    id: Union[str, PreprintPublicationId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    publication_type: Union[str, List[str]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, PreprintPublicationId):\n",
      "            self.id = PreprintPublicationId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class DrugLabel(Publication):\n",
      "    \"\"\"\n",
      "    a document accompanying a drug or its container that provides written, printed or graphic information about the\n",
      "    drug, including drug contents, specific instructions or warnings for administration, storage and disposal\n",
      "    instructions, etc.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"DrugLabel\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:DrugLabel\"\n",
      "    class_name: ClassVar[str] = \"drug label\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.DrugLabel\n",
      "\n",
      "    id: Union[str, DrugLabelId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    publication_type: Union[str, List[str]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, DrugLabelId):\n",
      "            self.id = DrugLabelId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class RetrievalSource(InformationContentEntity):\n",
      "    \"\"\"\n",
      "    Provides information about how a particular InformationResource served as a source from which knowledge expressed\n",
      "    in an Edge, or data used to generate this knowledge, was retrieved.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"RetrievalSource\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:RetrievalSource\"\n",
      "    class_name: ClassVar[str] = \"retrieval source\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.RetrievalSource\n",
      "\n",
      "    id: Union[str, RetrievalSourceId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    resource_id: Union[str, URIorCURIE] = None\n",
      "    resource_role: Union[str, \"ResourceRoleEnum\"] = None\n",
      "    upstream_resource_ids: Optional[Union[str, URIorCURIE]] = None\n",
      "    xref: Optional[Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]]] = empty_list()\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, RetrievalSourceId):\n",
      "            self.id = RetrievalSourceId(self.id)\n",
      "\n",
      "        if self._is_empty(self.resource_id):\n",
      "            self.MissingRequiredField(\"resource_id\")\n",
      "        if not isinstance(self.resource_id, URIorCURIE):\n",
      "            self.resource_id = URIorCURIE(self.resource_id)\n",
      "\n",
      "        if self._is_empty(self.resource_role):\n",
      "            self.MissingRequiredField(\"resource_role\")\n",
      "        if not isinstance(self.resource_role, ResourceRoleEnum):\n",
      "            self.resource_role = ResourceRoleEnum(self.resource_role)\n",
      "\n",
      "        if self.upstream_resource_ids is not None and not isinstance(self.upstream_resource_ids, URIorCURIE):\n",
      "            self.upstream_resource_ids = URIorCURIE(self.upstream_resource_ids)\n",
      "\n",
      "        if not isinstance(self.xref, list):\n",
      "            self.xref = [self.xref] if self.xref is not None else []\n",
      "        self.xref = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.xref]\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "class PhysicalEssenceOrOccurrent(YAMLRoot):\n",
      "    \"\"\"\n",
      "    Either a physical or processual entity.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"PhysicalEssenceOrOccurrent\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:PhysicalEssenceOrOccurrent\"\n",
      "    class_name: ClassVar[str] = \"physical essence or occurrent\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.PhysicalEssenceOrOccurrent\n",
      "\n",
      "\n",
      "class PhysicalEssence(PhysicalEssenceOrOccurrent):\n",
      "    \"\"\"\n",
      "    Semantic mixin concept.  Pertains to entities that have physical properties such as mass, volume, or charge.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"PhysicalEssence\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:PhysicalEssence\"\n",
      "    class_name: ClassVar[str] = \"physical essence\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.PhysicalEssence\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class PhysicalEntity(NamedThing):\n",
      "    \"\"\"\n",
      "    An entity that has material reality (a.k.a. physical essence).\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"PhysicalEntity\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:PhysicalEntity\"\n",
      "    class_name: ClassVar[str] = \"physical entity\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.PhysicalEntity\n",
      "\n",
      "    id: Union[str, PhysicalEntityId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, PhysicalEntityId):\n",
      "            self.id = PhysicalEntityId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "class Occurrent(PhysicalEssenceOrOccurrent):\n",
      "    \"\"\"\n",
      "    A processual entity.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Occurrent\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Occurrent\"\n",
      "    class_name: ClassVar[str] = \"occurrent\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Occurrent\n",
      "\n",
      "\n",
      "class ActivityAndBehavior(Occurrent):\n",
      "    \"\"\"\n",
      "    Activity or behavior of any independent integral living, organization or mechanical actor in the world\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ActivityAndBehavior\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ActivityAndBehavior\"\n",
      "    class_name: ClassVar[str] = \"activity and behavior\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ActivityAndBehavior\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Activity(NamedThing):\n",
      "    \"\"\"\n",
      "    An activity is something that occurs over a period of time and acts upon or with entities; it may include\n",
      "    consuming, processing, transforming, modifying, relocating, using, or generating entities.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Activity\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Activity\"\n",
      "    class_name: ClassVar[str] = \"activity\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Activity\n",
      "\n",
      "    id: Union[str, ActivityId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ActivityId):\n",
      "            self.id = ActivityId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Study(Activity):\n",
      "    \"\"\"\n",
      "    a detailed investigation and/or analysis\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Study\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Study\"\n",
      "    class_name: ClassVar[str] = \"study\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Study\n",
      "\n",
      "    id: Union[str, StudyId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, StudyId):\n",
      "            self.id = StudyId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Procedure(NamedThing):\n",
      "    \"\"\"\n",
      "    A series of actions conducted in a certain order or manner\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Procedure\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Procedure\"\n",
      "    class_name: ClassVar[str] = \"procedure\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Procedure\n",
      "\n",
      "    id: Union[str, ProcedureId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ProcedureId):\n",
      "            self.id = ProcedureId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Phenomenon(NamedThing):\n",
      "    \"\"\"\n",
      "    a fact or situation that is observed to exist or happen, especially one whose cause or explanation is in question\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Phenomenon\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Phenomenon\"\n",
      "    class_name: ClassVar[str] = \"phenomenon\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Phenomenon\n",
      "\n",
      "    id: Union[str, PhenomenonId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, PhenomenonId):\n",
      "            self.id = PhenomenonId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Device(NamedThing):\n",
      "    \"\"\"\n",
      "    A thing made or adapted for a particular purpose, especially a piece of mechanical or electronic equipment\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Device\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Device\"\n",
      "    class_name: ClassVar[str] = \"device\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Device\n",
      "\n",
      "    id: Union[str, DeviceId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, DeviceId):\n",
      "            self.id = DeviceId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class DiagnosticAid(NamedThing):\n",
      "    \"\"\"\n",
      "    A device or substance used to help diagnose disease or injury\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"DiagnosticAid\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:DiagnosticAid\"\n",
      "    class_name: ClassVar[str] = \"diagnostic aid\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.DiagnosticAid\n",
      "\n",
      "    id: Union[str, DiagnosticAidId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, DiagnosticAidId):\n",
      "            self.id = DiagnosticAidId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "class SubjectOfInvestigation(YAMLRoot):\n",
      "    \"\"\"\n",
      "    An entity that has the role of being studied in an investigation, study, or experiment\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"SubjectOfInvestigation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:SubjectOfInvestigation\"\n",
      "    class_name: ClassVar[str] = \"subject of investigation\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.SubjectOfInvestigation\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class MaterialSample(PhysicalEntity):\n",
      "    \"\"\"\n",
      "    A sample is a limited quantity of something (e.g. an individual or set of individuals from a population, or a\n",
      "    portion of a substance) to be used for testing, analysis, inspection, investigation, demonstration, or trial use.\n",
      "    [SIO]\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"MaterialSample\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:MaterialSample\"\n",
      "    class_name: ClassVar[str] = \"material sample\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.MaterialSample\n",
      "\n",
      "    id: Union[str, MaterialSampleId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, MaterialSampleId):\n",
      "            self.id = MaterialSampleId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class PlanetaryEntity(NamedThing):\n",
      "    \"\"\"\n",
      "    Any entity or process that exists at the level of the whole planet\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"PlanetaryEntity\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:PlanetaryEntity\"\n",
      "    class_name: ClassVar[str] = \"planetary entity\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.PlanetaryEntity\n",
      "\n",
      "    id: Union[str, PlanetaryEntityId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, PlanetaryEntityId):\n",
      "            self.id = PlanetaryEntityId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class EnvironmentalProcess(PlanetaryEntity):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"EnvironmentalProcess\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:EnvironmentalProcess\"\n",
      "    class_name: ClassVar[str] = \"environmental process\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.EnvironmentalProcess\n",
      "\n",
      "    id: Union[str, EnvironmentalProcessId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, EnvironmentalProcessId):\n",
      "            self.id = EnvironmentalProcessId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class EnvironmentalFeature(PlanetaryEntity):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"EnvironmentalFeature\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:EnvironmentalFeature\"\n",
      "    class_name: ClassVar[str] = \"environmental feature\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.EnvironmentalFeature\n",
      "\n",
      "    id: Union[str, EnvironmentalFeatureId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, EnvironmentalFeatureId):\n",
      "            self.id = EnvironmentalFeatureId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GeographicLocation(PlanetaryEntity):\n",
      "    \"\"\"\n",
      "    a location that can be described in lat/long coordinates\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GeographicLocation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GeographicLocation\"\n",
      "    class_name: ClassVar[str] = \"geographic location\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GeographicLocation\n",
      "\n",
      "    id: Union[str, GeographicLocationId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    latitude: Optional[float] = None\n",
      "    longitude: Optional[float] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GeographicLocationId):\n",
      "            self.id = GeographicLocationId(self.id)\n",
      "\n",
      "        if self.latitude is not None and not isinstance(self.latitude, float):\n",
      "            self.latitude = float(self.latitude)\n",
      "\n",
      "        if self.longitude is not None and not isinstance(self.longitude, float):\n",
      "            self.longitude = float(self.longitude)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GeographicLocationAtTime(GeographicLocation):\n",
      "    \"\"\"\n",
      "    a location that can be described in lat/long coordinates, for a particular time\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GeographicLocationAtTime\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GeographicLocationAtTime\"\n",
      "    class_name: ClassVar[str] = \"geographic location at time\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GeographicLocationAtTime\n",
      "\n",
      "    id: Union[str, GeographicLocationAtTimeId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    timepoint: Optional[Union[str, TimeType]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GeographicLocationAtTimeId):\n",
      "            self.id = GeographicLocationAtTimeId(self.id)\n",
      "\n",
      "        if self.timepoint is not None and not isinstance(self.timepoint, TimeType):\n",
      "            self.timepoint = TimeType(self.timepoint)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ThingWithTaxon(YAMLRoot):\n",
      "    \"\"\"\n",
      "    A mixin that can be used on any entity that can be taxonomically classified. This includes individual organisms;\n",
      "    genes, their products and other molecular entities; body parts; biological processes\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ThingWithTaxon\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ThingWithTaxon\"\n",
      "    class_name: ClassVar[str] = \"thing with taxon\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ThingWithTaxon\n",
      "\n",
      "    in_taxon: Optional[Union[Union[str, OrganismTaxonId], List[Union[str, OrganismTaxonId]]]] = empty_list()\n",
      "    in_taxon_label: Optional[Union[str, LabelType]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if not isinstance(self.in_taxon, list):\n",
      "            self.in_taxon = [self.in_taxon] if self.in_taxon is not None else []\n",
      "        self.in_taxon = [v if isinstance(v, OrganismTaxonId) else OrganismTaxonId(v) for v in self.in_taxon]\n",
      "\n",
      "        if self.in_taxon_label is not None and not isinstance(self.in_taxon_label, LabelType):\n",
      "            self.in_taxon_label = LabelType(self.in_taxon_label)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class BiologicalEntity(NamedThing):\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"BiologicalEntity\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:BiologicalEntity\"\n",
      "    class_name: ClassVar[str] = \"biological entity\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.BiologicalEntity\n",
      "\n",
      "    id: Union[str, BiologicalEntityId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    in_taxon: Optional[Union[Union[str, OrganismTaxonId], List[Union[str, OrganismTaxonId]]]] = empty_list()\n",
      "    in_taxon_label: Optional[Union[str, LabelType]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if not isinstance(self.in_taxon, list):\n",
      "            self.in_taxon = [self.in_taxon] if self.in_taxon is not None else []\n",
      "        self.in_taxon = [v if isinstance(v, OrganismTaxonId) else OrganismTaxonId(v) for v in self.in_taxon]\n",
      "\n",
      "        if self.in_taxon_label is not None and not isinstance(self.in_taxon_label, LabelType):\n",
      "            self.in_taxon_label = LabelType(self.in_taxon_label)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GenomicEntity(YAMLRoot):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GenomicEntity\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GenomicEntity\"\n",
      "    class_name: ClassVar[str] = \"genomic entity\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GenomicEntity\n",
      "\n",
      "    has_biological_sequence: Optional[Union[str, BiologicalSequence]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self.has_biological_sequence is not None and not isinstance(self.has_biological_sequence, BiologicalSequence):\n",
      "            self.has_biological_sequence = BiologicalSequence(self.has_biological_sequence)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class EpigenomicEntity(YAMLRoot):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"EpigenomicEntity\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:EpigenomicEntity\"\n",
      "    class_name: ClassVar[str] = \"epigenomic entity\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.EpigenomicEntity\n",
      "\n",
      "    has_biological_sequence: Optional[Union[str, BiologicalSequence]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self.has_biological_sequence is not None and not isinstance(self.has_biological_sequence, BiologicalSequence):\n",
      "            self.has_biological_sequence = BiologicalSequence(self.has_biological_sequence)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ChemicalEntity(NamedThing):\n",
      "    \"\"\"\n",
      "    A chemical entity is a physical entity that pertains to chemistry or biochemistry.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"has_chemical_role\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ChemicalEntity\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ChemicalEntity\"\n",
      "    class_name: ClassVar[str] = \"chemical entity\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ChemicalEntity\n",
      "\n",
      "    id: Union[str, ChemicalEntityId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    trade_name: Optional[str] = None\n",
      "    available_from: Optional[Union[Union[str, \"DrugAvailabilityEnum\"], List[Union[str, \"DrugAvailabilityEnum\"]]]] = empty_list()\n",
      "    max_tolerated_dose: Optional[str] = None\n",
      "    is_toxic: Optional[Union[bool, Bool]] = None\n",
      "    has_chemical_role: Optional[Union[Union[str, ChemicalRoleId], List[Union[str, ChemicalRoleId]]]] = empty_list()\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ChemicalEntityId):\n",
      "            self.id = ChemicalEntityId(self.id)\n",
      "\n",
      "        if self.trade_name is not None and not isinstance(self.trade_name, str):\n",
      "            self.trade_name = str(self.trade_name)\n",
      "\n",
      "        if not isinstance(self.available_from, list):\n",
      "            self.available_from = [self.available_from] if self.available_from is not None else []\n",
      "        self.available_from = [v if isinstance(v, DrugAvailabilityEnum) else DrugAvailabilityEnum(v) for v in self.available_from]\n",
      "\n",
      "        if self.max_tolerated_dose is not None and not isinstance(self.max_tolerated_dose, str):\n",
      "            self.max_tolerated_dose = str(self.max_tolerated_dose)\n",
      "\n",
      "        if self.is_toxic is not None and not isinstance(self.is_toxic, Bool):\n",
      "            self.is_toxic = Bool(self.is_toxic)\n",
      "\n",
      "        if not isinstance(self.has_chemical_role, list):\n",
      "            self.has_chemical_role = [self.has_chemical_role] if self.has_chemical_role is not None else []\n",
      "        self.has_chemical_role = [v if isinstance(v, ChemicalRoleId) else ChemicalRoleId(v) for v in self.has_chemical_role]\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class MolecularEntity(ChemicalEntity):\n",
      "    \"\"\"\n",
      "    A molecular entity is a chemical entity composed of individual or covalently bonded atoms.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"has_chemical_role\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"MolecularEntity\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:MolecularEntity\"\n",
      "    class_name: ClassVar[str] = \"molecular entity\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.MolecularEntity\n",
      "\n",
      "    id: Union[str, MolecularEntityId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    is_metabolite: Optional[Union[bool, Bool]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, MolecularEntityId):\n",
      "            self.id = MolecularEntityId(self.id)\n",
      "\n",
      "        if self.is_metabolite is not None and not isinstance(self.is_metabolite, Bool):\n",
      "            self.is_metabolite = Bool(self.is_metabolite)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class SmallMolecule(MolecularEntity):\n",
      "    \"\"\"\n",
      "    A small molecule entity is a molecular entity characterized by availability in small-molecule databases of SMILES,\n",
      "    InChI, IUPAC, or other unambiguous representation of its precise chemical structure; for convenience of\n",
      "    representation, any valid chemical representation is included, even if it is not strictly molecular (e.g., sodium\n",
      "    ion).\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"has_chemical_role\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"SmallMolecule\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:SmallMolecule\"\n",
      "    class_name: ClassVar[str] = \"small molecule\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.SmallMolecule\n",
      "\n",
      "    id: Union[str, SmallMoleculeId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, SmallMoleculeId):\n",
      "            self.id = SmallMoleculeId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ChemicalMixture(ChemicalEntity):\n",
      "    \"\"\"\n",
      "    A chemical mixture is a chemical entity composed of two or more molecular entities.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"has_chemical_role\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ChemicalMixture\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ChemicalMixture\"\n",
      "    class_name: ClassVar[str] = \"chemical mixture\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ChemicalMixture\n",
      "\n",
      "    id: Union[str, ChemicalMixtureId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    is_supplement: Optional[str] = None\n",
      "    highest_FDA_approval_status: Optional[Union[str, \"ApprovalStatusEnum\"]] = None\n",
      "    drug_regulatory_status_world_wide: Optional[Union[str, \"ApprovalStatusEnum\"]] = None\n",
      "    routes_of_delivery: Optional[Union[Union[str, \"DrugDeliveryEnum\"], List[Union[str, \"DrugDeliveryEnum\"]]]] = empty_list()\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ChemicalMixtureId):\n",
      "            self.id = ChemicalMixtureId(self.id)\n",
      "\n",
      "        if self.is_supplement is not None and not isinstance(self.is_supplement, str):\n",
      "            self.is_supplement = str(self.is_supplement)\n",
      "\n",
      "        if self.highest_FDA_approval_status is not None and not isinstance(self.highest_FDA_approval_status, ApprovalStatusEnum):\n",
      "            self.highest_FDA_approval_status = ApprovalStatusEnum(self.highest_FDA_approval_status)\n",
      "\n",
      "        if self.drug_regulatory_status_world_wide is not None and not isinstance(self.drug_regulatory_status_world_wide, ApprovalStatusEnum):\n",
      "            self.drug_regulatory_status_world_wide = ApprovalStatusEnum(self.drug_regulatory_status_world_wide)\n",
      "\n",
      "        if not isinstance(self.routes_of_delivery, list):\n",
      "            self.routes_of_delivery = [self.routes_of_delivery] if self.routes_of_delivery is not None else []\n",
      "        self.routes_of_delivery = [v if isinstance(v, DrugDeliveryEnum) else DrugDeliveryEnum(v) for v in self.routes_of_delivery]\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class NucleicAcidEntity(MolecularEntity):\n",
      "    \"\"\"\n",
      "    A nucleic acid entity is a molecular entity characterized by availability in gene databases of nucleotide-based\n",
      "    sequence representations of its precise sequence; for convenience of representation, partial sequences of various\n",
      "    kinds are included.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"has_chemical_role\", \"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"NucleicAcidEntity\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:NucleicAcidEntity\"\n",
      "    class_name: ClassVar[str] = \"nucleic acid entity\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.NucleicAcidEntity\n",
      "\n",
      "    id: Union[str, NucleicAcidEntityId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_biological_sequence: Optional[Union[str, BiologicalSequence]] = None\n",
      "    in_taxon: Optional[Union[Union[str, OrganismTaxonId], List[Union[str, OrganismTaxonId]]]] = empty_list()\n",
      "    in_taxon_label: Optional[Union[str, LabelType]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, NucleicAcidEntityId):\n",
      "            self.id = NucleicAcidEntityId(self.id)\n",
      "\n",
      "        if self.has_biological_sequence is not None and not isinstance(self.has_biological_sequence, BiologicalSequence):\n",
      "            self.has_biological_sequence = BiologicalSequence(self.has_biological_sequence)\n",
      "\n",
      "        if not isinstance(self.in_taxon, list):\n",
      "            self.in_taxon = [self.in_taxon] if self.in_taxon is not None else []\n",
      "        self.in_taxon = [v if isinstance(v, OrganismTaxonId) else OrganismTaxonId(v) for v in self.in_taxon]\n",
      "\n",
      "        if self.in_taxon_label is not None and not isinstance(self.in_taxon_label, LabelType):\n",
      "            self.in_taxon_label = LabelType(self.in_taxon_label)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class RegulatoryRegion(BiologicalEntity):\n",
      "    \"\"\"\n",
      "    A region (or regions) of the genome that contains known or putative regulatory elements that act in cis- or trans-\n",
      "    to affect the transcription of gene\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"RegulatoryRegion\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:RegulatoryRegion\"\n",
      "    class_name: ClassVar[str] = \"regulatory region\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.RegulatoryRegion\n",
      "\n",
      "    id: Union[str, RegulatoryRegionId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_biological_sequence: Optional[Union[str, BiologicalSequence]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, RegulatoryRegionId):\n",
      "            self.id = RegulatoryRegionId(self.id)\n",
      "\n",
      "        if self.has_biological_sequence is not None and not isinstance(self.has_biological_sequence, BiologicalSequence):\n",
      "            self.has_biological_sequence = BiologicalSequence(self.has_biological_sequence)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class AccessibleDnaRegion(RegulatoryRegion):\n",
      "    \"\"\"\n",
      "    A region (or regions) of a chromatinized genome that has been measured to be more accessible to an enzyme such as\n",
      "    DNase-I or Tn5 Transpose\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"AccessibleDnaRegion\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:AccessibleDnaRegion\"\n",
      "    class_name: ClassVar[str] = \"accessible dna region\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.AccessibleDnaRegion\n",
      "\n",
      "    id: Union[str, AccessibleDnaRegionId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_biological_sequence: Optional[Union[str, BiologicalSequence]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, AccessibleDnaRegionId):\n",
      "            self.id = AccessibleDnaRegionId(self.id)\n",
      "\n",
      "        if self.has_biological_sequence is not None and not isinstance(self.has_biological_sequence, BiologicalSequence):\n",
      "            self.has_biological_sequence = BiologicalSequence(self.has_biological_sequence)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class TranscriptionFactorBindingSite(RegulatoryRegion):\n",
      "    \"\"\"\n",
      "    A region (or regions) of the genome that contains a region of DNA known or predicted to bind a protein that\n",
      "    modulates gene transcription\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"TranscriptionFactorBindingSite\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:TranscriptionFactorBindingSite\"\n",
      "    class_name: ClassVar[str] = \"transcription factor binding site\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.TranscriptionFactorBindingSite\n",
      "\n",
      "    id: Union[str, TranscriptionFactorBindingSiteId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_biological_sequence: Optional[Union[str, BiologicalSequence]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, TranscriptionFactorBindingSiteId):\n",
      "            self.id = TranscriptionFactorBindingSiteId(self.id)\n",
      "\n",
      "        if self.has_biological_sequence is not None and not isinstance(self.has_biological_sequence, BiologicalSequence):\n",
      "            self.has_biological_sequence = BiologicalSequence(self.has_biological_sequence)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class MolecularMixture(ChemicalMixture):\n",
      "    \"\"\"\n",
      "    A molecular mixture is a chemical mixture composed of two or more molecular entities with known concentration and\n",
      "    stoichiometry.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"has_chemical_role\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"MolecularMixture\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:MolecularMixture\"\n",
      "    class_name: ClassVar[str] = \"molecular mixture\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.MolecularMixture\n",
      "\n",
      "    id: Union[str, MolecularMixtureId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, MolecularMixtureId):\n",
      "            self.id = MolecularMixtureId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ComplexMolecularMixture(ChemicalMixture):\n",
      "    \"\"\"\n",
      "    A complex molecular mixture is a chemical mixture composed of two or more molecular entities with unknown\n",
      "    concentration and stoichiometry.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"has_chemical_role\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ComplexMolecularMixture\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ComplexMolecularMixture\"\n",
      "    class_name: ClassVar[str] = \"complex molecular mixture\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ComplexMolecularMixture\n",
      "\n",
      "    id: Union[str, ComplexMolecularMixtureId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ComplexMolecularMixtureId):\n",
      "            self.id = ComplexMolecularMixtureId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class BiologicalProcessOrActivity(BiologicalEntity):\n",
      "    \"\"\"\n",
      "    Either an individual molecular activity, or a collection of causally connected molecular activities in a\n",
      "    biological system.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\", \"has_input\", \"has_output\", \"enabled_by\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"BiologicalProcessOrActivity\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:BiologicalProcessOrActivity\"\n",
      "    class_name: ClassVar[str] = \"biological process or activity\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.BiologicalProcessOrActivity\n",
      "\n",
      "    id: Union[str, BiologicalProcessOrActivityId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_input: Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]] = empty_list()\n",
      "    has_output: Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]] = empty_list()\n",
      "    enabled_by: Optional[Union[Union[str, PhysicalEntityId], List[Union[str, PhysicalEntityId]]]] = empty_list()\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, BiologicalProcessOrActivityId):\n",
      "            self.id = BiologicalProcessOrActivityId(self.id)\n",
      "\n",
      "        if not isinstance(self.has_input, list):\n",
      "            self.has_input = [self.has_input] if self.has_input is not None else []\n",
      "        self.has_input = [v if isinstance(v, NamedThingId) else NamedThingId(v) for v in self.has_input]\n",
      "\n",
      "        if not isinstance(self.has_output, list):\n",
      "            self.has_output = [self.has_output] if self.has_output is not None else []\n",
      "        self.has_output = [v if isinstance(v, NamedThingId) else NamedThingId(v) for v in self.has_output]\n",
      "\n",
      "        if not isinstance(self.enabled_by, list):\n",
      "            self.enabled_by = [self.enabled_by] if self.enabled_by is not None else []\n",
      "        self.enabled_by = [v if isinstance(v, PhysicalEntityId) else PhysicalEntityId(v) for v in self.enabled_by]\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class MolecularActivity(BiologicalProcessOrActivity):\n",
      "    \"\"\"\n",
      "    An execution of a molecular function carried out by a gene product or macromolecular complex.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\", \"has_input\", \"has_output\", \"enabled_by\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"MolecularActivity\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:MolecularActivity\"\n",
      "    class_name: ClassVar[str] = \"molecular activity\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.MolecularActivity\n",
      "\n",
      "    id: Union[str, MolecularActivityId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_input: Optional[Union[Union[str, MolecularEntityId], List[Union[str, MolecularEntityId]]]] = empty_list()\n",
      "    has_output: Optional[Union[Union[str, MolecularEntityId], List[Union[str, MolecularEntityId]]]] = empty_list()\n",
      "    enabled_by: Optional[Union[Union[dict, \"MacromolecularMachineMixin\"], List[Union[dict, \"MacromolecularMachineMixin\"]]]] = empty_list()\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, MolecularActivityId):\n",
      "            self.id = MolecularActivityId(self.id)\n",
      "\n",
      "        if not isinstance(self.has_input, list):\n",
      "            self.has_input = [self.has_input] if self.has_input is not None else []\n",
      "        self.has_input = [v if isinstance(v, MolecularEntityId) else MolecularEntityId(v) for v in self.has_input]\n",
      "\n",
      "        if not isinstance(self.has_output, list):\n",
      "            self.has_output = [self.has_output] if self.has_output is not None else []\n",
      "        self.has_output = [v if isinstance(v, MolecularEntityId) else MolecularEntityId(v) for v in self.has_output]\n",
      "\n",
      "        if not isinstance(self.enabled_by, list):\n",
      "            self.enabled_by = [self.enabled_by] if self.enabled_by is not None else []\n",
      "        self.enabled_by = [v if isinstance(v, MacromolecularMachineMixin) else MacromolecularMachineMixin(**as_dict(v)) for v in self.enabled_by]\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class BiologicalProcess(BiologicalProcessOrActivity):\n",
      "    \"\"\"\n",
      "    One or more causally connected executions of molecular functions\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\", \"has_input\", \"has_output\", \"enabled_by\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"BiologicalProcess\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:BiologicalProcess\"\n",
      "    class_name: ClassVar[str] = \"biological process\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.BiologicalProcess\n",
      "\n",
      "    id: Union[str, BiologicalProcessId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, BiologicalProcessId):\n",
      "            self.id = BiologicalProcessId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Pathway(BiologicalProcess):\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\", \"has_input\", \"has_output\", \"enabled_by\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Pathway\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Pathway\"\n",
      "    class_name: ClassVar[str] = \"pathway\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Pathway\n",
      "\n",
      "    id: Union[str, PathwayId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, PathwayId):\n",
      "            self.id = PathwayId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class PhysiologicalProcess(BiologicalProcess):\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\", \"has_input\", \"has_output\", \"enabled_by\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"PhysiologicalProcess\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:PhysiologicalProcess\"\n",
      "    class_name: ClassVar[str] = \"physiological process\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.PhysiologicalProcess\n",
      "\n",
      "    id: Union[str, PhysiologicalProcessId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, PhysiologicalProcessId):\n",
      "            self.id = PhysiologicalProcessId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Behavior(BiologicalProcess):\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\", \"has_input\", \"has_output\", \"enabled_by\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Behavior\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Behavior\"\n",
      "    class_name: ClassVar[str] = \"behavior\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Behavior\n",
      "\n",
      "    id: Union[str, BehaviorId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, BehaviorId):\n",
      "            self.id = BehaviorId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ProcessedMaterial(ChemicalMixture):\n",
      "    \"\"\"\n",
      "    A chemical entity (often a mixture) processed for consumption for nutritional, medical or technical use. Is a\n",
      "    material entity that is created or changed during material processing.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"has_chemical_role\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ProcessedMaterial\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ProcessedMaterial\"\n",
      "    class_name: ClassVar[str] = \"processed material\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ProcessedMaterial\n",
      "\n",
      "    id: Union[str, ProcessedMaterialId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ProcessedMaterialId):\n",
      "            self.id = ProcessedMaterialId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Drug(MolecularMixture):\n",
      "    \"\"\"\n",
      "    A substance intended for use in the diagnosis, cure, mitigation, treatment, or prevention of disease\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"has_chemical_role\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Drug\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Drug\"\n",
      "    class_name: ClassVar[str] = \"drug\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Drug\n",
      "\n",
      "    id: Union[str, DrugId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, DrugId):\n",
      "            self.id = DrugId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class EnvironmentalFoodContaminant(ChemicalEntity):\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"has_chemical_role\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"EnvironmentalFoodContaminant\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:EnvironmentalFoodContaminant\"\n",
      "    class_name: ClassVar[str] = \"environmental food contaminant\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.EnvironmentalFoodContaminant\n",
      "\n",
      "    id: Union[str, EnvironmentalFoodContaminantId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, EnvironmentalFoodContaminantId):\n",
      "            self.id = EnvironmentalFoodContaminantId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class FoodAdditive(ChemicalEntity):\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"has_chemical_role\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"FoodAdditive\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:FoodAdditive\"\n",
      "    class_name: ClassVar[str] = \"food additive\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.FoodAdditive\n",
      "\n",
      "    id: Union[str, FoodAdditiveId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, FoodAdditiveId):\n",
      "            self.id = FoodAdditiveId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Food(ChemicalMixture):\n",
      "    \"\"\"\n",
      "    A substance consumed by a living organism as a source of nutrition\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"has_chemical_role\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Food\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Food\"\n",
      "    class_name: ClassVar[str] = \"food\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Food\n",
      "\n",
      "    id: Union[str, FoodId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, FoodId):\n",
      "            self.id = FoodId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class OrganismAttribute(Attribute):\n",
      "    \"\"\"\n",
      "    describes a characteristic of an organismal entity.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"OrganismAttribute\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:OrganismAttribute\"\n",
      "    class_name: ClassVar[str] = \"organism attribute\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.OrganismAttribute\n",
      "\n",
      "    id: Union[str, OrganismAttributeId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_attribute_type: Union[str, OntologyClassId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, OrganismAttributeId):\n",
      "            self.id = OrganismAttributeId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class PhenotypicQuality(OrganismAttribute):\n",
      "    \"\"\"\n",
      "    A property of a phenotype\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"PhenotypicQuality\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:PhenotypicQuality\"\n",
      "    class_name: ClassVar[str] = \"phenotypic quality\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.PhenotypicQuality\n",
      "\n",
      "    id: Union[str, PhenotypicQualityId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_attribute_type: Union[str, OntologyClassId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, PhenotypicQualityId):\n",
      "            self.id = PhenotypicQualityId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GeneticInheritance(BiologicalEntity):\n",
      "    \"\"\"\n",
      "    The pattern or 'mode' in which a particular genetic trait or disorder is passed from one generation to the next,\n",
      "    e.g. autosomal dominant, autosomal recessive, etc.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GeneticInheritance\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GeneticInheritance\"\n",
      "    class_name: ClassVar[str] = \"genetic inheritance\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GeneticInheritance\n",
      "\n",
      "    id: Union[str, GeneticInheritanceId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GeneticInheritanceId):\n",
      "            self.id = GeneticInheritanceId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class OrganismalEntity(BiologicalEntity):\n",
      "    \"\"\"\n",
      "    A named entity that is either a part of an organism, a whole organism, population or clade of organisms, excluding\n",
      "    chemical entities\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"OrganismalEntity\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:OrganismalEntity\"\n",
      "    class_name: ClassVar[str] = \"organismal entity\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.OrganismalEntity\n",
      "\n",
      "    id: Union[str, OrganismalEntityId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_attribute: Optional[Union[Union[str, AttributeId], List[Union[str, AttributeId]]]] = empty_list()\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if not isinstance(self.has_attribute, list):\n",
      "            self.has_attribute = [self.has_attribute] if self.has_attribute is not None else []\n",
      "        self.has_attribute = [v if isinstance(v, AttributeId) else AttributeId(v) for v in self.has_attribute]\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Bacterium(OrganismalEntity):\n",
      "    \"\"\"\n",
      "    A member of a group of unicellular microorganisms lacking a nuclear membrane, that reproduce by binary fission and\n",
      "    are often motile.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Bacterium\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Bacterium\"\n",
      "    class_name: ClassVar[str] = \"bacterium\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Bacterium\n",
      "\n",
      "    id: Union[str, BacteriumId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, BacteriumId):\n",
      "            self.id = BacteriumId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Virus(OrganismalEntity):\n",
      "    \"\"\"\n",
      "    A virus is a microorganism that replicates itself as a microRNA and infects the host cell.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Virus\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Virus\"\n",
      "    class_name: ClassVar[str] = \"virus\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Virus\n",
      "\n",
      "    id: Union[str, VirusId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, VirusId):\n",
      "            self.id = VirusId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class CellularOrganism(OrganismalEntity):\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"CellularOrganism\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:CellularOrganism\"\n",
      "    class_name: ClassVar[str] = \"cellular organism\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.CellularOrganism\n",
      "\n",
      "    id: Union[str, CellularOrganismId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, CellularOrganismId):\n",
      "            self.id = CellularOrganismId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Mammal(CellularOrganism):\n",
      "    \"\"\"\n",
      "    A member of the class Mammalia, a clade of endothermic amniotes distinguished from reptiles and birds by the\n",
      "    possession of hair, three middle ear bones, mammary glands, and a neocortex\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Mammal\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Mammal\"\n",
      "    class_name: ClassVar[str] = \"mammal\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Mammal\n",
      "\n",
      "    id: Union[str, MammalId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, MammalId):\n",
      "            self.id = MammalId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Human(Mammal):\n",
      "    \"\"\"\n",
      "    A member of the the species Homo sapiens.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Human\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Human\"\n",
      "    class_name: ClassVar[str] = \"human\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Human\n",
      "\n",
      "    id: Union[str, HumanId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, HumanId):\n",
      "            self.id = HumanId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Plant(CellularOrganism):\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Plant\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Plant\"\n",
      "    class_name: ClassVar[str] = \"plant\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Plant\n",
      "\n",
      "    id: Union[str, PlantId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, PlantId):\n",
      "            self.id = PlantId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Invertebrate(CellularOrganism):\n",
      "    \"\"\"\n",
      "    An animal lacking a vertebral column. This group consists of 98% of all animal species.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Invertebrate\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Invertebrate\"\n",
      "    class_name: ClassVar[str] = \"invertebrate\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Invertebrate\n",
      "\n",
      "    id: Union[str, InvertebrateId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, InvertebrateId):\n",
      "            self.id = InvertebrateId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Vertebrate(CellularOrganism):\n",
      "    \"\"\"\n",
      "    A sub-phylum of animals consisting of those having a bony or cartilaginous vertebral column.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Vertebrate\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Vertebrate\"\n",
      "    class_name: ClassVar[str] = \"vertebrate\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Vertebrate\n",
      "\n",
      "    id: Union[str, VertebrateId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, VertebrateId):\n",
      "            self.id = VertebrateId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Fungus(CellularOrganism):\n",
      "    \"\"\"\n",
      "    A kingdom of eukaryotic, heterotrophic organisms that live as saprobes or parasites, including mushrooms, yeasts,\n",
      "    smuts, molds, etc. They reproduce either sexually or asexually, and have life cycles that range from simple to\n",
      "    complex. Filamentous fungi refer to those that grow as multicellular colonies (mushrooms and molds).\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Fungus\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Fungus\"\n",
      "    class_name: ClassVar[str] = \"fungus\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Fungus\n",
      "\n",
      "    id: Union[str, FungusId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, FungusId):\n",
      "            self.id = FungusId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class LifeStage(OrganismalEntity):\n",
      "    \"\"\"\n",
      "    A stage of development or growth of an organism, including post-natal adult stages\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"LifeStage\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:LifeStage\"\n",
      "    class_name: ClassVar[str] = \"life stage\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.LifeStage\n",
      "\n",
      "    id: Union[str, LifeStageId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, LifeStageId):\n",
      "            self.id = LifeStageId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class IndividualOrganism(OrganismalEntity):\n",
      "    \"\"\"\n",
      "    An instance of an organism. For example, Richard Nixon, Charles Darwin, my pet cat. Example ID:\n",
      "    ORCID:0000-0002-5355-2576\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"IndividualOrganism\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:IndividualOrganism\"\n",
      "    class_name: ClassVar[str] = \"individual organism\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.IndividualOrganism\n",
      "\n",
      "    id: Union[str, IndividualOrganismId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, IndividualOrganismId):\n",
      "            self.id = IndividualOrganismId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class PopulationOfIndividualOrganisms(OrganismalEntity):\n",
      "    \"\"\"\n",
      "    A collection of individuals from the same taxonomic class distinguished by one or more characteristics.\n",
      "    Characteristics can include, but are not limited to, shared geographic location, genetics, phenotypes.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"PopulationOfIndividualOrganisms\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:PopulationOfIndividualOrganisms\"\n",
      "    class_name: ClassVar[str] = \"population of individual organisms\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.PopulationOfIndividualOrganisms\n",
      "\n",
      "    id: Union[str, PopulationOfIndividualOrganismsId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, PopulationOfIndividualOrganismsId):\n",
      "            self.id = PopulationOfIndividualOrganismsId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class StudyPopulation(PopulationOfIndividualOrganisms):\n",
      "    \"\"\"\n",
      "    A group of people banded together or treated as a group as participants in a research study.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"StudyPopulation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:StudyPopulation\"\n",
      "    class_name: ClassVar[str] = \"study population\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.StudyPopulation\n",
      "\n",
      "    id: Union[str, StudyPopulationId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, StudyPopulationId):\n",
      "            self.id = StudyPopulationId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class DiseaseOrPhenotypicFeature(BiologicalEntity):\n",
      "    \"\"\"\n",
      "    Either one of a disease or an individual phenotypic feature. Some knowledge resources such as Monarch treat these\n",
      "    as distinct, others such as MESH conflate. Please see definitions of phenotypic feature and disease in this model\n",
      "    for their independent descriptions. This class is helpful to enforce domains and ranges that may involve either a\n",
      "    disease or a phenotypic feature.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"DiseaseOrPhenotypicFeature\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:DiseaseOrPhenotypicFeature\"\n",
      "    class_name: ClassVar[str] = \"disease or phenotypic feature\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.DiseaseOrPhenotypicFeature\n",
      "\n",
      "    id: Union[str, DiseaseOrPhenotypicFeatureId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, DiseaseOrPhenotypicFeatureId):\n",
      "            self.id = DiseaseOrPhenotypicFeatureId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Disease(DiseaseOrPhenotypicFeature):\n",
      "    \"\"\"\n",
      "    A disorder of structure or function, especially one that produces specific signs, phenotypes or symptoms or that\n",
      "    affects a specific location and is not simply a direct result of physical injury. A disposition to undergo\n",
      "    pathological processes that exists in an organism because of one or more disorders in that organism.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Disease\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Disease\"\n",
      "    class_name: ClassVar[str] = \"disease\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Disease\n",
      "\n",
      "    id: Union[str, DiseaseId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, DiseaseId):\n",
      "            self.id = DiseaseId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class PhenotypicFeature(DiseaseOrPhenotypicFeature):\n",
      "    \"\"\"\n",
      "    A combination of entity and quality that makes up a phenotyping statement. An observable characteristic of an\n",
      "    individual resulting from the interaction of its genotype with its molecular and physical environment.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"PhenotypicFeature\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:PhenotypicFeature\"\n",
      "    class_name: ClassVar[str] = \"phenotypic feature\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.PhenotypicFeature\n",
      "\n",
      "    id: Union[str, PhenotypicFeatureId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, PhenotypicFeatureId):\n",
      "            self.id = PhenotypicFeatureId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class BehavioralFeature(PhenotypicFeature):\n",
      "    \"\"\"\n",
      "    A phenotypic feature which is behavioral in nature.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"BehavioralFeature\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:BehavioralFeature\"\n",
      "    class_name: ClassVar[str] = \"behavioral feature\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.BehavioralFeature\n",
      "\n",
      "    id: Union[str, BehavioralFeatureId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, BehavioralFeatureId):\n",
      "            self.id = BehavioralFeatureId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class AnatomicalEntity(OrganismalEntity):\n",
      "    \"\"\"\n",
      "    A subcellular location, cell type or gross anatomical part\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"AnatomicalEntity\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:AnatomicalEntity\"\n",
      "    class_name: ClassVar[str] = \"anatomical entity\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.AnatomicalEntity\n",
      "\n",
      "    id: Union[str, AnatomicalEntityId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, AnatomicalEntityId):\n",
      "            self.id = AnatomicalEntityId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class CellularComponent(AnatomicalEntity):\n",
      "    \"\"\"\n",
      "    A location in or around a cell\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"CellularComponent\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:CellularComponent\"\n",
      "    class_name: ClassVar[str] = \"cellular component\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.CellularComponent\n",
      "\n",
      "    id: Union[str, CellularComponentId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, CellularComponentId):\n",
      "            self.id = CellularComponentId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Cell(AnatomicalEntity):\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Cell\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Cell\"\n",
      "    class_name: ClassVar[str] = \"cell\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Cell\n",
      "\n",
      "    id: Union[str, CellId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, CellId):\n",
      "            self.id = CellId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class CellLine(OrganismalEntity):\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"CellLine\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:CellLine\"\n",
      "    class_name: ClassVar[str] = \"cell line\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.CellLine\n",
      "\n",
      "    id: Union[str, CellLineId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, CellLineId):\n",
      "            self.id = CellLineId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GrossAnatomicalStructure(AnatomicalEntity):\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GrossAnatomicalStructure\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GrossAnatomicalStructure\"\n",
      "    class_name: ClassVar[str] = \"gross anatomical structure\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GrossAnatomicalStructure\n",
      "\n",
      "    id: Union[str, GrossAnatomicalStructureId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GrossAnatomicalStructureId):\n",
      "            self.id = GrossAnatomicalStructureId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "class ChemicalEntityOrGeneOrGeneProduct(YAMLRoot):\n",
      "    \"\"\"\n",
      "    A union of chemical entities and children, and gene or gene product. This mixin is helpful to use when searching\n",
      "    across chemical entities that must include genes and their children as chemical entities.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ChemicalEntityOrGeneOrGeneProduct\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ChemicalEntityOrGeneOrGeneProduct\"\n",
      "    class_name: ClassVar[str] = \"chemical entity or gene or gene product\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ChemicalEntityOrGeneOrGeneProduct\n",
      "\n",
      "\n",
      "class ChemicalEntityOrProteinOrPolypeptide(YAMLRoot):\n",
      "    \"\"\"\n",
      "    A union of chemical entities and children, and protein and polypeptide. This mixin is helpful to use when\n",
      "    searching across chemical entities that must include genes and their children as chemical entities.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ChemicalEntityOrProteinOrPolypeptide\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ChemicalEntityOrProteinOrPolypeptide\"\n",
      "    class_name: ClassVar[str] = \"chemical entity or protein or polypeptide\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ChemicalEntityOrProteinOrPolypeptide\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class MacromolecularMachineMixin(YAMLRoot):\n",
      "    \"\"\"\n",
      "    A union of gene locus, gene product, and macromolecular complex. These are the basic units of function in a cell.\n",
      "    They either carry out individual biological activities, or they encode molecules which do this.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"MacromolecularMachineMixin\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:MacromolecularMachineMixin\"\n",
      "    class_name: ClassVar[str] = \"macromolecular machine mixin\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.MacromolecularMachineMixin\n",
      "\n",
      "    name: Optional[Union[str, SymbolType]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self.name is not None and not isinstance(self.name, SymbolType):\n",
      "            self.name = SymbolType(self.name)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "class GeneOrGeneProduct(MacromolecularMachineMixin):\n",
      "    \"\"\"\n",
      "    A union of gene loci or gene products. Frequently an identifier for one will be used as proxy for another\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GeneOrGeneProduct\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GeneOrGeneProduct\"\n",
      "    class_name: ClassVar[str] = \"gene or gene product\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GeneOrGeneProduct\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Gene(BiologicalEntity):\n",
      "    \"\"\"\n",
      "    A region (or regions) that includes all of the sequence elements necessary to encode a functional transcript. A\n",
      "    gene locus may include regulatory regions, transcribed regions and/or other functional sequence regions.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Gene\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Gene\"\n",
      "    class_name: ClassVar[str] = \"gene\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Gene\n",
      "\n",
      "    id: Union[str, GeneId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    name: Optional[Union[str, LabelType]] = None\n",
      "    symbol: Optional[str] = None\n",
      "    xref: Optional[Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]]] = empty_list()\n",
      "    has_biological_sequence: Optional[Union[str, BiologicalSequence]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GeneId):\n",
      "            self.id = GeneId(self.id)\n",
      "\n",
      "        if self.name is not None and not isinstance(self.name, LabelType):\n",
      "            self.name = LabelType(self.name)\n",
      "\n",
      "        if self.symbol is not None and not isinstance(self.symbol, str):\n",
      "            self.symbol = str(self.symbol)\n",
      "\n",
      "        if not isinstance(self.xref, list):\n",
      "            self.xref = [self.xref] if self.xref is not None else []\n",
      "        self.xref = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.xref]\n",
      "\n",
      "        if self.has_biological_sequence is not None and not isinstance(self.has_biological_sequence, BiologicalSequence):\n",
      "            self.has_biological_sequence = BiologicalSequence(self.has_biological_sequence)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class MyGene(Gene):\n",
      "    \"\"\"\n",
      "    This is an example extension.  Doesn't do a lot\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = DIST[\"MyGene\"]\n",
      "    class_class_curie: ClassVar[str] = \"dist:MyGene\"\n",
      "    class_name: ClassVar[str] = \"my gene\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.MyGene\n",
      "\n",
      "    id: Union[str, MyGeneId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    gene_has_sequence: Union[str, GeneSequence] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, MyGeneId):\n",
      "            self.id = MyGeneId(self.id)\n",
      "\n",
      "        if self._is_empty(self.gene_has_sequence):\n",
      "            self.MissingRequiredField(\"gene_has_sequence\")\n",
      "        if not isinstance(self.gene_has_sequence, GeneSequence):\n",
      "            self.gene_has_sequence = GeneSequence(self.gene_has_sequence)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GeneProductMixin(GeneOrGeneProduct):\n",
      "    \"\"\"\n",
      "    The functional molecular product of a single gene locus. Gene products are either proteins or functional RNA\n",
      "    molecules.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GeneProductMixin\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GeneProductMixin\"\n",
      "    class_name: ClassVar[str] = \"gene product mixin\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GeneProductMixin\n",
      "\n",
      "    synonym: Optional[Union[Union[str, LabelType], List[Union[str, LabelType]]]] = empty_list()\n",
      "    xref: Optional[Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]]] = empty_list()\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if not isinstance(self.synonym, list):\n",
      "            self.synonym = [self.synonym] if self.synonym is not None else []\n",
      "        self.synonym = [v if isinstance(v, LabelType) else LabelType(v) for v in self.synonym]\n",
      "\n",
      "        if not isinstance(self.xref, list):\n",
      "            self.xref = [self.xref] if self.xref is not None else []\n",
      "        self.xref = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.xref]\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "class GeneProductIsoformMixin(GeneProductMixin):\n",
      "    \"\"\"\n",
      "    This is an abstract class that can be mixed in with different kinds of gene products to indicate that the gene\n",
      "    product is intended to represent a specific isoform rather than a canonical or reference or generic product. The\n",
      "    designation of canonical or reference may be arbitrary, or it may represent the superclass of all isoforms.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GeneProductIsoformMixin\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GeneProductIsoformMixin\"\n",
      "    class_name: ClassVar[str] = \"gene product isoform mixin\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GeneProductIsoformMixin\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class MacromolecularComplex(BiologicalEntity):\n",
      "    \"\"\"\n",
      "    A stable assembly of two or more macromolecules, i.e. proteins, nucleic acids, carbohydrates or lipids, in which\n",
      "    at least one component is a protein and the constituent parts function together.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"MacromolecularComplex\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:MacromolecularComplex\"\n",
      "    class_name: ClassVar[str] = \"macromolecular complex\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.MacromolecularComplex\n",
      "\n",
      "    id: Union[str, MacromolecularComplexId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    name: Optional[Union[str, LabelType]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, MacromolecularComplexId):\n",
      "            self.id = MacromolecularComplexId(self.id)\n",
      "\n",
      "        if self.name is not None and not isinstance(self.name, LabelType):\n",
      "            self.name = LabelType(self.name)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class NucleosomeModification(BiologicalEntity):\n",
      "    \"\"\"\n",
      "    A chemical modification of a histone protein within a nucleosome octomer or a substitution of a histone with a\n",
      "    variant histone isoform. e.g. Histone 4 Lysine 20 methylation (H4K20me), histone variant H2AZ substituting H2A.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"NucleosomeModification\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:NucleosomeModification\"\n",
      "    class_name: ClassVar[str] = \"nucleosome modification\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.NucleosomeModification\n",
      "\n",
      "    id: Union[str, NucleosomeModificationId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    name: Optional[Union[str, LabelType]] = None\n",
      "    xref: Optional[Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]]] = empty_list()\n",
      "    synonym: Optional[Union[Union[str, LabelType], List[Union[str, LabelType]]]] = empty_list()\n",
      "    has_biological_sequence: Optional[Union[str, BiologicalSequence]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, NucleosomeModificationId):\n",
      "            self.id = NucleosomeModificationId(self.id)\n",
      "\n",
      "        if self.name is not None and not isinstance(self.name, LabelType):\n",
      "            self.name = LabelType(self.name)\n",
      "\n",
      "        if not isinstance(self.xref, list):\n",
      "            self.xref = [self.xref] if self.xref is not None else []\n",
      "        self.xref = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.xref]\n",
      "\n",
      "        if not isinstance(self.synonym, list):\n",
      "            self.synonym = [self.synonym] if self.synonym is not None else []\n",
      "        self.synonym = [v if isinstance(v, LabelType) else LabelType(v) for v in self.synonym]\n",
      "\n",
      "        if self.has_biological_sequence is not None and not isinstance(self.has_biological_sequence, BiologicalSequence):\n",
      "            self.has_biological_sequence = BiologicalSequence(self.has_biological_sequence)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Genome(BiologicalEntity):\n",
      "    \"\"\"\n",
      "    A genome is the sum of genetic material within a cell or virion.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Genome\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Genome\"\n",
      "    class_name: ClassVar[str] = \"genome\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Genome\n",
      "\n",
      "    id: Union[str, GenomeId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_biological_sequence: Optional[Union[str, BiologicalSequence]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GenomeId):\n",
      "            self.id = GenomeId(self.id)\n",
      "\n",
      "        if self.has_biological_sequence is not None and not isinstance(self.has_biological_sequence, BiologicalSequence):\n",
      "            self.has_biological_sequence = BiologicalSequence(self.has_biological_sequence)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Exon(BiologicalEntity):\n",
      "    \"\"\"\n",
      "    A region of the transcript sequence within a gene which is not removed from the primary RNA transcript by RNA\n",
      "    splicing.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Exon\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Exon\"\n",
      "    class_name: ClassVar[str] = \"exon\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Exon\n",
      "\n",
      "    id: Union[str, ExonId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ExonId):\n",
      "            self.id = ExonId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Transcript(BiologicalEntity):\n",
      "    \"\"\"\n",
      "    An RNA synthesized on a DNA or RNA template by an RNA polymerase.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Transcript\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Transcript\"\n",
      "    class_name: ClassVar[str] = \"transcript\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Transcript\n",
      "\n",
      "    id: Union[str, TranscriptId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, TranscriptId):\n",
      "            self.id = TranscriptId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class CodingSequence(BiologicalEntity):\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"CodingSequence\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:CodingSequence\"\n",
      "    class_name: ClassVar[str] = \"coding sequence\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.CodingSequence\n",
      "\n",
      "    id: Union[str, CodingSequenceId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_biological_sequence: Optional[Union[str, BiologicalSequence]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, CodingSequenceId):\n",
      "            self.id = CodingSequenceId(self.id)\n",
      "\n",
      "        if self.has_biological_sequence is not None and not isinstance(self.has_biological_sequence, BiologicalSequence):\n",
      "            self.has_biological_sequence = BiologicalSequence(self.has_biological_sequence)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Polypeptide(BiologicalEntity):\n",
      "    \"\"\"\n",
      "    A polypeptide is a molecular entity characterized by availability in protein databases of amino-acid-based\n",
      "    sequence representations of its precise primary structure; for convenience of representation, partial sequences of\n",
      "    various kinds are included, even if they do not represent a physical molecule.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Polypeptide\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Polypeptide\"\n",
      "    class_name: ClassVar[str] = \"polypeptide\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Polypeptide\n",
      "\n",
      "    id: Union[str, PolypeptideId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, PolypeptideId):\n",
      "            self.id = PolypeptideId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Protein(Polypeptide):\n",
      "    \"\"\"\n",
      "    A gene product that is composed of a chain of amino acid sequences and is produced by ribosome-mediated\n",
      "    translation of mRNA\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Protein\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Protein\"\n",
      "    class_name: ClassVar[str] = \"protein\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Protein\n",
      "\n",
      "    id: Union[str, ProteinId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    name: Optional[Union[str, LabelType]] = None\n",
      "    xref: Optional[Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]]] = empty_list()\n",
      "    synonym: Optional[Union[Union[str, LabelType], List[Union[str, LabelType]]]] = empty_list()\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ProteinId):\n",
      "            self.id = ProteinId(self.id)\n",
      "\n",
      "        if self.name is not None and not isinstance(self.name, LabelType):\n",
      "            self.name = LabelType(self.name)\n",
      "\n",
      "        if not isinstance(self.xref, list):\n",
      "            self.xref = [self.xref] if self.xref is not None else []\n",
      "        self.xref = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.xref]\n",
      "\n",
      "        if not isinstance(self.synonym, list):\n",
      "            self.synonym = [self.synonym] if self.synonym is not None else []\n",
      "        self.synonym = [v if isinstance(v, LabelType) else LabelType(v) for v in self.synonym]\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ProteinIsoform(Protein):\n",
      "    \"\"\"\n",
      "    Represents a protein that is a specific isoform of the canonical or reference protein. See\n",
      "    https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4114032/\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ProteinIsoform\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ProteinIsoform\"\n",
      "    class_name: ClassVar[str] = \"protein isoform\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ProteinIsoform\n",
      "\n",
      "    id: Union[str, ProteinIsoformId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    name: Optional[Union[str, LabelType]] = None\n",
      "    xref: Optional[Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]]] = empty_list()\n",
      "    synonym: Optional[Union[Union[str, LabelType], List[Union[str, LabelType]]]] = empty_list()\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ProteinIsoformId):\n",
      "            self.id = ProteinIsoformId(self.id)\n",
      "\n",
      "        if self.name is not None and not isinstance(self.name, LabelType):\n",
      "            self.name = LabelType(self.name)\n",
      "\n",
      "        if not isinstance(self.xref, list):\n",
      "            self.xref = [self.xref] if self.xref is not None else []\n",
      "        self.xref = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.xref]\n",
      "\n",
      "        if not isinstance(self.synonym, list):\n",
      "            self.synonym = [self.synonym] if self.synonym is not None else []\n",
      "        self.synonym = [v if isinstance(v, LabelType) else LabelType(v) for v in self.synonym]\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ProteinDomain(BiologicalEntity):\n",
      "    \"\"\"\n",
      "    A conserved part of protein sequence and (tertiary) structure that can evolve, function, and exist independently\n",
      "    of the rest of the protein chain. Protein domains maintain their structure and function independently of the\n",
      "    proteins in which they are found. e.g. an SH3 domain.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ProteinDomain\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ProteinDomain\"\n",
      "    class_name: ClassVar[str] = \"protein domain\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ProteinDomain\n",
      "\n",
      "    id: Union[str, ProteinDomainId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_gene_or_gene_product: Optional[Union[Union[str, GeneId], List[Union[str, GeneId]]]] = empty_list()\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ProteinDomainId):\n",
      "            self.id = ProteinDomainId(self.id)\n",
      "\n",
      "        if not isinstance(self.has_gene_or_gene_product, list):\n",
      "            self.has_gene_or_gene_product = [self.has_gene_or_gene_product] if self.has_gene_or_gene_product is not None else []\n",
      "        self.has_gene_or_gene_product = [v if isinstance(v, GeneId) else GeneId(v) for v in self.has_gene_or_gene_product]\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class PosttranslationalModification(BiologicalEntity):\n",
      "    \"\"\"\n",
      "    A chemical modification of a polypeptide or protein that occurs after translation. e.g. polypeptide cleavage to\n",
      "    form separate proteins, methylation or acetylation of histone tail amino acids, protein ubiquitination.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"PosttranslationalModification\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:PosttranslationalModification\"\n",
      "    class_name: ClassVar[str] = \"posttranslational modification\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.PosttranslationalModification\n",
      "\n",
      "    id: Union[str, PosttranslationalModificationId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    name: Optional[Union[str, LabelType]] = None\n",
      "    xref: Optional[Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]]] = empty_list()\n",
      "    synonym: Optional[Union[Union[str, LabelType], List[Union[str, LabelType]]]] = empty_list()\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, PosttranslationalModificationId):\n",
      "            self.id = PosttranslationalModificationId(self.id)\n",
      "\n",
      "        if self.name is not None and not isinstance(self.name, LabelType):\n",
      "            self.name = LabelType(self.name)\n",
      "\n",
      "        if not isinstance(self.xref, list):\n",
      "            self.xref = [self.xref] if self.xref is not None else []\n",
      "        self.xref = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.xref]\n",
      "\n",
      "        if not isinstance(self.synonym, list):\n",
      "            self.synonym = [self.synonym] if self.synonym is not None else []\n",
      "        self.synonym = [v if isinstance(v, LabelType) else LabelType(v) for v in self.synonym]\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ProteinFamily(BiologicalEntity):\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ProteinFamily\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ProteinFamily\"\n",
      "    class_name: ClassVar[str] = \"protein family\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ProteinFamily\n",
      "\n",
      "    id: Union[str, ProteinFamilyId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_gene_or_gene_product: Optional[Union[Union[str, GeneId], List[Union[str, GeneId]]]] = empty_list()\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ProteinFamilyId):\n",
      "            self.id = ProteinFamilyId(self.id)\n",
      "\n",
      "        if not isinstance(self.has_gene_or_gene_product, list):\n",
      "            self.has_gene_or_gene_product = [self.has_gene_or_gene_product] if self.has_gene_or_gene_product is not None else []\n",
      "        self.has_gene_or_gene_product = [v if isinstance(v, GeneId) else GeneId(v) for v in self.has_gene_or_gene_product]\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class NucleicAcidSequenceMotif(BiologicalEntity):\n",
      "    \"\"\"\n",
      "    A linear nucleotide sequence pattern that is widespread and has, or is conjectured to have, a biological\n",
      "    significance. e.g. the TATA box promoter motif, transcription factor binding consensus sequences.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"NucleicAcidSequenceMotif\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:NucleicAcidSequenceMotif\"\n",
      "    class_name: ClassVar[str] = \"nucleic acid sequence motif\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.NucleicAcidSequenceMotif\n",
      "\n",
      "    id: Union[str, NucleicAcidSequenceMotifId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, NucleicAcidSequenceMotifId):\n",
      "            self.id = NucleicAcidSequenceMotifId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class RNAProduct(Transcript):\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"RNAProduct\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:RNAProduct\"\n",
      "    class_name: ClassVar[str] = \"RNA product\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.RNAProduct\n",
      "\n",
      "    id: Union[str, RNAProductId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    name: Optional[Union[str, LabelType]] = None\n",
      "    xref: Optional[Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]]] = empty_list()\n",
      "    synonym: Optional[Union[Union[str, LabelType], List[Union[str, LabelType]]]] = empty_list()\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, RNAProductId):\n",
      "            self.id = RNAProductId(self.id)\n",
      "\n",
      "        if self.name is not None and not isinstance(self.name, LabelType):\n",
      "            self.name = LabelType(self.name)\n",
      "\n",
      "        if not isinstance(self.xref, list):\n",
      "            self.xref = [self.xref] if self.xref is not None else []\n",
      "        self.xref = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.xref]\n",
      "\n",
      "        if not isinstance(self.synonym, list):\n",
      "            self.synonym = [self.synonym] if self.synonym is not None else []\n",
      "        self.synonym = [v if isinstance(v, LabelType) else LabelType(v) for v in self.synonym]\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class RNAProductIsoform(RNAProduct):\n",
      "    \"\"\"\n",
      "    Represents a protein that is a specific isoform of the canonical or reference RNA\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"RNAProductIsoform\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:RNAProductIsoform\"\n",
      "    class_name: ClassVar[str] = \"RNA product isoform\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.RNAProductIsoform\n",
      "\n",
      "    id: Union[str, RNAProductIsoformId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    name: Optional[Union[str, LabelType]] = None\n",
      "    xref: Optional[Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]]] = empty_list()\n",
      "    synonym: Optional[Union[Union[str, LabelType], List[Union[str, LabelType]]]] = empty_list()\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, RNAProductIsoformId):\n",
      "            self.id = RNAProductIsoformId(self.id)\n",
      "\n",
      "        if self.name is not None and not isinstance(self.name, LabelType):\n",
      "            self.name = LabelType(self.name)\n",
      "\n",
      "        if not isinstance(self.xref, list):\n",
      "            self.xref = [self.xref] if self.xref is not None else []\n",
      "        self.xref = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.xref]\n",
      "\n",
      "        if not isinstance(self.synonym, list):\n",
      "            self.synonym = [self.synonym] if self.synonym is not None else []\n",
      "        self.synonym = [v if isinstance(v, LabelType) else LabelType(v) for v in self.synonym]\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class NoncodingRNAProduct(RNAProduct):\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"NoncodingRNAProduct\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:NoncodingRNAProduct\"\n",
      "    class_name: ClassVar[str] = \"noncoding RNA product\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.NoncodingRNAProduct\n",
      "\n",
      "    id: Union[str, NoncodingRNAProductId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, NoncodingRNAProductId):\n",
      "            self.id = NoncodingRNAProductId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class MicroRNA(NoncodingRNAProduct):\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"MicroRNA\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:MicroRNA\"\n",
      "    class_name: ClassVar[str] = \"microRNA\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.MicroRNA\n",
      "\n",
      "    id: Union[str, MicroRNAId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, MicroRNAId):\n",
      "            self.id = MicroRNAId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class SiRNA(NoncodingRNAProduct):\n",
      "    \"\"\"\n",
      "    A small RNA molecule that is the product of a longer exogenous or endogenous dsRNA, which is either a bimolecular\n",
      "    duplex or very long hairpin, processed (via the Dicer pathway) such that numerous siRNAs accumulate from both\n",
      "    strands of the dsRNA. SRNAs trigger the cleavage of their target molecules.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"SiRNA\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:SiRNA\"\n",
      "    class_name: ClassVar[str] = \"siRNA\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.SiRNA\n",
      "\n",
      "    id: Union[str, SiRNAId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, SiRNAId):\n",
      "            self.id = SiRNAId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GeneGroupingMixin(YAMLRoot):\n",
      "    \"\"\"\n",
      "    any grouping of multiple genes or gene products\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GeneGroupingMixin\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GeneGroupingMixin\"\n",
      "    class_name: ClassVar[str] = \"gene grouping mixin\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GeneGroupingMixin\n",
      "\n",
      "    has_gene_or_gene_product: Optional[Union[Union[str, GeneId], List[Union[str, GeneId]]]] = empty_list()\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if not isinstance(self.has_gene_or_gene_product, list):\n",
      "            self.has_gene_or_gene_product = [self.has_gene_or_gene_product] if self.has_gene_or_gene_product is not None else []\n",
      "        self.has_gene_or_gene_product = [v if isinstance(v, GeneId) else GeneId(v) for v in self.has_gene_or_gene_product]\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GeneFamily(BiologicalEntity):\n",
      "    \"\"\"\n",
      "    any grouping of multiple genes or gene products related by common descent\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GeneFamily\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GeneFamily\"\n",
      "    class_name: ClassVar[str] = \"gene family\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GeneFamily\n",
      "\n",
      "    id: Union[str, GeneFamilyId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_gene_or_gene_product: Optional[Union[Union[str, GeneId], List[Union[str, GeneId]]]] = empty_list()\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GeneFamilyId):\n",
      "            self.id = GeneFamilyId(self.id)\n",
      "\n",
      "        if not isinstance(self.has_gene_or_gene_product, list):\n",
      "            self.has_gene_or_gene_product = [self.has_gene_or_gene_product] if self.has_gene_or_gene_product is not None else []\n",
      "        self.has_gene_or_gene_product = [v if isinstance(v, GeneId) else GeneId(v) for v in self.has_gene_or_gene_product]\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Zygosity(Attribute):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Zygosity\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Zygosity\"\n",
      "    class_name: ClassVar[str] = \"zygosity\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Zygosity\n",
      "\n",
      "    id: Union[str, ZygosityId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_attribute_type: Union[str, OntologyClassId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ZygosityId):\n",
      "            self.id = ZygosityId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Genotype(BiologicalEntity):\n",
      "    \"\"\"\n",
      "    An information content entity that describes a genome by specifying the total variation in genomic sequence and/or\n",
      "    gene expression, relative to some established background\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Genotype\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Genotype\"\n",
      "    class_name: ClassVar[str] = \"genotype\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Genotype\n",
      "\n",
      "    id: Union[str, GenotypeId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_zygosity: Optional[Union[str, ZygosityId]] = None\n",
      "    has_biological_sequence: Optional[Union[str, BiologicalSequence]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GenotypeId):\n",
      "            self.id = GenotypeId(self.id)\n",
      "\n",
      "        if self.has_zygosity is not None and not isinstance(self.has_zygosity, ZygosityId):\n",
      "            self.has_zygosity = ZygosityId(self.has_zygosity)\n",
      "\n",
      "        if self.has_biological_sequence is not None and not isinstance(self.has_biological_sequence, BiologicalSequence):\n",
      "            self.has_biological_sequence = BiologicalSequence(self.has_biological_sequence)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Haplotype(BiologicalEntity):\n",
      "    \"\"\"\n",
      "    A set of zero or more Alleles on a single instance of a Sequence[VMC]\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Haplotype\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Haplotype\"\n",
      "    class_name: ClassVar[str] = \"haplotype\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Haplotype\n",
      "\n",
      "    id: Union[str, HaplotypeId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_biological_sequence: Optional[Union[str, BiologicalSequence]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, HaplotypeId):\n",
      "            self.id = HaplotypeId(self.id)\n",
      "\n",
      "        if self.has_biological_sequence is not None and not isinstance(self.has_biological_sequence, BiologicalSequence):\n",
      "            self.has_biological_sequence = BiologicalSequence(self.has_biological_sequence)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class SequenceVariant(BiologicalEntity):\n",
      "    \"\"\"\n",
      "    A sequence_variant is a non exact copy of a sequence_feature or genome exhibiting one or more sequence_alteration.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"SequenceVariant\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:SequenceVariant\"\n",
      "    class_name: ClassVar[str] = \"sequence variant\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.SequenceVariant\n",
      "\n",
      "    id: Union[str, SequenceVariantId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_gene: Optional[Union[Union[str, GeneId], List[Union[str, GeneId]]]] = empty_list()\n",
      "    has_biological_sequence: Optional[Union[str, BiologicalSequence]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, SequenceVariantId):\n",
      "            self.id = SequenceVariantId(self.id)\n",
      "\n",
      "        if not isinstance(self.has_gene, list):\n",
      "            self.has_gene = [self.has_gene] if self.has_gene is not None else []\n",
      "        self.has_gene = [v if isinstance(v, GeneId) else GeneId(v) for v in self.has_gene]\n",
      "\n",
      "        if self.has_biological_sequence is not None and not isinstance(self.has_biological_sequence, BiologicalSequence):\n",
      "            self.has_biological_sequence = BiologicalSequence(self.has_biological_sequence)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Snv(SequenceVariant):\n",
      "    \"\"\"\n",
      "    SNVs are single nucleotide positions in genomic DNA at which different sequence alternatives exist\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Snv\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Snv\"\n",
      "    class_name: ClassVar[str] = \"snv\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Snv\n",
      "\n",
      "    id: Union[str, SnvId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, SnvId):\n",
      "            self.id = SnvId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ReagentTargetedGene(BiologicalEntity):\n",
      "    \"\"\"\n",
      "    A gene altered in its expression level in the context of some experiment as a result of being targeted by\n",
      "    gene-knockdown reagent(s) such as a morpholino or RNAi.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ReagentTargetedGene\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ReagentTargetedGene\"\n",
      "    class_name: ClassVar[str] = \"reagent targeted gene\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ReagentTargetedGene\n",
      "\n",
      "    id: Union[str, ReagentTargetedGeneId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_biological_sequence: Optional[Union[str, BiologicalSequence]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ReagentTargetedGeneId):\n",
      "            self.id = ReagentTargetedGeneId(self.id)\n",
      "\n",
      "        if self.has_biological_sequence is not None and not isinstance(self.has_biological_sequence, BiologicalSequence):\n",
      "            self.has_biological_sequence = BiologicalSequence(self.has_biological_sequence)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ClinicalAttribute(Attribute):\n",
      "    \"\"\"\n",
      "    Attributes relating to a clinical manifestation\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ClinicalAttribute\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ClinicalAttribute\"\n",
      "    class_name: ClassVar[str] = \"clinical attribute\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ClinicalAttribute\n",
      "\n",
      "    id: Union[str, ClinicalAttributeId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_attribute_type: Union[str, OntologyClassId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ClinicalAttributeId):\n",
      "            self.id = ClinicalAttributeId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ClinicalMeasurement(ClinicalAttribute):\n",
      "    \"\"\"\n",
      "    A clinical measurement is a special kind of attribute which results from a laboratory observation from a subject\n",
      "    individual or sample. Measurements can be connected to their subject by the 'has attribute' slot.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ClinicalMeasurement\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ClinicalMeasurement\"\n",
      "    class_name: ClassVar[str] = \"clinical measurement\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ClinicalMeasurement\n",
      "\n",
      "    id: Union[str, ClinicalMeasurementId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_attribute_type: Union[str, OntologyClassId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ClinicalMeasurementId):\n",
      "            self.id = ClinicalMeasurementId(self.id)\n",
      "\n",
      "        if self._is_empty(self.has_attribute_type):\n",
      "            self.MissingRequiredField(\"has_attribute_type\")\n",
      "        if not isinstance(self.has_attribute_type, OntologyClassId):\n",
      "            self.has_attribute_type = OntologyClassId(self.has_attribute_type)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ClinicalModifier(ClinicalAttribute):\n",
      "    \"\"\"\n",
      "    Used to characterize and specify the phenotypic abnormalities defined in the phenotypic abnormality sub-ontology,\n",
      "    with respect to severity, laterality, and other aspects\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ClinicalModifier\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ClinicalModifier\"\n",
      "    class_name: ClassVar[str] = \"clinical modifier\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ClinicalModifier\n",
      "\n",
      "    id: Union[str, ClinicalModifierId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_attribute_type: Union[str, OntologyClassId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ClinicalModifierId):\n",
      "            self.id = ClinicalModifierId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ClinicalCourse(ClinicalAttribute):\n",
      "    \"\"\"\n",
      "    The course a disease typically takes from its onset, progression in time, and eventual resolution or death of the\n",
      "    affected individual\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ClinicalCourse\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ClinicalCourse\"\n",
      "    class_name: ClassVar[str] = \"clinical course\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ClinicalCourse\n",
      "\n",
      "    id: Union[str, ClinicalCourseId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_attribute_type: Union[str, OntologyClassId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ClinicalCourseId):\n",
      "            self.id = ClinicalCourseId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Onset(ClinicalCourse):\n",
      "    \"\"\"\n",
      "    The age group in which (disease) symptom manifestations appear.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Onset\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Onset\"\n",
      "    class_name: ClassVar[str] = \"onset\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Onset\n",
      "\n",
      "    id: Union[str, OnsetId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_attribute_type: Union[str, OntologyClassId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, OnsetId):\n",
      "            self.id = OnsetId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ClinicalEntity(NamedThing):\n",
      "    \"\"\"\n",
      "    Any entity or process that exists in the clinical domain and outside the biological realm. Diseases are placed\n",
      "    under biological entities\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ClinicalEntity\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ClinicalEntity\"\n",
      "    class_name: ClassVar[str] = \"clinical entity\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ClinicalEntity\n",
      "\n",
      "    id: Union[str, ClinicalEntityId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ClinicalEntityId):\n",
      "            self.id = ClinicalEntityId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ClinicalTrial(ClinicalEntity):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ClinicalTrial\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ClinicalTrial\"\n",
      "    class_name: ClassVar[str] = \"clinical trial\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ClinicalTrial\n",
      "\n",
      "    id: Union[str, ClinicalTrialId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ClinicalTrialId):\n",
      "            self.id = ClinicalTrialId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ClinicalIntervention(ClinicalEntity):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ClinicalIntervention\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ClinicalIntervention\"\n",
      "    class_name: ClassVar[str] = \"clinical intervention\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ClinicalIntervention\n",
      "\n",
      "    id: Union[str, ClinicalInterventionId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ClinicalInterventionId):\n",
      "            self.id = ClinicalInterventionId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ClinicalFinding(PhenotypicFeature):\n",
      "    \"\"\"\n",
      "    this category is currently considered broad enough to tag clinical lab measurements and other biological\n",
      "    attributes taken as 'clinical traits' with some statistical score, for example, a p value in genetic associations.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ClinicalFinding\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ClinicalFinding\"\n",
      "    class_name: ClassVar[str] = \"clinical finding\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ClinicalFinding\n",
      "\n",
      "    id: Union[str, ClinicalFindingId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_attribute: Optional[Union[Union[str, ClinicalAttributeId], List[Union[str, ClinicalAttributeId]]]] = empty_list()\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ClinicalFindingId):\n",
      "            self.id = ClinicalFindingId(self.id)\n",
      "\n",
      "        if not isinstance(self.has_attribute, list):\n",
      "            self.has_attribute = [self.has_attribute] if self.has_attribute is not None else []\n",
      "        self.has_attribute = [v if isinstance(v, ClinicalAttributeId) else ClinicalAttributeId(v) for v in self.has_attribute]\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Hospitalization(ClinicalIntervention):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Hospitalization\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Hospitalization\"\n",
      "    class_name: ClassVar[str] = \"hospitalization\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Hospitalization\n",
      "\n",
      "    id: Union[str, HospitalizationId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, HospitalizationId):\n",
      "            self.id = HospitalizationId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class SocioeconomicAttribute(Attribute):\n",
      "    \"\"\"\n",
      "    Attributes relating to a socioeconomic manifestation\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"SocioeconomicAttribute\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:SocioeconomicAttribute\"\n",
      "    class_name: ClassVar[str] = \"socioeconomic attribute\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.SocioeconomicAttribute\n",
      "\n",
      "    id: Union[str, SocioeconomicAttributeId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_attribute_type: Union[str, OntologyClassId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, SocioeconomicAttributeId):\n",
      "            self.id = SocioeconomicAttributeId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Case(IndividualOrganism):\n",
      "    \"\"\"\n",
      "    An individual (human) organism that has a patient role in some clinical context.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Case\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Case\"\n",
      "    class_name: ClassVar[str] = \"case\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Case\n",
      "\n",
      "    id: Union[str, CaseId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, CaseId):\n",
      "            self.id = CaseId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Cohort(StudyPopulation):\n",
      "    \"\"\"\n",
      "    A group of people banded together or treated as a group who share common characteristics. A cohort 'study' is a\n",
      "    particular form of longitudinal study that samples a cohort, performing a cross-section at intervals through time.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Cohort\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Cohort\"\n",
      "    class_name: ClassVar[str] = \"cohort\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Cohort\n",
      "\n",
      "    id: Union[str, CohortId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, CohortId):\n",
      "            self.id = CohortId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ExposureEvent(OntologyClass):\n",
      "    \"\"\"\n",
      "    A (possibly time bounded) incidence of a feature of the environment of an organism that influences one or more\n",
      "    phenotypic features of that organism, potentially mediated by genes\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ExposureEvent\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ExposureEvent\"\n",
      "    class_name: ClassVar[str] = \"exposure event\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ExposureEvent\n",
      "\n",
      "    id: Union[str, ExposureEventId] = None\n",
      "    timepoint: Optional[Union[str, TimeType]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self.timepoint is not None and not isinstance(self.timepoint, TimeType):\n",
      "            self.timepoint = TimeType(self.timepoint)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GenomicBackgroundExposure(Attribute):\n",
      "    \"\"\"\n",
      "    A genomic background exposure is where an individual's specific genomic background of genes, sequence variants or\n",
      "    other pre-existing genomic conditions constitute a kind of 'exposure' to the organism, leading to or influencing\n",
      "    an outcome.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GenomicBackgroundExposure\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GenomicBackgroundExposure\"\n",
      "    class_name: ClassVar[str] = \"genomic background exposure\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GenomicBackgroundExposure\n",
      "\n",
      "    id: Union[str, GenomicBackgroundExposureId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_attribute_type: Union[str, OntologyClassId] = None\n",
      "    timepoint: Optional[Union[str, TimeType]] = None\n",
      "    has_gene_or_gene_product: Optional[Union[Union[str, GeneId], List[Union[str, GeneId]]]] = empty_list()\n",
      "    has_biological_sequence: Optional[Union[str, BiologicalSequence]] = None\n",
      "    in_taxon: Optional[Union[Union[str, OrganismTaxonId], List[Union[str, OrganismTaxonId]]]] = empty_list()\n",
      "    in_taxon_label: Optional[Union[str, LabelType]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GenomicBackgroundExposureId):\n",
      "            self.id = GenomicBackgroundExposureId(self.id)\n",
      "\n",
      "        if self.timepoint is not None and not isinstance(self.timepoint, TimeType):\n",
      "            self.timepoint = TimeType(self.timepoint)\n",
      "\n",
      "        if not isinstance(self.has_gene_or_gene_product, list):\n",
      "            self.has_gene_or_gene_product = [self.has_gene_or_gene_product] if self.has_gene_or_gene_product is not None else []\n",
      "        self.has_gene_or_gene_product = [v if isinstance(v, GeneId) else GeneId(v) for v in self.has_gene_or_gene_product]\n",
      "\n",
      "        if self.has_biological_sequence is not None and not isinstance(self.has_biological_sequence, BiologicalSequence):\n",
      "            self.has_biological_sequence = BiologicalSequence(self.has_biological_sequence)\n",
      "\n",
      "        if not isinstance(self.in_taxon, list):\n",
      "            self.in_taxon = [self.in_taxon] if self.in_taxon is not None else []\n",
      "        self.in_taxon = [v if isinstance(v, OrganismTaxonId) else OrganismTaxonId(v) for v in self.in_taxon]\n",
      "\n",
      "        if self.in_taxon_label is not None and not isinstance(self.in_taxon_label, LabelType):\n",
      "            self.in_taxon_label = LabelType(self.in_taxon_label)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "class PathologicalEntityMixin(YAMLRoot):\n",
      "    \"\"\"\n",
      "    A pathological (abnormal) structure or process.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"PathologicalEntityMixin\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:PathologicalEntityMixin\"\n",
      "    class_name: ClassVar[str] = \"pathological entity mixin\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.PathologicalEntityMixin\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class PathologicalProcess(BiologicalProcess):\n",
      "    \"\"\"\n",
      "    A biologic function or a process having an abnormal or deleterious effect at the subcellular, cellular,\n",
      "    multicellular, or organismal level.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\", \"has_input\", \"has_output\", \"enabled_by\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"PathologicalProcess\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:PathologicalProcess\"\n",
      "    class_name: ClassVar[str] = \"pathological process\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.PathologicalProcess\n",
      "\n",
      "    id: Union[str, PathologicalProcessId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, PathologicalProcessId):\n",
      "            self.id = PathologicalProcessId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class PathologicalProcessExposure(Attribute):\n",
      "    \"\"\"\n",
      "    A pathological process, when viewed as an exposure, representing a precondition, leading to or influencing an\n",
      "    outcome, e.g. autoimmunity leading to disease.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"PathologicalProcessExposure\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:PathologicalProcessExposure\"\n",
      "    class_name: ClassVar[str] = \"pathological process exposure\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.PathologicalProcessExposure\n",
      "\n",
      "    id: Union[str, PathologicalProcessExposureId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_attribute_type: Union[str, OntologyClassId] = None\n",
      "    timepoint: Optional[Union[str, TimeType]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, PathologicalProcessExposureId):\n",
      "            self.id = PathologicalProcessExposureId(self.id)\n",
      "\n",
      "        if self.timepoint is not None and not isinstance(self.timepoint, TimeType):\n",
      "            self.timepoint = TimeType(self.timepoint)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class PathologicalAnatomicalStructure(AnatomicalEntity):\n",
      "    \"\"\"\n",
      "    An anatomical structure with the potential of have an abnormal or deleterious effect at the subcellular, cellular,\n",
      "    multicellular, or organismal level.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = [\"in_taxon\"]\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"PathologicalAnatomicalStructure\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:PathologicalAnatomicalStructure\"\n",
      "    class_name: ClassVar[str] = \"pathological anatomical structure\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.PathologicalAnatomicalStructure\n",
      "\n",
      "    id: Union[str, PathologicalAnatomicalStructureId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, PathologicalAnatomicalStructureId):\n",
      "            self.id = PathologicalAnatomicalStructureId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class PathologicalAnatomicalExposure(Attribute):\n",
      "    \"\"\"\n",
      "    An abnormal anatomical structure, when viewed as an exposure, representing an precondition, leading to or\n",
      "    influencing an outcome, e.g. thrombosis leading to an ischemic disease outcome.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"PathologicalAnatomicalExposure\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:PathologicalAnatomicalExposure\"\n",
      "    class_name: ClassVar[str] = \"pathological anatomical exposure\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.PathologicalAnatomicalExposure\n",
      "\n",
      "    id: Union[str, PathologicalAnatomicalExposureId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_attribute_type: Union[str, OntologyClassId] = None\n",
      "    timepoint: Optional[Union[str, TimeType]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, PathologicalAnatomicalExposureId):\n",
      "            self.id = PathologicalAnatomicalExposureId(self.id)\n",
      "\n",
      "        if self.timepoint is not None and not isinstance(self.timepoint, TimeType):\n",
      "            self.timepoint = TimeType(self.timepoint)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class DiseaseOrPhenotypicFeatureExposure(Attribute):\n",
      "    \"\"\"\n",
      "    A disease or phenotypic feature state, when viewed as an exposure, represents an precondition, leading to or\n",
      "    influencing an outcome, e.g. HIV predisposing an individual to infections; a relative deficiency of skin\n",
      "    pigmentation predisposing an individual to skin cancer.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"DiseaseOrPhenotypicFeatureExposure\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:DiseaseOrPhenotypicFeatureExposure\"\n",
      "    class_name: ClassVar[str] = \"disease or phenotypic feature exposure\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.DiseaseOrPhenotypicFeatureExposure\n",
      "\n",
      "    id: Union[str, DiseaseOrPhenotypicFeatureExposureId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_attribute_type: Union[str, OntologyClassId] = None\n",
      "    timepoint: Optional[Union[str, TimeType]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, DiseaseOrPhenotypicFeatureExposureId):\n",
      "            self.id = DiseaseOrPhenotypicFeatureExposureId(self.id)\n",
      "\n",
      "        if self.timepoint is not None and not isinstance(self.timepoint, TimeType):\n",
      "            self.timepoint = TimeType(self.timepoint)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ChemicalExposure(Attribute):\n",
      "    \"\"\"\n",
      "    A chemical exposure is an intake of a particular chemical entity.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ChemicalExposure\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ChemicalExposure\"\n",
      "    class_name: ClassVar[str] = \"chemical exposure\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ChemicalExposure\n",
      "\n",
      "    id: Union[str, ChemicalExposureId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_attribute_type: Union[str, OntologyClassId] = None\n",
      "    has_quantitative_value: Optional[Union[Union[dict, QuantityValue], List[Union[dict, QuantityValue]]]] = empty_list()\n",
      "    timepoint: Optional[Union[str, TimeType]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ChemicalExposureId):\n",
      "            self.id = ChemicalExposureId(self.id)\n",
      "\n",
      "        if not isinstance(self.has_quantitative_value, list):\n",
      "            self.has_quantitative_value = [self.has_quantitative_value] if self.has_quantitative_value is not None else []\n",
      "        self.has_quantitative_value = [v if isinstance(v, QuantityValue) else QuantityValue(**as_dict(v)) for v in self.has_quantitative_value]\n",
      "\n",
      "        if self.timepoint is not None and not isinstance(self.timepoint, TimeType):\n",
      "            self.timepoint = TimeType(self.timepoint)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ComplexChemicalExposure(Attribute):\n",
      "    \"\"\"\n",
      "    A complex chemical exposure is an intake of a chemical mixture (e.g. gasoline), other than a drug.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ComplexChemicalExposure\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ComplexChemicalExposure\"\n",
      "    class_name: ClassVar[str] = \"complex chemical exposure\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ComplexChemicalExposure\n",
      "\n",
      "    id: Union[str, ComplexChemicalExposureId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_attribute_type: Union[str, OntologyClassId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ComplexChemicalExposureId):\n",
      "            self.id = ComplexChemicalExposureId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class DrugExposure(ChemicalExposure):\n",
      "    \"\"\"\n",
      "    A drug exposure is an intake of a particular drug.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"DrugExposure\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:DrugExposure\"\n",
      "    class_name: ClassVar[str] = \"drug exposure\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.DrugExposure\n",
      "\n",
      "    id: Union[str, DrugExposureId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_attribute_type: Union[str, OntologyClassId] = None\n",
      "    timepoint: Optional[Union[str, TimeType]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, DrugExposureId):\n",
      "            self.id = DrugExposureId(self.id)\n",
      "\n",
      "        if self.timepoint is not None and not isinstance(self.timepoint, TimeType):\n",
      "            self.timepoint = TimeType(self.timepoint)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class DrugToGeneInteractionExposure(DrugExposure):\n",
      "    \"\"\"\n",
      "    drug to gene interaction exposure is a drug exposure is where the interactions of the drug with specific genes are\n",
      "    known to constitute an 'exposure' to the organism, leading to or influencing an outcome.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"DrugToGeneInteractionExposure\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:DrugToGeneInteractionExposure\"\n",
      "    class_name: ClassVar[str] = \"drug to gene interaction exposure\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.DrugToGeneInteractionExposure\n",
      "\n",
      "    id: Union[str, DrugToGeneInteractionExposureId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_attribute_type: Union[str, OntologyClassId] = None\n",
      "    has_gene_or_gene_product: Optional[Union[Union[str, GeneId], List[Union[str, GeneId]]]] = empty_list()\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, DrugToGeneInteractionExposureId):\n",
      "            self.id = DrugToGeneInteractionExposureId(self.id)\n",
      "\n",
      "        if not isinstance(self.has_gene_or_gene_product, list):\n",
      "            self.has_gene_or_gene_product = [self.has_gene_or_gene_product] if self.has_gene_or_gene_product is not None else []\n",
      "        self.has_gene_or_gene_product = [v if isinstance(v, GeneId) else GeneId(v) for v in self.has_gene_or_gene_product]\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Treatment(NamedThing):\n",
      "    \"\"\"\n",
      "    A treatment is targeted at a disease or phenotype and may involve multiple drug 'exposures', medical devices\n",
      "    and/or procedures\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Treatment\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Treatment\"\n",
      "    class_name: ClassVar[str] = \"treatment\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Treatment\n",
      "\n",
      "    id: Union[str, TreatmentId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_drug: Optional[Union[Union[str, DrugId], List[Union[str, DrugId]]]] = empty_list()\n",
      "    has_device: Optional[Union[Union[str, DeviceId], List[Union[str, DeviceId]]]] = empty_list()\n",
      "    has_procedure: Optional[Union[Union[str, ProcedureId], List[Union[str, ProcedureId]]]] = empty_list()\n",
      "    timepoint: Optional[Union[str, TimeType]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, TreatmentId):\n",
      "            self.id = TreatmentId(self.id)\n",
      "\n",
      "        if not isinstance(self.has_drug, list):\n",
      "            self.has_drug = [self.has_drug] if self.has_drug is not None else []\n",
      "        self.has_drug = [v if isinstance(v, DrugId) else DrugId(v) for v in self.has_drug]\n",
      "\n",
      "        if not isinstance(self.has_device, list):\n",
      "            self.has_device = [self.has_device] if self.has_device is not None else []\n",
      "        self.has_device = [v if isinstance(v, DeviceId) else DeviceId(v) for v in self.has_device]\n",
      "\n",
      "        if not isinstance(self.has_procedure, list):\n",
      "            self.has_procedure = [self.has_procedure] if self.has_procedure is not None else []\n",
      "        self.has_procedure = [v if isinstance(v, ProcedureId) else ProcedureId(v) for v in self.has_procedure]\n",
      "\n",
      "        if self.timepoint is not None and not isinstance(self.timepoint, TimeType):\n",
      "            self.timepoint = TimeType(self.timepoint)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class BioticExposure(Attribute):\n",
      "    \"\"\"\n",
      "    An external biotic exposure is an intake of (sometimes pathological) biological organisms (including viruses).\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"BioticExposure\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:BioticExposure\"\n",
      "    class_name: ClassVar[str] = \"biotic exposure\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.BioticExposure\n",
      "\n",
      "    id: Union[str, BioticExposureId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_attribute_type: Union[str, OntologyClassId] = None\n",
      "    timepoint: Optional[Union[str, TimeType]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, BioticExposureId):\n",
      "            self.id = BioticExposureId(self.id)\n",
      "\n",
      "        if self.timepoint is not None and not isinstance(self.timepoint, TimeType):\n",
      "            self.timepoint = TimeType(self.timepoint)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class EnvironmentalExposure(Attribute):\n",
      "    \"\"\"\n",
      "    A environmental exposure is a factor relating to abiotic processes in the environment including sunlight (UV-B),\n",
      "    atmospheric (heat, cold, general pollution) and water-born contaminants.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"EnvironmentalExposure\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:EnvironmentalExposure\"\n",
      "    class_name: ClassVar[str] = \"environmental exposure\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.EnvironmentalExposure\n",
      "\n",
      "    id: Union[str, EnvironmentalExposureId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_attribute_type: Union[str, OntologyClassId] = None\n",
      "    timepoint: Optional[Union[str, TimeType]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, EnvironmentalExposureId):\n",
      "            self.id = EnvironmentalExposureId(self.id)\n",
      "\n",
      "        if self.timepoint is not None and not isinstance(self.timepoint, TimeType):\n",
      "            self.timepoint = TimeType(self.timepoint)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GeographicExposure(EnvironmentalExposure):\n",
      "    \"\"\"\n",
      "    A geographic exposure is a factor relating to geographic proximity to some impactful entity.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GeographicExposure\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GeographicExposure\"\n",
      "    class_name: ClassVar[str] = \"geographic exposure\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GeographicExposure\n",
      "\n",
      "    id: Union[str, GeographicExposureId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_attribute_type: Union[str, OntologyClassId] = None\n",
      "    timepoint: Optional[Union[str, TimeType]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GeographicExposureId):\n",
      "            self.id = GeographicExposureId(self.id)\n",
      "\n",
      "        if self.timepoint is not None and not isinstance(self.timepoint, TimeType):\n",
      "            self.timepoint = TimeType(self.timepoint)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class BehavioralExposure(Attribute):\n",
      "    \"\"\"\n",
      "    A behavioral exposure is a factor relating to behavior impacting an individual.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"BehavioralExposure\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:BehavioralExposure\"\n",
      "    class_name: ClassVar[str] = \"behavioral exposure\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.BehavioralExposure\n",
      "\n",
      "    id: Union[str, BehavioralExposureId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_attribute_type: Union[str, OntologyClassId] = None\n",
      "    timepoint: Optional[Union[str, TimeType]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, BehavioralExposureId):\n",
      "            self.id = BehavioralExposureId(self.id)\n",
      "\n",
      "        if self.timepoint is not None and not isinstance(self.timepoint, TimeType):\n",
      "            self.timepoint = TimeType(self.timepoint)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class SocioeconomicExposure(Attribute):\n",
      "    \"\"\"\n",
      "    A socioeconomic exposure is a factor relating to social and financial status of an affected individual (e.g.\n",
      "    poverty).\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"SocioeconomicExposure\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:SocioeconomicExposure\"\n",
      "    class_name: ClassVar[str] = \"socioeconomic exposure\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.SocioeconomicExposure\n",
      "\n",
      "    id: Union[str, SocioeconomicExposureId] = None\n",
      "    category: Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]] = None\n",
      "    has_attribute_type: Union[str, OntologyClassId] = None\n",
      "    has_attribute: Union[Union[str, SocioeconomicAttributeId], List[Union[str, SocioeconomicAttributeId]]] = None\n",
      "    timepoint: Optional[Union[str, TimeType]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, SocioeconomicExposureId):\n",
      "            self.id = SocioeconomicExposureId(self.id)\n",
      "\n",
      "        if self._is_empty(self.has_attribute):\n",
      "            self.MissingRequiredField(\"has_attribute\")\n",
      "        if not isinstance(self.has_attribute, list):\n",
      "            self.has_attribute = [self.has_attribute] if self.has_attribute is not None else []\n",
      "        self.has_attribute = [v if isinstance(v, SocioeconomicAttributeId) else SocioeconomicAttributeId(v) for v in self.has_attribute]\n",
      "\n",
      "        if self.timepoint is not None and not isinstance(self.timepoint, TimeType):\n",
      "            self.timepoint = TimeType(self.timepoint)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if self._is_empty(self.category):\n",
      "            self.MissingRequiredField(\"category\")\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "class Outcome(YAMLRoot):\n",
      "    \"\"\"\n",
      "    An entity that has the role of being the consequence of an exposure event. This is an abstract mixin grouping of\n",
      "    various categories of possible biological or non-biological (e.g. clinical) outcomes.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Outcome\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Outcome\"\n",
      "    class_name: ClassVar[str] = \"outcome\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Outcome\n",
      "\n",
      "\n",
      "class PathologicalProcessOutcome(YAMLRoot):\n",
      "    \"\"\"\n",
      "    An outcome resulting from an exposure event which is the manifestation of a pathological process.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"PathologicalProcessOutcome\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:PathologicalProcessOutcome\"\n",
      "    class_name: ClassVar[str] = \"pathological process outcome\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.PathologicalProcessOutcome\n",
      "\n",
      "\n",
      "class PathologicalAnatomicalOutcome(YAMLRoot):\n",
      "    \"\"\"\n",
      "    An outcome resulting from an exposure event which is the manifestation of an abnormal anatomical structure.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"PathologicalAnatomicalOutcome\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:PathologicalAnatomicalOutcome\"\n",
      "    class_name: ClassVar[str] = \"pathological anatomical outcome\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.PathologicalAnatomicalOutcome\n",
      "\n",
      "\n",
      "class DiseaseOrPhenotypicFeatureOutcome(YAMLRoot):\n",
      "    \"\"\"\n",
      "    Physiological outcomes resulting from an exposure event which is the manifestation of a disease or other\n",
      "    characteristic phenotype.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"DiseaseOrPhenotypicFeatureOutcome\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:DiseaseOrPhenotypicFeatureOutcome\"\n",
      "    class_name: ClassVar[str] = \"disease or phenotypic feature outcome\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.DiseaseOrPhenotypicFeatureOutcome\n",
      "\n",
      "\n",
      "class BehavioralOutcome(YAMLRoot):\n",
      "    \"\"\"\n",
      "    An outcome resulting from an exposure event which is the manifestation of human behavior.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"BehavioralOutcome\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:BehavioralOutcome\"\n",
      "    class_name: ClassVar[str] = \"behavioral outcome\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.BehavioralOutcome\n",
      "\n",
      "\n",
      "class HospitalizationOutcome(YAMLRoot):\n",
      "    \"\"\"\n",
      "    An outcome resulting from an exposure event which is the increased manifestation of acute (e.g. emergency room\n",
      "    visit) or chronic (inpatient) hospitalization.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"HospitalizationOutcome\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:HospitalizationOutcome\"\n",
      "    class_name: ClassVar[str] = \"hospitalization outcome\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.HospitalizationOutcome\n",
      "\n",
      "\n",
      "class MortalityOutcome(YAMLRoot):\n",
      "    \"\"\"\n",
      "    An outcome of death from resulting from an exposure event.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"MortalityOutcome\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:MortalityOutcome\"\n",
      "    class_name: ClassVar[str] = \"mortality outcome\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.MortalityOutcome\n",
      "\n",
      "\n",
      "class EpidemiologicalOutcome(YAMLRoot):\n",
      "    \"\"\"\n",
      "    An epidemiological outcome, such as societal disease burden, resulting from an exposure event.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"EpidemiologicalOutcome\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:EpidemiologicalOutcome\"\n",
      "    class_name: ClassVar[str] = \"epidemiological outcome\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.EpidemiologicalOutcome\n",
      "\n",
      "\n",
      "class SocioeconomicOutcome(YAMLRoot):\n",
      "    \"\"\"\n",
      "    An general social or economic outcome, such as healthcare costs, utilization, etc., resulting from an exposure\n",
      "    event\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"SocioeconomicOutcome\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:SocioeconomicOutcome\"\n",
      "    class_name: ClassVar[str] = \"socioeconomic outcome\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.SocioeconomicOutcome\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class Association(Entity):\n",
      "    \"\"\"\n",
      "    A typed association between two entities, supported by evidence\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"Association\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:Association\"\n",
      "    class_name: ClassVar[str] = \"association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.Association\n",
      "\n",
      "    id: Union[str, AssociationId] = None\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    negated: Optional[Union[bool, Bool]] = None\n",
      "    qualifier: Optional[str] = None\n",
      "    qualifiers: Optional[Union[Union[str, OntologyClassId], List[Union[str, OntologyClassId]]]] = empty_list()\n",
      "    publications: Optional[Union[Union[str, PublicationId], List[Union[str, PublicationId]]]] = empty_list()\n",
      "    has_evidence: Optional[Union[Union[str, EvidenceTypeId], List[Union[str, EvidenceTypeId]]]] = empty_list()\n",
      "    knowledge_source: Optional[str] = None\n",
      "    primary_knowledge_source: Optional[str] = None\n",
      "    aggregator_knowledge_source: Optional[Union[str, List[str]]] = empty_list()\n",
      "    timepoint: Optional[Union[str, TimeType]] = None\n",
      "    original_subject: Optional[str] = None\n",
      "    original_predicate: Optional[Union[str, URIorCURIE]] = None\n",
      "    original_object: Optional[str] = None\n",
      "    subject_category: Optional[Union[str, OntologyClassId]] = None\n",
      "    object_category: Optional[Union[str, OntologyClassId]] = None\n",
      "    subject_closure: Optional[Union[str, List[str]]] = empty_list()\n",
      "    object_closure: Optional[Union[str, List[str]]] = empty_list()\n",
      "    subject_category_closure: Optional[Union[Union[str, OntologyClassId], List[Union[str, OntologyClassId]]]] = empty_list()\n",
      "    object_category_closure: Optional[Union[Union[str, OntologyClassId], List[Union[str, OntologyClassId]]]] = empty_list()\n",
      "    subject_namespace: Optional[str] = None\n",
      "    object_namespace: Optional[str] = None\n",
      "    subject_label_closure: Optional[Union[str, List[str]]] = empty_list()\n",
      "    object_label_closure: Optional[Union[str, List[str]]] = empty_list()\n",
      "    retrieval_source_ids: Optional[Union[Union[str, RetrievalSourceId], List[Union[str, RetrievalSourceId]]]] = empty_list()\n",
      "    p_value: Optional[float] = None\n",
      "    adjusted_p_value: Optional[float] = None\n",
      "    has_supporting_studies: Optional[Union[Union[str, StudyId], List[Union[str, StudyId]]]] = empty_list()\n",
      "    type: Optional[Union[str, List[str]]] = empty_list()\n",
      "    category: Optional[Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]]] = empty_list()\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, AssociationId):\n",
      "            self.id = AssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, NamedThingId):\n",
      "            self.subject = NamedThingId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        if self._is_empty(self.knowledge_level):\n",
      "            self.MissingRequiredField(\"knowledge_level\")\n",
      "        if not isinstance(self.knowledge_level, KnowledgeLevelEnum):\n",
      "            self.knowledge_level = KnowledgeLevelEnum(self.knowledge_level)\n",
      "\n",
      "        if self._is_empty(self.agent_type):\n",
      "            self.MissingRequiredField(\"agent_type\")\n",
      "        if not isinstance(self.agent_type, AgentTypeEnum):\n",
      "            self.agent_type = AgentTypeEnum(self.agent_type)\n",
      "\n",
      "        if self.negated is not None and not isinstance(self.negated, Bool):\n",
      "            self.negated = Bool(self.negated)\n",
      "\n",
      "        if self.qualifier is not None and not isinstance(self.qualifier, str):\n",
      "            self.qualifier = str(self.qualifier)\n",
      "\n",
      "        if not isinstance(self.qualifiers, list):\n",
      "            self.qualifiers = [self.qualifiers] if self.qualifiers is not None else []\n",
      "        self.qualifiers = [v if isinstance(v, OntologyClassId) else OntologyClassId(v) for v in self.qualifiers]\n",
      "\n",
      "        if not isinstance(self.publications, list):\n",
      "            self.publications = [self.publications] if self.publications is not None else []\n",
      "        self.publications = [v if isinstance(v, PublicationId) else PublicationId(v) for v in self.publications]\n",
      "\n",
      "        if not isinstance(self.has_evidence, list):\n",
      "            self.has_evidence = [self.has_evidence] if self.has_evidence is not None else []\n",
      "        self.has_evidence = [v if isinstance(v, EvidenceTypeId) else EvidenceTypeId(v) for v in self.has_evidence]\n",
      "\n",
      "        if self.knowledge_source is not None and not isinstance(self.knowledge_source, str):\n",
      "            self.knowledge_source = str(self.knowledge_source)\n",
      "\n",
      "        if self.primary_knowledge_source is not None and not isinstance(self.primary_knowledge_source, str):\n",
      "            self.primary_knowledge_source = str(self.primary_knowledge_source)\n",
      "\n",
      "        if not isinstance(self.aggregator_knowledge_source, list):\n",
      "            self.aggregator_knowledge_source = [self.aggregator_knowledge_source] if self.aggregator_knowledge_source is not None else []\n",
      "        self.aggregator_knowledge_source = [v if isinstance(v, str) else str(v) for v in self.aggregator_knowledge_source]\n",
      "\n",
      "        if self.timepoint is not None and not isinstance(self.timepoint, TimeType):\n",
      "            self.timepoint = TimeType(self.timepoint)\n",
      "\n",
      "        if self.original_subject is not None and not isinstance(self.original_subject, str):\n",
      "            self.original_subject = str(self.original_subject)\n",
      "\n",
      "        if self.original_predicate is not None and not isinstance(self.original_predicate, URIorCURIE):\n",
      "            self.original_predicate = URIorCURIE(self.original_predicate)\n",
      "\n",
      "        if self.original_object is not None and not isinstance(self.original_object, str):\n",
      "            self.original_object = str(self.original_object)\n",
      "\n",
      "        if self.subject_category is not None and not isinstance(self.subject_category, OntologyClassId):\n",
      "            self.subject_category = OntologyClassId(self.subject_category)\n",
      "\n",
      "        if self.object_category is not None and not isinstance(self.object_category, OntologyClassId):\n",
      "            self.object_category = OntologyClassId(self.object_category)\n",
      "\n",
      "        if not isinstance(self.subject_closure, list):\n",
      "            self.subject_closure = [self.subject_closure] if self.subject_closure is not None else []\n",
      "        self.subject_closure = [v if isinstance(v, str) else str(v) for v in self.subject_closure]\n",
      "\n",
      "        if not isinstance(self.object_closure, list):\n",
      "            self.object_closure = [self.object_closure] if self.object_closure is not None else []\n",
      "        self.object_closure = [v if isinstance(v, str) else str(v) for v in self.object_closure]\n",
      "\n",
      "        if not isinstance(self.subject_category_closure, list):\n",
      "            self.subject_category_closure = [self.subject_category_closure] if self.subject_category_closure is not None else []\n",
      "        self.subject_category_closure = [v if isinstance(v, OntologyClassId) else OntologyClassId(v) for v in self.subject_category_closure]\n",
      "\n",
      "        if not isinstance(self.object_category_closure, list):\n",
      "            self.object_category_closure = [self.object_category_closure] if self.object_category_closure is not None else []\n",
      "        self.object_category_closure = [v if isinstance(v, OntologyClassId) else OntologyClassId(v) for v in self.object_category_closure]\n",
      "\n",
      "        if self.subject_namespace is not None and not isinstance(self.subject_namespace, str):\n",
      "            self.subject_namespace = str(self.subject_namespace)\n",
      "\n",
      "        if self.object_namespace is not None and not isinstance(self.object_namespace, str):\n",
      "            self.object_namespace = str(self.object_namespace)\n",
      "\n",
      "        if not isinstance(self.subject_label_closure, list):\n",
      "            self.subject_label_closure = [self.subject_label_closure] if self.subject_label_closure is not None else []\n",
      "        self.subject_label_closure = [v if isinstance(v, str) else str(v) for v in self.subject_label_closure]\n",
      "\n",
      "        if not isinstance(self.object_label_closure, list):\n",
      "            self.object_label_closure = [self.object_label_closure] if self.object_label_closure is not None else []\n",
      "        self.object_label_closure = [v if isinstance(v, str) else str(v) for v in self.object_label_closure]\n",
      "\n",
      "        if not isinstance(self.retrieval_source_ids, list):\n",
      "            self.retrieval_source_ids = [self.retrieval_source_ids] if self.retrieval_source_ids is not None else []\n",
      "        self.retrieval_source_ids = [v if isinstance(v, RetrievalSourceId) else RetrievalSourceId(v) for v in self.retrieval_source_ids]\n",
      "\n",
      "        if self.p_value is not None and not isinstance(self.p_value, float):\n",
      "            self.p_value = float(self.p_value)\n",
      "\n",
      "        if self.adjusted_p_value is not None and not isinstance(self.adjusted_p_value, float):\n",
      "            self.adjusted_p_value = float(self.adjusted_p_value)\n",
      "\n",
      "        if not isinstance(self.has_supporting_studies, list):\n",
      "            self.has_supporting_studies = [self.has_supporting_studies] if self.has_supporting_studies is not None else []\n",
      "        self.has_supporting_studies = [v if isinstance(v, StudyId) else StudyId(v) for v in self.has_supporting_studies]\n",
      "\n",
      "        if not isinstance(self.type, list):\n",
      "            self.type = [self.type] if self.type is not None else []\n",
      "        self.type = [v if isinstance(v, str) else str(v) for v in self.type]\n",
      "\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "    def __new__(cls, *args, **kwargs):\n",
      "\n",
      "        type_designator = \"category\"\n",
      "        if not type_designator in kwargs:\n",
      "            return super().__new__(cls,*args,**kwargs)\n",
      "        else:\n",
      "            type_designator_value = kwargs[type_designator]\n",
      "            target_cls = cls._class_for(\"class_class_curie\", type_designator_value)\n",
      "\n",
      "\n",
      "            if target_cls is None:\n",
      "                target_cls = cls._class_for(\"class_class_uri\", type_designator_value)\n",
      "\n",
      "\n",
      "            if target_cls is None:\n",
      "                target_cls = cls._class_for(\"class_model_uri\", type_designator_value)\n",
      "\n",
      "\n",
      "            if target_cls is None:\n",
      "                raise ValueError(f\"Wrong type designator value: class {cls.__name__} \"\n",
      "                                 f\"has no subclass with ['class_class_curie', 'class_class_uri', 'class_model_uri']='{kwargs[type_designator]}'\")\n",
      "            return super().__new__(target_cls,*args,**kwargs)\n",
      "\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ChemicalEntityAssessesNamedThingAssociation(Association):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ChemicalEntityAssessesNamedThingAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ChemicalEntityAssessesNamedThingAssociation\"\n",
      "    class_name: ClassVar[str] = \"chemical entity assesses named thing association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ChemicalEntityAssessesNamedThingAssociation\n",
      "\n",
      "    id: Union[str, ChemicalEntityAssessesNamedThingAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, ChemicalEntityId] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ChemicalEntityAssessesNamedThingAssociationId):\n",
      "            self.id = ChemicalEntityAssessesNamedThingAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, ChemicalEntityId):\n",
      "            self.subject = ChemicalEntityId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ContributorAssociation(Association):\n",
      "    \"\"\"\n",
      "    Any association between an entity (such as a publication) and various agents that contribute to its realisation\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ContributorAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ContributorAssociation\"\n",
      "    class_name: ClassVar[str] = \"contributor association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ContributorAssociation\n",
      "\n",
      "    id: Union[str, ContributorAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, InformationContentEntityId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, AgentId] = None\n",
      "    qualifiers: Optional[Union[Union[str, OntologyClassId], List[Union[str, OntologyClassId]]]] = empty_list()\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ContributorAssociationId):\n",
      "            self.id = ContributorAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, InformationContentEntityId):\n",
      "            self.subject = InformationContentEntityId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, AgentId):\n",
      "            self.object = AgentId(self.object)\n",
      "\n",
      "        if not isinstance(self.qualifiers, list):\n",
      "            self.qualifiers = [self.qualifiers] if self.qualifiers is not None else []\n",
      "        self.qualifiers = [v if isinstance(v, OntologyClassId) else OntologyClassId(v) for v in self.qualifiers]\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GenotypeToGenotypePartAssociation(Association):\n",
      "    \"\"\"\n",
      "    Any association between one genotype and a genotypic entity that is a sub-component of it\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GenotypeToGenotypePartAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GenotypeToGenotypePartAssociation\"\n",
      "    class_name: ClassVar[str] = \"genotype to genotype part association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GenotypeToGenotypePartAssociation\n",
      "\n",
      "    id: Union[str, GenotypeToGenotypePartAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    subject: Union[str, GenotypeId] = None\n",
      "    object: Union[str, GenotypeId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GenotypeToGenotypePartAssociationId):\n",
      "            self.id = GenotypeToGenotypePartAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, GenotypeId):\n",
      "            self.subject = GenotypeId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, GenotypeId):\n",
      "            self.object = GenotypeId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GenotypeToGeneAssociation(Association):\n",
      "    \"\"\"\n",
      "    Any association between a genotype and a gene. The genotype have have multiple variants in that gene or a single\n",
      "    one. There is no assumption of cardinality\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GenotypeToGeneAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GenotypeToGeneAssociation\"\n",
      "    class_name: ClassVar[str] = \"genotype to gene association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GenotypeToGeneAssociation\n",
      "\n",
      "    id: Union[str, GenotypeToGeneAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    subject: Union[str, GenotypeId] = None\n",
      "    object: Union[str, GeneId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GenotypeToGeneAssociationId):\n",
      "            self.id = GenotypeToGeneAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, GenotypeId):\n",
      "            self.subject = GenotypeId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, GeneId):\n",
      "            self.object = GeneId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GenotypeToVariantAssociation(Association):\n",
      "    \"\"\"\n",
      "    Any association between a genotype and a sequence variant.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GenotypeToVariantAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GenotypeToVariantAssociation\"\n",
      "    class_name: ClassVar[str] = \"genotype to variant association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GenotypeToVariantAssociation\n",
      "\n",
      "    id: Union[str, GenotypeToVariantAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    subject: Union[str, GenotypeId] = None\n",
      "    object: Union[str, SequenceVariantId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GenotypeToVariantAssociationId):\n",
      "            self.id = GenotypeToVariantAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, GenotypeId):\n",
      "            self.subject = GenotypeId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, SequenceVariantId):\n",
      "            self.object = SequenceVariantId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GeneToGeneAssociation(Association):\n",
      "    \"\"\"\n",
      "    abstract parent class for different kinds of gene-gene or gene product to gene product relationships. Includes\n",
      "    homology and interaction.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GeneToGeneAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GeneToGeneAssociation\"\n",
      "    class_name: ClassVar[str] = \"gene to gene association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GeneToGeneAssociation\n",
      "\n",
      "    id: Union[str, GeneToGeneAssociationId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[dict, GeneOrGeneProduct] = None\n",
      "    object: Union[dict, GeneOrGeneProduct] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, GeneOrGeneProduct):\n",
      "            self.subject = GeneOrGeneProduct(**as_dict(self.subject))\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, GeneOrGeneProduct):\n",
      "            self.object = GeneOrGeneProduct(**as_dict(self.object))\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GeneToGeneHomologyAssociation(GeneToGeneAssociation):\n",
      "    \"\"\"\n",
      "    A homology association between two genes. May be orthology (in which case the species of subject and object should\n",
      "    differ) or paralogy (in which case the species may be the same)\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GeneToGeneHomologyAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GeneToGeneHomologyAssociation\"\n",
      "    class_name: ClassVar[str] = \"gene to gene homology association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GeneToGeneHomologyAssociation\n",
      "\n",
      "    id: Union[str, GeneToGeneHomologyAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[dict, GeneOrGeneProduct] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[dict, GeneOrGeneProduct] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GeneToGeneHomologyAssociationId):\n",
      "            self.id = GeneToGeneHomologyAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, GeneOrGeneProduct):\n",
      "            self.subject = GeneOrGeneProduct(**as_dict(self.subject))\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, GeneOrGeneProduct):\n",
      "            self.object = GeneOrGeneProduct(**as_dict(self.object))\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GeneToGeneFamilyAssociation(Association):\n",
      "    \"\"\"\n",
      "    Set membership of a gene in a family of genes related by common evolutionary ancestry usually inferred by sequence\n",
      "    comparisons. The genes in a given family generally share common sequence motifs which generally map onto shared\n",
      "    gene product structure-function relationships.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GeneToGeneFamilyAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GeneToGeneFamilyAssociation\"\n",
      "    class_name: ClassVar[str] = \"gene to gene family association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GeneToGeneFamilyAssociation\n",
      "\n",
      "    id: Union[str, GeneToGeneFamilyAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, GeneId] = None\n",
      "    object: Union[str, GeneFamilyId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GeneToGeneFamilyAssociationId):\n",
      "            self.id = GeneToGeneFamilyAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, GeneId):\n",
      "            self.subject = GeneId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, GeneFamilyId):\n",
      "            self.object = GeneFamilyId(self.object)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GeneExpressionMixin(YAMLRoot):\n",
      "    \"\"\"\n",
      "    Observed gene expression intensity, context (site, stage) and associated phenotypic status within which the\n",
      "    expression occurs.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GeneExpressionMixin\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GeneExpressionMixin\"\n",
      "    class_name: ClassVar[str] = \"gene expression mixin\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GeneExpressionMixin\n",
      "\n",
      "    quantifier_qualifier: Optional[Union[str, OntologyClassId]] = None\n",
      "    expression_site: Optional[Union[str, AnatomicalEntityId]] = None\n",
      "    stage_qualifier: Optional[Union[str, LifeStageId]] = None\n",
      "    phenotypic_state: Optional[Union[str, DiseaseOrPhenotypicFeatureId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self.quantifier_qualifier is not None and not isinstance(self.quantifier_qualifier, OntologyClassId):\n",
      "            self.quantifier_qualifier = OntologyClassId(self.quantifier_qualifier)\n",
      "\n",
      "        if self.expression_site is not None and not isinstance(self.expression_site, AnatomicalEntityId):\n",
      "            self.expression_site = AnatomicalEntityId(self.expression_site)\n",
      "\n",
      "        if self.stage_qualifier is not None and not isinstance(self.stage_qualifier, LifeStageId):\n",
      "            self.stage_qualifier = LifeStageId(self.stage_qualifier)\n",
      "\n",
      "        if self.phenotypic_state is not None and not isinstance(self.phenotypic_state, DiseaseOrPhenotypicFeatureId):\n",
      "            self.phenotypic_state = DiseaseOrPhenotypicFeatureId(self.phenotypic_state)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GeneToGeneCoexpressionAssociation(GeneToGeneAssociation):\n",
      "    \"\"\"\n",
      "    Indicates that two genes are co-expressed, generally under the same conditions.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GeneToGeneCoexpressionAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GeneToGeneCoexpressionAssociation\"\n",
      "    class_name: ClassVar[str] = \"gene to gene coexpression association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GeneToGeneCoexpressionAssociation\n",
      "\n",
      "    id: Union[str, GeneToGeneCoexpressionAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[dict, GeneOrGeneProduct] = None\n",
      "    object: Union[dict, GeneOrGeneProduct] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    quantifier_qualifier: Optional[Union[str, OntologyClassId]] = None\n",
      "    expression_site: Optional[Union[str, AnatomicalEntityId]] = None\n",
      "    stage_qualifier: Optional[Union[str, LifeStageId]] = None\n",
      "    phenotypic_state: Optional[Union[str, DiseaseOrPhenotypicFeatureId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GeneToGeneCoexpressionAssociationId):\n",
      "            self.id = GeneToGeneCoexpressionAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self.quantifier_qualifier is not None and not isinstance(self.quantifier_qualifier, OntologyClassId):\n",
      "            self.quantifier_qualifier = OntologyClassId(self.quantifier_qualifier)\n",
      "\n",
      "        if self.expression_site is not None and not isinstance(self.expression_site, AnatomicalEntityId):\n",
      "            self.expression_site = AnatomicalEntityId(self.expression_site)\n",
      "\n",
      "        if self.stage_qualifier is not None and not isinstance(self.stage_qualifier, LifeStageId):\n",
      "            self.stage_qualifier = LifeStageId(self.stage_qualifier)\n",
      "\n",
      "        if self.phenotypic_state is not None and not isinstance(self.phenotypic_state, DiseaseOrPhenotypicFeatureId):\n",
      "            self.phenotypic_state = DiseaseOrPhenotypicFeatureId(self.phenotypic_state)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class PairwiseGeneToGeneInteraction(GeneToGeneAssociation):\n",
      "    \"\"\"\n",
      "    An interaction between two genes or two gene products. May be physical (e.g. protein binding) or genetic (between\n",
      "    genes). May be symmetric (e.g. protein interaction) or directed (e.g. phosphorylation)\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"PairwiseGeneToGeneInteraction\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:PairwiseGeneToGeneInteraction\"\n",
      "    class_name: ClassVar[str] = \"pairwise gene to gene interaction\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.PairwiseGeneToGeneInteraction\n",
      "\n",
      "    id: Union[str, PairwiseGeneToGeneInteractionId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[dict, GeneOrGeneProduct] = None\n",
      "    object: Union[dict, GeneOrGeneProduct] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, PairwiseGeneToGeneInteractionId):\n",
      "            self.id = PairwiseGeneToGeneInteractionId(self.id)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class PairwiseMolecularInteraction(PairwiseGeneToGeneInteraction):\n",
      "    \"\"\"\n",
      "    An interaction at the molecular level between two physical entities\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"PairwiseMolecularInteraction\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:PairwiseMolecularInteraction\"\n",
      "    class_name: ClassVar[str] = \"pairwise molecular interaction\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.PairwiseMolecularInteraction\n",
      "\n",
      "    id: Union[str, PairwiseMolecularInteractionId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, MolecularEntityId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, MolecularEntityId] = None\n",
      "    interacting_molecules_category: Optional[Union[str, OntologyClassId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, PairwiseMolecularInteractionId):\n",
      "            self.id = PairwiseMolecularInteractionId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, MolecularEntityId):\n",
      "            self.subject = MolecularEntityId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, MolecularEntityId):\n",
      "            self.object = MolecularEntityId(self.object)\n",
      "\n",
      "        if self.interacting_molecules_category is not None and not isinstance(self.interacting_molecules_category, OntologyClassId):\n",
      "            self.interacting_molecules_category = OntologyClassId(self.interacting_molecules_category)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class CellLineToEntityAssociationMixin(YAMLRoot):\n",
      "    \"\"\"\n",
      "    An relationship between a cell line and another entity\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"CellLineToEntityAssociationMixin\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:CellLineToEntityAssociationMixin\"\n",
      "    class_name: ClassVar[str] = \"cell line to entity association mixin\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.CellLineToEntityAssociationMixin\n",
      "\n",
      "    subject: Union[str, CellLineId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, CellLineId):\n",
      "            self.subject = CellLineId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class CellLineToDiseaseOrPhenotypicFeatureAssociation(Association):\n",
      "    \"\"\"\n",
      "    An relationship between a cell line and a disease or a phenotype, where the cell line is derived from an\n",
      "    individual with that disease or phenotype.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"CellLineToDiseaseOrPhenotypicFeatureAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:CellLineToDiseaseOrPhenotypicFeatureAssociation\"\n",
      "    class_name: ClassVar[str] = \"cell line to disease or phenotypic feature association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.CellLineToDiseaseOrPhenotypicFeatureAssociation\n",
      "\n",
      "    id: Union[str, CellLineToDiseaseOrPhenotypicFeatureAssociationId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, DiseaseOrPhenotypicFeatureId] = None\n",
      "    disease_context_qualifier: Optional[Union[str, DiseaseId]] = None\n",
      "    subject_specialization_qualifier: Optional[Union[str, URIorCURIE]] = None\n",
      "    object_specialization_qualifier: Optional[Union[str, URIorCURIE]] = None\n",
      "    anatomical_context_qualifier: Optional[str] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, CellLineToDiseaseOrPhenotypicFeatureAssociationId):\n",
      "            self.id = CellLineToDiseaseOrPhenotypicFeatureAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, DiseaseOrPhenotypicFeatureId):\n",
      "            self.subject = DiseaseOrPhenotypicFeatureId(self.subject)\n",
      "\n",
      "        if self.disease_context_qualifier is not None and not isinstance(self.disease_context_qualifier, DiseaseId):\n",
      "            self.disease_context_qualifier = DiseaseId(self.disease_context_qualifier)\n",
      "\n",
      "        if self.subject_specialization_qualifier is not None and not isinstance(self.subject_specialization_qualifier, URIorCURIE):\n",
      "            self.subject_specialization_qualifier = URIorCURIE(self.subject_specialization_qualifier)\n",
      "\n",
      "        if self.object_specialization_qualifier is not None and not isinstance(self.object_specialization_qualifier, URIorCURIE):\n",
      "            self.object_specialization_qualifier = URIorCURIE(self.object_specialization_qualifier)\n",
      "\n",
      "        if self.anatomical_context_qualifier is not None and not isinstance(self.anatomical_context_qualifier, str):\n",
      "            self.anatomical_context_qualifier = str(self.anatomical_context_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ChemicalEntityToEntityAssociationMixin(YAMLRoot):\n",
      "    \"\"\"\n",
      "    An interaction between a chemical entity and another entity\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ChemicalEntityToEntityAssociationMixin\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ChemicalEntityToEntityAssociationMixin\"\n",
      "    class_name: ClassVar[str] = \"chemical entity to entity association mixin\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ChemicalEntityToEntityAssociationMixin\n",
      "\n",
      "    subject: Union[dict, ChemicalEntityOrGeneOrGeneProduct] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, ChemicalEntityOrGeneOrGeneProduct):\n",
      "            self.subject = ChemicalEntityOrGeneOrGeneProduct()\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class DrugToEntityAssociationMixin(ChemicalEntityToEntityAssociationMixin):\n",
      "    \"\"\"\n",
      "    An interaction between a drug and another entity\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"DrugToEntityAssociationMixin\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:DrugToEntityAssociationMixin\"\n",
      "    class_name: ClassVar[str] = \"drug to entity association mixin\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.DrugToEntityAssociationMixin\n",
      "\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, NamedThingId):\n",
      "            self.subject = NamedThingId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, DrugId):\n",
      "            self.subject = DrugId(self.subject)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ChemicalToEntityAssociationMixin(ChemicalEntityToEntityAssociationMixin):\n",
      "    \"\"\"\n",
      "    An interaction between a chemical entity and another entity\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ChemicalToEntityAssociationMixin\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ChemicalToEntityAssociationMixin\"\n",
      "    class_name: ClassVar[str] = \"chemical to entity association mixin\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ChemicalToEntityAssociationMixin\n",
      "\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, NamedThingId):\n",
      "            self.subject = NamedThingId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, ChemicalEntityOrGeneOrGeneProduct):\n",
      "            self.subject = ChemicalEntityOrGeneOrGeneProduct()\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class CaseToEntityAssociationMixin(YAMLRoot):\n",
      "    \"\"\"\n",
      "    An abstract association for use where the case is the subject\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"CaseToEntityAssociationMixin\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:CaseToEntityAssociationMixin\"\n",
      "    class_name: ClassVar[str] = \"case to entity association mixin\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.CaseToEntityAssociationMixin\n",
      "\n",
      "    subject: Union[str, CaseId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, CaseId):\n",
      "            self.subject = CaseId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ChemicalToChemicalAssociation(Association):\n",
      "    \"\"\"\n",
      "    A relationship between two chemical entities. This can encompass actual interactions as well as temporal causal\n",
      "    edges, e.g. one chemical converted to another.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ChemicalToChemicalAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ChemicalToChemicalAssociation\"\n",
      "    class_name: ClassVar[str] = \"chemical to chemical association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ChemicalToChemicalAssociation\n",
      "\n",
      "    id: Union[str, ChemicalToChemicalAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, ChemicalEntityId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ChemicalToChemicalAssociationId):\n",
      "            self.id = ChemicalToChemicalAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, NamedThingId):\n",
      "            self.subject = NamedThingId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, ChemicalEntityId):\n",
      "            self.object = ChemicalEntityId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ReactionToParticipantAssociation(ChemicalToChemicalAssociation):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ReactionToParticipantAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ReactionToParticipantAssociation\"\n",
      "    class_name: ClassVar[str] = \"reaction to participant association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ReactionToParticipantAssociation\n",
      "\n",
      "    id: Union[str, ReactionToParticipantAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, ChemicalEntityId] = None\n",
      "    subject: Union[str, MolecularEntityId] = None\n",
      "    stoichiometry: Optional[int] = None\n",
      "    reaction_direction: Optional[Union[str, \"ReactionDirectionEnum\"]] = None\n",
      "    reaction_side: Optional[Union[str, \"ReactionSideEnum\"]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ReactionToParticipantAssociationId):\n",
      "            self.id = ReactionToParticipantAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, MolecularEntityId):\n",
      "            self.subject = MolecularEntityId(self.subject)\n",
      "\n",
      "        if self.stoichiometry is not None and not isinstance(self.stoichiometry, int):\n",
      "            self.stoichiometry = int(self.stoichiometry)\n",
      "\n",
      "        if self.reaction_direction is not None and not isinstance(self.reaction_direction, ReactionDirectionEnum):\n",
      "            self.reaction_direction = ReactionDirectionEnum(self.reaction_direction)\n",
      "\n",
      "        if self.reaction_side is not None and not isinstance(self.reaction_side, ReactionSideEnum):\n",
      "            self.reaction_side = ReactionSideEnum(self.reaction_side)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ReactionToCatalystAssociation(ReactionToParticipantAssociation):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ReactionToCatalystAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ReactionToCatalystAssociation\"\n",
      "    class_name: ClassVar[str] = \"reaction to catalyst association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ReactionToCatalystAssociation\n",
      "\n",
      "    id: Union[str, ReactionToCatalystAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    subject: Union[str, MolecularEntityId] = None\n",
      "    object: Union[dict, GeneOrGeneProduct] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ReactionToCatalystAssociationId):\n",
      "            self.id = ReactionToCatalystAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, GeneOrGeneProduct):\n",
      "            self.object = GeneOrGeneProduct(**as_dict(self.object))\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ChemicalToChemicalDerivationAssociation(ChemicalToChemicalAssociation):\n",
      "    \"\"\"\n",
      "    A causal relationship between two chemical entities, where the subject represents the upstream entity and the\n",
      "    object represents the downstream. For any such association there is an implicit reaction: IF R has-input C1 AND R\n",
      "    has-output C2 AND R enabled-by P AND R type Reaction THEN C1 derives-into C2 catalyst qualifier P\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ChemicalToChemicalDerivationAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ChemicalToChemicalDerivationAssociation\"\n",
      "    class_name: ClassVar[str] = \"chemical to chemical derivation association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ChemicalToChemicalDerivationAssociation\n",
      "\n",
      "    id: Union[str, ChemicalToChemicalDerivationAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, ChemicalEntityId] = None\n",
      "    object: Union[str, ChemicalEntityId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    catalyst_qualifier: Optional[Union[Union[dict, MacromolecularMachineMixin], List[Union[dict, MacromolecularMachineMixin]]]] = empty_list()\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ChemicalToChemicalDerivationAssociationId):\n",
      "            self.id = ChemicalToChemicalDerivationAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, ChemicalEntityId):\n",
      "            self.subject = ChemicalEntityId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, ChemicalEntityId):\n",
      "            self.object = ChemicalEntityId(self.object)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if not isinstance(self.catalyst_qualifier, list):\n",
      "            self.catalyst_qualifier = [self.catalyst_qualifier] if self.catalyst_qualifier is not None else []\n",
      "        self.catalyst_qualifier = [v if isinstance(v, MacromolecularMachineMixin) else MacromolecularMachineMixin(**as_dict(v)) for v in self.catalyst_qualifier]\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ChemicalToDiseaseOrPhenotypicFeatureAssociation(Association):\n",
      "    \"\"\"\n",
      "    An interaction between a chemical entity and a phenotype or disease, where the presence of the chemical gives rise\n",
      "    to or exacerbates the phenotype.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ChemicalToDiseaseOrPhenotypicFeatureAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ChemicalToDiseaseOrPhenotypicFeatureAssociation\"\n",
      "    class_name: ClassVar[str] = \"chemical to disease or phenotypic feature association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ChemicalToDiseaseOrPhenotypicFeatureAssociation\n",
      "\n",
      "    id: Union[str, ChemicalToDiseaseOrPhenotypicFeatureAssociationId] = None\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    object: Union[str, DiseaseOrPhenotypicFeatureId] = None\n",
      "    disease_context_qualifier: Optional[Union[str, DiseaseId]] = None\n",
      "    subject_specialization_qualifier: Optional[Union[str, URIorCURIE]] = None\n",
      "    object_specialization_qualifier: Optional[Union[str, URIorCURIE]] = None\n",
      "    anatomical_context_qualifier: Optional[str] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ChemicalToDiseaseOrPhenotypicFeatureAssociationId):\n",
      "            self.id = ChemicalToDiseaseOrPhenotypicFeatureAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, NamedThingId):\n",
      "            self.subject = NamedThingId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, DiseaseOrPhenotypicFeatureId):\n",
      "            self.object = DiseaseOrPhenotypicFeatureId(self.object)\n",
      "\n",
      "        if self.disease_context_qualifier is not None and not isinstance(self.disease_context_qualifier, DiseaseId):\n",
      "            self.disease_context_qualifier = DiseaseId(self.disease_context_qualifier)\n",
      "\n",
      "        if self.subject_specialization_qualifier is not None and not isinstance(self.subject_specialization_qualifier, URIorCURIE):\n",
      "            self.subject_specialization_qualifier = URIorCURIE(self.subject_specialization_qualifier)\n",
      "\n",
      "        if self.object_specialization_qualifier is not None and not isinstance(self.object_specialization_qualifier, URIorCURIE):\n",
      "            self.object_specialization_qualifier = URIorCURIE(self.object_specialization_qualifier)\n",
      "\n",
      "        if self.anatomical_context_qualifier is not None and not isinstance(self.anatomical_context_qualifier, str):\n",
      "            self.anatomical_context_qualifier = str(self.anatomical_context_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ChemicalOrDrugOrTreatmentToDiseaseOrPhenotypicFeatureAssociation(Association):\n",
      "    \"\"\"\n",
      "    This association defines a relationship between a chemical or treatment (or procedure) and a disease or phenotypic\n",
      "    feature where the disease or phenotypic feature is a secondary undesirable effect.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ChemicalOrDrugOrTreatmentToDiseaseOrPhenotypicFeatureAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ChemicalOrDrugOrTreatmentToDiseaseOrPhenotypicFeatureAssociation\"\n",
      "    class_name: ClassVar[str] = \"chemical or drug or treatment to disease or phenotypic feature association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ChemicalOrDrugOrTreatmentToDiseaseOrPhenotypicFeatureAssociation\n",
      "\n",
      "    id: Union[str, ChemicalOrDrugOrTreatmentToDiseaseOrPhenotypicFeatureAssociationId] = None\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    FDA_adverse_event_level: Optional[Union[str, \"FDAIDAAdverseEventEnum\"]] = None\n",
      "    disease_context_qualifier: Optional[Union[str, DiseaseId]] = None\n",
      "    subject_specialization_qualifier: Optional[Union[str, URIorCURIE]] = None\n",
      "    object_specialization_qualifier: Optional[Union[str, URIorCURIE]] = None\n",
      "    anatomical_context_qualifier: Optional[str] = None\n",
      "    frequency_qualifier: Optional[Union[str, FrequencyValue]] = None\n",
      "    subject_aspect_qualifier: Optional[str] = None\n",
      "    subject_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    object_aspect_qualifier: Optional[str] = None\n",
      "    object_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    qualified_predicate: Optional[str] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ChemicalOrDrugOrTreatmentToDiseaseOrPhenotypicFeatureAssociationId):\n",
      "            self.id = ChemicalOrDrugOrTreatmentToDiseaseOrPhenotypicFeatureAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, NamedThingId):\n",
      "            self.subject = NamedThingId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self.FDA_adverse_event_level is not None and not isinstance(self.FDA_adverse_event_level, FDAIDAAdverseEventEnum):\n",
      "            self.FDA_adverse_event_level = FDAIDAAdverseEventEnum(self.FDA_adverse_event_level)\n",
      "\n",
      "        if self.disease_context_qualifier is not None and not isinstance(self.disease_context_qualifier, DiseaseId):\n",
      "            self.disease_context_qualifier = DiseaseId(self.disease_context_qualifier)\n",
      "\n",
      "        if self.subject_specialization_qualifier is not None and not isinstance(self.subject_specialization_qualifier, URIorCURIE):\n",
      "            self.subject_specialization_qualifier = URIorCURIE(self.subject_specialization_qualifier)\n",
      "\n",
      "        if self.object_specialization_qualifier is not None and not isinstance(self.object_specialization_qualifier, URIorCURIE):\n",
      "            self.object_specialization_qualifier = URIorCURIE(self.object_specialization_qualifier)\n",
      "\n",
      "        if self.anatomical_context_qualifier is not None and not isinstance(self.anatomical_context_qualifier, str):\n",
      "            self.anatomical_context_qualifier = str(self.anatomical_context_qualifier)\n",
      "\n",
      "        if self.frequency_qualifier is not None and not isinstance(self.frequency_qualifier, FrequencyValue):\n",
      "            self.frequency_qualifier = FrequencyValue(self.frequency_qualifier)\n",
      "\n",
      "        if self.subject_aspect_qualifier is not None and not isinstance(self.subject_aspect_qualifier, str):\n",
      "            self.subject_aspect_qualifier = str(self.subject_aspect_qualifier)\n",
      "\n",
      "        if self.subject_direction_qualifier is not None and not isinstance(self.subject_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.subject_direction_qualifier = DirectionQualifierEnum(self.subject_direction_qualifier)\n",
      "\n",
      "        if self.object_aspect_qualifier is not None and not isinstance(self.object_aspect_qualifier, str):\n",
      "            self.object_aspect_qualifier = str(self.object_aspect_qualifier)\n",
      "\n",
      "        if self.object_direction_qualifier is not None and not isinstance(self.object_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.object_direction_qualifier = DirectionQualifierEnum(self.object_direction_qualifier)\n",
      "\n",
      "        if self.qualified_predicate is not None and not isinstance(self.qualified_predicate, str):\n",
      "            self.qualified_predicate = str(self.qualified_predicate)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ChemicalOrDrugOrTreatmentSideEffectDiseaseOrPhenotypicFeatureAssociation(ChemicalOrDrugOrTreatmentToDiseaseOrPhenotypicFeatureAssociation):\n",
      "    \"\"\"\n",
      "    This association defines a relationship between a chemical or treatment (or procedure) and a disease or phenotypic\n",
      "    feature where the disesae or phenotypic feature is a secondary, typically (but not always) undesirable effect.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ChemicalOrDrugOrTreatmentSideEffectDiseaseOrPhenotypicFeatureAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ChemicalOrDrugOrTreatmentSideEffectDiseaseOrPhenotypicFeatureAssociation\"\n",
      "    class_name: ClassVar[str] = \"chemical or drug or treatment side effect disease or phenotypic feature association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ChemicalOrDrugOrTreatmentSideEffectDiseaseOrPhenotypicFeatureAssociation\n",
      "\n",
      "    id: Union[str, ChemicalOrDrugOrTreatmentSideEffectDiseaseOrPhenotypicFeatureAssociationId] = None\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    disease_context_qualifier: Optional[Union[str, DiseaseId]] = None\n",
      "    subject_specialization_qualifier: Optional[Union[str, URIorCURIE]] = None\n",
      "    object_specialization_qualifier: Optional[Union[str, URIorCURIE]] = None\n",
      "    anatomical_context_qualifier: Optional[str] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ChemicalOrDrugOrTreatmentSideEffectDiseaseOrPhenotypicFeatureAssociationId):\n",
      "            self.id = ChemicalOrDrugOrTreatmentSideEffectDiseaseOrPhenotypicFeatureAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, NamedThingId):\n",
      "            self.subject = NamedThingId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self.disease_context_qualifier is not None and not isinstance(self.disease_context_qualifier, DiseaseId):\n",
      "            self.disease_context_qualifier = DiseaseId(self.disease_context_qualifier)\n",
      "\n",
      "        if self.subject_specialization_qualifier is not None and not isinstance(self.subject_specialization_qualifier, URIorCURIE):\n",
      "            self.subject_specialization_qualifier = URIorCURIE(self.subject_specialization_qualifier)\n",
      "\n",
      "        if self.object_specialization_qualifier is not None and not isinstance(self.object_specialization_qualifier, URIorCURIE):\n",
      "            self.object_specialization_qualifier = URIorCURIE(self.object_specialization_qualifier)\n",
      "\n",
      "        if self.anatomical_context_qualifier is not None and not isinstance(self.anatomical_context_qualifier, str):\n",
      "            self.anatomical_context_qualifier = str(self.anatomical_context_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GeneToPathwayAssociation(Association):\n",
      "    \"\"\"\n",
      "    An interaction between a gene or gene product and a biological process or pathway.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GeneToPathwayAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GeneToPathwayAssociation\"\n",
      "    class_name: ClassVar[str] = \"gene to pathway association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GeneToPathwayAssociation\n",
      "\n",
      "    id: Union[str, GeneToPathwayAssociationId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[dict, GeneOrGeneProduct] = None\n",
      "    object: Union[str, PathwayId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GeneToPathwayAssociationId):\n",
      "            self.id = GeneToPathwayAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, GeneOrGeneProduct):\n",
      "            self.subject = GeneOrGeneProduct(**as_dict(self.subject))\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, PathwayId):\n",
      "            self.object = PathwayId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class MolecularActivityToPathwayAssociation(Association):\n",
      "    \"\"\"\n",
      "    Association that holds the relationship between a reaction and the pathway it participates in.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"MolecularActivityToPathwayAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:MolecularActivityToPathwayAssociation\"\n",
      "    class_name: ClassVar[str] = \"molecular activity to pathway association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.MolecularActivityToPathwayAssociation\n",
      "\n",
      "    id: Union[str, MolecularActivityToPathwayAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, MolecularActivityId] = None\n",
      "    object: Union[str, PathwayId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, MolecularActivityToPathwayAssociationId):\n",
      "            self.id = MolecularActivityToPathwayAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, MolecularActivityId):\n",
      "            self.subject = MolecularActivityId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, PathwayId):\n",
      "            self.object = PathwayId(self.object)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ChemicalToPathwayAssociation(Association):\n",
      "    \"\"\"\n",
      "    An interaction between a chemical entity and a biological process or pathway.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ChemicalToPathwayAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ChemicalToPathwayAssociation\"\n",
      "    class_name: ClassVar[str] = \"chemical to pathway association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ChemicalToPathwayAssociation\n",
      "\n",
      "    id: Union[str, ChemicalToPathwayAssociationId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, ChemicalEntityId] = None\n",
      "    object: Union[str, PathwayId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ChemicalToPathwayAssociationId):\n",
      "            self.id = ChemicalToPathwayAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, ChemicalEntityId):\n",
      "            self.subject = ChemicalEntityId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, PathwayId):\n",
      "            self.object = PathwayId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class NamedThingAssociatedWithLikelihoodOfNamedThingAssociation(Association):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"NamedThingAssociatedWithLikelihoodOfNamedThingAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:NamedThingAssociatedWithLikelihoodOfNamedThingAssociation\"\n",
      "    class_name: ClassVar[str] = \"named thing associated with likelihood of named thing association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.NamedThingAssociatedWithLikelihoodOfNamedThingAssociation\n",
      "\n",
      "    id: Union[str, NamedThingAssociatedWithLikelihoodOfNamedThingAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    subject_aspect_qualifier: Optional[str] = None\n",
      "    subject_context_qualifier: Optional[Union[str, OntologyClassId]] = None\n",
      "    object_aspect_qualifier: Optional[str] = None\n",
      "    object_context_qualifier: Optional[Union[str, OntologyClassId]] = None\n",
      "    population_context_qualifier: Optional[Union[str, PopulationOfIndividualOrganismsId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, NamedThingAssociatedWithLikelihoodOfNamedThingAssociationId):\n",
      "            self.id = NamedThingAssociatedWithLikelihoodOfNamedThingAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, NamedThingId):\n",
      "            self.subject = NamedThingId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        if self.subject_aspect_qualifier is not None and not isinstance(self.subject_aspect_qualifier, str):\n",
      "            self.subject_aspect_qualifier = str(self.subject_aspect_qualifier)\n",
      "\n",
      "        if self.subject_context_qualifier is not None and not isinstance(self.subject_context_qualifier, OntologyClassId):\n",
      "            self.subject_context_qualifier = OntologyClassId(self.subject_context_qualifier)\n",
      "\n",
      "        if self.object_aspect_qualifier is not None and not isinstance(self.object_aspect_qualifier, str):\n",
      "            self.object_aspect_qualifier = str(self.object_aspect_qualifier)\n",
      "\n",
      "        if self.object_context_qualifier is not None and not isinstance(self.object_context_qualifier, OntologyClassId):\n",
      "            self.object_context_qualifier = OntologyClassId(self.object_context_qualifier)\n",
      "\n",
      "        if self.population_context_qualifier is not None and not isinstance(self.population_context_qualifier, PopulationOfIndividualOrganismsId):\n",
      "            self.population_context_qualifier = PopulationOfIndividualOrganismsId(self.population_context_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ChemicalGeneInteractionAssociation(Association):\n",
      "    \"\"\"\n",
      "    describes a physical interaction between a chemical entity and a gene or gene product. Any biological or chemical\n",
      "    effect resulting from such an interaction are out of scope, and covered by the ChemicalAffectsGeneAssociation type\n",
      "    (e.g. impact of a chemical on the abundance, activity, structure, etc, of either participant in the interaction)\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ChemicalGeneInteractionAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ChemicalGeneInteractionAssociation\"\n",
      "    class_name: ClassVar[str] = \"chemical gene interaction association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ChemicalGeneInteractionAssociation\n",
      "\n",
      "    id: Union[str, ChemicalGeneInteractionAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, ChemicalEntityId] = None\n",
      "    object: Union[dict, GeneOrGeneProduct] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    subject_form_or_variant_qualifier: Optional[Union[str, \"ChemicalOrGeneOrGeneProductFormOrVariantEnum\"]] = None\n",
      "    subject_part_qualifier: Optional[Union[str, \"GeneOrGeneProductOrChemicalPartQualifierEnum\"]] = None\n",
      "    subject_derivative_qualifier: Optional[Union[str, \"ChemicalEntityDerivativeEnum\"]] = None\n",
      "    subject_context_qualifier: Optional[Union[str, AnatomicalEntityId]] = None\n",
      "    object_form_or_variant_qualifier: Optional[Union[str, \"ChemicalOrGeneOrGeneProductFormOrVariantEnum\"]] = None\n",
      "    object_part_qualifier: Optional[Union[str, \"GeneOrGeneProductOrChemicalPartQualifierEnum\"]] = None\n",
      "    object_context_qualifier: Optional[Union[str, AnatomicalEntityId]] = None\n",
      "    anatomical_context_qualifier: Optional[Union[str, AnatomicalEntityId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ChemicalGeneInteractionAssociationId):\n",
      "            self.id = ChemicalGeneInteractionAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, ChemicalEntityId):\n",
      "            self.subject = ChemicalEntityId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, GeneOrGeneProduct):\n",
      "            self.object = GeneOrGeneProduct(**as_dict(self.object))\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self.subject_form_or_variant_qualifier is not None and not isinstance(self.subject_form_or_variant_qualifier, ChemicalOrGeneOrGeneProductFormOrVariantEnum):\n",
      "            self.subject_form_or_variant_qualifier = ChemicalOrGeneOrGeneProductFormOrVariantEnum(self.subject_form_or_variant_qualifier)\n",
      "\n",
      "        if self.subject_part_qualifier is not None and not isinstance(self.subject_part_qualifier, GeneOrGeneProductOrChemicalPartQualifierEnum):\n",
      "            self.subject_part_qualifier = GeneOrGeneProductOrChemicalPartQualifierEnum(self.subject_part_qualifier)\n",
      "\n",
      "        if self.subject_derivative_qualifier is not None and not isinstance(self.subject_derivative_qualifier, ChemicalEntityDerivativeEnum):\n",
      "            self.subject_derivative_qualifier = ChemicalEntityDerivativeEnum(self.subject_derivative_qualifier)\n",
      "\n",
      "        if self.subject_context_qualifier is not None and not isinstance(self.subject_context_qualifier, AnatomicalEntityId):\n",
      "            self.subject_context_qualifier = AnatomicalEntityId(self.subject_context_qualifier)\n",
      "\n",
      "        if self.object_form_or_variant_qualifier is not None and not isinstance(self.object_form_or_variant_qualifier, ChemicalOrGeneOrGeneProductFormOrVariantEnum):\n",
      "            self.object_form_or_variant_qualifier = ChemicalOrGeneOrGeneProductFormOrVariantEnum(self.object_form_or_variant_qualifier)\n",
      "\n",
      "        if self.object_part_qualifier is not None and not isinstance(self.object_part_qualifier, GeneOrGeneProductOrChemicalPartQualifierEnum):\n",
      "            self.object_part_qualifier = GeneOrGeneProductOrChemicalPartQualifierEnum(self.object_part_qualifier)\n",
      "\n",
      "        if self.object_context_qualifier is not None and not isinstance(self.object_context_qualifier, AnatomicalEntityId):\n",
      "            self.object_context_qualifier = AnatomicalEntityId(self.object_context_qualifier)\n",
      "\n",
      "        if self.anatomical_context_qualifier is not None and not isinstance(self.anatomical_context_qualifier, AnatomicalEntityId):\n",
      "            self.anatomical_context_qualifier = AnatomicalEntityId(self.anatomical_context_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GeneRegulatesGeneAssociation(Association):\n",
      "    \"\"\"\n",
      "    Describes a regulatory relationship between two genes or gene products.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GeneRegulatesGeneAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GeneRegulatesGeneAssociation\"\n",
      "    class_name: ClassVar[str] = \"gene regulates gene association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GeneRegulatesGeneAssociation\n",
      "\n",
      "    id: Union[str, GeneRegulatesGeneAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    object_aspect_qualifier: Union[str, \"GeneOrGeneProductOrChemicalEntityAspectEnum\"] = None\n",
      "    object_direction_qualifier: Union[str, \"DirectionQualifierEnum\"] = None\n",
      "    qualified_predicate: str = None\n",
      "    subject: Union[dict, GeneOrGeneProduct] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[dict, GeneOrGeneProduct] = None\n",
      "    species_context_qualifier: Optional[Union[str, OrganismTaxonId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GeneRegulatesGeneAssociationId):\n",
      "            self.id = GeneRegulatesGeneAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.object_aspect_qualifier):\n",
      "            self.MissingRequiredField(\"object_aspect_qualifier\")\n",
      "        if not isinstance(self.object_aspect_qualifier, GeneOrGeneProductOrChemicalEntityAspectEnum):\n",
      "            self.object_aspect_qualifier = GeneOrGeneProductOrChemicalEntityAspectEnum(self.object_aspect_qualifier)\n",
      "\n",
      "        if self._is_empty(self.object_direction_qualifier):\n",
      "            self.MissingRequiredField(\"object_direction_qualifier\")\n",
      "        if not isinstance(self.object_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.object_direction_qualifier = DirectionQualifierEnum(self.object_direction_qualifier)\n",
      "\n",
      "        if self._is_empty(self.qualified_predicate):\n",
      "            self.MissingRequiredField(\"qualified_predicate\")\n",
      "        if not isinstance(self.qualified_predicate, str):\n",
      "            self.qualified_predicate = str(self.qualified_predicate)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, GeneOrGeneProduct):\n",
      "            self.subject = GeneOrGeneProduct(**as_dict(self.subject))\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, GeneOrGeneProduct):\n",
      "            self.object = GeneOrGeneProduct(**as_dict(self.object))\n",
      "\n",
      "        if self.species_context_qualifier is not None and not isinstance(self.species_context_qualifier, OrganismTaxonId):\n",
      "            self.species_context_qualifier = OrganismTaxonId(self.species_context_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ProcessRegulatesProcessAssociation(Association):\n",
      "    \"\"\"\n",
      "    Describes a regulatory relationship between two genes or gene products.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ProcessRegulatesProcessAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ProcessRegulatesProcessAssociation\"\n",
      "    class_name: ClassVar[str] = \"process regulates process association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ProcessRegulatesProcessAssociation\n",
      "\n",
      "    id: Union[str, ProcessRegulatesProcessAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, BiologicalProcessId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, BiologicalProcessId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ProcessRegulatesProcessAssociationId):\n",
      "            self.id = ProcessRegulatesProcessAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, BiologicalProcessId):\n",
      "            self.subject = BiologicalProcessId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, BiologicalProcessId):\n",
      "            self.object = BiologicalProcessId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ChemicalAffectsGeneAssociation(Association):\n",
      "    \"\"\"\n",
      "    Describes an effect that a chemical has on a gene or gene product (e.g. an impact of on its abundance,\n",
      "    activity,localization, processing, expression, etc.)\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ChemicalAffectsGeneAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ChemicalAffectsGeneAssociation\"\n",
      "    class_name: ClassVar[str] = \"chemical affects gene association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ChemicalAffectsGeneAssociation\n",
      "\n",
      "    id: Union[str, ChemicalAffectsGeneAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, ChemicalEntityId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[dict, GeneOrGeneProduct] = None\n",
      "    subject_form_or_variant_qualifier: Optional[Union[str, \"ChemicalOrGeneOrGeneProductFormOrVariantEnum\"]] = None\n",
      "    subject_part_qualifier: Optional[Union[str, \"GeneOrGeneProductOrChemicalPartQualifierEnum\"]] = None\n",
      "    subject_derivative_qualifier: Optional[Union[str, \"ChemicalEntityDerivativeEnum\"]] = None\n",
      "    subject_aspect_qualifier: Optional[Union[str, \"GeneOrGeneProductOrChemicalEntityAspectEnum\"]] = None\n",
      "    subject_context_qualifier: Optional[Union[str, AnatomicalEntityId]] = None\n",
      "    subject_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    object_form_or_variant_qualifier: Optional[Union[str, \"ChemicalOrGeneOrGeneProductFormOrVariantEnum\"]] = None\n",
      "    object_part_qualifier: Optional[Union[str, \"GeneOrGeneProductOrChemicalPartQualifierEnum\"]] = None\n",
      "    object_aspect_qualifier: Optional[Union[str, \"GeneOrGeneProductOrChemicalEntityAspectEnum\"]] = None\n",
      "    object_context_qualifier: Optional[Union[str, AnatomicalEntityId]] = None\n",
      "    object_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    causal_mechanism_qualifier: Optional[Union[str, \"CausalMechanismQualifierEnum\"]] = None\n",
      "    anatomical_context_qualifier: Optional[Union[str, AnatomicalEntityId]] = None\n",
      "    qualified_predicate: Optional[str] = None\n",
      "    species_context_qualifier: Optional[Union[str, OrganismTaxonId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ChemicalAffectsGeneAssociationId):\n",
      "            self.id = ChemicalAffectsGeneAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, ChemicalEntityId):\n",
      "            self.subject = ChemicalEntityId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, GeneOrGeneProduct):\n",
      "            self.object = GeneOrGeneProduct(**as_dict(self.object))\n",
      "\n",
      "        if self.subject_form_or_variant_qualifier is not None and not isinstance(self.subject_form_or_variant_qualifier, ChemicalOrGeneOrGeneProductFormOrVariantEnum):\n",
      "            self.subject_form_or_variant_qualifier = ChemicalOrGeneOrGeneProductFormOrVariantEnum(self.subject_form_or_variant_qualifier)\n",
      "\n",
      "        if self.subject_part_qualifier is not None and not isinstance(self.subject_part_qualifier, GeneOrGeneProductOrChemicalPartQualifierEnum):\n",
      "            self.subject_part_qualifier = GeneOrGeneProductOrChemicalPartQualifierEnum(self.subject_part_qualifier)\n",
      "\n",
      "        if self.subject_derivative_qualifier is not None and not isinstance(self.subject_derivative_qualifier, ChemicalEntityDerivativeEnum):\n",
      "            self.subject_derivative_qualifier = ChemicalEntityDerivativeEnum(self.subject_derivative_qualifier)\n",
      "\n",
      "        if self.subject_aspect_qualifier is not None and not isinstance(self.subject_aspect_qualifier, GeneOrGeneProductOrChemicalEntityAspectEnum):\n",
      "            self.subject_aspect_qualifier = GeneOrGeneProductOrChemicalEntityAspectEnum(self.subject_aspect_qualifier)\n",
      "\n",
      "        if self.subject_context_qualifier is not None and not isinstance(self.subject_context_qualifier, AnatomicalEntityId):\n",
      "            self.subject_context_qualifier = AnatomicalEntityId(self.subject_context_qualifier)\n",
      "\n",
      "        if self.subject_direction_qualifier is not None and not isinstance(self.subject_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.subject_direction_qualifier = DirectionQualifierEnum(self.subject_direction_qualifier)\n",
      "\n",
      "        if self.object_form_or_variant_qualifier is not None and not isinstance(self.object_form_or_variant_qualifier, ChemicalOrGeneOrGeneProductFormOrVariantEnum):\n",
      "            self.object_form_or_variant_qualifier = ChemicalOrGeneOrGeneProductFormOrVariantEnum(self.object_form_or_variant_qualifier)\n",
      "\n",
      "        if self.object_part_qualifier is not None and not isinstance(self.object_part_qualifier, GeneOrGeneProductOrChemicalPartQualifierEnum):\n",
      "            self.object_part_qualifier = GeneOrGeneProductOrChemicalPartQualifierEnum(self.object_part_qualifier)\n",
      "\n",
      "        if self.object_aspect_qualifier is not None and not isinstance(self.object_aspect_qualifier, GeneOrGeneProductOrChemicalEntityAspectEnum):\n",
      "            self.object_aspect_qualifier = GeneOrGeneProductOrChemicalEntityAspectEnum(self.object_aspect_qualifier)\n",
      "\n",
      "        if self.object_context_qualifier is not None and not isinstance(self.object_context_qualifier, AnatomicalEntityId):\n",
      "            self.object_context_qualifier = AnatomicalEntityId(self.object_context_qualifier)\n",
      "\n",
      "        if self.object_direction_qualifier is not None and not isinstance(self.object_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.object_direction_qualifier = DirectionQualifierEnum(self.object_direction_qualifier)\n",
      "\n",
      "        if self.causal_mechanism_qualifier is not None and not isinstance(self.causal_mechanism_qualifier, CausalMechanismQualifierEnum):\n",
      "            self.causal_mechanism_qualifier = CausalMechanismQualifierEnum(self.causal_mechanism_qualifier)\n",
      "\n",
      "        if self.anatomical_context_qualifier is not None and not isinstance(self.anatomical_context_qualifier, AnatomicalEntityId):\n",
      "            self.anatomical_context_qualifier = AnatomicalEntityId(self.anatomical_context_qualifier)\n",
      "\n",
      "        if self.qualified_predicate is not None and not isinstance(self.qualified_predicate, str):\n",
      "            self.qualified_predicate = str(self.qualified_predicate)\n",
      "\n",
      "        if self.species_context_qualifier is not None and not isinstance(self.species_context_qualifier, OrganismTaxonId):\n",
      "            self.species_context_qualifier = OrganismTaxonId(self.species_context_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GeneAffectsChemicalAssociation(Association):\n",
      "    \"\"\"\n",
      "    Describes an effect that a gene or gene product has on a chemical entity (e.g. an impact of on its abundance,\n",
      "    activity, localization, processing, transport, etc.)\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GeneAffectsChemicalAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GeneAffectsChemicalAssociation\"\n",
      "    class_name: ClassVar[str] = \"gene affects chemical association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GeneAffectsChemicalAssociation\n",
      "\n",
      "    id: Union[str, GeneAffectsChemicalAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[dict, GeneOrGeneProduct] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, ChemicalEntityId] = None\n",
      "    subject_form_or_variant_qualifier: Optional[Union[str, \"ChemicalOrGeneOrGeneProductFormOrVariantEnum\"]] = None\n",
      "    subject_part_qualifier: Optional[Union[str, \"GeneOrGeneProductOrChemicalPartQualifierEnum\"]] = None\n",
      "    subject_derivative_qualifier: Optional[str] = None\n",
      "    subject_aspect_qualifier: Optional[Union[str, \"GeneOrGeneProductOrChemicalEntityAspectEnum\"]] = None\n",
      "    subject_context_qualifier: Optional[Union[str, AnatomicalEntityId]] = None\n",
      "    subject_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    object_form_or_variant_qualifier: Optional[Union[str, \"ChemicalOrGeneOrGeneProductFormOrVariantEnum\"]] = None\n",
      "    object_part_qualifier: Optional[Union[str, \"GeneOrGeneProductOrChemicalPartQualifierEnum\"]] = None\n",
      "    object_aspect_qualifier: Optional[Union[str, \"GeneOrGeneProductOrChemicalEntityAspectEnum\"]] = None\n",
      "    object_context_qualifier: Optional[Union[str, AnatomicalEntityId]] = None\n",
      "    object_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    object_derivative_qualifier: Optional[Union[str, \"ChemicalEntityDerivativeEnum\"]] = None\n",
      "    causal_mechanism_qualifier: Optional[Union[str, \"CausalMechanismQualifierEnum\"]] = None\n",
      "    anatomical_context_qualifier: Optional[Union[str, AnatomicalEntityId]] = None\n",
      "    qualified_predicate: Optional[str] = None\n",
      "    species_context_qualifier: Optional[Union[str, OrganismTaxonId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GeneAffectsChemicalAssociationId):\n",
      "            self.id = GeneAffectsChemicalAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, GeneOrGeneProduct):\n",
      "            self.subject = GeneOrGeneProduct(**as_dict(self.subject))\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, ChemicalEntityId):\n",
      "            self.object = ChemicalEntityId(self.object)\n",
      "\n",
      "        if self.subject_form_or_variant_qualifier is not None and not isinstance(self.subject_form_or_variant_qualifier, ChemicalOrGeneOrGeneProductFormOrVariantEnum):\n",
      "            self.subject_form_or_variant_qualifier = ChemicalOrGeneOrGeneProductFormOrVariantEnum(self.subject_form_or_variant_qualifier)\n",
      "\n",
      "        if self.subject_part_qualifier is not None and not isinstance(self.subject_part_qualifier, GeneOrGeneProductOrChemicalPartQualifierEnum):\n",
      "            self.subject_part_qualifier = GeneOrGeneProductOrChemicalPartQualifierEnum(self.subject_part_qualifier)\n",
      "\n",
      "        if self.subject_derivative_qualifier is not None and not isinstance(self.subject_derivative_qualifier, str):\n",
      "            self.subject_derivative_qualifier = str(self.subject_derivative_qualifier)\n",
      "\n",
      "        if self.subject_aspect_qualifier is not None and not isinstance(self.subject_aspect_qualifier, GeneOrGeneProductOrChemicalEntityAspectEnum):\n",
      "            self.subject_aspect_qualifier = GeneOrGeneProductOrChemicalEntityAspectEnum(self.subject_aspect_qualifier)\n",
      "\n",
      "        if self.subject_context_qualifier is not None and not isinstance(self.subject_context_qualifier, AnatomicalEntityId):\n",
      "            self.subject_context_qualifier = AnatomicalEntityId(self.subject_context_qualifier)\n",
      "\n",
      "        if self.subject_direction_qualifier is not None and not isinstance(self.subject_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.subject_direction_qualifier = DirectionQualifierEnum(self.subject_direction_qualifier)\n",
      "\n",
      "        if self.object_form_or_variant_qualifier is not None and not isinstance(self.object_form_or_variant_qualifier, ChemicalOrGeneOrGeneProductFormOrVariantEnum):\n",
      "            self.object_form_or_variant_qualifier = ChemicalOrGeneOrGeneProductFormOrVariantEnum(self.object_form_or_variant_qualifier)\n",
      "\n",
      "        if self.object_part_qualifier is not None and not isinstance(self.object_part_qualifier, GeneOrGeneProductOrChemicalPartQualifierEnum):\n",
      "            self.object_part_qualifier = GeneOrGeneProductOrChemicalPartQualifierEnum(self.object_part_qualifier)\n",
      "\n",
      "        if self.object_aspect_qualifier is not None and not isinstance(self.object_aspect_qualifier, GeneOrGeneProductOrChemicalEntityAspectEnum):\n",
      "            self.object_aspect_qualifier = GeneOrGeneProductOrChemicalEntityAspectEnum(self.object_aspect_qualifier)\n",
      "\n",
      "        if self.object_context_qualifier is not None and not isinstance(self.object_context_qualifier, AnatomicalEntityId):\n",
      "            self.object_context_qualifier = AnatomicalEntityId(self.object_context_qualifier)\n",
      "\n",
      "        if self.object_direction_qualifier is not None and not isinstance(self.object_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.object_direction_qualifier = DirectionQualifierEnum(self.object_direction_qualifier)\n",
      "\n",
      "        if self.object_derivative_qualifier is not None and not isinstance(self.object_derivative_qualifier, ChemicalEntityDerivativeEnum):\n",
      "            self.object_derivative_qualifier = ChemicalEntityDerivativeEnum(self.object_derivative_qualifier)\n",
      "\n",
      "        if self.causal_mechanism_qualifier is not None and not isinstance(self.causal_mechanism_qualifier, CausalMechanismQualifierEnum):\n",
      "            self.causal_mechanism_qualifier = CausalMechanismQualifierEnum(self.causal_mechanism_qualifier)\n",
      "\n",
      "        if self.anatomical_context_qualifier is not None and not isinstance(self.anatomical_context_qualifier, AnatomicalEntityId):\n",
      "            self.anatomical_context_qualifier = AnatomicalEntityId(self.anatomical_context_qualifier)\n",
      "\n",
      "        if self.qualified_predicate is not None and not isinstance(self.qualified_predicate, str):\n",
      "            self.qualified_predicate = str(self.qualified_predicate)\n",
      "\n",
      "        if self.species_context_qualifier is not None and not isinstance(self.species_context_qualifier, OrganismTaxonId):\n",
      "            self.species_context_qualifier = OrganismTaxonId(self.species_context_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class DrugToGeneAssociation(Association):\n",
      "    \"\"\"\n",
      "    An interaction between a drug and a gene or gene product.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"DrugToGeneAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:DrugToGeneAssociation\"\n",
      "    class_name: ClassVar[str] = \"drug to gene association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.DrugToGeneAssociation\n",
      "\n",
      "    id: Union[str, DrugToGeneAssociationId] = None\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    object: Union[dict, GeneOrGeneProduct] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, DrugToGeneAssociationId):\n",
      "            self.id = DrugToGeneAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, NamedThingId):\n",
      "            self.subject = NamedThingId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, GeneOrGeneProduct):\n",
      "            self.object = GeneOrGeneProduct(**as_dict(self.object))\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class MaterialSampleToEntityAssociationMixin(YAMLRoot):\n",
      "    \"\"\"\n",
      "    An association between a material sample and something.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"MaterialSampleToEntityAssociationMixin\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:MaterialSampleToEntityAssociationMixin\"\n",
      "    class_name: ClassVar[str] = \"material sample to entity association mixin\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.MaterialSampleToEntityAssociationMixin\n",
      "\n",
      "    subject: Union[str, MaterialSampleId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, MaterialSampleId):\n",
      "            self.subject = MaterialSampleId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class MaterialSampleDerivationAssociation(Association):\n",
      "    \"\"\"\n",
      "    An association between a material sample and the material entity from which it is derived.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"MaterialSampleDerivationAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:MaterialSampleDerivationAssociation\"\n",
      "    class_name: ClassVar[str] = \"material sample derivation association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.MaterialSampleDerivationAssociation\n",
      "\n",
      "    id: Union[str, MaterialSampleDerivationAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, MaterialSampleId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, MaterialSampleDerivationAssociationId):\n",
      "            self.id = MaterialSampleDerivationAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, MaterialSampleId):\n",
      "            self.subject = MaterialSampleId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class MaterialSampleToDiseaseOrPhenotypicFeatureAssociation(Association):\n",
      "    \"\"\"\n",
      "    An association between a material sample and a disease or phenotype.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"MaterialSampleToDiseaseOrPhenotypicFeatureAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:MaterialSampleToDiseaseOrPhenotypicFeatureAssociation\"\n",
      "    class_name: ClassVar[str] = \"material sample to disease or phenotypic feature association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.MaterialSampleToDiseaseOrPhenotypicFeatureAssociation\n",
      "\n",
      "    id: Union[str, MaterialSampleToDiseaseOrPhenotypicFeatureAssociationId] = None\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    disease_context_qualifier: Optional[Union[str, DiseaseId]] = None\n",
      "    subject_specialization_qualifier: Optional[Union[str, URIorCURIE]] = None\n",
      "    object_specialization_qualifier: Optional[Union[str, URIorCURIE]] = None\n",
      "    anatomical_context_qualifier: Optional[str] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, MaterialSampleToDiseaseOrPhenotypicFeatureAssociationId):\n",
      "            self.id = MaterialSampleToDiseaseOrPhenotypicFeatureAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, NamedThingId):\n",
      "            self.subject = NamedThingId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        if self.disease_context_qualifier is not None and not isinstance(self.disease_context_qualifier, DiseaseId):\n",
      "            self.disease_context_qualifier = DiseaseId(self.disease_context_qualifier)\n",
      "\n",
      "        if self.subject_specialization_qualifier is not None and not isinstance(self.subject_specialization_qualifier, URIorCURIE):\n",
      "            self.subject_specialization_qualifier = URIorCURIE(self.subject_specialization_qualifier)\n",
      "\n",
      "        if self.object_specialization_qualifier is not None and not isinstance(self.object_specialization_qualifier, URIorCURIE):\n",
      "            self.object_specialization_qualifier = URIorCURIE(self.object_specialization_qualifier)\n",
      "\n",
      "        if self.anatomical_context_qualifier is not None and not isinstance(self.anatomical_context_qualifier, str):\n",
      "            self.anatomical_context_qualifier = str(self.anatomical_context_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class DiseaseToEntityAssociationMixin(YAMLRoot):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"DiseaseToEntityAssociationMixin\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:DiseaseToEntityAssociationMixin\"\n",
      "    class_name: ClassVar[str] = \"disease to entity association mixin\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.DiseaseToEntityAssociationMixin\n",
      "\n",
      "    subject: Union[str, DiseaseId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, DiseaseId):\n",
      "            self.subject = DiseaseId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class EntityToExposureEventAssociationMixin(YAMLRoot):\n",
      "    \"\"\"\n",
      "    An association between some entity and an exposure event.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"EntityToExposureEventAssociationMixin\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:EntityToExposureEventAssociationMixin\"\n",
      "    class_name: ClassVar[str] = \"entity to exposure event association mixin\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.EntityToExposureEventAssociationMixin\n",
      "\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, ExposureEventId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, NamedThingId):\n",
      "            self.subject = NamedThingId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, ExposureEventId):\n",
      "            self.object = ExposureEventId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class DiseaseToExposureEventAssociation(Association):\n",
      "    \"\"\"\n",
      "    An association between an exposure event and a disease.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"DiseaseToExposureEventAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:DiseaseToExposureEventAssociation\"\n",
      "    class_name: ClassVar[str] = \"disease to exposure event association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.DiseaseToExposureEventAssociation\n",
      "\n",
      "    id: Union[str, DiseaseToExposureEventAssociationId] = None\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, DiseaseToExposureEventAssociationId):\n",
      "            self.id = DiseaseToExposureEventAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, NamedThingId):\n",
      "            self.subject = NamedThingId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class EntityToOutcomeAssociationMixin(YAMLRoot):\n",
      "    \"\"\"\n",
      "    An association between some entity and an outcome\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"EntityToOutcomeAssociationMixin\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:EntityToOutcomeAssociationMixin\"\n",
      "    class_name: ClassVar[str] = \"entity to outcome association mixin\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.EntityToOutcomeAssociationMixin\n",
      "\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[dict, Outcome] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, NamedThingId):\n",
      "            self.subject = NamedThingId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, Outcome):\n",
      "            self.object = Outcome()\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ExposureEventToOutcomeAssociation(Association):\n",
      "    \"\"\"\n",
      "    An association between an exposure event and an outcome.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ExposureEventToOutcomeAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ExposureEventToOutcomeAssociation\"\n",
      "    class_name: ClassVar[str] = \"exposure event to outcome association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ExposureEventToOutcomeAssociation\n",
      "\n",
      "    id: Union[str, ExposureEventToOutcomeAssociationId] = None\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    population_context_qualifier: Optional[Union[str, PopulationOfIndividualOrganismsId]] = None\n",
      "    temporal_context_qualifier: Optional[Union[str, TimeType]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ExposureEventToOutcomeAssociationId):\n",
      "            self.id = ExposureEventToOutcomeAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, NamedThingId):\n",
      "            self.subject = NamedThingId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        if self.population_context_qualifier is not None and not isinstance(self.population_context_qualifier, PopulationOfIndividualOrganismsId):\n",
      "            self.population_context_qualifier = PopulationOfIndividualOrganismsId(self.population_context_qualifier)\n",
      "\n",
      "        if self.temporal_context_qualifier is not None and not isinstance(self.temporal_context_qualifier, TimeType):\n",
      "            self.temporal_context_qualifier = TimeType(self.temporal_context_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class FrequencyQualifierMixin(YAMLRoot):\n",
      "    \"\"\"\n",
      "    Qualifier for frequency type associations\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"FrequencyQualifierMixin\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:FrequencyQualifierMixin\"\n",
      "    class_name: ClassVar[str] = \"frequency qualifier mixin\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.FrequencyQualifierMixin\n",
      "\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    frequency_qualifier: Optional[Union[str, FrequencyValue]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, NamedThingId):\n",
      "            self.subject = NamedThingId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        if self.frequency_qualifier is not None and not isinstance(self.frequency_qualifier, FrequencyValue):\n",
      "            self.frequency_qualifier = FrequencyValue(self.frequency_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class EntityToFeatureOrDiseaseQualifiersMixin(FrequencyQualifierMixin):\n",
      "    \"\"\"\n",
      "    Qualifiers for entity to disease or phenotype associations.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"EntityToFeatureOrDiseaseQualifiersMixin\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:EntityToFeatureOrDiseaseQualifiersMixin\"\n",
      "    class_name: ClassVar[str] = \"entity to feature or disease qualifiers mixin\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.EntityToFeatureOrDiseaseQualifiersMixin\n",
      "\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    subject_aspect_qualifier: Optional[str] = None\n",
      "    subject_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    object_aspect_qualifier: Optional[str] = None\n",
      "    object_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    qualified_predicate: Optional[str] = None\n",
      "    disease_context_qualifier: Optional[Union[str, DiseaseId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self.subject_aspect_qualifier is not None and not isinstance(self.subject_aspect_qualifier, str):\n",
      "            self.subject_aspect_qualifier = str(self.subject_aspect_qualifier)\n",
      "\n",
      "        if self.subject_direction_qualifier is not None and not isinstance(self.subject_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.subject_direction_qualifier = DirectionQualifierEnum(self.subject_direction_qualifier)\n",
      "\n",
      "        if self.object_aspect_qualifier is not None and not isinstance(self.object_aspect_qualifier, str):\n",
      "            self.object_aspect_qualifier = str(self.object_aspect_qualifier)\n",
      "\n",
      "        if self.object_direction_qualifier is not None and not isinstance(self.object_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.object_direction_qualifier = DirectionQualifierEnum(self.object_direction_qualifier)\n",
      "\n",
      "        if self.qualified_predicate is not None and not isinstance(self.qualified_predicate, str):\n",
      "            self.qualified_predicate = str(self.qualified_predicate)\n",
      "\n",
      "        if self.disease_context_qualifier is not None and not isinstance(self.disease_context_qualifier, DiseaseId):\n",
      "            self.disease_context_qualifier = DiseaseId(self.disease_context_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class FeatureOrDiseaseQualifiersToEntityMixin(FrequencyQualifierMixin):\n",
      "    \"\"\"\n",
      "    Qualifiers for disease or phenotype to entity associations.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"FeatureOrDiseaseQualifiersToEntityMixin\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:FeatureOrDiseaseQualifiersToEntityMixin\"\n",
      "    class_name: ClassVar[str] = \"feature or disease qualifiers to entity mixin\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.FeatureOrDiseaseQualifiersToEntityMixin\n",
      "\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    subject_aspect_qualifier: Optional[str] = None\n",
      "    subject_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    object_aspect_qualifier: Optional[str] = None\n",
      "    object_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    qualified_predicate: Optional[str] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self.subject_aspect_qualifier is not None and not isinstance(self.subject_aspect_qualifier, str):\n",
      "            self.subject_aspect_qualifier = str(self.subject_aspect_qualifier)\n",
      "\n",
      "        if self.subject_direction_qualifier is not None and not isinstance(self.subject_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.subject_direction_qualifier = DirectionQualifierEnum(self.subject_direction_qualifier)\n",
      "\n",
      "        if self.object_aspect_qualifier is not None and not isinstance(self.object_aspect_qualifier, str):\n",
      "            self.object_aspect_qualifier = str(self.object_aspect_qualifier)\n",
      "\n",
      "        if self.object_direction_qualifier is not None and not isinstance(self.object_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.object_direction_qualifier = DirectionQualifierEnum(self.object_direction_qualifier)\n",
      "\n",
      "        if self.qualified_predicate is not None and not isinstance(self.qualified_predicate, str):\n",
      "            self.qualified_predicate = str(self.qualified_predicate)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class EntityToPhenotypicFeatureAssociationMixin(EntityToFeatureOrDiseaseQualifiersMixin):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"EntityToPhenotypicFeatureAssociationMixin\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:EntityToPhenotypicFeatureAssociationMixin\"\n",
      "    class_name: ClassVar[str] = \"entity to phenotypic feature association mixin\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.EntityToPhenotypicFeatureAssociationMixin\n",
      "\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, PhenotypicFeatureId] = None\n",
      "    sex_qualifier: Optional[Union[str, BiologicalSexId]] = None\n",
      "    has_count: Optional[int] = None\n",
      "    has_total: Optional[int] = None\n",
      "    has_quotient: Optional[float] = None\n",
      "    has_percentage: Optional[float] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, NamedThingId):\n",
      "            self.subject = NamedThingId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, PhenotypicFeatureId):\n",
      "            self.object = PhenotypicFeatureId(self.object)\n",
      "\n",
      "        if self.sex_qualifier is not None and not isinstance(self.sex_qualifier, BiologicalSexId):\n",
      "            self.sex_qualifier = BiologicalSexId(self.sex_qualifier)\n",
      "\n",
      "        if self.has_count is not None and not isinstance(self.has_count, int):\n",
      "            self.has_count = int(self.has_count)\n",
      "\n",
      "        if self.has_total is not None and not isinstance(self.has_total, int):\n",
      "            self.has_total = int(self.has_total)\n",
      "\n",
      "        if self.has_quotient is not None and not isinstance(self.has_quotient, float):\n",
      "            self.has_quotient = float(self.has_quotient)\n",
      "\n",
      "        if self.has_percentage is not None and not isinstance(self.has_percentage, float):\n",
      "            self.has_percentage = float(self.has_percentage)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class PhenotypicFeatureToEntityAssociationMixin(FeatureOrDiseaseQualifiersToEntityMixin):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"PhenotypicFeatureToEntityAssociationMixin\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:PhenotypicFeatureToEntityAssociationMixin\"\n",
      "    class_name: ClassVar[str] = \"phenotypic feature to entity association mixin\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.PhenotypicFeatureToEntityAssociationMixin\n",
      "\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    subject: Union[str, PhenotypicFeatureId] = None\n",
      "    sex_qualifier: Optional[Union[str, BiologicalSexId]] = None\n",
      "    has_count: Optional[int] = None\n",
      "    has_total: Optional[int] = None\n",
      "    has_quotient: Optional[float] = None\n",
      "    has_percentage: Optional[float] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, PhenotypicFeatureId):\n",
      "            self.subject = PhenotypicFeatureId(self.subject)\n",
      "\n",
      "        if self.sex_qualifier is not None and not isinstance(self.sex_qualifier, BiologicalSexId):\n",
      "            self.sex_qualifier = BiologicalSexId(self.sex_qualifier)\n",
      "\n",
      "        if self.has_count is not None and not isinstance(self.has_count, int):\n",
      "            self.has_count = int(self.has_count)\n",
      "\n",
      "        if self.has_total is not None and not isinstance(self.has_total, int):\n",
      "            self.has_total = int(self.has_total)\n",
      "\n",
      "        if self.has_quotient is not None and not isinstance(self.has_quotient, float):\n",
      "            self.has_quotient = float(self.has_quotient)\n",
      "\n",
      "        if self.has_percentage is not None and not isinstance(self.has_percentage, float):\n",
      "            self.has_percentage = float(self.has_percentage)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class PhenotypicFeatureToPhenotypicFeatureAssociation(Association):\n",
      "    \"\"\"\n",
      "    Association between two concept nodes of phenotypic character, qualified by the predicate used. This association\n",
      "    may typically be used to specify 'similar_to' or 'member_of' relationships.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"PhenotypicFeatureToPhenotypicFeatureAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:PhenotypicFeatureToPhenotypicFeatureAssociation\"\n",
      "    class_name: ClassVar[str] = \"phenotypic feature to phenotypic feature association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.PhenotypicFeatureToPhenotypicFeatureAssociation\n",
      "\n",
      "    id: Union[str, PhenotypicFeatureToPhenotypicFeatureAssociationId] = None\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    frequency_qualifier: Optional[Union[str, FrequencyValue]] = None\n",
      "    subject_aspect_qualifier: Optional[str] = None\n",
      "    subject_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    object_aspect_qualifier: Optional[str] = None\n",
      "    object_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    qualified_predicate: Optional[str] = None\n",
      "    sex_qualifier: Optional[Union[str, BiologicalSexId]] = None\n",
      "    disease_context_qualifier: Optional[Union[str, DiseaseId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, PhenotypicFeatureToPhenotypicFeatureAssociationId):\n",
      "            self.id = PhenotypicFeatureToPhenotypicFeatureAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, NamedThingId):\n",
      "            self.subject = NamedThingId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        if self.frequency_qualifier is not None and not isinstance(self.frequency_qualifier, FrequencyValue):\n",
      "            self.frequency_qualifier = FrequencyValue(self.frequency_qualifier)\n",
      "\n",
      "        if self.subject_aspect_qualifier is not None and not isinstance(self.subject_aspect_qualifier, str):\n",
      "            self.subject_aspect_qualifier = str(self.subject_aspect_qualifier)\n",
      "\n",
      "        if self.subject_direction_qualifier is not None and not isinstance(self.subject_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.subject_direction_qualifier = DirectionQualifierEnum(self.subject_direction_qualifier)\n",
      "\n",
      "        if self.object_aspect_qualifier is not None and not isinstance(self.object_aspect_qualifier, str):\n",
      "            self.object_aspect_qualifier = str(self.object_aspect_qualifier)\n",
      "\n",
      "        if self.object_direction_qualifier is not None and not isinstance(self.object_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.object_direction_qualifier = DirectionQualifierEnum(self.object_direction_qualifier)\n",
      "\n",
      "        if self.qualified_predicate is not None and not isinstance(self.qualified_predicate, str):\n",
      "            self.qualified_predicate = str(self.qualified_predicate)\n",
      "\n",
      "        if self.sex_qualifier is not None and not isinstance(self.sex_qualifier, BiologicalSexId):\n",
      "            self.sex_qualifier = BiologicalSexId(self.sex_qualifier)\n",
      "\n",
      "        if self.disease_context_qualifier is not None and not isinstance(self.disease_context_qualifier, DiseaseId):\n",
      "            self.disease_context_qualifier = DiseaseId(self.disease_context_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class InformationContentEntityToNamedThingAssociation(Association):\n",
      "    \"\"\"\n",
      "    association between a named thing and a information content entity where the specific context of the relationship\n",
      "    between that named thing and the publication is unknown. For example, model organisms databases often capture the\n",
      "    knowledge that a gene is found in a journal article, but not specifically the context in which that gene was\n",
      "    documented in the article. In these cases, this association with the accompanying predicate 'mentions' could be\n",
      "    used. Conversely, for more specific associations (like 'gene to disease association', the publication should be\n",
      "    captured as an edge property).\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"InformationContentEntityToNamedThingAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:InformationContentEntityToNamedThingAssociation\"\n",
      "    class_name: ClassVar[str] = \"information content entity to named thing association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.InformationContentEntityToNamedThingAssociation\n",
      "\n",
      "    id: Union[str, InformationContentEntityToNamedThingAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, InformationContentEntityToNamedThingAssociationId):\n",
      "            self.id = InformationContentEntityToNamedThingAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, NamedThingId):\n",
      "            self.subject = NamedThingId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class EntityToDiseaseAssociationMixin(EntityToFeatureOrDiseaseQualifiersMixin):\n",
      "    \"\"\"\n",
      "    mixin class for any association whose object (target node) is a disease\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"EntityToDiseaseAssociationMixin\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:EntityToDiseaseAssociationMixin\"\n",
      "    class_name: ClassVar[str] = \"entity to disease association mixin\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.EntityToDiseaseAssociationMixin\n",
      "\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, DiseaseId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, DiseaseId):\n",
      "            self.object = DiseaseId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class DiseaseOrPhenotypicFeatureToEntityAssociationMixin(YAMLRoot):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"DiseaseOrPhenotypicFeatureToEntityAssociationMixin\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:DiseaseOrPhenotypicFeatureToEntityAssociationMixin\"\n",
      "    class_name: ClassVar[str] = \"disease or phenotypic feature to entity association mixin\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.DiseaseOrPhenotypicFeatureToEntityAssociationMixin\n",
      "\n",
      "    subject: Union[str, DiseaseOrPhenotypicFeatureId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, DiseaseOrPhenotypicFeatureId):\n",
      "            self.subject = DiseaseOrPhenotypicFeatureId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class DiseaseOrPhenotypicFeatureToLocationAssociation(Association):\n",
      "    \"\"\"\n",
      "    An association between either a disease or a phenotypic feature and an anatomical entity, where the\n",
      "    disease/feature manifests in that site.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"DiseaseOrPhenotypicFeatureToLocationAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:DiseaseOrPhenotypicFeatureToLocationAssociation\"\n",
      "    class_name: ClassVar[str] = \"disease or phenotypic feature to location association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.DiseaseOrPhenotypicFeatureToLocationAssociation\n",
      "\n",
      "    id: Union[str, DiseaseOrPhenotypicFeatureToLocationAssociationId] = None\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    object: Union[str, AnatomicalEntityId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, DiseaseOrPhenotypicFeatureToLocationAssociationId):\n",
      "            self.id = DiseaseOrPhenotypicFeatureToLocationAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, NamedThingId):\n",
      "            self.subject = NamedThingId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, AnatomicalEntityId):\n",
      "            self.object = AnatomicalEntityId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class DiseaseOrPhenotypicFeatureToGeneticInheritanceAssociation(Association):\n",
      "    \"\"\"\n",
      "    An association between either a disease or a phenotypic feature and its mode of (genetic) inheritance.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"DiseaseOrPhenotypicFeatureToGeneticInheritanceAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:DiseaseOrPhenotypicFeatureToGeneticInheritanceAssociation\"\n",
      "    class_name: ClassVar[str] = \"disease or phenotypic feature to genetic inheritance association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.DiseaseOrPhenotypicFeatureToGeneticInheritanceAssociation\n",
      "\n",
      "    id: Union[str, DiseaseOrPhenotypicFeatureToGeneticInheritanceAssociationId] = None\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, GeneticInheritanceId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, DiseaseOrPhenotypicFeatureToGeneticInheritanceAssociationId):\n",
      "            self.id = DiseaseOrPhenotypicFeatureToGeneticInheritanceAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, NamedThingId):\n",
      "            self.subject = NamedThingId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, GeneticInheritanceId):\n",
      "            self.object = GeneticInheritanceId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class EntityToDiseaseOrPhenotypicFeatureAssociationMixin(YAMLRoot):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"EntityToDiseaseOrPhenotypicFeatureAssociationMixin\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:EntityToDiseaseOrPhenotypicFeatureAssociationMixin\"\n",
      "    class_name: ClassVar[str] = \"entity to disease or phenotypic feature association mixin\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.EntityToDiseaseOrPhenotypicFeatureAssociationMixin\n",
      "\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, DiseaseOrPhenotypicFeatureId] = None\n",
      "    disease_context_qualifier: Optional[Union[str, DiseaseId]] = None\n",
      "    subject_specialization_qualifier: Optional[Union[str, URIorCURIE]] = None\n",
      "    object_specialization_qualifier: Optional[Union[str, URIorCURIE]] = None\n",
      "    anatomical_context_qualifier: Optional[str] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, NamedThingId):\n",
      "            self.subject = NamedThingId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, DiseaseOrPhenotypicFeatureId):\n",
      "            self.object = DiseaseOrPhenotypicFeatureId(self.object)\n",
      "\n",
      "        if self.disease_context_qualifier is not None and not isinstance(self.disease_context_qualifier, DiseaseId):\n",
      "            self.disease_context_qualifier = DiseaseId(self.disease_context_qualifier)\n",
      "\n",
      "        if self.subject_specialization_qualifier is not None and not isinstance(self.subject_specialization_qualifier, URIorCURIE):\n",
      "            self.subject_specialization_qualifier = URIorCURIE(self.subject_specialization_qualifier)\n",
      "\n",
      "        if self.object_specialization_qualifier is not None and not isinstance(self.object_specialization_qualifier, URIorCURIE):\n",
      "            self.object_specialization_qualifier = URIorCURIE(self.object_specialization_qualifier)\n",
      "\n",
      "        if self.anatomical_context_qualifier is not None and not isinstance(self.anatomical_context_qualifier, str):\n",
      "            self.anatomical_context_qualifier = str(self.anatomical_context_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GenotypeToEntityAssociationMixin(YAMLRoot):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GenotypeToEntityAssociationMixin\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GenotypeToEntityAssociationMixin\"\n",
      "    class_name: ClassVar[str] = \"genotype to entity association mixin\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GenotypeToEntityAssociationMixin\n",
      "\n",
      "    subject: Union[str, GenotypeId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, GenotypeId):\n",
      "            self.subject = GenotypeId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GenotypeToPhenotypicFeatureAssociation(Association):\n",
      "    \"\"\"\n",
      "    Any association between one genotype and a phenotypic feature, where having the genotype confers the phenotype,\n",
      "    either in isolation or through environment\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GenotypeToPhenotypicFeatureAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GenotypeToPhenotypicFeatureAssociation\"\n",
      "    class_name: ClassVar[str] = \"genotype to phenotypic feature association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GenotypeToPhenotypicFeatureAssociation\n",
      "\n",
      "    id: Union[str, GenotypeToPhenotypicFeatureAssociationId] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    subject: Union[str, GenotypeId] = None\n",
      "    frequency_qualifier: Optional[Union[str, FrequencyValue]] = None\n",
      "    subject_aspect_qualifier: Optional[str] = None\n",
      "    subject_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    object_aspect_qualifier: Optional[str] = None\n",
      "    object_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    qualified_predicate: Optional[str] = None\n",
      "    disease_context_qualifier: Optional[Union[str, DiseaseId]] = None\n",
      "    sex_qualifier: Optional[Union[str, BiologicalSexId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GenotypeToPhenotypicFeatureAssociationId):\n",
      "            self.id = GenotypeToPhenotypicFeatureAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, GenotypeId):\n",
      "            self.subject = GenotypeId(self.subject)\n",
      "\n",
      "        if self.frequency_qualifier is not None and not isinstance(self.frequency_qualifier, FrequencyValue):\n",
      "            self.frequency_qualifier = FrequencyValue(self.frequency_qualifier)\n",
      "\n",
      "        if self.subject_aspect_qualifier is not None and not isinstance(self.subject_aspect_qualifier, str):\n",
      "            self.subject_aspect_qualifier = str(self.subject_aspect_qualifier)\n",
      "\n",
      "        if self.subject_direction_qualifier is not None and not isinstance(self.subject_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.subject_direction_qualifier = DirectionQualifierEnum(self.subject_direction_qualifier)\n",
      "\n",
      "        if self.object_aspect_qualifier is not None and not isinstance(self.object_aspect_qualifier, str):\n",
      "            self.object_aspect_qualifier = str(self.object_aspect_qualifier)\n",
      "\n",
      "        if self.object_direction_qualifier is not None and not isinstance(self.object_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.object_direction_qualifier = DirectionQualifierEnum(self.object_direction_qualifier)\n",
      "\n",
      "        if self.qualified_predicate is not None and not isinstance(self.qualified_predicate, str):\n",
      "            self.qualified_predicate = str(self.qualified_predicate)\n",
      "\n",
      "        if self.disease_context_qualifier is not None and not isinstance(self.disease_context_qualifier, DiseaseId):\n",
      "            self.disease_context_qualifier = DiseaseId(self.disease_context_qualifier)\n",
      "\n",
      "        if self.sex_qualifier is not None and not isinstance(self.sex_qualifier, BiologicalSexId):\n",
      "            self.sex_qualifier = BiologicalSexId(self.sex_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ExposureEventToPhenotypicFeatureAssociation(Association):\n",
      "    \"\"\"\n",
      "    Any association between an environment and a phenotypic feature, where being in the environment influences the\n",
      "    phenotype.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ExposureEventToPhenotypicFeatureAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ExposureEventToPhenotypicFeatureAssociation\"\n",
      "    class_name: ClassVar[str] = \"exposure event to phenotypic feature association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ExposureEventToPhenotypicFeatureAssociation\n",
      "\n",
      "    id: Union[str, ExposureEventToPhenotypicFeatureAssociationId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, ExposureEventId] = None\n",
      "    frequency_qualifier: Optional[Union[str, FrequencyValue]] = None\n",
      "    subject_aspect_qualifier: Optional[str] = None\n",
      "    subject_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    object_aspect_qualifier: Optional[str] = None\n",
      "    object_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    qualified_predicate: Optional[str] = None\n",
      "    disease_context_qualifier: Optional[Union[str, DiseaseId]] = None\n",
      "    sex_qualifier: Optional[Union[str, BiologicalSexId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ExposureEventToPhenotypicFeatureAssociationId):\n",
      "            self.id = ExposureEventToPhenotypicFeatureAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, ExposureEventId):\n",
      "            self.subject = ExposureEventId(self.subject)\n",
      "\n",
      "        if self.frequency_qualifier is not None and not isinstance(self.frequency_qualifier, FrequencyValue):\n",
      "            self.frequency_qualifier = FrequencyValue(self.frequency_qualifier)\n",
      "\n",
      "        if self.subject_aspect_qualifier is not None and not isinstance(self.subject_aspect_qualifier, str):\n",
      "            self.subject_aspect_qualifier = str(self.subject_aspect_qualifier)\n",
      "\n",
      "        if self.subject_direction_qualifier is not None and not isinstance(self.subject_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.subject_direction_qualifier = DirectionQualifierEnum(self.subject_direction_qualifier)\n",
      "\n",
      "        if self.object_aspect_qualifier is not None and not isinstance(self.object_aspect_qualifier, str):\n",
      "            self.object_aspect_qualifier = str(self.object_aspect_qualifier)\n",
      "\n",
      "        if self.object_direction_qualifier is not None and not isinstance(self.object_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.object_direction_qualifier = DirectionQualifierEnum(self.object_direction_qualifier)\n",
      "\n",
      "        if self.qualified_predicate is not None and not isinstance(self.qualified_predicate, str):\n",
      "            self.qualified_predicate = str(self.qualified_predicate)\n",
      "\n",
      "        if self.disease_context_qualifier is not None and not isinstance(self.disease_context_qualifier, DiseaseId):\n",
      "            self.disease_context_qualifier = DiseaseId(self.disease_context_qualifier)\n",
      "\n",
      "        if self.sex_qualifier is not None and not isinstance(self.sex_qualifier, BiologicalSexId):\n",
      "            self.sex_qualifier = BiologicalSexId(self.sex_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class DiseaseToPhenotypicFeatureAssociation(Association):\n",
      "    \"\"\"\n",
      "    An association between a disease and a phenotypic feature in which the phenotypic feature is associated with the\n",
      "    disease in some way.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"DiseaseToPhenotypicFeatureAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:DiseaseToPhenotypicFeatureAssociation\"\n",
      "    class_name: ClassVar[str] = \"disease to phenotypic feature association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.DiseaseToPhenotypicFeatureAssociation\n",
      "\n",
      "    id: Union[str, DiseaseToPhenotypicFeatureAssociationId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, DiseaseId] = None\n",
      "    object: Union[str, PhenotypicFeatureId] = None\n",
      "    onset_qualifier: Optional[Union[str, OnsetId]] = None\n",
      "    has_count: Optional[int] = None\n",
      "    has_total: Optional[int] = None\n",
      "    has_quotient: Optional[float] = None\n",
      "    has_percentage: Optional[float] = None\n",
      "    frequency_qualifier: Optional[Union[str, FrequencyValue]] = None\n",
      "    subject_aspect_qualifier: Optional[str] = None\n",
      "    subject_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    object_aspect_qualifier: Optional[str] = None\n",
      "    object_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    qualified_predicate: Optional[str] = None\n",
      "    disease_context_qualifier: Optional[Union[str, DiseaseId]] = None\n",
      "    sex_qualifier: Optional[Union[str, BiologicalSexId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, DiseaseToPhenotypicFeatureAssociationId):\n",
      "            self.id = DiseaseToPhenotypicFeatureAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, DiseaseId):\n",
      "            self.subject = DiseaseId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, PhenotypicFeatureId):\n",
      "            self.object = PhenotypicFeatureId(self.object)\n",
      "\n",
      "        if self.onset_qualifier is not None and not isinstance(self.onset_qualifier, OnsetId):\n",
      "            self.onset_qualifier = OnsetId(self.onset_qualifier)\n",
      "\n",
      "        if self.has_count is not None and not isinstance(self.has_count, int):\n",
      "            self.has_count = int(self.has_count)\n",
      "\n",
      "        if self.has_total is not None and not isinstance(self.has_total, int):\n",
      "            self.has_total = int(self.has_total)\n",
      "\n",
      "        if self.has_quotient is not None and not isinstance(self.has_quotient, float):\n",
      "            self.has_quotient = float(self.has_quotient)\n",
      "\n",
      "        if self.has_percentage is not None and not isinstance(self.has_percentage, float):\n",
      "            self.has_percentage = float(self.has_percentage)\n",
      "\n",
      "        if self.frequency_qualifier is not None and not isinstance(self.frequency_qualifier, FrequencyValue):\n",
      "            self.frequency_qualifier = FrequencyValue(self.frequency_qualifier)\n",
      "\n",
      "        if self.subject_aspect_qualifier is not None and not isinstance(self.subject_aspect_qualifier, str):\n",
      "            self.subject_aspect_qualifier = str(self.subject_aspect_qualifier)\n",
      "\n",
      "        if self.subject_direction_qualifier is not None and not isinstance(self.subject_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.subject_direction_qualifier = DirectionQualifierEnum(self.subject_direction_qualifier)\n",
      "\n",
      "        if self.object_aspect_qualifier is not None and not isinstance(self.object_aspect_qualifier, str):\n",
      "            self.object_aspect_qualifier = str(self.object_aspect_qualifier)\n",
      "\n",
      "        if self.object_direction_qualifier is not None and not isinstance(self.object_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.object_direction_qualifier = DirectionQualifierEnum(self.object_direction_qualifier)\n",
      "\n",
      "        if self.qualified_predicate is not None and not isinstance(self.qualified_predicate, str):\n",
      "            self.qualified_predicate = str(self.qualified_predicate)\n",
      "\n",
      "        if self.disease_context_qualifier is not None and not isinstance(self.disease_context_qualifier, DiseaseId):\n",
      "            self.disease_context_qualifier = DiseaseId(self.disease_context_qualifier)\n",
      "\n",
      "        if self.sex_qualifier is not None and not isinstance(self.sex_qualifier, BiologicalSexId):\n",
      "            self.sex_qualifier = BiologicalSexId(self.sex_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class CaseToPhenotypicFeatureAssociation(Association):\n",
      "    \"\"\"\n",
      "    An association between a case (e.g. individual patient) and a phenotypic feature in which the individual has or\n",
      "    has had the phenotype.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"CaseToPhenotypicFeatureAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:CaseToPhenotypicFeatureAssociation\"\n",
      "    class_name: ClassVar[str] = \"case to phenotypic feature association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.CaseToPhenotypicFeatureAssociation\n",
      "\n",
      "    id: Union[str, CaseToPhenotypicFeatureAssociationId] = None\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    frequency_qualifier: Optional[Union[str, FrequencyValue]] = None\n",
      "    subject_aspect_qualifier: Optional[str] = None\n",
      "    subject_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    object_aspect_qualifier: Optional[str] = None\n",
      "    object_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    qualified_predicate: Optional[str] = None\n",
      "    disease_context_qualifier: Optional[Union[str, DiseaseId]] = None\n",
      "    sex_qualifier: Optional[Union[str, BiologicalSexId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, CaseToPhenotypicFeatureAssociationId):\n",
      "            self.id = CaseToPhenotypicFeatureAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, NamedThingId):\n",
      "            self.subject = NamedThingId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        if self.frequency_qualifier is not None and not isinstance(self.frequency_qualifier, FrequencyValue):\n",
      "            self.frequency_qualifier = FrequencyValue(self.frequency_qualifier)\n",
      "\n",
      "        if self.subject_aspect_qualifier is not None and not isinstance(self.subject_aspect_qualifier, str):\n",
      "            self.subject_aspect_qualifier = str(self.subject_aspect_qualifier)\n",
      "\n",
      "        if self.subject_direction_qualifier is not None and not isinstance(self.subject_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.subject_direction_qualifier = DirectionQualifierEnum(self.subject_direction_qualifier)\n",
      "\n",
      "        if self.object_aspect_qualifier is not None and not isinstance(self.object_aspect_qualifier, str):\n",
      "            self.object_aspect_qualifier = str(self.object_aspect_qualifier)\n",
      "\n",
      "        if self.object_direction_qualifier is not None and not isinstance(self.object_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.object_direction_qualifier = DirectionQualifierEnum(self.object_direction_qualifier)\n",
      "\n",
      "        if self.qualified_predicate is not None and not isinstance(self.qualified_predicate, str):\n",
      "            self.qualified_predicate = str(self.qualified_predicate)\n",
      "\n",
      "        if self.disease_context_qualifier is not None and not isinstance(self.disease_context_qualifier, DiseaseId):\n",
      "            self.disease_context_qualifier = DiseaseId(self.disease_context_qualifier)\n",
      "\n",
      "        if self.sex_qualifier is not None and not isinstance(self.sex_qualifier, BiologicalSexId):\n",
      "            self.sex_qualifier = BiologicalSexId(self.sex_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class BehaviorToBehavioralFeatureAssociation(Association):\n",
      "    \"\"\"\n",
      "    An association between an mixture behavior and a behavioral feature manifested by the individual exhibited or has\n",
      "    exhibited the behavior.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"BehaviorToBehavioralFeatureAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:BehaviorToBehavioralFeatureAssociation\"\n",
      "    class_name: ClassVar[str] = \"behavior to behavioral feature association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.BehaviorToBehavioralFeatureAssociation\n",
      "\n",
      "    id: Union[str, BehaviorToBehavioralFeatureAssociationId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, BehaviorId] = None\n",
      "    object: Union[str, BehavioralFeatureId] = None\n",
      "    frequency_qualifier: Optional[Union[str, FrequencyValue]] = None\n",
      "    subject_aspect_qualifier: Optional[str] = None\n",
      "    subject_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    object_aspect_qualifier: Optional[str] = None\n",
      "    object_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    qualified_predicate: Optional[str] = None\n",
      "    disease_context_qualifier: Optional[Union[str, DiseaseId]] = None\n",
      "    sex_qualifier: Optional[Union[str, BiologicalSexId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, BehaviorToBehavioralFeatureAssociationId):\n",
      "            self.id = BehaviorToBehavioralFeatureAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, BehaviorId):\n",
      "            self.subject = BehaviorId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, BehavioralFeatureId):\n",
      "            self.object = BehavioralFeatureId(self.object)\n",
      "\n",
      "        if self.frequency_qualifier is not None and not isinstance(self.frequency_qualifier, FrequencyValue):\n",
      "            self.frequency_qualifier = FrequencyValue(self.frequency_qualifier)\n",
      "\n",
      "        if self.subject_aspect_qualifier is not None and not isinstance(self.subject_aspect_qualifier, str):\n",
      "            self.subject_aspect_qualifier = str(self.subject_aspect_qualifier)\n",
      "\n",
      "        if self.subject_direction_qualifier is not None and not isinstance(self.subject_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.subject_direction_qualifier = DirectionQualifierEnum(self.subject_direction_qualifier)\n",
      "\n",
      "        if self.object_aspect_qualifier is not None and not isinstance(self.object_aspect_qualifier, str):\n",
      "            self.object_aspect_qualifier = str(self.object_aspect_qualifier)\n",
      "\n",
      "        if self.object_direction_qualifier is not None and not isinstance(self.object_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.object_direction_qualifier = DirectionQualifierEnum(self.object_direction_qualifier)\n",
      "\n",
      "        if self.qualified_predicate is not None and not isinstance(self.qualified_predicate, str):\n",
      "            self.qualified_predicate = str(self.qualified_predicate)\n",
      "\n",
      "        if self.disease_context_qualifier is not None and not isinstance(self.disease_context_qualifier, DiseaseId):\n",
      "            self.disease_context_qualifier = DiseaseId(self.disease_context_qualifier)\n",
      "\n",
      "        if self.sex_qualifier is not None and not isinstance(self.sex_qualifier, BiologicalSexId):\n",
      "            self.sex_qualifier = BiologicalSexId(self.sex_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GeneToEntityAssociationMixin(YAMLRoot):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GeneToEntityAssociationMixin\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GeneToEntityAssociationMixin\"\n",
      "    class_name: ClassVar[str] = \"gene to entity association mixin\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GeneToEntityAssociationMixin\n",
      "\n",
      "    subject: Union[dict, GeneOrGeneProduct] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, GeneOrGeneProduct):\n",
      "            self.subject = GeneOrGeneProduct(**as_dict(self.subject))\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class VariantToEntityAssociationMixin(YAMLRoot):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"VariantToEntityAssociationMixin\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:VariantToEntityAssociationMixin\"\n",
      "    class_name: ClassVar[str] = \"variant to entity association mixin\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.VariantToEntityAssociationMixin\n",
      "\n",
      "    subject: Union[str, SequenceVariantId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, SequenceVariantId):\n",
      "            self.subject = SequenceVariantId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GeneToDiseaseOrPhenotypicFeatureAssociation(Association):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GeneToDiseaseOrPhenotypicFeatureAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GeneToDiseaseOrPhenotypicFeatureAssociation\"\n",
      "    class_name: ClassVar[str] = \"gene to disease or phenotypic feature association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GeneToDiseaseOrPhenotypicFeatureAssociation\n",
      "\n",
      "    id: Union[str, GeneToDiseaseOrPhenotypicFeatureAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[dict, GeneOrGeneProduct] = None\n",
      "    object: Union[str, DiseaseOrPhenotypicFeatureId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    subject_aspect_qualifier: Optional[Union[str, \"GeneOrGeneProductOrChemicalEntityAspectEnum\"]] = None\n",
      "    object_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    frequency_qualifier: Optional[Union[str, FrequencyValue]] = None\n",
      "    subject_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    object_aspect_qualifier: Optional[str] = None\n",
      "    qualified_predicate: Optional[str] = None\n",
      "    disease_context_qualifier: Optional[Union[str, DiseaseId]] = None\n",
      "    sex_qualifier: Optional[Union[str, BiologicalSexId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GeneToDiseaseOrPhenotypicFeatureAssociationId):\n",
      "            self.id = GeneToDiseaseOrPhenotypicFeatureAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, GeneOrGeneProduct):\n",
      "            self.subject = GeneOrGeneProduct(**as_dict(self.subject))\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, DiseaseOrPhenotypicFeatureId):\n",
      "            self.object = DiseaseOrPhenotypicFeatureId(self.object)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self.subject_aspect_qualifier is not None and not isinstance(self.subject_aspect_qualifier, GeneOrGeneProductOrChemicalEntityAspectEnum):\n",
      "            self.subject_aspect_qualifier = GeneOrGeneProductOrChemicalEntityAspectEnum(self.subject_aspect_qualifier)\n",
      "\n",
      "        if self.object_direction_qualifier is not None and not isinstance(self.object_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.object_direction_qualifier = DirectionQualifierEnum(self.object_direction_qualifier)\n",
      "\n",
      "        if self.frequency_qualifier is not None and not isinstance(self.frequency_qualifier, FrequencyValue):\n",
      "            self.frequency_qualifier = FrequencyValue(self.frequency_qualifier)\n",
      "\n",
      "        if self.subject_direction_qualifier is not None and not isinstance(self.subject_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.subject_direction_qualifier = DirectionQualifierEnum(self.subject_direction_qualifier)\n",
      "\n",
      "        if self.object_aspect_qualifier is not None and not isinstance(self.object_aspect_qualifier, str):\n",
      "            self.object_aspect_qualifier = str(self.object_aspect_qualifier)\n",
      "\n",
      "        if self.qualified_predicate is not None and not isinstance(self.qualified_predicate, str):\n",
      "            self.qualified_predicate = str(self.qualified_predicate)\n",
      "\n",
      "        if self.disease_context_qualifier is not None and not isinstance(self.disease_context_qualifier, DiseaseId):\n",
      "            self.disease_context_qualifier = DiseaseId(self.disease_context_qualifier)\n",
      "\n",
      "        if self.sex_qualifier is not None and not isinstance(self.sex_qualifier, BiologicalSexId):\n",
      "            self.sex_qualifier = BiologicalSexId(self.sex_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GeneToPhenotypicFeatureAssociation(GeneToDiseaseOrPhenotypicFeatureAssociation):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GeneToPhenotypicFeatureAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GeneToPhenotypicFeatureAssociation\"\n",
      "    class_name: ClassVar[str] = \"gene to phenotypic feature association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GeneToPhenotypicFeatureAssociation\n",
      "\n",
      "    id: Union[str, GeneToPhenotypicFeatureAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    subject: Union[dict, GeneOrGeneProduct] = None\n",
      "    object: Union[str, PhenotypicFeatureId] = None\n",
      "    frequency_qualifier: Optional[Union[str, FrequencyValue]] = None\n",
      "    subject_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    object_aspect_qualifier: Optional[str] = None\n",
      "    qualified_predicate: Optional[str] = None\n",
      "    disease_context_qualifier: Optional[Union[str, DiseaseId]] = None\n",
      "    sex_qualifier: Optional[Union[str, BiologicalSexId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GeneToPhenotypicFeatureAssociationId):\n",
      "            self.id = GeneToPhenotypicFeatureAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, GeneOrGeneProduct):\n",
      "            self.subject = GeneOrGeneProduct(**as_dict(self.subject))\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, PhenotypicFeatureId):\n",
      "            self.object = PhenotypicFeatureId(self.object)\n",
      "\n",
      "        if self.frequency_qualifier is not None and not isinstance(self.frequency_qualifier, FrequencyValue):\n",
      "            self.frequency_qualifier = FrequencyValue(self.frequency_qualifier)\n",
      "\n",
      "        if self.subject_direction_qualifier is not None and not isinstance(self.subject_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.subject_direction_qualifier = DirectionQualifierEnum(self.subject_direction_qualifier)\n",
      "\n",
      "        if self.object_aspect_qualifier is not None and not isinstance(self.object_aspect_qualifier, str):\n",
      "            self.object_aspect_qualifier = str(self.object_aspect_qualifier)\n",
      "\n",
      "        if self.qualified_predicate is not None and not isinstance(self.qualified_predicate, str):\n",
      "            self.qualified_predicate = str(self.qualified_predicate)\n",
      "\n",
      "        if self.disease_context_qualifier is not None and not isinstance(self.disease_context_qualifier, DiseaseId):\n",
      "            self.disease_context_qualifier = DiseaseId(self.disease_context_qualifier)\n",
      "\n",
      "        if self.sex_qualifier is not None and not isinstance(self.sex_qualifier, BiologicalSexId):\n",
      "            self.sex_qualifier = BiologicalSexId(self.sex_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GeneToDiseaseAssociation(GeneToDiseaseOrPhenotypicFeatureAssociation):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GeneToDiseaseAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GeneToDiseaseAssociation\"\n",
      "    class_name: ClassVar[str] = \"gene to disease association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GeneToDiseaseAssociation\n",
      "\n",
      "    id: Union[str, GeneToDiseaseAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    subject: Union[dict, GeneOrGeneProduct] = None\n",
      "    object: Union[str, DiseaseId] = None\n",
      "    frequency_qualifier: Optional[Union[str, FrequencyValue]] = None\n",
      "    subject_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    object_aspect_qualifier: Optional[str] = None\n",
      "    qualified_predicate: Optional[str] = None\n",
      "    disease_context_qualifier: Optional[Union[str, DiseaseId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GeneToDiseaseAssociationId):\n",
      "            self.id = GeneToDiseaseAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, GeneOrGeneProduct):\n",
      "            self.subject = GeneOrGeneProduct(**as_dict(self.subject))\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, DiseaseId):\n",
      "            self.object = DiseaseId(self.object)\n",
      "\n",
      "        if self.frequency_qualifier is not None and not isinstance(self.frequency_qualifier, FrequencyValue):\n",
      "            self.frequency_qualifier = FrequencyValue(self.frequency_qualifier)\n",
      "\n",
      "        if self.subject_direction_qualifier is not None and not isinstance(self.subject_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.subject_direction_qualifier = DirectionQualifierEnum(self.subject_direction_qualifier)\n",
      "\n",
      "        if self.object_aspect_qualifier is not None and not isinstance(self.object_aspect_qualifier, str):\n",
      "            self.object_aspect_qualifier = str(self.object_aspect_qualifier)\n",
      "\n",
      "        if self.qualified_predicate is not None and not isinstance(self.qualified_predicate, str):\n",
      "            self.qualified_predicate = str(self.qualified_predicate)\n",
      "\n",
      "        if self.disease_context_qualifier is not None and not isinstance(self.disease_context_qualifier, DiseaseId):\n",
      "            self.disease_context_qualifier = DiseaseId(self.disease_context_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class CausalGeneToDiseaseAssociation(GeneToDiseaseAssociation):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"CausalGeneToDiseaseAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:CausalGeneToDiseaseAssociation\"\n",
      "    class_name: ClassVar[str] = \"causal gene to disease association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.CausalGeneToDiseaseAssociation\n",
      "\n",
      "    id: Union[str, CausalGeneToDiseaseAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    subject: Union[dict, GeneOrGeneProduct] = None\n",
      "    object: Union[str, DiseaseId] = None\n",
      "    frequency_qualifier: Optional[Union[str, FrequencyValue]] = None\n",
      "    subject_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    object_aspect_qualifier: Optional[str] = None\n",
      "    qualified_predicate: Optional[str] = None\n",
      "    disease_context_qualifier: Optional[Union[str, DiseaseId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, CausalGeneToDiseaseAssociationId):\n",
      "            self.id = CausalGeneToDiseaseAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, GeneOrGeneProduct):\n",
      "            self.subject = GeneOrGeneProduct(**as_dict(self.subject))\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, DiseaseId):\n",
      "            self.object = DiseaseId(self.object)\n",
      "\n",
      "        if self.frequency_qualifier is not None and not isinstance(self.frequency_qualifier, FrequencyValue):\n",
      "            self.frequency_qualifier = FrequencyValue(self.frequency_qualifier)\n",
      "\n",
      "        if self.subject_direction_qualifier is not None and not isinstance(self.subject_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.subject_direction_qualifier = DirectionQualifierEnum(self.subject_direction_qualifier)\n",
      "\n",
      "        if self.object_aspect_qualifier is not None and not isinstance(self.object_aspect_qualifier, str):\n",
      "            self.object_aspect_qualifier = str(self.object_aspect_qualifier)\n",
      "\n",
      "        if self.qualified_predicate is not None and not isinstance(self.qualified_predicate, str):\n",
      "            self.qualified_predicate = str(self.qualified_predicate)\n",
      "\n",
      "        if self.disease_context_qualifier is not None and not isinstance(self.disease_context_qualifier, DiseaseId):\n",
      "            self.disease_context_qualifier = DiseaseId(self.disease_context_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class CorrelatedGeneToDiseaseAssociation(GeneToDiseaseAssociation):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"CorrelatedGeneToDiseaseAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:CorrelatedGeneToDiseaseAssociation\"\n",
      "    class_name: ClassVar[str] = \"correlated gene to disease association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.CorrelatedGeneToDiseaseAssociation\n",
      "\n",
      "    id: Union[str, CorrelatedGeneToDiseaseAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    subject: Union[dict, GeneOrGeneProduct] = None\n",
      "    object: Union[str, DiseaseId] = None\n",
      "    frequency_qualifier: Optional[Union[str, FrequencyValue]] = None\n",
      "    subject_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    object_aspect_qualifier: Optional[str] = None\n",
      "    qualified_predicate: Optional[str] = None\n",
      "    disease_context_qualifier: Optional[Union[str, DiseaseId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, CorrelatedGeneToDiseaseAssociationId):\n",
      "            self.id = CorrelatedGeneToDiseaseAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, GeneOrGeneProduct):\n",
      "            self.subject = GeneOrGeneProduct(**as_dict(self.subject))\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, DiseaseId):\n",
      "            self.object = DiseaseId(self.object)\n",
      "\n",
      "        if self.frequency_qualifier is not None and not isinstance(self.frequency_qualifier, FrequencyValue):\n",
      "            self.frequency_qualifier = FrequencyValue(self.frequency_qualifier)\n",
      "\n",
      "        if self.subject_direction_qualifier is not None and not isinstance(self.subject_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.subject_direction_qualifier = DirectionQualifierEnum(self.subject_direction_qualifier)\n",
      "\n",
      "        if self.object_aspect_qualifier is not None and not isinstance(self.object_aspect_qualifier, str):\n",
      "            self.object_aspect_qualifier = str(self.object_aspect_qualifier)\n",
      "\n",
      "        if self.qualified_predicate is not None and not isinstance(self.qualified_predicate, str):\n",
      "            self.qualified_predicate = str(self.qualified_predicate)\n",
      "\n",
      "        if self.disease_context_qualifier is not None and not isinstance(self.disease_context_qualifier, DiseaseId):\n",
      "            self.disease_context_qualifier = DiseaseId(self.disease_context_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class DruggableGeneToDiseaseAssociation(GeneToDiseaseAssociation):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"DruggableGeneToDiseaseAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:DruggableGeneToDiseaseAssociation\"\n",
      "    class_name: ClassVar[str] = \"druggable gene to disease association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.DruggableGeneToDiseaseAssociation\n",
      "\n",
      "    id: Union[str, DruggableGeneToDiseaseAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    object: Union[str, DiseaseId] = None\n",
      "    subject: Union[dict, GeneOrGeneProduct] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    frequency_qualifier: Optional[Union[str, FrequencyValue]] = None\n",
      "    subject_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    object_aspect_qualifier: Optional[str] = None\n",
      "    qualified_predicate: Optional[str] = None\n",
      "    disease_context_qualifier: Optional[Union[str, DiseaseId]] = None\n",
      "    has_evidence: Optional[Union[Union[str, \"DruggableGeneCategoryEnum\"], List[Union[str, \"DruggableGeneCategoryEnum\"]]]] = empty_list()\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, DruggableGeneToDiseaseAssociationId):\n",
      "            self.id = DruggableGeneToDiseaseAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, GeneOrGeneProduct):\n",
      "            self.subject = GeneOrGeneProduct(**as_dict(self.subject))\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self.frequency_qualifier is not None and not isinstance(self.frequency_qualifier, FrequencyValue):\n",
      "            self.frequency_qualifier = FrequencyValue(self.frequency_qualifier)\n",
      "\n",
      "        if self.subject_direction_qualifier is not None and not isinstance(self.subject_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.subject_direction_qualifier = DirectionQualifierEnum(self.subject_direction_qualifier)\n",
      "\n",
      "        if self.object_aspect_qualifier is not None and not isinstance(self.object_aspect_qualifier, str):\n",
      "            self.object_aspect_qualifier = str(self.object_aspect_qualifier)\n",
      "\n",
      "        if self.qualified_predicate is not None and not isinstance(self.qualified_predicate, str):\n",
      "            self.qualified_predicate = str(self.qualified_predicate)\n",
      "\n",
      "        if self.disease_context_qualifier is not None and not isinstance(self.disease_context_qualifier, DiseaseId):\n",
      "            self.disease_context_qualifier = DiseaseId(self.disease_context_qualifier)\n",
      "\n",
      "        if not isinstance(self.has_evidence, list):\n",
      "            self.has_evidence = [self.has_evidence] if self.has_evidence is not None else []\n",
      "        self.has_evidence = [v if isinstance(v, DruggableGeneCategoryEnum) else DruggableGeneCategoryEnum(v) for v in self.has_evidence]\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class PhenotypicFeatureToDiseaseAssociation(Association):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"PhenotypicFeatureToDiseaseAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:PhenotypicFeatureToDiseaseAssociation\"\n",
      "    class_name: ClassVar[str] = \"phenotypic feature to disease association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.PhenotypicFeatureToDiseaseAssociation\n",
      "\n",
      "    id: Union[str, PhenotypicFeatureToDiseaseAssociationId] = None\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    frequency_qualifier: Optional[Union[str, FrequencyValue]] = None\n",
      "    subject_aspect_qualifier: Optional[str] = None\n",
      "    subject_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    object_aspect_qualifier: Optional[str] = None\n",
      "    object_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    qualified_predicate: Optional[str] = None\n",
      "    disease_context_qualifier: Optional[Union[str, DiseaseId]] = None\n",
      "    sex_qualifier: Optional[Union[str, BiologicalSexId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, PhenotypicFeatureToDiseaseAssociationId):\n",
      "            self.id = PhenotypicFeatureToDiseaseAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, NamedThingId):\n",
      "            self.subject = NamedThingId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self.frequency_qualifier is not None and not isinstance(self.frequency_qualifier, FrequencyValue):\n",
      "            self.frequency_qualifier = FrequencyValue(self.frequency_qualifier)\n",
      "\n",
      "        if self.subject_aspect_qualifier is not None and not isinstance(self.subject_aspect_qualifier, str):\n",
      "            self.subject_aspect_qualifier = str(self.subject_aspect_qualifier)\n",
      "\n",
      "        if self.subject_direction_qualifier is not None and not isinstance(self.subject_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.subject_direction_qualifier = DirectionQualifierEnum(self.subject_direction_qualifier)\n",
      "\n",
      "        if self.object_aspect_qualifier is not None and not isinstance(self.object_aspect_qualifier, str):\n",
      "            self.object_aspect_qualifier = str(self.object_aspect_qualifier)\n",
      "\n",
      "        if self.object_direction_qualifier is not None and not isinstance(self.object_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.object_direction_qualifier = DirectionQualifierEnum(self.object_direction_qualifier)\n",
      "\n",
      "        if self.qualified_predicate is not None and not isinstance(self.qualified_predicate, str):\n",
      "            self.qualified_predicate = str(self.qualified_predicate)\n",
      "\n",
      "        if self.disease_context_qualifier is not None and not isinstance(self.disease_context_qualifier, DiseaseId):\n",
      "            self.disease_context_qualifier = DiseaseId(self.disease_context_qualifier)\n",
      "\n",
      "        if self.sex_qualifier is not None and not isinstance(self.sex_qualifier, BiologicalSexId):\n",
      "            self.sex_qualifier = BiologicalSexId(self.sex_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class VariantToGeneAssociation(Association):\n",
      "    \"\"\"\n",
      "    An association between a variant and a gene, where the variant has a genetic association with the gene (i.e. is in\n",
      "    linkage disequilibrium)\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"VariantToGeneAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:VariantToGeneAssociation\"\n",
      "    class_name: ClassVar[str] = \"variant to gene association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.VariantToGeneAssociation\n",
      "\n",
      "    id: Union[str, VariantToGeneAssociationId] = None\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    object: Union[str, GeneId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, VariantToGeneAssociationId):\n",
      "            self.id = VariantToGeneAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, NamedThingId):\n",
      "            self.subject = NamedThingId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, GeneId):\n",
      "            self.object = GeneId(self.object)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class VariantToGeneExpressionAssociation(VariantToGeneAssociation):\n",
      "    \"\"\"\n",
      "    An association between a variant and expression of a gene (i.e. e-QTL)\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"VariantToGeneExpressionAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:VariantToGeneExpressionAssociation\"\n",
      "    class_name: ClassVar[str] = \"variant to gene expression association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.VariantToGeneExpressionAssociation\n",
      "\n",
      "    id: Union[str, VariantToGeneExpressionAssociationId] = None\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    object: Union[str, GeneId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    quantifier_qualifier: Optional[Union[str, OntologyClassId]] = None\n",
      "    expression_site: Optional[Union[str, AnatomicalEntityId]] = None\n",
      "    stage_qualifier: Optional[Union[str, LifeStageId]] = None\n",
      "    phenotypic_state: Optional[Union[str, DiseaseOrPhenotypicFeatureId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, VariantToGeneExpressionAssociationId):\n",
      "            self.id = VariantToGeneExpressionAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self.quantifier_qualifier is not None and not isinstance(self.quantifier_qualifier, OntologyClassId):\n",
      "            self.quantifier_qualifier = OntologyClassId(self.quantifier_qualifier)\n",
      "\n",
      "        if self.expression_site is not None and not isinstance(self.expression_site, AnatomicalEntityId):\n",
      "            self.expression_site = AnatomicalEntityId(self.expression_site)\n",
      "\n",
      "        if self.stage_qualifier is not None and not isinstance(self.stage_qualifier, LifeStageId):\n",
      "            self.stage_qualifier = LifeStageId(self.stage_qualifier)\n",
      "\n",
      "        if self.phenotypic_state is not None and not isinstance(self.phenotypic_state, DiseaseOrPhenotypicFeatureId):\n",
      "            self.phenotypic_state = DiseaseOrPhenotypicFeatureId(self.phenotypic_state)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class VariantToPopulationAssociation(Association):\n",
      "    \"\"\"\n",
      "    An association between a variant and a population, where the variant has particular frequency in the population\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"VariantToPopulationAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:VariantToPopulationAssociation\"\n",
      "    class_name: ClassVar[str] = \"variant to population association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.VariantToPopulationAssociation\n",
      "\n",
      "    id: Union[str, VariantToPopulationAssociationId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, SequenceVariantId] = None\n",
      "    object: Union[str, PopulationOfIndividualOrganismsId] = None\n",
      "    has_quotient: Optional[float] = None\n",
      "    has_count: Optional[int] = None\n",
      "    has_total: Optional[int] = None\n",
      "    has_percentage: Optional[float] = None\n",
      "    frequency_qualifier: Optional[Union[str, FrequencyValue]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, VariantToPopulationAssociationId):\n",
      "            self.id = VariantToPopulationAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, SequenceVariantId):\n",
      "            self.subject = SequenceVariantId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, PopulationOfIndividualOrganismsId):\n",
      "            self.object = PopulationOfIndividualOrganismsId(self.object)\n",
      "\n",
      "        if self.has_quotient is not None and not isinstance(self.has_quotient, float):\n",
      "            self.has_quotient = float(self.has_quotient)\n",
      "\n",
      "        if self.has_count is not None and not isinstance(self.has_count, int):\n",
      "            self.has_count = int(self.has_count)\n",
      "\n",
      "        if self.has_total is not None and not isinstance(self.has_total, int):\n",
      "            self.has_total = int(self.has_total)\n",
      "\n",
      "        if self.has_percentage is not None and not isinstance(self.has_percentage, float):\n",
      "            self.has_percentage = float(self.has_percentage)\n",
      "\n",
      "        if self.frequency_qualifier is not None and not isinstance(self.frequency_qualifier, FrequencyValue):\n",
      "            self.frequency_qualifier = FrequencyValue(self.frequency_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class PopulationToPopulationAssociation(Association):\n",
      "    \"\"\"\n",
      "    An association between a two populations\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"PopulationToPopulationAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:PopulationToPopulationAssociation\"\n",
      "    class_name: ClassVar[str] = \"population to population association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.PopulationToPopulationAssociation\n",
      "\n",
      "    id: Union[str, PopulationToPopulationAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, PopulationOfIndividualOrganismsId] = None\n",
      "    object: Union[str, PopulationOfIndividualOrganismsId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, PopulationToPopulationAssociationId):\n",
      "            self.id = PopulationToPopulationAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, PopulationOfIndividualOrganismsId):\n",
      "            self.subject = PopulationOfIndividualOrganismsId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, PopulationOfIndividualOrganismsId):\n",
      "            self.object = PopulationOfIndividualOrganismsId(self.object)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class VariantToPhenotypicFeatureAssociation(Association):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"VariantToPhenotypicFeatureAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:VariantToPhenotypicFeatureAssociation\"\n",
      "    class_name: ClassVar[str] = \"variant to phenotypic feature association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.VariantToPhenotypicFeatureAssociation\n",
      "\n",
      "    id: Union[str, VariantToPhenotypicFeatureAssociationId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, SequenceVariantId] = None\n",
      "    frequency_qualifier: Optional[Union[str, FrequencyValue]] = None\n",
      "    subject_aspect_qualifier: Optional[str] = None\n",
      "    subject_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    object_aspect_qualifier: Optional[str] = None\n",
      "    object_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    qualified_predicate: Optional[str] = None\n",
      "    disease_context_qualifier: Optional[Union[str, DiseaseId]] = None\n",
      "    sex_qualifier: Optional[Union[str, BiologicalSexId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, VariantToPhenotypicFeatureAssociationId):\n",
      "            self.id = VariantToPhenotypicFeatureAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, SequenceVariantId):\n",
      "            self.subject = SequenceVariantId(self.subject)\n",
      "\n",
      "        if self.frequency_qualifier is not None and not isinstance(self.frequency_qualifier, FrequencyValue):\n",
      "            self.frequency_qualifier = FrequencyValue(self.frequency_qualifier)\n",
      "\n",
      "        if self.subject_aspect_qualifier is not None and not isinstance(self.subject_aspect_qualifier, str):\n",
      "            self.subject_aspect_qualifier = str(self.subject_aspect_qualifier)\n",
      "\n",
      "        if self.subject_direction_qualifier is not None and not isinstance(self.subject_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.subject_direction_qualifier = DirectionQualifierEnum(self.subject_direction_qualifier)\n",
      "\n",
      "        if self.object_aspect_qualifier is not None and not isinstance(self.object_aspect_qualifier, str):\n",
      "            self.object_aspect_qualifier = str(self.object_aspect_qualifier)\n",
      "\n",
      "        if self.object_direction_qualifier is not None and not isinstance(self.object_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.object_direction_qualifier = DirectionQualifierEnum(self.object_direction_qualifier)\n",
      "\n",
      "        if self.qualified_predicate is not None and not isinstance(self.qualified_predicate, str):\n",
      "            self.qualified_predicate = str(self.qualified_predicate)\n",
      "\n",
      "        if self.disease_context_qualifier is not None and not isinstance(self.disease_context_qualifier, DiseaseId):\n",
      "            self.disease_context_qualifier = DiseaseId(self.disease_context_qualifier)\n",
      "\n",
      "        if self.sex_qualifier is not None and not isinstance(self.sex_qualifier, BiologicalSexId):\n",
      "            self.sex_qualifier = BiologicalSexId(self.sex_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class VariantToDiseaseAssociation(Association):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"VariantToDiseaseAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:VariantToDiseaseAssociation\"\n",
      "    class_name: ClassVar[str] = \"variant to disease association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.VariantToDiseaseAssociation\n",
      "\n",
      "    id: Union[str, VariantToDiseaseAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    frequency_qualifier: Optional[Union[str, FrequencyValue]] = None\n",
      "    subject_aspect_qualifier: Optional[str] = None\n",
      "    subject_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    object_aspect_qualifier: Optional[str] = None\n",
      "    object_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    qualified_predicate: Optional[str] = None\n",
      "    disease_context_qualifier: Optional[Union[str, DiseaseId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, VariantToDiseaseAssociationId):\n",
      "            self.id = VariantToDiseaseAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, NamedThingId):\n",
      "            self.subject = NamedThingId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        if self.frequency_qualifier is not None and not isinstance(self.frequency_qualifier, FrequencyValue):\n",
      "            self.frequency_qualifier = FrequencyValue(self.frequency_qualifier)\n",
      "\n",
      "        if self.subject_aspect_qualifier is not None and not isinstance(self.subject_aspect_qualifier, str):\n",
      "            self.subject_aspect_qualifier = str(self.subject_aspect_qualifier)\n",
      "\n",
      "        if self.subject_direction_qualifier is not None and not isinstance(self.subject_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.subject_direction_qualifier = DirectionQualifierEnum(self.subject_direction_qualifier)\n",
      "\n",
      "        if self.object_aspect_qualifier is not None and not isinstance(self.object_aspect_qualifier, str):\n",
      "            self.object_aspect_qualifier = str(self.object_aspect_qualifier)\n",
      "\n",
      "        if self.object_direction_qualifier is not None and not isinstance(self.object_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.object_direction_qualifier = DirectionQualifierEnum(self.object_direction_qualifier)\n",
      "\n",
      "        if self.qualified_predicate is not None and not isinstance(self.qualified_predicate, str):\n",
      "            self.qualified_predicate = str(self.qualified_predicate)\n",
      "\n",
      "        if self.disease_context_qualifier is not None and not isinstance(self.disease_context_qualifier, DiseaseId):\n",
      "            self.disease_context_qualifier = DiseaseId(self.disease_context_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GenotypeToDiseaseAssociation(Association):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GenotypeToDiseaseAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GenotypeToDiseaseAssociation\"\n",
      "    class_name: ClassVar[str] = \"genotype to disease association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GenotypeToDiseaseAssociation\n",
      "\n",
      "    id: Union[str, GenotypeToDiseaseAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    frequency_qualifier: Optional[Union[str, FrequencyValue]] = None\n",
      "    subject_aspect_qualifier: Optional[str] = None\n",
      "    subject_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    object_aspect_qualifier: Optional[str] = None\n",
      "    object_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    qualified_predicate: Optional[str] = None\n",
      "    disease_context_qualifier: Optional[Union[str, DiseaseId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GenotypeToDiseaseAssociationId):\n",
      "            self.id = GenotypeToDiseaseAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, NamedThingId):\n",
      "            self.subject = NamedThingId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        if self.frequency_qualifier is not None and not isinstance(self.frequency_qualifier, FrequencyValue):\n",
      "            self.frequency_qualifier = FrequencyValue(self.frequency_qualifier)\n",
      "\n",
      "        if self.subject_aspect_qualifier is not None and not isinstance(self.subject_aspect_qualifier, str):\n",
      "            self.subject_aspect_qualifier = str(self.subject_aspect_qualifier)\n",
      "\n",
      "        if self.subject_direction_qualifier is not None and not isinstance(self.subject_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.subject_direction_qualifier = DirectionQualifierEnum(self.subject_direction_qualifier)\n",
      "\n",
      "        if self.object_aspect_qualifier is not None and not isinstance(self.object_aspect_qualifier, str):\n",
      "            self.object_aspect_qualifier = str(self.object_aspect_qualifier)\n",
      "\n",
      "        if self.object_direction_qualifier is not None and not isinstance(self.object_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.object_direction_qualifier = DirectionQualifierEnum(self.object_direction_qualifier)\n",
      "\n",
      "        if self.qualified_predicate is not None and not isinstance(self.qualified_predicate, str):\n",
      "            self.qualified_predicate = str(self.qualified_predicate)\n",
      "\n",
      "        if self.disease_context_qualifier is not None and not isinstance(self.disease_context_qualifier, DiseaseId):\n",
      "            self.disease_context_qualifier = DiseaseId(self.disease_context_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ModelToDiseaseAssociationMixin(YAMLRoot):\n",
      "    \"\"\"\n",
      "    This mixin is used for any association class for which the subject (source node) plays the role of a 'model', in\n",
      "    that it recapitulates some features of the disease in a way that is useful for studying the disease outside a\n",
      "    patient carrying the disease\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ModelToDiseaseAssociationMixin\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ModelToDiseaseAssociationMixin\"\n",
      "    class_name: ClassVar[str] = \"model to disease association mixin\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ModelToDiseaseAssociationMixin\n",
      "\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, NamedThingId):\n",
      "            self.subject = NamedThingId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GeneAsAModelOfDiseaseAssociation(GeneToDiseaseAssociation):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GeneAsAModelOfDiseaseAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GeneAsAModelOfDiseaseAssociation\"\n",
      "    class_name: ClassVar[str] = \"gene as a model of disease association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GeneAsAModelOfDiseaseAssociation\n",
      "\n",
      "    id: Union[str, GeneAsAModelOfDiseaseAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, DiseaseId] = None\n",
      "    subject: Union[dict, GeneOrGeneProduct] = None\n",
      "    frequency_qualifier: Optional[Union[str, FrequencyValue]] = None\n",
      "    subject_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    object_aspect_qualifier: Optional[str] = None\n",
      "    qualified_predicate: Optional[str] = None\n",
      "    disease_context_qualifier: Optional[Union[str, DiseaseId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GeneAsAModelOfDiseaseAssociationId):\n",
      "            self.id = GeneAsAModelOfDiseaseAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, GeneOrGeneProduct):\n",
      "            self.subject = GeneOrGeneProduct(**as_dict(self.subject))\n",
      "\n",
      "        if self.frequency_qualifier is not None and not isinstance(self.frequency_qualifier, FrequencyValue):\n",
      "            self.frequency_qualifier = FrequencyValue(self.frequency_qualifier)\n",
      "\n",
      "        if self.subject_direction_qualifier is not None and not isinstance(self.subject_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.subject_direction_qualifier = DirectionQualifierEnum(self.subject_direction_qualifier)\n",
      "\n",
      "        if self.object_aspect_qualifier is not None and not isinstance(self.object_aspect_qualifier, str):\n",
      "            self.object_aspect_qualifier = str(self.object_aspect_qualifier)\n",
      "\n",
      "        if self.qualified_predicate is not None and not isinstance(self.qualified_predicate, str):\n",
      "            self.qualified_predicate = str(self.qualified_predicate)\n",
      "\n",
      "        if self.disease_context_qualifier is not None and not isinstance(self.disease_context_qualifier, DiseaseId):\n",
      "            self.disease_context_qualifier = DiseaseId(self.disease_context_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class VariantAsAModelOfDiseaseAssociation(VariantToDiseaseAssociation):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"VariantAsAModelOfDiseaseAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:VariantAsAModelOfDiseaseAssociation\"\n",
      "    class_name: ClassVar[str] = \"variant as a model of disease association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.VariantAsAModelOfDiseaseAssociation\n",
      "\n",
      "    id: Union[str, VariantAsAModelOfDiseaseAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    subject: Union[str, SequenceVariantId] = None\n",
      "    frequency_qualifier: Optional[Union[str, FrequencyValue]] = None\n",
      "    subject_aspect_qualifier: Optional[str] = None\n",
      "    subject_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    object_aspect_qualifier: Optional[str] = None\n",
      "    object_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    qualified_predicate: Optional[str] = None\n",
      "    disease_context_qualifier: Optional[Union[str, DiseaseId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, VariantAsAModelOfDiseaseAssociationId):\n",
      "            self.id = VariantAsAModelOfDiseaseAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, SequenceVariantId):\n",
      "            self.subject = SequenceVariantId(self.subject)\n",
      "\n",
      "        if self.frequency_qualifier is not None and not isinstance(self.frequency_qualifier, FrequencyValue):\n",
      "            self.frequency_qualifier = FrequencyValue(self.frequency_qualifier)\n",
      "\n",
      "        if self.subject_aspect_qualifier is not None and not isinstance(self.subject_aspect_qualifier, str):\n",
      "            self.subject_aspect_qualifier = str(self.subject_aspect_qualifier)\n",
      "\n",
      "        if self.subject_direction_qualifier is not None and not isinstance(self.subject_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.subject_direction_qualifier = DirectionQualifierEnum(self.subject_direction_qualifier)\n",
      "\n",
      "        if self.object_aspect_qualifier is not None and not isinstance(self.object_aspect_qualifier, str):\n",
      "            self.object_aspect_qualifier = str(self.object_aspect_qualifier)\n",
      "\n",
      "        if self.object_direction_qualifier is not None and not isinstance(self.object_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.object_direction_qualifier = DirectionQualifierEnum(self.object_direction_qualifier)\n",
      "\n",
      "        if self.qualified_predicate is not None and not isinstance(self.qualified_predicate, str):\n",
      "            self.qualified_predicate = str(self.qualified_predicate)\n",
      "\n",
      "        if self.disease_context_qualifier is not None and not isinstance(self.disease_context_qualifier, DiseaseId):\n",
      "            self.disease_context_qualifier = DiseaseId(self.disease_context_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GenotypeAsAModelOfDiseaseAssociation(GenotypeToDiseaseAssociation):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GenotypeAsAModelOfDiseaseAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GenotypeAsAModelOfDiseaseAssociation\"\n",
      "    class_name: ClassVar[str] = \"genotype as a model of disease association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GenotypeAsAModelOfDiseaseAssociation\n",
      "\n",
      "    id: Union[str, GenotypeAsAModelOfDiseaseAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    subject: Union[str, GenotypeId] = None\n",
      "    frequency_qualifier: Optional[Union[str, FrequencyValue]] = None\n",
      "    subject_aspect_qualifier: Optional[str] = None\n",
      "    subject_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    object_aspect_qualifier: Optional[str] = None\n",
      "    object_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    qualified_predicate: Optional[str] = None\n",
      "    disease_context_qualifier: Optional[Union[str, DiseaseId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GenotypeAsAModelOfDiseaseAssociationId):\n",
      "            self.id = GenotypeAsAModelOfDiseaseAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, GenotypeId):\n",
      "            self.subject = GenotypeId(self.subject)\n",
      "\n",
      "        if self.frequency_qualifier is not None and not isinstance(self.frequency_qualifier, FrequencyValue):\n",
      "            self.frequency_qualifier = FrequencyValue(self.frequency_qualifier)\n",
      "\n",
      "        if self.subject_aspect_qualifier is not None and not isinstance(self.subject_aspect_qualifier, str):\n",
      "            self.subject_aspect_qualifier = str(self.subject_aspect_qualifier)\n",
      "\n",
      "        if self.subject_direction_qualifier is not None and not isinstance(self.subject_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.subject_direction_qualifier = DirectionQualifierEnum(self.subject_direction_qualifier)\n",
      "\n",
      "        if self.object_aspect_qualifier is not None and not isinstance(self.object_aspect_qualifier, str):\n",
      "            self.object_aspect_qualifier = str(self.object_aspect_qualifier)\n",
      "\n",
      "        if self.object_direction_qualifier is not None and not isinstance(self.object_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.object_direction_qualifier = DirectionQualifierEnum(self.object_direction_qualifier)\n",
      "\n",
      "        if self.qualified_predicate is not None and not isinstance(self.qualified_predicate, str):\n",
      "            self.qualified_predicate = str(self.qualified_predicate)\n",
      "\n",
      "        if self.disease_context_qualifier is not None and not isinstance(self.disease_context_qualifier, DiseaseId):\n",
      "            self.disease_context_qualifier = DiseaseId(self.disease_context_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class CellLineAsAModelOfDiseaseAssociation(CellLineToDiseaseOrPhenotypicFeatureAssociation):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"CellLineAsAModelOfDiseaseAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:CellLineAsAModelOfDiseaseAssociation\"\n",
      "    class_name: ClassVar[str] = \"cell line as a model of disease association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.CellLineAsAModelOfDiseaseAssociation\n",
      "\n",
      "    id: Union[str, CellLineAsAModelOfDiseaseAssociationId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, CellLineId] = None\n",
      "    disease_context_qualifier: Optional[Union[str, DiseaseId]] = None\n",
      "    frequency_qualifier: Optional[Union[str, FrequencyValue]] = None\n",
      "    subject_aspect_qualifier: Optional[str] = None\n",
      "    subject_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    object_aspect_qualifier: Optional[str] = None\n",
      "    object_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    qualified_predicate: Optional[str] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, CellLineAsAModelOfDiseaseAssociationId):\n",
      "            self.id = CellLineAsAModelOfDiseaseAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, CellLineId):\n",
      "            self.subject = CellLineId(self.subject)\n",
      "\n",
      "        if self.disease_context_qualifier is not None and not isinstance(self.disease_context_qualifier, DiseaseId):\n",
      "            self.disease_context_qualifier = DiseaseId(self.disease_context_qualifier)\n",
      "\n",
      "        if self.frequency_qualifier is not None and not isinstance(self.frequency_qualifier, FrequencyValue):\n",
      "            self.frequency_qualifier = FrequencyValue(self.frequency_qualifier)\n",
      "\n",
      "        if self.subject_aspect_qualifier is not None and not isinstance(self.subject_aspect_qualifier, str):\n",
      "            self.subject_aspect_qualifier = str(self.subject_aspect_qualifier)\n",
      "\n",
      "        if self.subject_direction_qualifier is not None and not isinstance(self.subject_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.subject_direction_qualifier = DirectionQualifierEnum(self.subject_direction_qualifier)\n",
      "\n",
      "        if self.object_aspect_qualifier is not None and not isinstance(self.object_aspect_qualifier, str):\n",
      "            self.object_aspect_qualifier = str(self.object_aspect_qualifier)\n",
      "\n",
      "        if self.object_direction_qualifier is not None and not isinstance(self.object_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.object_direction_qualifier = DirectionQualifierEnum(self.object_direction_qualifier)\n",
      "\n",
      "        if self.qualified_predicate is not None and not isinstance(self.qualified_predicate, str):\n",
      "            self.qualified_predicate = str(self.qualified_predicate)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class OrganismalEntityAsAModelOfDiseaseAssociation(Association):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"OrganismalEntityAsAModelOfDiseaseAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:OrganismalEntityAsAModelOfDiseaseAssociation\"\n",
      "    class_name: ClassVar[str] = \"organismal entity as a model of disease association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.OrganismalEntityAsAModelOfDiseaseAssociation\n",
      "\n",
      "    id: Union[str, OrganismalEntityAsAModelOfDiseaseAssociationId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, OrganismalEntityId] = None\n",
      "    frequency_qualifier: Optional[Union[str, FrequencyValue]] = None\n",
      "    subject_aspect_qualifier: Optional[str] = None\n",
      "    subject_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    object_aspect_qualifier: Optional[str] = None\n",
      "    object_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "    qualified_predicate: Optional[str] = None\n",
      "    disease_context_qualifier: Optional[Union[str, DiseaseId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, OrganismalEntityAsAModelOfDiseaseAssociationId):\n",
      "            self.id = OrganismalEntityAsAModelOfDiseaseAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, OrganismalEntityId):\n",
      "            self.subject = OrganismalEntityId(self.subject)\n",
      "\n",
      "        if self.frequency_qualifier is not None and not isinstance(self.frequency_qualifier, FrequencyValue):\n",
      "            self.frequency_qualifier = FrequencyValue(self.frequency_qualifier)\n",
      "\n",
      "        if self.subject_aspect_qualifier is not None and not isinstance(self.subject_aspect_qualifier, str):\n",
      "            self.subject_aspect_qualifier = str(self.subject_aspect_qualifier)\n",
      "\n",
      "        if self.subject_direction_qualifier is not None and not isinstance(self.subject_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.subject_direction_qualifier = DirectionQualifierEnum(self.subject_direction_qualifier)\n",
      "\n",
      "        if self.object_aspect_qualifier is not None and not isinstance(self.object_aspect_qualifier, str):\n",
      "            self.object_aspect_qualifier = str(self.object_aspect_qualifier)\n",
      "\n",
      "        if self.object_direction_qualifier is not None and not isinstance(self.object_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.object_direction_qualifier = DirectionQualifierEnum(self.object_direction_qualifier)\n",
      "\n",
      "        if self.qualified_predicate is not None and not isinstance(self.qualified_predicate, str):\n",
      "            self.qualified_predicate = str(self.qualified_predicate)\n",
      "\n",
      "        if self.disease_context_qualifier is not None and not isinstance(self.disease_context_qualifier, DiseaseId):\n",
      "            self.disease_context_qualifier = DiseaseId(self.disease_context_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class OrganismToOrganismAssociation(Association):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"OrganismToOrganismAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:OrganismToOrganismAssociation\"\n",
      "    class_name: ClassVar[str] = \"organism to organism association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.OrganismToOrganismAssociation\n",
      "\n",
      "    id: Union[str, OrganismToOrganismAssociationId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, IndividualOrganismId] = None\n",
      "    object: Union[str, IndividualOrganismId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, OrganismToOrganismAssociationId):\n",
      "            self.id = OrganismToOrganismAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, IndividualOrganismId):\n",
      "            self.subject = IndividualOrganismId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, IndividualOrganismId):\n",
      "            self.object = IndividualOrganismId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class TaxonToTaxonAssociation(Association):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"TaxonToTaxonAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:TaxonToTaxonAssociation\"\n",
      "    class_name: ClassVar[str] = \"taxon to taxon association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.TaxonToTaxonAssociation\n",
      "\n",
      "    id: Union[str, TaxonToTaxonAssociationId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, OrganismTaxonId] = None\n",
      "    object: Union[str, OrganismTaxonId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, TaxonToTaxonAssociationId):\n",
      "            self.id = TaxonToTaxonAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, OrganismTaxonId):\n",
      "            self.subject = OrganismTaxonId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, OrganismTaxonId):\n",
      "            self.object = OrganismTaxonId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GeneHasVariantThatContributesToDiseaseAssociation(GeneToDiseaseAssociation):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GeneHasVariantThatContributesToDiseaseAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GeneHasVariantThatContributesToDiseaseAssociation\"\n",
      "    class_name: ClassVar[str] = \"gene has variant that contributes to disease association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GeneHasVariantThatContributesToDiseaseAssociation\n",
      "\n",
      "    id: Union[str, GeneHasVariantThatContributesToDiseaseAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[dict, GeneOrGeneProduct] = None\n",
      "    object: Union[str, DiseaseId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    subject_form_or_variant_qualifier: Optional[str] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GeneHasVariantThatContributesToDiseaseAssociationId):\n",
      "            self.id = GeneHasVariantThatContributesToDiseaseAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, GeneOrGeneProduct):\n",
      "            self.subject = GeneOrGeneProduct(**as_dict(self.subject))\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, DiseaseId):\n",
      "            self.object = DiseaseId(self.object)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self.subject_form_or_variant_qualifier is not None and not isinstance(self.subject_form_or_variant_qualifier, str):\n",
      "            self.subject_form_or_variant_qualifier = str(self.subject_form_or_variant_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GeneToExpressionSiteAssociation(Association):\n",
      "    \"\"\"\n",
      "    An association between a gene and a gene expression site, possibly qualified by stage/timing info.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GeneToExpressionSiteAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GeneToExpressionSiteAssociation\"\n",
      "    class_name: ClassVar[str] = \"gene to expression site association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GeneToExpressionSiteAssociation\n",
      "\n",
      "    id: Union[str, GeneToExpressionSiteAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[dict, GeneOrGeneProduct] = None\n",
      "    object: Union[str, AnatomicalEntityId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    stage_qualifier: Optional[Union[str, LifeStageId]] = None\n",
      "    quantifier_qualifier: Optional[Union[str, OntologyClassId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GeneToExpressionSiteAssociationId):\n",
      "            self.id = GeneToExpressionSiteAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, GeneOrGeneProduct):\n",
      "            self.subject = GeneOrGeneProduct(**as_dict(self.subject))\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, AnatomicalEntityId):\n",
      "            self.object = AnatomicalEntityId(self.object)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self.stage_qualifier is not None and not isinstance(self.stage_qualifier, LifeStageId):\n",
      "            self.stage_qualifier = LifeStageId(self.stage_qualifier)\n",
      "\n",
      "        if self.quantifier_qualifier is not None and not isinstance(self.quantifier_qualifier, OntologyClassId):\n",
      "            self.quantifier_qualifier = OntologyClassId(self.quantifier_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class SequenceVariantModulatesTreatmentAssociation(Association):\n",
      "    \"\"\"\n",
      "    An association between a sequence variant and a treatment or health intervention. The treatment object itself\n",
      "    encompasses both the disease and the drug used.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"SequenceVariantModulatesTreatmentAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:SequenceVariantModulatesTreatmentAssociation\"\n",
      "    class_name: ClassVar[str] = \"sequence variant modulates treatment association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.SequenceVariantModulatesTreatmentAssociation\n",
      "\n",
      "    id: Union[str, SequenceVariantModulatesTreatmentAssociationId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, SequenceVariantId] = None\n",
      "    object: Union[str, TreatmentId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, SequenceVariantId):\n",
      "            self.subject = SequenceVariantId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, TreatmentId):\n",
      "            self.object = TreatmentId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class FunctionalAssociation(Association):\n",
      "    \"\"\"\n",
      "    An association between a macromolecular machine mixin (gene, gene product or complex of gene products) and either\n",
      "    a molecular activity, a biological process or a cellular location in which a function is executed.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"FunctionalAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:FunctionalAssociation\"\n",
      "    class_name: ClassVar[str] = \"functional association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.FunctionalAssociation\n",
      "\n",
      "    id: Union[str, FunctionalAssociationId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[dict, MacromolecularMachineMixin] = None\n",
      "    object: Union[str, OntologyClassId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, FunctionalAssociationId):\n",
      "            self.id = FunctionalAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, MacromolecularMachineMixin):\n",
      "            self.subject = MacromolecularMachineMixin(**as_dict(self.subject))\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, OntologyClassId):\n",
      "            self.object = OntologyClassId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class MacromolecularMachineToEntityAssociationMixin(YAMLRoot):\n",
      "    \"\"\"\n",
      "    an association which has a macromolecular machine mixin as a subject\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"MacromolecularMachineToEntityAssociationMixin\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:MacromolecularMachineToEntityAssociationMixin\"\n",
      "    class_name: ClassVar[str] = \"macromolecular machine to entity association mixin\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.MacromolecularMachineToEntityAssociationMixin\n",
      "\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    species_context_qualifier: Optional[Union[str, OrganismTaxonId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, NamedThingId):\n",
      "            self.subject = NamedThingId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        if self.species_context_qualifier is not None and not isinstance(self.species_context_qualifier, OrganismTaxonId):\n",
      "            self.species_context_qualifier = OrganismTaxonId(self.species_context_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class MacromolecularMachineToMolecularActivityAssociation(FunctionalAssociation):\n",
      "    \"\"\"\n",
      "    A functional association between a macromolecular machine (gene, gene product or complex) and a molecular activity\n",
      "    (as represented in the GO molecular function branch), where the entity carries out the activity, or contributes to\n",
      "    its execution.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"MacromolecularMachineToMolecularActivityAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:MacromolecularMachineToMolecularActivityAssociation\"\n",
      "    class_name: ClassVar[str] = \"macromolecular machine to molecular activity association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.MacromolecularMachineToMolecularActivityAssociation\n",
      "\n",
      "    id: Union[str, MacromolecularMachineToMolecularActivityAssociationId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[dict, MacromolecularMachineMixin] = None\n",
      "    object: Union[str, MolecularActivityId] = None\n",
      "    species_context_qualifier: Optional[Union[str, OrganismTaxonId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, MacromolecularMachineToMolecularActivityAssociationId):\n",
      "            self.id = MacromolecularMachineToMolecularActivityAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, MolecularActivityId):\n",
      "            self.object = MolecularActivityId(self.object)\n",
      "\n",
      "        if self.species_context_qualifier is not None and not isinstance(self.species_context_qualifier, OrganismTaxonId):\n",
      "            self.species_context_qualifier = OrganismTaxonId(self.species_context_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class MacromolecularMachineToBiologicalProcessAssociation(FunctionalAssociation):\n",
      "    \"\"\"\n",
      "    A functional association between a macromolecular machine (gene, gene product or complex) and a biological process\n",
      "    or pathway (as represented in the GO biological process branch), where the entity carries out some part of the\n",
      "    process, regulates it, or acts upstream of it.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"MacromolecularMachineToBiologicalProcessAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:MacromolecularMachineToBiologicalProcessAssociation\"\n",
      "    class_name: ClassVar[str] = \"macromolecular machine to biological process association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.MacromolecularMachineToBiologicalProcessAssociation\n",
      "\n",
      "    id: Union[str, MacromolecularMachineToBiologicalProcessAssociationId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[dict, MacromolecularMachineMixin] = None\n",
      "    object: Union[str, BiologicalProcessId] = None\n",
      "    species_context_qualifier: Optional[Union[str, OrganismTaxonId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, MacromolecularMachineToBiologicalProcessAssociationId):\n",
      "            self.id = MacromolecularMachineToBiologicalProcessAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, BiologicalProcessId):\n",
      "            self.object = BiologicalProcessId(self.object)\n",
      "\n",
      "        if self.species_context_qualifier is not None and not isinstance(self.species_context_qualifier, OrganismTaxonId):\n",
      "            self.species_context_qualifier = OrganismTaxonId(self.species_context_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class MacromolecularMachineToCellularComponentAssociation(FunctionalAssociation):\n",
      "    \"\"\"\n",
      "    A functional association between a macromolecular machine (gene, gene product or complex) and a cellular component\n",
      "    (as represented in the GO cellular component branch), where the entity carries out its function in the cellular\n",
      "    component.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"MacromolecularMachineToCellularComponentAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:MacromolecularMachineToCellularComponentAssociation\"\n",
      "    class_name: ClassVar[str] = \"macromolecular machine to cellular component association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.MacromolecularMachineToCellularComponentAssociation\n",
      "\n",
      "    id: Union[str, MacromolecularMachineToCellularComponentAssociationId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[dict, MacromolecularMachineMixin] = None\n",
      "    object: Union[str, CellularComponentId] = None\n",
      "    species_context_qualifier: Optional[Union[str, OrganismTaxonId]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, MacromolecularMachineToCellularComponentAssociationId):\n",
      "            self.id = MacromolecularMachineToCellularComponentAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, CellularComponentId):\n",
      "            self.object = CellularComponentId(self.object)\n",
      "\n",
      "        if self.species_context_qualifier is not None and not isinstance(self.species_context_qualifier, OrganismTaxonId):\n",
      "            self.species_context_qualifier = OrganismTaxonId(self.species_context_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class MolecularActivityToChemicalEntityAssociation(Association):\n",
      "    \"\"\"\n",
      "    Added in response to capturing relationship between microbiome activities as measured via measurements of blood\n",
      "    analytes as collected via blood and stool samples\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"MolecularActivityToChemicalEntityAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:MolecularActivityToChemicalEntityAssociation\"\n",
      "    class_name: ClassVar[str] = \"molecular activity to chemical entity association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.MolecularActivityToChemicalEntityAssociation\n",
      "\n",
      "    id: Union[str, MolecularActivityToChemicalEntityAssociationId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, MolecularActivityId] = None\n",
      "    object: Union[str, ChemicalEntityId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, MolecularActivityToChemicalEntityAssociationId):\n",
      "            self.id = MolecularActivityToChemicalEntityAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, MolecularActivityId):\n",
      "            self.subject = MolecularActivityId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, ChemicalEntityId):\n",
      "            self.object = ChemicalEntityId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class MolecularActivityToMolecularActivityAssociation(Association):\n",
      "    \"\"\"\n",
      "    Added in response to capturing relationship between microbiome activities as measured via measurements of blood\n",
      "    analytes as collected via blood and stool samples\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"MolecularActivityToMolecularActivityAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:MolecularActivityToMolecularActivityAssociation\"\n",
      "    class_name: ClassVar[str] = \"molecular activity to molecular activity association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.MolecularActivityToMolecularActivityAssociation\n",
      "\n",
      "    id: Union[str, MolecularActivityToMolecularActivityAssociationId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, MolecularActivityId] = None\n",
      "    object: Union[str, MolecularActivityId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, MolecularActivityToMolecularActivityAssociationId):\n",
      "            self.id = MolecularActivityToMolecularActivityAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, MolecularActivityId):\n",
      "            self.subject = MolecularActivityId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, MolecularActivityId):\n",
      "            self.object = MolecularActivityId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GeneToGoTermAssociation(FunctionalAssociation):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GeneToGoTermAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GeneToGoTermAssociation\"\n",
      "    class_name: ClassVar[str] = \"gene to go term association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GeneToGoTermAssociation\n",
      "\n",
      "    id: Union[str, GeneToGoTermAssociationId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, GeneId] = None\n",
      "    object: Union[str, OntologyClassId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GeneToGoTermAssociationId):\n",
      "            self.id = GeneToGoTermAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, GeneId):\n",
      "            self.subject = GeneId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, OntologyClassId):\n",
      "            self.object = OntologyClassId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class EntityToDiseaseAssociation(Association):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"EntityToDiseaseAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:EntityToDiseaseAssociation\"\n",
      "    class_name: ClassVar[str] = \"entity to disease association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.EntityToDiseaseAssociation\n",
      "\n",
      "    id: Union[str, EntityToDiseaseAssociationId] = None\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    clinical_approval_status: Optional[Union[str, \"ClinicalApprovalStatusEnum\"]] = None\n",
      "    max_research_phase: Optional[Union[str, \"ResearchPhaseEnum\"]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, EntityToDiseaseAssociationId):\n",
      "            self.id = EntityToDiseaseAssociationId(self.id)\n",
      "\n",
      "        if self.clinical_approval_status is not None and not isinstance(self.clinical_approval_status, ClinicalApprovalStatusEnum):\n",
      "            self.clinical_approval_status = ClinicalApprovalStatusEnum(self.clinical_approval_status)\n",
      "\n",
      "        if self.max_research_phase is not None and not isinstance(self.max_research_phase, ResearchPhaseEnum):\n",
      "            self.max_research_phase = ResearchPhaseEnum(self.max_research_phase)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class EntityToPhenotypicFeatureAssociation(Association):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"EntityToPhenotypicFeatureAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:EntityToPhenotypicFeatureAssociation\"\n",
      "    class_name: ClassVar[str] = \"entity to phenotypic feature association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.EntityToPhenotypicFeatureAssociation\n",
      "\n",
      "    id: Union[str, EntityToPhenotypicFeatureAssociationId] = None\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    clinical_approval_status: Optional[Union[str, \"ClinicalApprovalStatusEnum\"]] = None\n",
      "    max_research_phase: Optional[Union[str, \"ResearchPhaseEnum\"]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, EntityToPhenotypicFeatureAssociationId):\n",
      "            self.id = EntityToPhenotypicFeatureAssociationId(self.id)\n",
      "\n",
      "        if self.clinical_approval_status is not None and not isinstance(self.clinical_approval_status, ClinicalApprovalStatusEnum):\n",
      "            self.clinical_approval_status = ClinicalApprovalStatusEnum(self.clinical_approval_status)\n",
      "\n",
      "        if self.max_research_phase is not None and not isinstance(self.max_research_phase, ResearchPhaseEnum):\n",
      "            self.max_research_phase = ResearchPhaseEnum(self.max_research_phase)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class SequenceAssociation(Association):\n",
      "    \"\"\"\n",
      "    An association between a sequence feature and a nucleic acid entity it is localized to.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"SequenceAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:SequenceAssociation\"\n",
      "    class_name: ClassVar[str] = \"sequence association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.SequenceAssociation\n",
      "\n",
      "    id: Union[str, SequenceAssociationId] = None\n",
      "    subject: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, SequenceAssociationId):\n",
      "            self.id = SequenceAssociationId(self.id)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GenomicSequenceLocalization(SequenceAssociation):\n",
      "    \"\"\"\n",
      "    A relationship between a sequence feature and a nucleic acid entity it is localized to. The reference entity may\n",
      "    be a chromosome, chromosome region or information entity such as a contig.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GenomicSequenceLocalization\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GenomicSequenceLocalization\"\n",
      "    class_name: ClassVar[str] = \"genomic sequence localization\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GenomicSequenceLocalization\n",
      "\n",
      "    id: Union[str, GenomicSequenceLocalizationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, NucleicAcidEntityId] = None\n",
      "    object: Union[str, NucleicAcidEntityId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    start_interbase_coordinate: Optional[int] = None\n",
      "    end_interbase_coordinate: Optional[int] = None\n",
      "    genome_build: Optional[Union[str, \"StrandEnum\"]] = None\n",
      "    strand: Optional[Union[str, \"StrandEnum\"]] = None\n",
      "    phase: Optional[Union[str, \"PhaseEnum\"]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GenomicSequenceLocalizationId):\n",
      "            self.id = GenomicSequenceLocalizationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, NucleicAcidEntityId):\n",
      "            self.subject = NucleicAcidEntityId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NucleicAcidEntityId):\n",
      "            self.object = NucleicAcidEntityId(self.object)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self.start_interbase_coordinate is not None and not isinstance(self.start_interbase_coordinate, int):\n",
      "            self.start_interbase_coordinate = int(self.start_interbase_coordinate)\n",
      "\n",
      "        if self.end_interbase_coordinate is not None and not isinstance(self.end_interbase_coordinate, int):\n",
      "            self.end_interbase_coordinate = int(self.end_interbase_coordinate)\n",
      "\n",
      "        if self.genome_build is not None and not isinstance(self.genome_build, StrandEnum):\n",
      "            self.genome_build = StrandEnum(self.genome_build)\n",
      "\n",
      "        if self.strand is not None and not isinstance(self.strand, StrandEnum):\n",
      "            self.strand = StrandEnum(self.strand)\n",
      "\n",
      "        if self.phase is not None and not isinstance(self.phase, PhaseEnum):\n",
      "            self.phase = PhaseEnum(self.phase)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class SequenceFeatureRelationship(Association):\n",
      "    \"\"\"\n",
      "    For example, a particular exon is part of a particular transcript or gene\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"SequenceFeatureRelationship\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:SequenceFeatureRelationship\"\n",
      "    class_name: ClassVar[str] = \"sequence feature relationship\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.SequenceFeatureRelationship\n",
      "\n",
      "    id: Union[str, SequenceFeatureRelationshipId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, NucleicAcidEntityId] = None\n",
      "    object: Union[str, NucleicAcidEntityId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, SequenceFeatureRelationshipId):\n",
      "            self.id = SequenceFeatureRelationshipId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, NucleicAcidEntityId):\n",
      "            self.subject = NucleicAcidEntityId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NucleicAcidEntityId):\n",
      "            self.object = NucleicAcidEntityId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class TranscriptToGeneRelationship(SequenceFeatureRelationship):\n",
      "    \"\"\"\n",
      "    A gene is a collection of transcripts\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"TranscriptToGeneRelationship\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:TranscriptToGeneRelationship\"\n",
      "    class_name: ClassVar[str] = \"transcript to gene relationship\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.TranscriptToGeneRelationship\n",
      "\n",
      "    id: Union[str, TranscriptToGeneRelationshipId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, TranscriptId] = None\n",
      "    object: Union[str, GeneId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, TranscriptToGeneRelationshipId):\n",
      "            self.id = TranscriptToGeneRelationshipId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, TranscriptId):\n",
      "            self.subject = TranscriptId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, GeneId):\n",
      "            self.object = GeneId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class GeneToGeneProductRelationship(SequenceFeatureRelationship):\n",
      "    \"\"\"\n",
      "    A gene is transcribed and potentially translated to a gene product\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"GeneToGeneProductRelationship\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:GeneToGeneProductRelationship\"\n",
      "    class_name: ClassVar[str] = \"gene to gene product relationship\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.GeneToGeneProductRelationship\n",
      "\n",
      "    id: Union[str, GeneToGeneProductRelationshipId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, GeneId] = None\n",
      "    object: Union[dict, GeneProductMixin] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, GeneToGeneProductRelationshipId):\n",
      "            self.id = GeneToGeneProductRelationshipId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, GeneId):\n",
      "            self.subject = GeneId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, GeneProductMixin):\n",
      "            self.object = GeneProductMixin(**as_dict(self.object))\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ExonToTranscriptRelationship(SequenceFeatureRelationship):\n",
      "    \"\"\"\n",
      "    A transcript is formed from multiple exons\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ExonToTranscriptRelationship\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ExonToTranscriptRelationship\"\n",
      "    class_name: ClassVar[str] = \"exon to transcript relationship\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ExonToTranscriptRelationship\n",
      "\n",
      "    id: Union[str, ExonToTranscriptRelationshipId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, ExonId] = None\n",
      "    object: Union[str, TranscriptId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ExonToTranscriptRelationshipId):\n",
      "            self.id = ExonToTranscriptRelationshipId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, ExonId):\n",
      "            self.subject = ExonId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, TranscriptId):\n",
      "            self.object = TranscriptId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class ChemicalEntityOrGeneOrGeneProductRegulatesGeneAssociation(Association):\n",
      "    \"\"\"\n",
      "    A regulatory relationship between two genes\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"ChemicalEntityOrGeneOrGeneProductRegulatesGeneAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:ChemicalEntityOrGeneOrGeneProductRegulatesGeneAssociation\"\n",
      "    class_name: ClassVar[str] = \"chemical entity or gene or gene product regulates gene association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.ChemicalEntityOrGeneOrGeneProductRegulatesGeneAssociation\n",
      "\n",
      "    id: Union[str, ChemicalEntityOrGeneOrGeneProductRegulatesGeneAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    subject: Union[dict, ChemicalEntityOrGeneOrGeneProduct] = None\n",
      "    object: Union[dict, GeneOrGeneProduct] = None\n",
      "    object_direction_qualifier: Optional[Union[str, \"DirectionQualifierEnum\"]] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, ChemicalEntityOrGeneOrGeneProductRegulatesGeneAssociationId):\n",
      "            self.id = ChemicalEntityOrGeneOrGeneProductRegulatesGeneAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, ChemicalEntityOrGeneOrGeneProduct):\n",
      "            self.subject = ChemicalEntityOrGeneOrGeneProduct()\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, GeneOrGeneProduct):\n",
      "            self.object = GeneOrGeneProduct(**as_dict(self.object))\n",
      "\n",
      "        if self.object_direction_qualifier is not None and not isinstance(self.object_direction_qualifier, DirectionQualifierEnum):\n",
      "            self.object_direction_qualifier = DirectionQualifierEnum(self.object_direction_qualifier)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class AnatomicalEntityToAnatomicalEntityAssociation(Association):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"AnatomicalEntityToAnatomicalEntityAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:AnatomicalEntityToAnatomicalEntityAssociation\"\n",
      "    class_name: ClassVar[str] = \"anatomical entity to anatomical entity association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.AnatomicalEntityToAnatomicalEntityAssociation\n",
      "\n",
      "    id: Union[str, AnatomicalEntityToAnatomicalEntityAssociationId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, AnatomicalEntityId] = None\n",
      "    object: Union[str, AnatomicalEntityId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, AnatomicalEntityId):\n",
      "            self.subject = AnatomicalEntityId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, AnatomicalEntityId):\n",
      "            self.object = AnatomicalEntityId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class AnatomicalEntityToAnatomicalEntityPartOfAssociation(AnatomicalEntityToAnatomicalEntityAssociation):\n",
      "    \"\"\"\n",
      "    A relationship between two anatomical entities where the relationship is mereological, i.e the two entities are\n",
      "    related by parthood. This includes relationships between cellular components and cells, between cells and tissues,\n",
      "    tissues and whole organisms\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"AnatomicalEntityToAnatomicalEntityPartOfAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:AnatomicalEntityToAnatomicalEntityPartOfAssociation\"\n",
      "    class_name: ClassVar[str] = \"anatomical entity to anatomical entity part of association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.AnatomicalEntityToAnatomicalEntityPartOfAssociation\n",
      "\n",
      "    id: Union[str, AnatomicalEntityToAnatomicalEntityPartOfAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, AnatomicalEntityId] = None\n",
      "    object: Union[str, AnatomicalEntityId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, AnatomicalEntityToAnatomicalEntityPartOfAssociationId):\n",
      "            self.id = AnatomicalEntityToAnatomicalEntityPartOfAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, AnatomicalEntityId):\n",
      "            self.subject = AnatomicalEntityId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, AnatomicalEntityId):\n",
      "            self.object = AnatomicalEntityId(self.object)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class AnatomicalEntityToAnatomicalEntityOntogenicAssociation(AnatomicalEntityToAnatomicalEntityAssociation):\n",
      "    \"\"\"\n",
      "    A relationship between two anatomical entities where the relationship is ontogenic, i.e. the two entities are\n",
      "    related by development. A number of different relationship types can be used to specify the precise nature of the\n",
      "    relationship.\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"AnatomicalEntityToAnatomicalEntityOntogenicAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:AnatomicalEntityToAnatomicalEntityOntogenicAssociation\"\n",
      "    class_name: ClassVar[str] = \"anatomical entity to anatomical entity ontogenic association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.AnatomicalEntityToAnatomicalEntityOntogenicAssociation\n",
      "\n",
      "    id: Union[str, AnatomicalEntityToAnatomicalEntityOntogenicAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, AnatomicalEntityId] = None\n",
      "    object: Union[str, AnatomicalEntityId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, AnatomicalEntityToAnatomicalEntityOntogenicAssociationId):\n",
      "            self.id = AnatomicalEntityToAnatomicalEntityOntogenicAssociationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, AnatomicalEntityId):\n",
      "            self.subject = AnatomicalEntityId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, AnatomicalEntityId):\n",
      "            self.object = AnatomicalEntityId(self.object)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class OrganismTaxonToEntityAssociation(YAMLRoot):\n",
      "    \"\"\"\n",
      "    An association between an organism taxon and another entity\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"OrganismTaxonToEntityAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:OrganismTaxonToEntityAssociation\"\n",
      "    class_name: ClassVar[str] = \"organism taxon to entity association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.OrganismTaxonToEntityAssociation\n",
      "\n",
      "    subject: Union[str, OrganismTaxonId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, OrganismTaxonId):\n",
      "            self.subject = OrganismTaxonId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class OrganismTaxonToOrganismTaxonAssociation(Association):\n",
      "    \"\"\"\n",
      "    A relationship between two organism taxon nodes\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"OrganismTaxonToOrganismTaxonAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:OrganismTaxonToOrganismTaxonAssociation\"\n",
      "    class_name: ClassVar[str] = \"organism taxon to organism taxon association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.OrganismTaxonToOrganismTaxonAssociation\n",
      "\n",
      "    id: Union[str, OrganismTaxonToOrganismTaxonAssociationId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, OrganismTaxonId] = None\n",
      "    object: Union[str, OrganismTaxonId] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, OrganismTaxonId):\n",
      "            self.subject = OrganismTaxonId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, OrganismTaxonId):\n",
      "            self.object = OrganismTaxonId(self.object)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class OrganismTaxonToOrganismTaxonSpecialization(OrganismTaxonToOrganismTaxonAssociation):\n",
      "    \"\"\"\n",
      "    A child-parent relationship between two taxa. For example: Homo sapiens subclass_of Homo\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"OrganismTaxonToOrganismTaxonSpecialization\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:OrganismTaxonToOrganismTaxonSpecialization\"\n",
      "    class_name: ClassVar[str] = \"organism taxon to organism taxon specialization\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.OrganismTaxonToOrganismTaxonSpecialization\n",
      "\n",
      "    id: Union[str, OrganismTaxonToOrganismTaxonSpecializationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, OrganismTaxonId] = None\n",
      "    object: Union[str, OrganismTaxonId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, OrganismTaxonToOrganismTaxonSpecializationId):\n",
      "            self.id = OrganismTaxonToOrganismTaxonSpecializationId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, OrganismTaxonId):\n",
      "            self.subject = OrganismTaxonId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, OrganismTaxonId):\n",
      "            self.object = OrganismTaxonId(self.object)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class OrganismTaxonToOrganismTaxonInteraction(OrganismTaxonToOrganismTaxonAssociation):\n",
      "    \"\"\"\n",
      "    An interaction relationship between two taxa. This may be a symbiotic relationship (encompassing mutualism and\n",
      "    parasitism), or it may be non-symbiotic. Example: plague transmitted_by flea; cattle domesticated_by Homo sapiens;\n",
      "    plague infects Homo sapiens\n",
      "    \"\"\"\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"OrganismTaxonToOrganismTaxonInteraction\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:OrganismTaxonToOrganismTaxonInteraction\"\n",
      "    class_name: ClassVar[str] = \"organism taxon to organism taxon interaction\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.OrganismTaxonToOrganismTaxonInteraction\n",
      "\n",
      "    id: Union[str, OrganismTaxonToOrganismTaxonInteractionId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, OrganismTaxonId] = None\n",
      "    object: Union[str, OrganismTaxonId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "    associated_environmental_context: Optional[str] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.id):\n",
      "            self.MissingRequiredField(\"id\")\n",
      "        if not isinstance(self.id, OrganismTaxonToOrganismTaxonInteractionId):\n",
      "            self.id = OrganismTaxonToOrganismTaxonInteractionId(self.id)\n",
      "\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, OrganismTaxonId):\n",
      "            self.subject = OrganismTaxonId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, OrganismTaxonId):\n",
      "            self.object = OrganismTaxonId(self.object)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        if self.associated_environmental_context is not None and not isinstance(self.associated_environmental_context, str):\n",
      "            self.associated_environmental_context = str(self.associated_environmental_context)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "@dataclass(repr=False)\n",
      "class OrganismTaxonToEnvironmentAssociation(Association):\n",
      "    _inherited_slots: ClassVar[List[str]] = []\n",
      "\n",
      "    class_class_uri: ClassVar[URIRef] = BIOLINK[\"OrganismTaxonToEnvironmentAssociation\"]\n",
      "    class_class_curie: ClassVar[str] = \"biolink:OrganismTaxonToEnvironmentAssociation\"\n",
      "    class_name: ClassVar[str] = \"organism taxon to environment association\"\n",
      "    class_model_uri: ClassVar[URIRef] = DIST.OrganismTaxonToEnvironmentAssociation\n",
      "\n",
      "    id: Union[str, OrganismTaxonToEnvironmentAssociationId] = None\n",
      "    knowledge_level: Union[str, \"KnowledgeLevelEnum\"] = None\n",
      "    agent_type: Union[str, \"AgentTypeEnum\"] = None\n",
      "    subject: Union[str, OrganismTaxonId] = None\n",
      "    object: Union[str, NamedThingId] = None\n",
      "    predicate: Union[str, PredicateType] = None\n",
      "\n",
      "    def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):\n",
      "        if self._is_empty(self.subject):\n",
      "            self.MissingRequiredField(\"subject\")\n",
      "        if not isinstance(self.subject, OrganismTaxonId):\n",
      "            self.subject = OrganismTaxonId(self.subject)\n",
      "\n",
      "        if self._is_empty(self.object):\n",
      "            self.MissingRequiredField(\"object\")\n",
      "        if not isinstance(self.object, NamedThingId):\n",
      "            self.object = NamedThingId(self.object)\n",
      "\n",
      "        if self._is_empty(self.predicate):\n",
      "            self.MissingRequiredField(\"predicate\")\n",
      "        if not isinstance(self.predicate, PredicateType):\n",
      "            self.predicate = PredicateType(self.predicate)\n",
      "\n",
      "        super().__post_init__(**kwargs)\n",
      "        if not isinstance(self.category, list):\n",
      "            self.category = [self.category] if self.category is not None else []\n",
      "        self.category = [v if isinstance(v, URIorCURIE) else URIorCURIE(v) for v in self.category]\n",
      "\n",
      "\n",
      "# Enumerations\n",
      "class ApprovalStatusEnum(EnumDefinitionImpl):\n",
      "\n",
      "    discovery_and_development_phase = PermissibleValue(\n",
      "        text=\"discovery_and_development_phase\",\n",
      "        description=\"\"\"Discovery & Development Phase. Discovery involves researchers finding new possibilities for medication through testing molecular compounds, noting unexpected effects from existing treatments, or the creation of new technology that allows novel ways of targeting medical products to sites in the body. Drug development occurs after researchers identify potential compounds for experiments.\"\"\")\n",
      "    preclinical_research_phase = PermissibleValue(\n",
      "        text=\"preclinical_research_phase\",\n",
      "        description=\"\"\"Preclinical Research Phase.  Once researchers have examined the possibilities a new drug may contain, they must do preliminary research to determine its potential for harm (toxicity). This is categorized as preclinical research and can be one of two types: in vitro or in vivo.\"\"\")\n",
      "    fda_clinical_research_phase = PermissibleValue(\n",
      "        text=\"fda_clinical_research_phase\",\n",
      "        description=\"\"\"Clinical Research Phase. Clinical research involves trials of the drug on people, and it is one of the most involved stages in the drug development and approval process. Clinical trials must answer specific questions and follow a protocol determined by the drug researcher or manufacturer.\"\"\")\n",
      "    fda_review_phase_4 = PermissibleValue(\n",
      "        text=\"fda_review_phase_4\",\n",
      "        description=\"FDA Review\")\n",
      "    fda_post_market_safety_review = PermissibleValue(\n",
      "        text=\"fda_post_market_safety_review\",\n",
      "        description=\"\"\"FDA Post-Market Safety Monitoring.  The last phase of drug approval is an ongoing one while the drug is on the marketplace. If a developer wants to change anything about the drug formulation or approve it for a new use, they must apply with the FDA. The FDA also frequently reviews the drugs advertising and its manufacturing facility to make sure everything involved in its creation and marketing is in compliance with regulations.\"\"\")\n",
      "    fda_clinical_research_phase_1 = PermissibleValue(\n",
      "        text=\"fda_clinical_research_phase_1\",\n",
      "        description=\"\"\"In the FDA Clinical Research Phase, the Clinical Research Phase 1 involves 20  100 study participants and lasts several months. This phase is used to determine the safety and dosage of the drug, and about 70% of these drugs move on to the next clinical research phase.\"\"\")\n",
      "    fda_clinical_research_phase_2 = PermissibleValue(\n",
      "        text=\"fda_clinical_research_phase_2\",\n",
      "        description=\"\"\"In the FDA Clinical Research Phase, the Clinical Research Phase 2 involves up to several hundred people, who must have the disease or condition the drug supposes to treat. This phase can last from a few months to two years, and its purpose is to monitor the efficacy of the drug, as well as note side effects that may occur.\"\"\")\n",
      "    fda_clinical_research_phase_3 = PermissibleValue(\n",
      "        text=\"fda_clinical_research_phase_3\",\n",
      "        description=\"\"\"In the FDA Clinical Research Phase, the Clinical Research Phase 3 involves 300  3000 volunteers and can last up to four years. It is used to continue monitoring the efficacy of the drug, as well as exploring any longer-term adverse reactions.\"\"\")\n",
      "    fda_clinical_research_phase_4 = PermissibleValue(\n",
      "        text=\"fda_clinical_research_phase_4\",\n",
      "        description=\"\"\"In the FDA Clinical Research Phase, the Clinical Research Phase 4 involves several thousands of volunteers who have the disease or condition and continues to monitor safety and efficacy. If a drug passes this phase, it goes on to FDA review.\"\"\")\n",
      "    fda_fast_track = PermissibleValue(\n",
      "        text=\"fda_fast_track\",\n",
      "        description=\"\"\"Fast track is a process designed to facilitate the development, and expedite the review of drugs to treat serious conditions and fill an unmet medical need. The purpose is to get important new drugs to the patient earlier. Fast Track addresses a broad range of serious conditions. For more information https://www.fda.gov/patients/fast-track-breakthrough-therapy-accelerated-approval-priority-review/fast-track\"\"\")\n",
      "    fda_breakthrough_therapy = PermissibleValue(\n",
      "        text=\"fda_breakthrough_therapy\",\n",
      "        description=\"\"\"Breakthrough Therapy designation is a process designed to expedite the development and review of drugs that are intended to treat a serious condition and preliminary clinical evidence indicates that the drug may demonstrate substantial improvement over available therapy on a clinically significant endpoint(s). For more information https://www.fda.gov/patients/fast-track-breakthrough-therapy-accelerated-approval-priority-review/breakthrough-therapy\"\"\")\n",
      "    fda_accelerated_approval = PermissibleValue(\n",
      "        text=\"fda_accelerated_approval\",\n",
      "        description=\"\"\"When studying a new drug, it can sometimes take many years to learn whether a drug actually provides a real effect on how a patient survives, feels, or functions. A positive therapeutic effect that is clinically meaningful in the context of a given disease is known as clinical benefit. Mindful of the fact that it may take an extended period of time to measure a drugs intended clinical benefit, in 1992 FDA instituted the Accelerated Approval regulations. These regulations allowed drugs for serious conditions that filled an unmet medical need to be approved based on a surrogate endpoint. Using a surrogate endpoint enabled the FDA to approve these drugs faster. For more information https://www.fda.gov/patients/fast-track-breakthrough-therapy-accelerated-approval-priority-review/accelerated-approval\"\"\")\n",
      "    fda_priority_review = PermissibleValue(\n",
      "        text=\"fda_priority_review\",\n",
      "        description=\"\"\"Prior to approval, each drug marketed in the United States must go through a detailed FDA review process. In 1992, under the Prescription Drug User Act (PDUFA), FDA agreed to specific goals for improving the drug review time and created a two-tiered system of review times  Standard Review and Priority Review. A Priority Review designation means FDAs goal is to take action on an application within 6 months (compared to 10 months under standard review). For more information https://www.fda.gov/patients/fast-track-breakthrough-therapy-accelerated-approval-priority-review/priority-review\"\"\")\n",
      "    regular_fda_approval = PermissibleValue(\n",
      "        text=\"regular_fda_approval\",\n",
      "        description=\"\"\"Regular FDA Approval.  The last phase of drug approval is an ongoing one while the drug is on the marketplace. If a developer wants to change anything about the drug formulation or approve it for a new use, they must apply with the FDA. The FDA also frequently reviews the drugs advertising and its manufacturing facility to make sure everything involved in its creation and marketing is in compliance with regulations.\"\"\")\n",
      "    post_approval_withdrawal = PermissibleValue(text=\"post_approval_withdrawal\")\n",
      "\n",
      "    _defn = EnumDefinition(\n",
      "        name=\"ApprovalStatusEnum\",\n",
      "    )\n",
      "\n",
      "class ClinicalApprovalStatusEnum(EnumDefinitionImpl):\n",
      "\n",
      "    approved_for_condition = PermissibleValue(text=\"approved_for_condition\")\n",
      "    fda_approved_for_condition = PermissibleValue(text=\"fda_approved_for_condition\")\n",
      "    not_approved_for_condition = PermissibleValue(text=\"not_approved_for_condition\")\n",
      "    post_approval_withdrawal = PermissibleValue(text=\"post_approval_withdrawal\")\n",
      "    off_label_use = PermissibleValue(text=\"off_label_use\")\n",
      "    not_provided = PermissibleValue(text=\"not_provided\")\n",
      "\n",
      "    _defn = EnumDefinition(\n",
      "        name=\"ClinicalApprovalStatusEnum\",\n",
      "    )\n",
      "\n",
      "class ResearchPhaseEnum(EnumDefinitionImpl):\n",
      "\n",
      "    pre_clinical_research_phase = PermissibleValue(\n",
      "        text=\"pre_clinical_research_phase\",\n",
      "        description=\"\"\"Biolink 'pre_clinical_research' is the union of both the `FDA discovery and development phase` and `FDA preclinical research phase`. Discovery involves researchers finding new possibilities for medication through testing molecular compounds, noting unexpected effects from existing treatments, or the creation of new technology that allows novel ways of targeting medical products to sites in the body. Drug development occurs after researchers identify potential compounds for experiments Preclinical Research Phase. Once researchers have examined the possibilities a new drug may contain, they must do preliminary research to determine its potential for harm (toxicity). This is categorized as preclinical research and can be one of two types: in vitro or in vivo.\"\"\")\n",
      "    clinical_trial_phase = PermissibleValue(\n",
      "        text=\"clinical_trial_phase\",\n",
      "        description=\"\"\"Clinical research involves trials of the drug on people, and it is one of the most involved stages in the drug development and approval process. Clinical trials must answer specific questions and follow a protocol determined by the drug researcher or manufacturer.\"\"\")\n",
      "    clinical_trial_phase_1 = PermissibleValue(\n",
      "        text=\"clinical_trial_phase_1\",\n",
      "        description=\"\"\"In the FDA Clinical Trial Phase, the Clinical Trial Phase 1 involves 20  100 study participants and lasts several months. This phase is used to determine the safety and dosage of the drug, and about 70% of these drugs move on to the next clinical research phase.\"\"\")\n",
      "    clinical_trial_phase_2 = PermissibleValue(\n",
      "        text=\"clinical_trial_phase_2\",\n",
      "        description=\"\"\"In the FDA Clinical Trial Phase, the Clinical Trial Phase 2 involves up to several hundred people, who must have the disease or condition the drug supposes to treat. This phase can last from a few months to two years, and its purpose is to monitor the efficacy of the drug, as well as note side effects that may occur.\"\"\")\n",
      "    clinical_trial_phase_3 = PermissibleValue(\n",
      "        text=\"clinical_trial_phase_3\",\n",
      "        description=\"\"\"In the FDA Clinical Trial Phase, the Clinical Trial Phase 3 involves 300  3000 volunteers and can last up to four years. It is used to continue monitoring the efficacy of the drug, as well as exploring any longer-term adverse reactions.\"\"\")\n",
      "    clinical_trial_phase_4 = PermissibleValue(\n",
      "        text=\"clinical_trial_phase_4\",\n",
      "        description=\"\"\"In the FDA Clinical Trial Phase, the Clinical Trial Phase 4 involves several thousands of volunteers who have the disease or condition and continues to monitor safety and efficacy. If a drug passes this phase, it goes on to FDA review.\"\"\")\n",
      "    not_provided = PermissibleValue(text=\"not_provided\")\n",
      "\n",
      "    _defn = EnumDefinition(\n",
      "        name=\"ResearchPhaseEnum\",\n",
      "    )\n",
      "\n",
      "class DirectionQualifierEnum(EnumDefinitionImpl):\n",
      "\n",
      "    increased = PermissibleValue(text=\"increased\")\n",
      "    upregulated = PermissibleValue(text=\"upregulated\")\n",
      "    decreased = PermissibleValue(text=\"decreased\")\n",
      "    downregulated = PermissibleValue(text=\"downregulated\")\n",
      "\n",
      "    _defn = EnumDefinition(\n",
      "        name=\"DirectionQualifierEnum\",\n",
      "    )\n",
      "\n",
      "class ChemicalEntityDerivativeEnum(EnumDefinitionImpl):\n",
      "\n",
      "    metabolite = PermissibleValue(text=\"metabolite\")\n",
      "\n",
      "    _defn = EnumDefinition(\n",
      "        name=\"ChemicalEntityDerivativeEnum\",\n",
      "    )\n",
      "\n",
      "class ChemicalOrGeneOrGeneProductFormOrVariantEnum(EnumDefinitionImpl):\n",
      "\n",
      "    genetic_variant_form = PermissibleValue(text=\"genetic_variant_form\")\n",
      "    modified_form = PermissibleValue(text=\"modified_form\")\n",
      "    loss_of_function_variant_form = PermissibleValue(text=\"loss_of_function_variant_form\")\n",
      "    gain_of_function_variant_form = PermissibleValue(text=\"gain_of_function_variant_form\")\n",
      "    polymorphic_form = PermissibleValue(text=\"polymorphic_form\")\n",
      "    snp_form = PermissibleValue(text=\"snp_form\")\n",
      "    analog_form = PermissibleValue(text=\"analog_form\")\n",
      "\n",
      "    _defn = EnumDefinition(\n",
      "        name=\"ChemicalOrGeneOrGeneProductFormOrVariantEnum\",\n",
      "    )\n",
      "\n",
      "class GeneOrGeneProductOrChemicalPartQualifierEnum(EnumDefinitionImpl):\n",
      "\n",
      "    polya_tail = PermissibleValue(text=\"polya_tail\")\n",
      "    promoter = PermissibleValue(text=\"promoter\")\n",
      "    enhancer = PermissibleValue(text=\"enhancer\")\n",
      "    exon = PermissibleValue(text=\"exon\")\n",
      "    intron = PermissibleValue(text=\"intron\")\n",
      "\n",
      "    _defn = EnumDefinition(\n",
      "        name=\"GeneOrGeneProductOrChemicalPartQualifierEnum\",\n",
      "    )\n",
      "\n",
      "    @classmethod\n",
      "    def _addvals(cls):\n",
      "        setattr(cls, \"3_prime_utr\",\n",
      "            PermissibleValue(text=\"3_prime_utr\"))\n",
      "        setattr(cls, \"5_prime_utr\",\n",
      "            PermissibleValue(text=\"5_prime_utr\"))\n",
      "\n",
      "class GeneOrGeneProductOrChemicalEntityAspectEnum(EnumDefinitionImpl):\n",
      "\n",
      "    activity_or_abundance = PermissibleValue(\n",
      "        text=\"activity_or_abundance\",\n",
      "        description=\"\"\"Used in cases where the specificity of the relationship can not be determined to be either activity or abundance.  In general, a more specific value from this enumeration should be used.\"\"\")\n",
      "    abundance = PermissibleValue(text=\"abundance\")\n",
      "    activity = PermissibleValue(text=\"activity\")\n",
      "    expression = PermissibleValue(text=\"expression\")\n",
      "    synthesis = PermissibleValue(text=\"synthesis\")\n",
      "    degradation = PermissibleValue(text=\"degradation\")\n",
      "    cleavage = PermissibleValue(text=\"cleavage\")\n",
      "    hydrolysis = PermissibleValue(text=\"hydrolysis\")\n",
      "    metabolic_processing = PermissibleValue(text=\"metabolic_processing\")\n",
      "    mutation_rate = PermissibleValue(text=\"mutation_rate\")\n",
      "    stability = PermissibleValue(text=\"stability\")\n",
      "    folding = PermissibleValue(text=\"folding\")\n",
      "    localization = PermissibleValue(text=\"localization\")\n",
      "    transport = PermissibleValue(text=\"transport\")\n",
      "    absorption = PermissibleValue(text=\"absorption\")\n",
      "    aggregation = PermissibleValue(text=\"aggregation\")\n",
      "    interaction = PermissibleValue(text=\"interaction\")\n",
      "    release = PermissibleValue(text=\"release\")\n",
      "    secretion = PermissibleValue(text=\"secretion\")\n",
      "    uptake = PermissibleValue(text=\"uptake\")\n",
      "    splicing = PermissibleValue(text=\"splicing\")\n",
      "    molecular_interaction = PermissibleValue(text=\"molecular_interaction\")\n",
      "    guanyl_nucleotide_exchange = PermissibleValue(text=\"guanyl_nucleotide_exchange\")\n",
      "    adenyl_nucleotide_exchange = PermissibleValue(text=\"adenyl_nucleotide_exchange\")\n",
      "    molecular_modification = PermissibleValue(text=\"molecular_modification\")\n",
      "    acetylation = PermissibleValue(text=\"acetylation\")\n",
      "    acylation = PermissibleValue(text=\"acylation\")\n",
      "    alkylation = PermissibleValue(text=\"alkylation\")\n",
      "    amination = PermissibleValue(text=\"amination\")\n",
      "    carbamoylation = PermissibleValue(text=\"carbamoylation\")\n",
      "    ethylation = PermissibleValue(text=\"ethylation\")\n",
      "    glutathionylation = PermissibleValue(text=\"glutathionylation\")\n",
      "    glycation = PermissibleValue(text=\"glycation\")\n",
      "    glycosylation = PermissibleValue(text=\"glycosylation\")\n",
      "    glucuronidation = PermissibleValue(text=\"glucuronidation\")\n",
      "    n_linked_glycosylation = PermissibleValue(text=\"n_linked_glycosylation\")\n",
      "    o_linked_glycosylation = PermissibleValue(text=\"o_linked_glycosylation\")\n",
      "    hydroxylation = PermissibleValue(text=\"hydroxylation\")\n",
      "    lipidation = PermissibleValue(text=\"lipidation\")\n",
      "    farnesylation = PermissibleValue(text=\"farnesylation\")\n",
      "    geranoylation = PermissibleValue(text=\"geranoylation\")\n",
      "    myristoylation = PermissibleValue(text=\"myristoylation\")\n",
      "    palmitoylation = PermissibleValue(text=\"palmitoylation\")\n",
      "    prenylation = PermissibleValue(text=\"prenylation\")\n",
      "    methylation = PermissibleValue(text=\"methylation\")\n",
      "    nitrosation = PermissibleValue(text=\"nitrosation\")\n",
      "    nucleotidylation = PermissibleValue(text=\"nucleotidylation\")\n",
      "    phosphorylation = PermissibleValue(text=\"phosphorylation\")\n",
      "    ribosylation = PermissibleValue(text=\"ribosylation\")\n",
      "    sulfation = PermissibleValue(text=\"sulfation\")\n",
      "    sumoylation = PermissibleValue(text=\"sumoylation\")\n",
      "    ubiquitination = PermissibleValue(text=\"ubiquitination\")\n",
      "    oxidation = PermissibleValue(text=\"oxidation\")\n",
      "    reduction = PermissibleValue(text=\"reduction\")\n",
      "    carboxylation = PermissibleValue(text=\"carboxylation\")\n",
      "\n",
      "    _defn = EnumDefinition(\n",
      "        name=\"GeneOrGeneProductOrChemicalEntityAspectEnum\",\n",
      "    )\n",
      "\n",
      "    @classmethod\n",
      "    def _addvals(cls):\n",
      "        setattr(cls, \"ADP-ribosylation\",\n",
      "            PermissibleValue(text=\"ADP-ribosylation\"))\n",
      "\n",
      "class CausalMechanismQualifierEnum(EnumDefinitionImpl):\n",
      "\n",
      "    binding = PermissibleValue(\n",
      "        text=\"binding\",\n",
      "        description=\"\"\"A causal mechanism mediated by the direct contact between effector and target chemical or biomolecular entity, which form a stable physical interaction.\"\"\")\n",
      "    inhibition = PermissibleValue(\n",
      "        text=\"inhibition\",\n",
      "        description=\"\"\"A causal mechanism in which the effector binds to the target and negatively effects its normal function, e.g. prevention of enzymatic reaction or activation of downstream pathway.\"\"\")\n",
      "    antibody_inhibition = PermissibleValue(\n",
      "        text=\"antibody_inhibition\",\n",
      "        description=\"A causal mechanism in which an antibody specifically binds to and interferes with the target.\")\n",
      "    antagonism = PermissibleValue(\n",
      "        text=\"antagonism\",\n",
      "        description=\"\"\"A causal mechanism in which the effector binds to a receptor and prevents activation by an agonist through competing for the binding site.\"\"\")\n",
      "    molecular_channel_blockage = PermissibleValue(\n",
      "        text=\"molecular_channel_blockage\",\n",
      "        description=\"\"\"A causal mechanism in which the effector binds to a molecular channel and prevents or reduces transport of ions through it.\"\"\")\n",
      "    inverse_agonism = PermissibleValue(\n",
      "        text=\"inverse_agonism\",\n",
      "        description=\"\"\"A causal mechanism in which the effector binds to the same receptor-binding site as an agonist and antagonizes its effects, often exerting the opposite effect of the agonist by suppressing spontaneous receptor signaling.\"\"\")\n",
      "    negative_allosteric_modulation = PermissibleValue(\n",
      "        text=\"negative_allosteric_modulation\",\n",
      "        description=\"\"\"A causal mechanism in which the effector reduces or prevents the action of the endogenous ligand of a receptor by binding to a site distinct from that ligand (i.e. non-competitive inhibition)\"\"\")\n",
      "    agonism = PermissibleValue(\n",
      "        text=\"agonism\",\n",
      "        description=\"\"\"A causal mechanism in which the effector binds and activates a receptor to mimic the effect of an endogenous ligand.\"\"\")\n",
      "    molecular_channel_opening = PermissibleValue(\n",
      "        text=\"molecular_channel_opening\",\n",
      "        description=\"\"\"A causal mechanism in which the effector binds to a molecular channel and facilitates transport of ions through it.\"\"\")\n",
      "    positive_allosteric_modulation = PermissibleValue(\n",
      "        text=\"positive_allosteric_modulation\",\n",
      "        description=\"\"\"A causal mechanism in which the effector enhances the action of the endogenous ligand of a receptor by binding to a site distinct from that ligand (i.e. non-competitive inhibition)\"\"\")\n",
      "    potentiation = PermissibleValue(\n",
      "        text=\"potentiation\",\n",
      "        description=\"\"\"A causal mechanism in which the effector  binds to and enhances or intensifies the effect of some other chemical or drug on its target.\"\"\")\n",
      "    activation = PermissibleValue(\n",
      "        text=\"activation\",\n",
      "        description=\"\"\"A causal mechanism in which the effector binds to and positively affects the normal functioning of its target.\"\"\")\n",
      "    inducer = PermissibleValue(\n",
      "        text=\"inducer\",\n",
      "        description=\"\"\"A causal mechanism in which the effector binds to and increases the activity/rate of an enzyme that processes drugs in the body.\"\"\")\n",
      "    transcriptional_regulation = PermissibleValue(\n",
      "        text=\"transcriptional_regulation\",\n",
      "        description=\"A causal mechanism mediated by through the control of target gene transcription\")\n",
      "    signaling_mediated_control = PermissibleValue(\n",
      "        text=\"signaling_mediated_control\",\n",
      "        description=\"\"\"A causal mechanism mediated by the activation or control of signaling events that influence the some aspect of the target entity (e.g. its activity, processing, transport, etc)\"\"\")\n",
      "    stabilization = PermissibleValue(text=\"stabilization\")\n",
      "    stimulation = PermissibleValue(text=\"stimulation\")\n",
      "    releasing_activity = PermissibleValue(text=\"releasing_activity\")\n",
      "\n",
      "    _defn = EnumDefinition(\n",
      "        name=\"CausalMechanismQualifierEnum\",\n",
      "    )\n",
      "\n",
      "class LogicalInterpretationEnum(EnumDefinitionImpl):\n",
      "\n",
      "    some_some = PermissibleValue(\n",
      "        text=\"some_some\",\n",
      "        description=\"A modifier on a triple that causes the triple to be interpreted as a some-some statement\",\n",
      "        meaning=OS[\"SomeSomeInterpretation\"])\n",
      "    all_some = PermissibleValue(\n",
      "        text=\"all_some\",\n",
      "        description=\"A modifier on a triple that causes the triple to be interpreted as an all-some statement.\",\n",
      "        meaning=OS[\"AllSomeInterpretation\"])\n",
      "    inverse_all_some = PermissibleValue(text=\"inverse_all_some\")\n",
      "\n",
      "    _defn = EnumDefinition(\n",
      "        name=\"LogicalInterpretationEnum\",\n",
      "    )\n",
      "\n",
      "class ReactionDirectionEnum(EnumDefinitionImpl):\n",
      "\n",
      "    left_to_right = PermissibleValue(text=\"left_to_right\")\n",
      "    right_to_left = PermissibleValue(text=\"right_to_left\")\n",
      "    bidirectional = PermissibleValue(text=\"bidirectional\")\n",
      "    neutral = PermissibleValue(text=\"neutral\")\n",
      "\n",
      "    _defn = EnumDefinition(\n",
      "        name=\"ReactionDirectionEnum\",\n",
      "    )\n",
      "\n",
      "class ReactionSideEnum(EnumDefinitionImpl):\n",
      "\n",
      "    left = PermissibleValue(text=\"left\")\n",
      "    right = PermissibleValue(text=\"right\")\n",
      "\n",
      "    _defn = EnumDefinition(\n",
      "        name=\"ReactionSideEnum\",\n",
      "    )\n",
      "\n",
      "class PhaseEnum(EnumDefinitionImpl):\n",
      "    \"\"\"\n",
      "    phase\n",
      "    \"\"\"\n",
      "    _defn = EnumDefinition(\n",
      "        name=\"PhaseEnum\",\n",
      "        description=\"phase\",\n",
      "    )\n",
      "\n",
      "    @classmethod\n",
      "    def _addvals(cls):\n",
      "        setattr(cls, \"0\",\n",
      "            PermissibleValue(text=\"0\"))\n",
      "        setattr(cls, \"1\",\n",
      "            PermissibleValue(text=\"1\"))\n",
      "        setattr(cls, \"2\",\n",
      "            PermissibleValue(text=\"2\"))\n",
      "\n",
      "class StrandEnum(EnumDefinitionImpl):\n",
      "    \"\"\"\n",
      "    strand\n",
      "    \"\"\"\n",
      "    _defn = EnumDefinition(\n",
      "        name=\"StrandEnum\",\n",
      "        description=\"strand\",\n",
      "    )\n",
      "\n",
      "    @classmethod\n",
      "    def _addvals(cls):\n",
      "        setattr(cls, \"+\",\n",
      "            PermissibleValue(\n",
      "                text=\"+\",\n",
      "                description=\"Positive\"))\n",
      "        setattr(cls, \"-\",\n",
      "            PermissibleValue(\n",
      "                text=\"-\",\n",
      "                description=\"Negative\"))\n",
      "        setattr(cls, \".\",\n",
      "            PermissibleValue(\n",
      "                text=\".\",\n",
      "                description=\"Unstranded\"))\n",
      "        setattr(cls, \"?\",\n",
      "            PermissibleValue(\n",
      "                text=\"?\",\n",
      "                description=\"Unknown\"))\n",
      "\n",
      "class SequenceEnum(EnumDefinitionImpl):\n",
      "    \"\"\"\n",
      "    type of sequence\n",
      "    \"\"\"\n",
      "    na = PermissibleValue(\n",
      "        text=\"na\",\n",
      "        description=\"nucleic acid\")\n",
      "    aa = PermissibleValue(\n",
      "        text=\"aa\",\n",
      "        description=\"amino acid\")\n",
      "\n",
      "    _defn = EnumDefinition(\n",
      "        name=\"SequenceEnum\",\n",
      "        description=\"type of sequence\",\n",
      "    )\n",
      "\n",
      "class DruggableGeneCategoryEnum(EnumDefinitionImpl):\n",
      "\n",
      "    tclin = PermissibleValue(\n",
      "        text=\"tclin\",\n",
      "        description=\"\"\"These targets have activities in DrugCentral (ie. approved drugs) with known mechanism of action.\"\"\")\n",
      "    tbio = PermissibleValue(\n",
      "        text=\"tbio\",\n",
      "        description=\"\"\"These targets have activities in ChEMBL, Guide to Pharmacology or DrugCentral that satisfy the activity thresholds detailed below.\"\"\")\n",
      "    tchem = PermissibleValue(\n",
      "        text=\"tchem\",\n",
      "        description=\"\"\"These targets do not have known drug or small molecule activities that satisfy the activity thresholds detailed below AND satisfy one or more of the following criteria: target is above the cutoff criteria for Tdark target is annotated with a Gene Ontology Molecular Function or Biological Process leaf term(s) with an Experimental Evidence code\"\"\")\n",
      "    tdark = PermissibleValue(\n",
      "        text=\"tdark\",\n",
      "        description=\"\"\"These are targets about which virtually nothing is known. They do not have known drug or small molecule activities that satisfy the activity thresholds detailed below AND satisfy two or more of the following criteria: A PubMed text-mining score from Jensen Lab less than 5, greater than or equal TO 3 Gene RIFs, or less than or equal to 50 Antibodies available according to http://antibodypedia.com.\"\"\")\n",
      "\n",
      "    _defn = EnumDefinition(\n",
      "        name=\"DruggableGeneCategoryEnum\",\n",
      "    )\n",
      "\n",
      "class DrugAvailabilityEnum(EnumDefinitionImpl):\n",
      "\n",
      "    over_the_counter = PermissibleValue(\n",
      "        text=\"over_the_counter\",\n",
      "        description=\"chemical entity is available over the counter without a prescription.\")\n",
      "    prescription = PermissibleValue(\n",
      "        text=\"prescription\",\n",
      "        description=\"chemical entity is available by prescription.\")\n",
      "\n",
      "    _defn = EnumDefinition(\n",
      "        name=\"DrugAvailabilityEnum\",\n",
      "    )\n",
      "\n",
      "class DrugDeliveryEnum(EnumDefinitionImpl):\n",
      "\n",
      "    inhalation = PermissibleValue(text=\"inhalation\")\n",
      "    oral = PermissibleValue(text=\"oral\")\n",
      "    absorption_through_the_skin = PermissibleValue(text=\"absorption_through_the_skin\")\n",
      "    intravenous_injection = PermissibleValue(text=\"intravenous_injection\")\n",
      "\n",
      "    _defn = EnumDefinition(\n",
      "        name=\"DrugDeliveryEnum\",\n",
      "    )\n",
      "\n",
      "class ResourceRoleEnum(EnumDefinitionImpl):\n",
      "    \"\"\"\n",
      "    The role played by the information reource in serving as a source for an edge in a TRAPI message. Note that a\n",
      "    given Edge should have one and only one 'primary' source, and may have any number of 'aggregator' or 'supporting\n",
      "    data' sources. This enumeration is found in Biolink Model, but is repeated here for convenience.\n",
      "    \"\"\"\n",
      "    primary_knowledge_source = PermissibleValue(text=\"primary_knowledge_source\")\n",
      "    aggregator_knowledge_source = PermissibleValue(text=\"aggregator_knowledge_source\")\n",
      "    supporting_data_source = PermissibleValue(text=\"supporting_data_source\")\n",
      "\n",
      "    _defn = EnumDefinition(\n",
      "        name=\"ResourceRoleEnum\",\n",
      "        description=\"\"\"The role played by the information reource in serving as a source for an edge in a TRAPI message. Note that a given Edge should have one and only one 'primary' source, and may have any number of 'aggregator' or 'supporting data' sources.  This enumeration is found in Biolink Model, but is repeated here for convenience.\"\"\",\n",
      "    )\n",
      "\n",
      "class FDAIDAAdverseEventEnum(EnumDefinitionImpl):\n",
      "    \"\"\"\n",
      "    please consult with the FDA guidelines as proposed in this document:\n",
      "    https://www.accessdata.fda.gov/scripts/cdrh/cfdocs/cfcfr/cfrsearch.cfm?fr=312.32\n",
      "    \"\"\"\n",
      "    life_threatening_adverse_event = PermissibleValue(\n",
      "        text=\"life_threatening_adverse_event\",\n",
      "        description=\"\"\"An adverse event or suspected adverse reaction is considered 'life-threatening' if, in the view of either the investigator or sponsor, its occurrence places the patient or subject at immediate risk of death. It does not include an adverse event or suspected adverse reaction that, had it occurred in a more severe form, might have caused death.\"\"\")\n",
      "    serious_adverse_event = PermissibleValue(\n",
      "        text=\"serious_adverse_event\",\n",
      "        description=\"\"\"An adverse event or suspected adverse reaction is considered 'serious' if, in the view of either the investigator or sponsor, it results in any of the following outcomes: Death, a life-threatening adverse event, inpatient hospitalization or prolongation of existing hospitalization, a persistent or significant incapacity or substantial disruption of the ability to conduct normal life functions, or a congenital anomaly/birth defect. Important medical events that may not result in death, be life-threatening, or require hospitalization may be considered serious when, based upon appropriate medical judgment, they may jeopardize the patient or subject and may require medical or surgical intervention to prevent one of the outcomes listed in this definition. Examples of such medical events include allergic bronchospasm requiring intensive treatment in an emergency room or at home, blood dyscrasias or convulsions that do not result in inpatient hospitalization, or the development of drug dependency or drug abuse.\"\"\")\n",
      "    suspected_adverse_reaction = PermissibleValue(\n",
      "        text=\"suspected_adverse_reaction\",\n",
      "        description=\"\"\"means any adverse event for which there is a reasonable possibility that the drug caused the adverse event. For the purposes of IND safety reporting, 'reasonable possibility' means there is evidence to suggest a causal relationship between the drug and the adverse event. Suspected adverse reaction implies a lesser degree of certainty about causality than adverse reaction, which means any adverse event caused by a drug.\"\"\")\n",
      "    unexpected_adverse_event = PermissibleValue(\n",
      "        text=\"unexpected_adverse_event\",\n",
      "        description=\"\"\"An adverse event or suspected adverse reaction is considered 'unexpected' if it is not listed in the investigator brochure or is not listed at the specificity or severity that has been observed; or, if an investigator brochure is not required or available, is not consistent with the risk information described in the general investigational plan or elsewhere in the current application, as amended. For example, under this definition, hepatic necrosis would be unexpected (by virtue of greater severity) if the investigator brochure referred only to elevated hepatic enzymes or hepatitis. Similarly, cerebral thromboembolism and cerebral vasculitis would be unexpected (by virtue of greater specificity) if the investigator brochure listed only cerebral vascular accidents. 'Unexpected', as used in this definition, also refers to adverse events or suspected adverse reactions that are mentioned in the investigator brochure as occurring with a class of drugs or as anticipated from the pharmacological properties of the drug, but are not specifically mentioned as occurring with the particular drug under investigation.\"\"\")\n",
      "\n",
      "    _defn = EnumDefinition(\n",
      "        name=\"FDAIDAAdverseEventEnum\",\n",
      "        description=\"\"\"please consult with the FDA guidelines as proposed in this document: https://www.accessdata.fda.gov/scripts/cdrh/cfdocs/cfcfr/cfrsearch.cfm?fr=312.32\"\"\",\n",
      "    )\n",
      "\n",
      "class AgentTypeEnum(EnumDefinitionImpl):\n",
      "\n",
      "    manual_agent = PermissibleValue(\n",
      "        text=\"manual_agent\",\n",
      "        description=\"\"\"A human agent who is responsible for generating a statement of knowledge. The human may utilize computationally generated information as evidence for the resulting knowledge,  but the human is the one who ultimately interprets/reasons with  this evidence to produce a statement of knowledge.\"\"\")\n",
      "    automated_agent = PermissibleValue(\n",
      "        text=\"automated_agent\",\n",
      "        description=\"\"\"An automated agent, typically a software program or tool, that is  responsible for generating a statement of knowledge. Human contribution  to the knowledge creation process ends with the definition and coding of algorithms or analysis pipelines that get executed by the automated agent.\"\"\")\n",
      "    data_analysis_pipeline = PermissibleValue(\n",
      "        text=\"data_analysis_pipeline\",\n",
      "        description=\"\"\"An automated agent that executes an analysis workflow over data and  reports the direct results of the analysis. These typically report  statistical associations/correlations between variables in the input dataset, and do not interpret/infer broader conclusions from associations the analysis reveals in the data.\"\"\")\n",
      "    computational_model = PermissibleValue(\n",
      "        text=\"computational_model\",\n",
      "        description=\"\"\"An automated agent that generates knowledge statements (typically predictions) based on rules/logic explicitly encoded in an algorithm (e.g. heuristic models, supervised classifiers), or learned from patterns  observed in data (e.g. ML models, unsupervised classifiers).\"\"\")\n",
      "    text_mining_agent = PermissibleValue(\n",
      "        text=\"text_mining_agent\",\n",
      "        description=\"\"\"An automated agent that uses Natural Language Processing to recognize concepts and/or relationships in text, and report them using formally encoded semantics (e.g. as an edge in a knowledge graph).\"\"\")\n",
      "    image_processing_agent = PermissibleValue(\n",
      "        text=\"image_processing_agent\",\n",
      "        description=\"\"\"An automated agent that processes images to generate textual statements of  knowledge derived from the image and/or expressed in text the image  depicts (e.g. via OCR).\"\"\")\n",
      "    manual_validation_of_automated_agent = PermissibleValue(\n",
      "        text=\"manual_validation_of_automated_agent\",\n",
      "        description=\"\"\"A human agent reviews and validates/approves the veracity of knowledge  that is initially generated by an automated agent.\"\"\")\n",
      "    not_provided = PermissibleValue(\n",
      "        text=\"not_provided\",\n",
      "        description=\"\"\"The agent type is not provided, typically because it cannot be determined from available information if the agent that generated the knowledge is  manual or automated.\"\"\")\n",
      "\n",
      "    _defn = EnumDefinition(\n",
      "        name=\"AgentTypeEnum\",\n",
      "    )\n",
      "\n",
      "class KnowledgeLevelEnum(EnumDefinitionImpl):\n",
      "\n",
      "    knowledge_assertion = PermissibleValue(\n",
      "        text=\"knowledge_assertion\",\n",
      "        description=\"\"\"A statement of purported fact that is put forth by an agent as true, based on assessment of direct evidence. Assertions are likely but not  definitively true.\"\"\")\n",
      "    logical_entailment = PermissibleValue(\n",
      "        text=\"logical_entailment\",\n",
      "        description=\"\"\"A statement reporting a conclusion that follows logically from premises representing established facts or knowledge assertions (e.g. fingernail part of finger, finger part of hand --> fingernail part of hand).\"\"\")\n",
      "    prediction = PermissibleValue(\n",
      "        text=\"prediction\",\n",
      "        description=\"\"\"A statement of a possible fact based on probabilistic forms of reasoning over more indirect forms of evidence, that lead to more speculative conclusions.\"\"\")\n",
      "    statistical_association = PermissibleValue(\n",
      "        text=\"statistical_association\",\n",
      "        description=\"\"\"A statement that reports concepts representing variables in a dataset to be statistically associated with each other in a particular cohort (e.g. 'Metformin Treatment (variable 1) is correlated with Diabetes Diagnosis (variable 2) in EHR dataset X').\"\"\")\n",
      "    observation = PermissibleValue(\n",
      "        text=\"observation\",\n",
      "        description=\"\"\"A statement reporting (and possibly quantifying) a phenomenon that was observed to occur -  absent any analysis or interpretation that generates a statistical association or supports a broader conclusion or inference.\"\"\")\n",
      "    not_provided = PermissibleValue(\n",
      "        text=\"not_provided\",\n",
      "        description=\"\"\"The knowledge level is not provided, typically because it cannot be determined from available. information.\"\"\")\n",
      "\n",
      "    _defn = EnumDefinition(\n",
      "        name=\"KnowledgeLevelEnum\",\n",
      "    )\n",
      "\n",
      "# Slots\n",
      "class slots:\n",
      "    pass\n",
      "\n",
      "slots.gene_has_sequence = Slot(uri=DIST.hasSeq, name=\"gene has sequence\", curie=DIST.curie('hasSeq'),\n",
      "                   model_uri=DIST.gene_has_sequence, domain=Gene, range=Union[str, GeneSequence])\n",
      "\n",
      "slots.has_attribute = Slot(uri=BIOLINK.has_attribute, name=\"has attribute\", curie=BIOLINK.curie('has_attribute'),\n",
      "                   model_uri=DIST.has_attribute, domain=Entity, range=Optional[Union[Union[str, AttributeId], List[Union[str, AttributeId]]]])\n",
      "\n",
      "slots.has_attribute_type = Slot(uri=BIOLINK.has_attribute_type, name=\"has attribute type\", curie=BIOLINK.curie('has_attribute_type'),\n",
      "                   model_uri=DIST.has_attribute_type, domain=Attribute, range=Union[str, OntologyClassId])\n",
      "\n",
      "slots.has_qualitative_value = Slot(uri=BIOLINK.has_qualitative_value, name=\"has qualitative value\", curie=BIOLINK.curie('has_qualitative_value'),\n",
      "                   model_uri=DIST.has_qualitative_value, domain=Attribute, range=Optional[Union[str, NamedThingId]])\n",
      "\n",
      "slots.has_quantitative_value = Slot(uri=BIOLINK.has_quantitative_value, name=\"has quantitative value\", curie=BIOLINK.curie('has_quantitative_value'),\n",
      "                   model_uri=DIST.has_quantitative_value, domain=Attribute, range=Optional[Union[Union[dict, QuantityValue], List[Union[dict, QuantityValue]]]])\n",
      "\n",
      "slots.has_numeric_value = Slot(uri=BIOLINK.has_numeric_value, name=\"has numeric value\", curie=BIOLINK.curie('has_numeric_value'),\n",
      "                   model_uri=DIST.has_numeric_value, domain=QuantityValue, range=Optional[float])\n",
      "\n",
      "slots.has_unit = Slot(uri=BIOLINK.has_unit, name=\"has unit\", curie=BIOLINK.curie('has_unit'),\n",
      "                   model_uri=DIST.has_unit, domain=QuantityValue, range=Optional[Union[str, Unit]])\n",
      "\n",
      "slots.base_coordinate = Slot(uri=BIOLINK.base_coordinate, name=\"base coordinate\", curie=BIOLINK.curie('base_coordinate'),\n",
      "                   model_uri=DIST.base_coordinate, domain=GenomicSequenceLocalization, range=Optional[int])\n",
      "\n",
      "slots.node_property = Slot(uri=BIOLINK.node_property, name=\"node property\", curie=BIOLINK.curie('node_property'),\n",
      "                   model_uri=DIST.node_property, domain=NamedThing, range=Optional[str])\n",
      "\n",
      "slots.id = Slot(uri=BIOLINK.id, name=\"id\", curie=BIOLINK.curie('id'),\n",
      "                   model_uri=DIST.id, domain=Entity, range=Union[str, EntityId])\n",
      "\n",
      "slots.iri = Slot(uri=BIOLINK.iri, name=\"iri\", curie=BIOLINK.curie('iri'),\n",
      "                   model_uri=DIST.iri, domain=None, range=Optional[Union[str, IriType]])\n",
      "\n",
      "slots.type = Slot(uri=RDF.type, name=\"type\", curie=RDF.curie('type'),\n",
      "                   model_uri=DIST.type, domain=Entity, range=Optional[Union[str, List[str]]])\n",
      "\n",
      "slots.category = Slot(uri=BIOLINK.category, name=\"category\", curie=BIOLINK.curie('category'),\n",
      "                   model_uri=DIST.category, domain=Entity, range=Optional[Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]]])\n",
      "\n",
      "slots.publication_type = Slot(uri=DCT.type, name=\"publication type\", curie=DCT.curie('type'),\n",
      "                   model_uri=DIST.publication_type, domain=None, range=Optional[Union[str, List[str]]])\n",
      "\n",
      "slots.name = Slot(uri=RDFS.label, name=\"name\", curie=RDFS.curie('label'),\n",
      "                   model_uri=DIST.name, domain=Entity, range=Optional[Union[str, LabelType]])\n",
      "\n",
      "slots.stoichiometry = Slot(uri=BIOLINK.stoichiometry, name=\"stoichiometry\", curie=BIOLINK.curie('stoichiometry'),\n",
      "                   model_uri=DIST.stoichiometry, domain=Association, range=Optional[int])\n",
      "\n",
      "slots.reaction_direction = Slot(uri=BIOLINK.reaction_direction, name=\"reaction direction\", curie=BIOLINK.curie('reaction_direction'),\n",
      "                   model_uri=DIST.reaction_direction, domain=Association, range=Optional[Union[str, \"ReactionDirectionEnum\"]])\n",
      "\n",
      "slots.reaction_balanced = Slot(uri=BIOLINK.reaction_balanced, name=\"reaction balanced\", curie=BIOLINK.curie('reaction_balanced'),\n",
      "                   model_uri=DIST.reaction_balanced, domain=Association, range=Optional[Union[bool, Bool]])\n",
      "\n",
      "slots.reaction_side = Slot(uri=BIOLINK.reaction_side, name=\"reaction side\", curie=BIOLINK.curie('reaction_side'),\n",
      "                   model_uri=DIST.reaction_side, domain=Association, range=Optional[Union[str, \"ReactionSideEnum\"]])\n",
      "\n",
      "slots.symbol = Slot(uri=BIOLINK.symbol, name=\"symbol\", curie=BIOLINK.curie('symbol'),\n",
      "                   model_uri=DIST.symbol, domain=NamedThing, range=Optional[str])\n",
      "\n",
      "slots.synonym = Slot(uri=BIOLINK.synonym, name=\"synonym\", curie=BIOLINK.curie('synonym'),\n",
      "                   model_uri=DIST.synonym, domain=NamedThing, range=Optional[Union[Union[str, LabelType], List[Union[str, LabelType]]]])\n",
      "\n",
      "slots.exact_synonym = Slot(uri=BIOLINK.exact_synonym, name=\"exact synonym\", curie=BIOLINK.curie('exact_synonym'),\n",
      "                   model_uri=DIST.exact_synonym, domain=NamedThing, range=Optional[Union[Union[str, LabelType], List[Union[str, LabelType]]]])\n",
      "\n",
      "slots.broad_synonym = Slot(uri=BIOLINK.broad_synonym, name=\"broad synonym\", curie=BIOLINK.curie('broad_synonym'),\n",
      "                   model_uri=DIST.broad_synonym, domain=NamedThing, range=Optional[Union[Union[str, LabelType], List[Union[str, LabelType]]]])\n",
      "\n",
      "slots.narrow_synonym = Slot(uri=BIOLINK.narrow_synonym, name=\"narrow synonym\", curie=BIOLINK.curie('narrow_synonym'),\n",
      "                   model_uri=DIST.narrow_synonym, domain=NamedThing, range=Optional[Union[Union[str, LabelType], List[Union[str, LabelType]]]])\n",
      "\n",
      "slots.related_synonym = Slot(uri=BIOLINK.related_synonym, name=\"related synonym\", curie=BIOLINK.curie('related_synonym'),\n",
      "                   model_uri=DIST.related_synonym, domain=NamedThing, range=Optional[Union[Union[str, LabelType], List[Union[str, LabelType]]]])\n",
      "\n",
      "slots.deprecated = Slot(uri=BIOLINK.deprecated, name=\"deprecated\", curie=BIOLINK.curie('deprecated'),\n",
      "                   model_uri=DIST.deprecated, domain=None, range=Optional[Union[bool, Bool]])\n",
      "\n",
      "slots.has_topic = Slot(uri=BIOLINK.has_topic, name=\"has topic\", curie=BIOLINK.curie('has_topic'),\n",
      "                   model_uri=DIST.has_topic, domain=NamedThing, range=Optional[Union[str, OntologyClassId]])\n",
      "\n",
      "slots.xref = Slot(uri=BIOLINK.xref, name=\"xref\", curie=BIOLINK.curie('xref'),\n",
      "                   model_uri=DIST.xref, domain=NamedThing, range=Optional[Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]]])\n",
      "\n",
      "slots.url = Slot(uri=BIOLINK.url, name=\"url\", curie=BIOLINK.curie('url'),\n",
      "                   model_uri=DIST.url, domain=Entity, range=Optional[str])\n",
      "\n",
      "slots.semmed_agreement_count = Slot(uri=BIOLINK.semmed_agreement_count, name=\"semmed agreement count\", curie=BIOLINK.curie('semmed_agreement_count'),\n",
      "                   model_uri=DIST.semmed_agreement_count, domain=Association, range=Optional[int])\n",
      "\n",
      "slots.support_graphs = Slot(uri=BIOLINK.support_graphs, name=\"support graphs\", curie=BIOLINK.curie('support_graphs'),\n",
      "                   model_uri=DIST.support_graphs, domain=Association, range=Optional[Union[str, List[str]]])\n",
      "\n",
      "slots.resource_id = Slot(uri=BIOLINK.resource_id, name=\"resource id\", curie=BIOLINK.curie('resource_id'),\n",
      "                   model_uri=DIST.resource_id, domain=RetrievalSource, range=Optional[Union[str, URIorCURIE]])\n",
      "\n",
      "slots.resource_role = Slot(uri=BIOLINK.resource_role, name=\"resource role\", curie=BIOLINK.curie('resource_role'),\n",
      "                   model_uri=DIST.resource_role, domain=RetrievalSource, range=Optional[Union[str, \"ResourceRoleEnum\"]])\n",
      "\n",
      "slots.retrieval_source_ids = Slot(uri=BIOLINK.retrieval_source_ids, name=\"retrieval source ids\", curie=BIOLINK.curie('retrieval_source_ids'),\n",
      "                   model_uri=DIST.retrieval_source_ids, domain=None, range=Optional[Union[Union[str, RetrievalSourceId], List[Union[str, RetrievalSourceId]]]])\n",
      "\n",
      "slots.full_name = Slot(uri=BIOLINK.full_name, name=\"full name\", curie=BIOLINK.curie('full_name'),\n",
      "                   model_uri=DIST.full_name, domain=NamedThing, range=Optional[Union[str, LabelType]])\n",
      "\n",
      "slots.upstream_resource_ids = Slot(uri=BIOLINK.upstream_resource_ids, name=\"upstream resource ids\", curie=BIOLINK.curie('upstream_resource_ids'),\n",
      "                   model_uri=DIST.upstream_resource_ids, domain=RetrievalSource, range=Optional[Union[str, URIorCURIE]])\n",
      "\n",
      "slots.description = Slot(uri=DCT.description, name=\"description\", curie=DCT.curie('description'),\n",
      "                   model_uri=DIST.description, domain=None, range=Optional[Union[str, NarrativeText]])\n",
      "\n",
      "slots.systematic_synonym = Slot(uri=GOP.systematic_synonym, name=\"systematic synonym\", curie=GOP.curie('systematic_synonym'),\n",
      "                   model_uri=DIST.systematic_synonym, domain=NamedThing, range=Optional[Union[Union[str, LabelType], List[Union[str, LabelType]]]])\n",
      "\n",
      "slots.affiliation = Slot(uri=BIOLINK.affiliation, name=\"affiliation\", curie=BIOLINK.curie('affiliation'),\n",
      "                   model_uri=DIST.affiliation, domain=Agent, range=Optional[Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]]])\n",
      "\n",
      "slots.address = Slot(uri=BIOLINK.address, name=\"address\", curie=BIOLINK.curie('address'),\n",
      "                   model_uri=DIST.address, domain=NamedThing, range=Optional[str])\n",
      "\n",
      "slots.latitude = Slot(uri=BIOLINK.latitude, name=\"latitude\", curie=BIOLINK.curie('latitude'),\n",
      "                   model_uri=DIST.latitude, domain=NamedThing, range=Optional[float])\n",
      "\n",
      "slots.longitude = Slot(uri=BIOLINK.longitude, name=\"longitude\", curie=BIOLINK.curie('longitude'),\n",
      "                   model_uri=DIST.longitude, domain=NamedThing, range=Optional[float])\n",
      "\n",
      "slots.timepoint = Slot(uri=BIOLINK.timepoint, name=\"timepoint\", curie=BIOLINK.curie('timepoint'),\n",
      "                   model_uri=DIST.timepoint, domain=None, range=Optional[Union[str, TimeType]])\n",
      "\n",
      "slots.creation_date = Slot(uri=BIOLINK.creation_date, name=\"creation date\", curie=BIOLINK.curie('creation_date'),\n",
      "                   model_uri=DIST.creation_date, domain=NamedThing, range=Optional[Union[str, XSDDate]])\n",
      "\n",
      "slots.update_date = Slot(uri=BIOLINK.update_date, name=\"update date\", curie=BIOLINK.curie('update_date'),\n",
      "                   model_uri=DIST.update_date, domain=NamedThing, range=Optional[Union[str, XSDDate]])\n",
      "\n",
      "slots.aggregate_statistic = Slot(uri=BIOLINK.aggregate_statistic, name=\"aggregate statistic\", curie=BIOLINK.curie('aggregate_statistic'),\n",
      "                   model_uri=DIST.aggregate_statistic, domain=NamedThing, range=Optional[str])\n",
      "\n",
      "slots.has_count = Slot(uri=BIOLINK.has_count, name=\"has count\", curie=BIOLINK.curie('has_count'),\n",
      "                   model_uri=DIST.has_count, domain=NamedThing, range=Optional[int])\n",
      "\n",
      "slots.has_total = Slot(uri=BIOLINK.has_total, name=\"has total\", curie=BIOLINK.curie('has_total'),\n",
      "                   model_uri=DIST.has_total, domain=NamedThing, range=Optional[int])\n",
      "\n",
      "slots.has_quotient = Slot(uri=BIOLINK.has_quotient, name=\"has quotient\", curie=BIOLINK.curie('has_quotient'),\n",
      "                   model_uri=DIST.has_quotient, domain=NamedThing, range=Optional[float])\n",
      "\n",
      "slots.has_percentage = Slot(uri=BIOLINK.has_percentage, name=\"has percentage\", curie=BIOLINK.curie('has_percentage'),\n",
      "                   model_uri=DIST.has_percentage, domain=NamedThing, range=Optional[float])\n",
      "\n",
      "slots.has_taxonomic_rank = Slot(uri=BIOLINK.has_taxonomic_rank, name=\"has taxonomic rank\", curie=BIOLINK.curie('has_taxonomic_rank'),\n",
      "                   model_uri=DIST.has_taxonomic_rank, domain=NamedThing, range=Optional[Union[str, TaxonomicRankId]], mappings = [WIKIDATA[\"P105\"]])\n",
      "\n",
      "slots.has_dataset = Slot(uri=BIOLINK.has_dataset, name=\"has dataset\", curie=BIOLINK.curie('has_dataset'),\n",
      "                   model_uri=DIST.has_dataset, domain=DatasetVersion, range=Optional[Union[str, DatasetId]])\n",
      "\n",
      "slots.source_web_page = Slot(uri=BIOLINK.source_web_page, name=\"source web page\", curie=BIOLINK.curie('source_web_page'),\n",
      "                   model_uri=DIST.source_web_page, domain=DatasetSummary, range=Optional[str])\n",
      "\n",
      "slots.source_logo = Slot(uri=SCHEMA.logo, name=\"source logo\", curie=SCHEMA.curie('logo'),\n",
      "                   model_uri=DIST.source_logo, domain=DatasetSummary, range=Optional[str])\n",
      "\n",
      "slots.retrieved_on = Slot(uri=BIOLINK.retrieved_on, name=\"retrieved on\", curie=BIOLINK.curie('retrieved_on'),\n",
      "                   model_uri=DIST.retrieved_on, domain=Dataset, range=Optional[Union[str, XSDDate]])\n",
      "\n",
      "slots.version_of = Slot(uri=BIOLINK.version_of, name=\"version of\", curie=BIOLINK.curie('version_of'),\n",
      "                   model_uri=DIST.version_of, domain=DatasetVersion, range=Optional[Union[str, DatasetSummaryId]])\n",
      "\n",
      "slots.version = Slot(uri=BIOLINK.version, name=\"version\", curie=BIOLINK.curie('version'),\n",
      "                   model_uri=DIST.version, domain=Dataset, range=Optional[str])\n",
      "\n",
      "slots.license = Slot(uri=BIOLINK.license, name=\"license\", curie=BIOLINK.curie('license'),\n",
      "                   model_uri=DIST.license, domain=InformationContentEntity, range=Optional[str])\n",
      "\n",
      "slots.rights = Slot(uri=BIOLINK.rights, name=\"rights\", curie=BIOLINK.curie('rights'),\n",
      "                   model_uri=DIST.rights, domain=InformationContentEntity, range=Optional[str])\n",
      "\n",
      "slots.format = Slot(uri=BIOLINK.format, name=\"format\", curie=BIOLINK.curie('format'),\n",
      "                   model_uri=DIST.format, domain=InformationContentEntity, range=Optional[str])\n",
      "\n",
      "slots.created_with = Slot(uri=BIOLINK.created_with, name=\"created with\", curie=BIOLINK.curie('created_with'),\n",
      "                   model_uri=DIST.created_with, domain=Dataset, range=Optional[str])\n",
      "\n",
      "slots.download_url = Slot(uri=DCAT.downloadURL, name=\"download url\", curie=DCAT.curie('downloadURL'),\n",
      "                   model_uri=DIST.download_url, domain=InformationContentEntity, range=Optional[str])\n",
      "\n",
      "slots.dataset_download_url = Slot(uri=DCAT.downloadURL, name=\"dataset download url\", curie=DCAT.curie('downloadURL'),\n",
      "                   model_uri=DIST.dataset_download_url, domain=Dataset, range=Optional[str])\n",
      "\n",
      "slots.distribution_download_url = Slot(uri=BIOLINK.distribution_download_url, name=\"distribution download url\", curie=BIOLINK.curie('distribution_download_url'),\n",
      "                   model_uri=DIST.distribution_download_url, domain=DatasetDistribution, range=Optional[str])\n",
      "\n",
      "slots.ingest_date = Slot(uri=BIOLINK.ingest_date, name=\"ingest date\", curie=BIOLINK.curie('ingest_date'),\n",
      "                   model_uri=DIST.ingest_date, domain=DatasetVersion, range=Optional[str])\n",
      "\n",
      "slots.has_distribution = Slot(uri=DCT.distribution, name=\"has distribution\", curie=DCT.curie('distribution'),\n",
      "                   model_uri=DIST.has_distribution, domain=DatasetVersion, range=Optional[Union[str, DatasetDistributionId]])\n",
      "\n",
      "slots.published_in = Slot(uri=BIOLINK.published_in, name=\"published in\", curie=BIOLINK.curie('published_in'),\n",
      "                   model_uri=DIST.published_in, domain=Publication, range=Optional[Union[str, URIorCURIE]])\n",
      "\n",
      "slots.iso_abbreviation = Slot(uri=BIOLINK.iso_abbreviation, name=\"iso abbreviation\", curie=BIOLINK.curie('iso_abbreviation'),\n",
      "                   model_uri=DIST.iso_abbreviation, domain=Publication, range=Optional[str])\n",
      "\n",
      "slots.authors = Slot(uri=BIOLINK.authors, name=\"authors\", curie=BIOLINK.curie('authors'),\n",
      "                   model_uri=DIST.authors, domain=Publication, range=Optional[Union[Union[str, AgentId], List[Union[str, AgentId]]]])\n",
      "\n",
      "slots.volume = Slot(uri=BIOLINK.volume, name=\"volume\", curie=BIOLINK.curie('volume'),\n",
      "                   model_uri=DIST.volume, domain=Publication, range=Optional[str])\n",
      "\n",
      "slots.chapter = Slot(uri=BIOLINK.chapter, name=\"chapter\", curie=BIOLINK.curie('chapter'),\n",
      "                   model_uri=DIST.chapter, domain=BookChapter, range=Optional[str])\n",
      "\n",
      "slots.issue = Slot(uri=BIOLINK.issue, name=\"issue\", curie=BIOLINK.curie('issue'),\n",
      "                   model_uri=DIST.issue, domain=Publication, range=Optional[str])\n",
      "\n",
      "slots.pages = Slot(uri=BIOLINK.pages, name=\"pages\", curie=BIOLINK.curie('pages'),\n",
      "                   model_uri=DIST.pages, domain=Publication, range=Optional[Union[str, List[str]]])\n",
      "\n",
      "slots.summary = Slot(uri=BIOLINK.summary, name=\"summary\", curie=BIOLINK.curie('summary'),\n",
      "                   model_uri=DIST.summary, domain=Publication, range=Optional[str])\n",
      "\n",
      "slots.keywords = Slot(uri=BIOLINK.keywords, name=\"keywords\", curie=BIOLINK.curie('keywords'),\n",
      "                   model_uri=DIST.keywords, domain=Publication, range=Optional[Union[str, List[str]]])\n",
      "\n",
      "slots.mesh_terms = Slot(uri=BIOLINK.mesh_terms, name=\"mesh terms\", curie=BIOLINK.curie('mesh_terms'),\n",
      "                   model_uri=DIST.mesh_terms, domain=Publication, range=Optional[Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]]])\n",
      "\n",
      "slots.has_biological_sequence = Slot(uri=BIOLINK.has_biological_sequence, name=\"has biological sequence\", curie=BIOLINK.curie('has_biological_sequence'),\n",
      "                   model_uri=DIST.has_biological_sequence, domain=NamedThing, range=Optional[Union[str, BiologicalSequence]])\n",
      "\n",
      "slots.has_gene_or_gene_product = Slot(uri=BIOLINK.has_gene_or_gene_product, name=\"has gene or gene product\", curie=BIOLINK.curie('has_gene_or_gene_product'),\n",
      "                   model_uri=DIST.has_gene_or_gene_product, domain=NamedThing, range=Optional[Union[Union[str, GeneId], List[Union[str, GeneId]]]])\n",
      "\n",
      "slots.has_gene = Slot(uri=BIOLINK.has_gene, name=\"has gene\", curie=BIOLINK.curie('has_gene'),\n",
      "                   model_uri=DIST.has_gene, domain=NamedThing, range=Optional[Union[Union[str, GeneId], List[Union[str, GeneId]]]])\n",
      "\n",
      "slots.has_zygosity = Slot(uri=BIOLINK.has_zygosity, name=\"has zygosity\", curie=BIOLINK.curie('has_zygosity'),\n",
      "                   model_uri=DIST.has_zygosity, domain=NucleicAcidEntity, range=Optional[Union[str, ZygosityId]])\n",
      "\n",
      "slots.has_chemical_formula = Slot(uri=BIOLINK.has_chemical_formula, name=\"has chemical formula\", curie=BIOLINK.curie('has_chemical_formula'),\n",
      "                   model_uri=DIST.has_chemical_formula, domain=NamedThing, range=Optional[str])\n",
      "\n",
      "slots.is_metabolite = Slot(uri=BIOLINK.is_metabolite, name=\"is metabolite\", curie=BIOLINK.curie('is_metabolite'),\n",
      "                   model_uri=DIST.is_metabolite, domain=MolecularEntity, range=Optional[Union[bool, Bool]])\n",
      "\n",
      "slots.has_constituent = Slot(uri=BIOLINK.has_constituent, name=\"has constituent\", curie=BIOLINK.curie('has_constituent'),\n",
      "                   model_uri=DIST.has_constituent, domain=NamedThing, range=Optional[Union[Union[str, MolecularEntityId], List[Union[str, MolecularEntityId]]]])\n",
      "\n",
      "slots.has_drug = Slot(uri=BIOLINK.has_drug, name=\"has drug\", curie=BIOLINK.curie('has_drug'),\n",
      "                   model_uri=DIST.has_drug, domain=NamedThing, range=Optional[Union[Union[str, DrugId], List[Union[str, DrugId]]]])\n",
      "\n",
      "slots.has_device = Slot(uri=BIOLINK.has_device, name=\"has device\", curie=BIOLINK.curie('has_device'),\n",
      "                   model_uri=DIST.has_device, domain=NamedThing, range=Optional[Union[Union[str, DeviceId], List[Union[str, DeviceId]]]])\n",
      "\n",
      "slots.has_procedure = Slot(uri=BIOLINK.has_procedure, name=\"has procedure\", curie=BIOLINK.curie('has_procedure'),\n",
      "                   model_uri=DIST.has_procedure, domain=NamedThing, range=Optional[Union[Union[str, ProcedureId], List[Union[str, ProcedureId]]]])\n",
      "\n",
      "slots.has_receptor = Slot(uri=BIOLINK.has_receptor, name=\"has receptor\", curie=BIOLINK.curie('has_receptor'),\n",
      "                   model_uri=DIST.has_receptor, domain=None, range=Optional[Union[str, OrganismalEntityId]])\n",
      "\n",
      "slots.has_stressor = Slot(uri=BIOLINK.has_stressor, name=\"has stressor\", curie=BIOLINK.curie('has_stressor'),\n",
      "                   model_uri=DIST.has_stressor, domain=None, range=Optional[str])\n",
      "\n",
      "slots.has_route = Slot(uri=BIOLINK.has_route, name=\"has route\", curie=BIOLINK.curie('has_route'),\n",
      "                   model_uri=DIST.has_route, domain=None, range=Optional[str])\n",
      "\n",
      "slots.population_context_qualifier = Slot(uri=BIOLINK.population_context_qualifier, name=\"population context qualifier\", curie=BIOLINK.curie('population_context_qualifier'),\n",
      "                   model_uri=DIST.population_context_qualifier, domain=Association, range=Optional[Union[str, PopulationOfIndividualOrganismsId]])\n",
      "\n",
      "slots.temporal_context_qualifier = Slot(uri=BIOLINK.temporal_context_qualifier, name=\"temporal context qualifier\", curie=BIOLINK.curie('temporal_context_qualifier'),\n",
      "                   model_uri=DIST.temporal_context_qualifier, domain=Association, range=Optional[Union[str, TimeType]])\n",
      "\n",
      "slots.temporal_interval_qualifier = Slot(uri=BIOLINK.temporal_interval_qualifier, name=\"temporal interval qualifier\", curie=BIOLINK.curie('temporal_interval_qualifier'),\n",
      "                   model_uri=DIST.temporal_interval_qualifier, domain=Association, range=Optional[Union[str, TimeType]])\n",
      "\n",
      "slots.is_supplement = Slot(uri=BIOLINK.is_supplement, name=\"is supplement\", curie=BIOLINK.curie('is_supplement'),\n",
      "                   model_uri=DIST.is_supplement, domain=ChemicalMixture, range=Optional[str])\n",
      "\n",
      "slots.trade_name = Slot(uri=BIOLINK.trade_name, name=\"trade name\", curie=BIOLINK.curie('trade_name'),\n",
      "                   model_uri=DIST.trade_name, domain=ChemicalEntity, range=Optional[str])\n",
      "\n",
      "slots.available_from = Slot(uri=BIOLINK.available_from, name=\"available from\", curie=BIOLINK.curie('available_from'),\n",
      "                   model_uri=DIST.available_from, domain=NamedThing, range=Optional[Union[Union[str, \"DrugAvailabilityEnum\"], List[Union[str, \"DrugAvailabilityEnum\"]]]])\n",
      "\n",
      "slots.is_toxic = Slot(uri=BIOLINK.is_toxic, name=\"is toxic\", curie=BIOLINK.curie('is_toxic'),\n",
      "                   model_uri=DIST.is_toxic, domain=NamedThing, range=Optional[Union[bool, Bool]])\n",
      "\n",
      "slots.has_chemical_role = Slot(uri=BIOLINK.has_chemical_role, name=\"has chemical role\", curie=BIOLINK.curie('has_chemical_role'),\n",
      "                   model_uri=DIST.has_chemical_role, domain=NamedThing, range=Optional[Union[Union[str, ChemicalRoleId], List[Union[str, ChemicalRoleId]]]])\n",
      "\n",
      "slots.max_tolerated_dose = Slot(uri=BIOLINK.max_tolerated_dose, name=\"max tolerated dose\", curie=BIOLINK.curie('max_tolerated_dose'),\n",
      "                   model_uri=DIST.max_tolerated_dose, domain=NamedThing, range=Optional[str])\n",
      "\n",
      "slots.animal_model_available_from = Slot(uri=BIOLINK.animal_model_available_from, name=\"animal model available from\", curie=BIOLINK.curie('animal_model_available_from'),\n",
      "                   model_uri=DIST.animal_model_available_from, domain=NamedThing, range=Optional[Union[Union[str, DiseaseOrPhenotypicFeatureId], List[Union[str, DiseaseOrPhenotypicFeatureId]]]])\n",
      "\n",
      "slots.FDA_adverse_event_level = Slot(uri=BIOLINK.FDA_adverse_event_level, name=\"FDA adverse event level\", curie=BIOLINK.curie('FDA_adverse_event_level'),\n",
      "                   model_uri=DIST.FDA_adverse_event_level, domain=Association, range=Optional[Union[str, \"FDAIDAAdverseEventEnum\"]])\n",
      "\n",
      "slots.highest_FDA_approval_status = Slot(uri=BIOLINK.highest_FDA_approval_status, name=\"highest FDA approval status\", curie=BIOLINK.curie('highest_FDA_approval_status'),\n",
      "                   model_uri=DIST.highest_FDA_approval_status, domain=None, range=Optional[Union[str, \"ApprovalStatusEnum\"]])\n",
      "\n",
      "slots.drug_regulatory_status_world_wide = Slot(uri=BIOLINK.drug_regulatory_status_world_wide, name=\"drug regulatory status world wide\", curie=BIOLINK.curie('drug_regulatory_status_world_wide'),\n",
      "                   model_uri=DIST.drug_regulatory_status_world_wide, domain=None, range=Optional[Union[str, \"ApprovalStatusEnum\"]])\n",
      "\n",
      "slots.routes_of_delivery = Slot(uri=BIOLINK.routes_of_delivery, name=\"routes of delivery\", curie=BIOLINK.curie('routes_of_delivery'),\n",
      "                   model_uri=DIST.routes_of_delivery, domain=None, range=Optional[Union[Union[str, \"DrugDeliveryEnum\"], List[Union[str, \"DrugDeliveryEnum\"]]]])\n",
      "\n",
      "slots.form_or_variant_qualifier = Slot(uri=BIOLINK.form_or_variant_qualifier, name=\"form or variant qualifier\", curie=BIOLINK.curie('form_or_variant_qualifier'),\n",
      "                   model_uri=DIST.form_or_variant_qualifier, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.specialization_qualifier = Slot(uri=BIOLINK.specialization_qualifier, name=\"specialization qualifier\", curie=BIOLINK.curie('specialization_qualifier'),\n",
      "                   model_uri=DIST.specialization_qualifier, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.aspect_qualifier = Slot(uri=BIOLINK.aspect_qualifier, name=\"aspect qualifier\", curie=BIOLINK.curie('aspect_qualifier'),\n",
      "                   model_uri=DIST.aspect_qualifier, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.derivative_qualifier = Slot(uri=BIOLINK.derivative_qualifier, name=\"derivative qualifier\", curie=BIOLINK.curie('derivative_qualifier'),\n",
      "                   model_uri=DIST.derivative_qualifier, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.part_qualifier = Slot(uri=BIOLINK.part_qualifier, name=\"part qualifier\", curie=BIOLINK.curie('part_qualifier'),\n",
      "                   model_uri=DIST.part_qualifier, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.context_qualifier = Slot(uri=BIOLINK.context_qualifier, name=\"context qualifier\", curie=BIOLINK.curie('context_qualifier'),\n",
      "                   model_uri=DIST.context_qualifier, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.direction_qualifier = Slot(uri=BIOLINK.direction_qualifier, name=\"direction qualifier\", curie=BIOLINK.curie('direction_qualifier'),\n",
      "                   model_uri=DIST.direction_qualifier, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.mapped_predicate = Slot(uri=BIOLINK.mapped_predicate, name=\"mapped predicate\", curie=BIOLINK.curie('mapped_predicate'),\n",
      "                   model_uri=DIST.mapped_predicate, domain=None, range=Optional[str])\n",
      "\n",
      "slots.predicate_mappings = Slot(uri=BIOLINK.predicate_mappings, name=\"predicate mappings\", curie=BIOLINK.curie('predicate_mappings'),\n",
      "                   model_uri=DIST.predicate_mappings, domain=None, range=Optional[Union[Union[dict, PredicateMapping], List[Union[dict, PredicateMapping]]]])\n",
      "\n",
      "slots.exact_matches = Slot(uri=BIOLINK.exact_matches, name=\"exact matches\", curie=BIOLINK.curie('exact_matches'),\n",
      "                   model_uri=DIST.exact_matches, domain=None, range=Optional[Union[str, List[str]]])\n",
      "\n",
      "slots.narrow_matches = Slot(uri=BIOLINK.narrow_matches, name=\"narrow matches\", curie=BIOLINK.curie('narrow_matches'),\n",
      "                   model_uri=DIST.narrow_matches, domain=None, range=Optional[Union[str, List[str]]])\n",
      "\n",
      "slots.broad_matches = Slot(uri=BIOLINK.broad_matches, name=\"broad matches\", curie=BIOLINK.curie('broad_matches'),\n",
      "                   model_uri=DIST.broad_matches, domain=None, range=Optional[Union[str, List[str]]])\n",
      "\n",
      "slots.subject_aspect_qualifier = Slot(uri=BIOLINK.subject_aspect_qualifier, name=\"subject aspect qualifier\", curie=BIOLINK.curie('subject_aspect_qualifier'),\n",
      "                   model_uri=DIST.subject_aspect_qualifier, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.subject_specialization_qualifier = Slot(uri=BIOLINK.subject_specialization_qualifier, name=\"subject specialization qualifier\", curie=BIOLINK.curie('subject_specialization_qualifier'),\n",
      "                   model_uri=DIST.subject_specialization_qualifier, domain=Association, range=Optional[Union[str, URIorCURIE]])\n",
      "\n",
      "slots.subject_form_or_variant_qualifier = Slot(uri=BIOLINK.subject_form_or_variant_qualifier, name=\"subject form or variant qualifier\", curie=BIOLINK.curie('subject_form_or_variant_qualifier'),\n",
      "                   model_uri=DIST.subject_form_or_variant_qualifier, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.subject_part_qualifier = Slot(uri=BIOLINK.subject_part_qualifier, name=\"subject part qualifier\", curie=BIOLINK.curie('subject_part_qualifier'),\n",
      "                   model_uri=DIST.subject_part_qualifier, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.subject_derivative_qualifier = Slot(uri=BIOLINK.subject_derivative_qualifier, name=\"subject derivative qualifier\", curie=BIOLINK.curie('subject_derivative_qualifier'),\n",
      "                   model_uri=DIST.subject_derivative_qualifier, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.subject_context_qualifier = Slot(uri=BIOLINK.subject_context_qualifier, name=\"subject context qualifier\", curie=BIOLINK.curie('subject_context_qualifier'),\n",
      "                   model_uri=DIST.subject_context_qualifier, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.subject_direction_qualifier = Slot(uri=BIOLINK.subject_direction_qualifier, name=\"subject direction qualifier\", curie=BIOLINK.curie('subject_direction_qualifier'),\n",
      "                   model_uri=DIST.subject_direction_qualifier, domain=Association, range=Optional[Union[str, \"DirectionQualifierEnum\"]])\n",
      "\n",
      "slots.object_aspect_qualifier = Slot(uri=BIOLINK.object_aspect_qualifier, name=\"object aspect qualifier\", curie=BIOLINK.curie('object_aspect_qualifier'),\n",
      "                   model_uri=DIST.object_aspect_qualifier, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.object_specialization_qualifier = Slot(uri=BIOLINK.object_specialization_qualifier, name=\"object specialization qualifier\", curie=BIOLINK.curie('object_specialization_qualifier'),\n",
      "                   model_uri=DIST.object_specialization_qualifier, domain=Association, range=Optional[Union[str, URIorCURIE]])\n",
      "\n",
      "slots.object_form_or_variant_qualifier = Slot(uri=BIOLINK.object_form_or_variant_qualifier, name=\"object form or variant qualifier\", curie=BIOLINK.curie('object_form_or_variant_qualifier'),\n",
      "                   model_uri=DIST.object_form_or_variant_qualifier, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.object_part_qualifier = Slot(uri=BIOLINK.object_part_qualifier, name=\"object part qualifier\", curie=BIOLINK.curie('object_part_qualifier'),\n",
      "                   model_uri=DIST.object_part_qualifier, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.object_derivative_qualifier = Slot(uri=BIOLINK.object_derivative_qualifier, name=\"object derivative qualifier\", curie=BIOLINK.curie('object_derivative_qualifier'),\n",
      "                   model_uri=DIST.object_derivative_qualifier, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.object_context_qualifier = Slot(uri=BIOLINK.object_context_qualifier, name=\"object context qualifier\", curie=BIOLINK.curie('object_context_qualifier'),\n",
      "                   model_uri=DIST.object_context_qualifier, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.object_direction_qualifier = Slot(uri=BIOLINK.object_direction_qualifier, name=\"object direction qualifier\", curie=BIOLINK.curie('object_direction_qualifier'),\n",
      "                   model_uri=DIST.object_direction_qualifier, domain=Association, range=Optional[Union[str, \"DirectionQualifierEnum\"]])\n",
      "\n",
      "slots.qualified_predicate = Slot(uri=BIOLINK.qualified_predicate, name=\"qualified predicate\", curie=BIOLINK.curie('qualified_predicate'),\n",
      "                   model_uri=DIST.qualified_predicate, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.statement_qualifier = Slot(uri=BIOLINK.statement_qualifier, name=\"statement qualifier\", curie=BIOLINK.curie('statement_qualifier'),\n",
      "                   model_uri=DIST.statement_qualifier, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.causal_mechanism_qualifier = Slot(uri=BIOLINK.causal_mechanism_qualifier, name=\"causal mechanism qualifier\", curie=BIOLINK.curie('causal_mechanism_qualifier'),\n",
      "                   model_uri=DIST.causal_mechanism_qualifier, domain=Association, range=Optional[Union[str, \"CausalMechanismQualifierEnum\"]])\n",
      "\n",
      "slots.anatomical_context_qualifier = Slot(uri=BIOLINK.anatomical_context_qualifier, name=\"anatomical context qualifier\", curie=BIOLINK.curie('anatomical_context_qualifier'),\n",
      "                   model_uri=DIST.anatomical_context_qualifier, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.species_context_qualifier = Slot(uri=BIOLINK.species_context_qualifier, name=\"species context qualifier\", curie=BIOLINK.curie('species_context_qualifier'),\n",
      "                   model_uri=DIST.species_context_qualifier, domain=Association, range=Optional[Union[str, OrganismTaxonId]])\n",
      "\n",
      "slots.disease_context_qualifier = Slot(uri=BIOLINK.disease_context_qualifier, name=\"disease context qualifier\", curie=BIOLINK.curie('disease_context_qualifier'),\n",
      "                   model_uri=DIST.disease_context_qualifier, domain=Association, range=Optional[Union[str, DiseaseId]])\n",
      "\n",
      "slots.qualifiers = Slot(uri=BIOLINK.qualifiers, name=\"qualifiers\", curie=BIOLINK.curie('qualifiers'),\n",
      "                   model_uri=DIST.qualifiers, domain=Association, range=Optional[Union[Union[str, OntologyClassId], List[Union[str, OntologyClassId]]]])\n",
      "\n",
      "slots.frequency_qualifier = Slot(uri=BIOLINK.frequency_qualifier, name=\"frequency qualifier\", curie=BIOLINK.curie('frequency_qualifier'),\n",
      "                   model_uri=DIST.frequency_qualifier, domain=Association, range=Optional[Union[str, FrequencyValue]])\n",
      "\n",
      "slots.severity_qualifier = Slot(uri=BIOLINK.severity_qualifier, name=\"severity qualifier\", curie=BIOLINK.curie('severity_qualifier'),\n",
      "                   model_uri=DIST.severity_qualifier, domain=Association, range=Optional[Union[str, SeverityValueId]])\n",
      "\n",
      "slots.sex_qualifier = Slot(uri=BIOLINK.sex_qualifier, name=\"sex qualifier\", curie=BIOLINK.curie('sex_qualifier'),\n",
      "                   model_uri=DIST.sex_qualifier, domain=Association, range=Optional[Union[str, BiologicalSexId]])\n",
      "\n",
      "slots.onset_qualifier = Slot(uri=BIOLINK.onset_qualifier, name=\"onset qualifier\", curie=BIOLINK.curie('onset_qualifier'),\n",
      "                   model_uri=DIST.onset_qualifier, domain=Association, range=Optional[Union[str, OnsetId]])\n",
      "\n",
      "slots.clinical_modifier_qualifier = Slot(uri=BIOLINK.clinical_modifier_qualifier, name=\"clinical modifier qualifier\", curie=BIOLINK.curie('clinical_modifier_qualifier'),\n",
      "                   model_uri=DIST.clinical_modifier_qualifier, domain=Association, range=Optional[Union[str, ClinicalModifierId]])\n",
      "\n",
      "slots.sequence_variant_qualifier = Slot(uri=BIOLINK.sequence_variant_qualifier, name=\"sequence variant qualifier\", curie=BIOLINK.curie('sequence_variant_qualifier'),\n",
      "                   model_uri=DIST.sequence_variant_qualifier, domain=Association, range=Optional[Union[str, SequenceVariantId]])\n",
      "\n",
      "slots.quantifier_qualifier = Slot(uri=BIOLINK.quantifier_qualifier, name=\"quantifier qualifier\", curie=BIOLINK.curie('quantifier_qualifier'),\n",
      "                   model_uri=DIST.quantifier_qualifier, domain=Association, range=Optional[Union[str, OntologyClassId]])\n",
      "\n",
      "slots.catalyst_qualifier = Slot(uri=BIOLINK.catalyst_qualifier, name=\"catalyst qualifier\", curie=BIOLINK.curie('catalyst_qualifier'),\n",
      "                   model_uri=DIST.catalyst_qualifier, domain=Association, range=Optional[Union[Union[dict, MacromolecularMachineMixin], List[Union[dict, MacromolecularMachineMixin]]]])\n",
      "\n",
      "slots.stage_qualifier = Slot(uri=BIOLINK.stage_qualifier, name=\"stage qualifier\", curie=BIOLINK.curie('stage_qualifier'),\n",
      "                   model_uri=DIST.stage_qualifier, domain=Association, range=Optional[Union[str, LifeStageId]])\n",
      "\n",
      "slots.related_to = Slot(uri=BIOLINK.related_to, name=\"related to\", curie=BIOLINK.curie('related_to'),\n",
      "                   model_uri=DIST.related_to, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.related_to_at_concept_level = Slot(uri=BIOLINK.related_to_at_concept_level, name=\"related to at concept level\", curie=BIOLINK.curie('related_to_at_concept_level'),\n",
      "                   model_uri=DIST.related_to_at_concept_level, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.related_to_at_instance_level = Slot(uri=BIOLINK.related_to_at_instance_level, name=\"related to at instance level\", curie=BIOLINK.curie('related_to_at_instance_level'),\n",
      "                   model_uri=DIST.related_to_at_instance_level, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.associated_with = Slot(uri=BIOLINK.associated_with, name=\"associated with\", curie=BIOLINK.curie('associated_with'),\n",
      "                   model_uri=DIST.associated_with, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.superclass_of = Slot(uri=BIOLINK.superclass_of, name=\"superclass of\", curie=BIOLINK.curie('superclass_of'),\n",
      "                   model_uri=DIST.superclass_of, domain=None, range=Optional[Union[Union[str, OntologyClassId], List[Union[str, OntologyClassId]]]])\n",
      "\n",
      "slots.subclass_of = Slot(uri=BIOLINK.subclass_of, name=\"subclass of\", curie=BIOLINK.curie('subclass_of'),\n",
      "                   model_uri=DIST.subclass_of, domain=None, range=Optional[Union[Union[str, OntologyClassId], List[Union[str, OntologyClassId]]]])\n",
      "\n",
      "slots.same_as = Slot(uri=BIOLINK.same_as, name=\"same as\", curie=BIOLINK.curie('same_as'),\n",
      "                   model_uri=DIST.same_as, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.close_match = Slot(uri=BIOLINK.close_match, name=\"close match\", curie=BIOLINK.curie('close_match'),\n",
      "                   model_uri=DIST.close_match, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.exact_match = Slot(uri=BIOLINK.exact_match, name=\"exact match\", curie=BIOLINK.curie('exact_match'),\n",
      "                   model_uri=DIST.exact_match, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.broad_match = Slot(uri=BIOLINK.broad_match, name=\"broad match\", curie=BIOLINK.curie('broad_match'),\n",
      "                   model_uri=DIST.broad_match, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.narrow_match = Slot(uri=BIOLINK.narrow_match, name=\"narrow match\", curie=BIOLINK.curie('narrow_match'),\n",
      "                   model_uri=DIST.narrow_match, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.member_of = Slot(uri=BIOLINK.member_of, name=\"member of\", curie=BIOLINK.curie('member_of'),\n",
      "                   model_uri=DIST.member_of, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.has_member = Slot(uri=BIOLINK.has_member, name=\"has member\", curie=BIOLINK.curie('has_member'),\n",
      "                   model_uri=DIST.has_member, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.opposite_of = Slot(uri=BIOLINK.opposite_of, name=\"opposite of\", curie=BIOLINK.curie('opposite_of'),\n",
      "                   model_uri=DIST.opposite_of, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.affects_likelihood_of = Slot(uri=BIOLINK.affects_likelihood_of, name=\"affects likelihood of\", curie=BIOLINK.curie('affects_likelihood_of'),\n",
      "                   model_uri=DIST.affects_likelihood_of, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.likelihood_affected_by = Slot(uri=BIOLINK.likelihood_affected_by, name=\"likelihood affected by\", curie=BIOLINK.curie('likelihood_affected_by'),\n",
      "                   model_uri=DIST.likelihood_affected_by, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.associated_with_likelihood_of = Slot(uri=BIOLINK.associated_with_likelihood_of, name=\"associated with likelihood of\", curie=BIOLINK.curie('associated_with_likelihood_of'),\n",
      "                   model_uri=DIST.associated_with_likelihood_of, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.likelihood_associated_with = Slot(uri=BIOLINK.likelihood_associated_with, name=\"likelihood associated with\", curie=BIOLINK.curie('likelihood_associated_with'),\n",
      "                   model_uri=DIST.likelihood_associated_with, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.associated_with_increased_likelihood_of = Slot(uri=BIOLINK.associated_with_increased_likelihood_of, name=\"associated with increased likelihood of\", curie=BIOLINK.curie('associated_with_increased_likelihood_of'),\n",
      "                   model_uri=DIST.associated_with_increased_likelihood_of, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.increased_likelihood_associated_with = Slot(uri=BIOLINK.increased_likelihood_associated_with, name=\"increased likelihood associated with\", curie=BIOLINK.curie('increased_likelihood_associated_with'),\n",
      "                   model_uri=DIST.increased_likelihood_associated_with, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.associated_with_decreased_likelihood_of = Slot(uri=BIOLINK.associated_with_decreased_likelihood_of, name=\"associated with decreased likelihood of\", curie=BIOLINK.curie('associated_with_decreased_likelihood_of'),\n",
      "                   model_uri=DIST.associated_with_decreased_likelihood_of, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.decreased_likelihood_associated_with = Slot(uri=BIOLINK.decreased_likelihood_associated_with, name=\"decreased likelihood associated with\", curie=BIOLINK.curie('decreased_likelihood_associated_with'),\n",
      "                   model_uri=DIST.decreased_likelihood_associated_with, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.target_for = Slot(uri=BIOLINK.target_for, name=\"target for\", curie=BIOLINK.curie('target_for'),\n",
      "                   model_uri=DIST.target_for, domain=Gene, range=Optional[Union[Union[str, DiseaseId], List[Union[str, DiseaseId]]]])\n",
      "\n",
      "slots.has_target = Slot(uri=BIOLINK.has_target, name=\"has target\", curie=BIOLINK.curie('has_target'),\n",
      "                   model_uri=DIST.has_target, domain=Disease, range=Optional[Union[Union[str, GeneId], List[Union[str, GeneId]]]])\n",
      "\n",
      "slots.active_in = Slot(uri=BIOLINK.active_in, name=\"active in\", curie=BIOLINK.curie('active_in'),\n",
      "                   model_uri=DIST.active_in, domain=None, range=Optional[Union[Union[str, CellularComponentId], List[Union[str, CellularComponentId]]]])\n",
      "\n",
      "slots.has_active_component = Slot(uri=BIOLINK.has_active_component, name=\"has active component\", curie=BIOLINK.curie('has_active_component'),\n",
      "                   model_uri=DIST.has_active_component, domain=CellularComponent, range=Optional[Union[Union[dict, \"GeneOrGeneProduct\"], List[Union[dict, \"GeneOrGeneProduct\"]]]])\n",
      "\n",
      "slots.acts_upstream_of = Slot(uri=BIOLINK.acts_upstream_of, name=\"acts upstream of\", curie=BIOLINK.curie('acts_upstream_of'),\n",
      "                   model_uri=DIST.acts_upstream_of, domain=None, range=Optional[Union[Union[str, BiologicalProcessId], List[Union[str, BiologicalProcessId]]]])\n",
      "\n",
      "slots.has_upstream_actor = Slot(uri=BIOLINK.has_upstream_actor, name=\"has upstream actor\", curie=BIOLINK.curie('has_upstream_actor'),\n",
      "                   model_uri=DIST.has_upstream_actor, domain=BiologicalProcess, range=Optional[Union[Union[dict, \"GeneOrGeneProduct\"], List[Union[dict, \"GeneOrGeneProduct\"]]]])\n",
      "\n",
      "slots.acts_upstream_of_positive_effect = Slot(uri=BIOLINK.acts_upstream_of_positive_effect, name=\"acts upstream of positive effect\", curie=BIOLINK.curie('acts_upstream_of_positive_effect'),\n",
      "                   model_uri=DIST.acts_upstream_of_positive_effect, domain=None, range=Optional[Union[Union[str, BiologicalProcessId], List[Union[str, BiologicalProcessId]]]])\n",
      "\n",
      "slots.has_positive_upstream_actor = Slot(uri=BIOLINK.has_positive_upstream_actor, name=\"has positive upstream actor\", curie=BIOLINK.curie('has_positive_upstream_actor'),\n",
      "                   model_uri=DIST.has_positive_upstream_actor, domain=BiologicalProcess, range=Optional[Union[Union[dict, \"GeneOrGeneProduct\"], List[Union[dict, \"GeneOrGeneProduct\"]]]])\n",
      "\n",
      "slots.acts_upstream_of_negative_effect = Slot(uri=BIOLINK.acts_upstream_of_negative_effect, name=\"acts upstream of negative effect\", curie=BIOLINK.curie('acts_upstream_of_negative_effect'),\n",
      "                   model_uri=DIST.acts_upstream_of_negative_effect, domain=None, range=Optional[Union[Union[str, BiologicalProcessId], List[Union[str, BiologicalProcessId]]]])\n",
      "\n",
      "slots.has_negative_upstream_actor = Slot(uri=BIOLINK.has_negative_upstream_actor, name=\"has negative upstream actor\", curie=BIOLINK.curie('has_negative_upstream_actor'),\n",
      "                   model_uri=DIST.has_negative_upstream_actor, domain=BiologicalProcess, range=Optional[Union[Union[dict, \"GeneOrGeneProduct\"], List[Union[dict, \"GeneOrGeneProduct\"]]]])\n",
      "\n",
      "slots.acts_upstream_of_or_within = Slot(uri=BIOLINK.acts_upstream_of_or_within, name=\"acts upstream of or within\", curie=BIOLINK.curie('acts_upstream_of_or_within'),\n",
      "                   model_uri=DIST.acts_upstream_of_or_within, domain=None, range=Optional[Union[Union[str, BiologicalProcessId], List[Union[str, BiologicalProcessId]]]])\n",
      "\n",
      "slots.has_upstream_or_within_actor = Slot(uri=BIOLINK.has_upstream_or_within_actor, name=\"has upstream or within actor\", curie=BIOLINK.curie('has_upstream_or_within_actor'),\n",
      "                   model_uri=DIST.has_upstream_or_within_actor, domain=BiologicalProcess, range=Optional[Union[Union[dict, \"GeneOrGeneProduct\"], List[Union[dict, \"GeneOrGeneProduct\"]]]])\n",
      "\n",
      "slots.acts_upstream_of_or_within_positive_effect = Slot(uri=BIOLINK.acts_upstream_of_or_within_positive_effect, name=\"acts upstream of or within positive effect\", curie=BIOLINK.curie('acts_upstream_of_or_within_positive_effect'),\n",
      "                   model_uri=DIST.acts_upstream_of_or_within_positive_effect, domain=None, range=Optional[Union[Union[str, BiologicalProcessId], List[Union[str, BiologicalProcessId]]]])\n",
      "\n",
      "slots.has_positive_upstream_or_within_actor = Slot(uri=BIOLINK.has_positive_upstream_or_within_actor, name=\"has positive upstream or within actor\", curie=BIOLINK.curie('has_positive_upstream_or_within_actor'),\n",
      "                   model_uri=DIST.has_positive_upstream_or_within_actor, domain=BiologicalProcess, range=Optional[Union[Union[dict, \"GeneOrGeneProduct\"], List[Union[dict, \"GeneOrGeneProduct\"]]]])\n",
      "\n",
      "slots.acts_upstream_of_or_within_negative_effect = Slot(uri=BIOLINK.acts_upstream_of_or_within_negative_effect, name=\"acts upstream of or within negative effect\", curie=BIOLINK.curie('acts_upstream_of_or_within_negative_effect'),\n",
      "                   model_uri=DIST.acts_upstream_of_or_within_negative_effect, domain=None, range=Optional[Union[Union[str, BiologicalProcessId], List[Union[str, BiologicalProcessId]]]])\n",
      "\n",
      "slots.has_negative_upstream_or_within_actor = Slot(uri=BIOLINK.has_negative_upstream_or_within_actor, name=\"has negative upstream or within actor\", curie=BIOLINK.curie('has_negative_upstream_or_within_actor'),\n",
      "                   model_uri=DIST.has_negative_upstream_or_within_actor, domain=BiologicalProcess, range=Optional[Union[Union[dict, \"GeneOrGeneProduct\"], List[Union[dict, \"GeneOrGeneProduct\"]]]])\n",
      "\n",
      "slots.mentions = Slot(uri=BIOLINK.mentions, name=\"mentions\", curie=BIOLINK.curie('mentions'),\n",
      "                   model_uri=DIST.mentions, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.mentioned_by = Slot(uri=BIOLINK.mentioned_by, name=\"mentioned by\", curie=BIOLINK.curie('mentioned_by'),\n",
      "                   model_uri=DIST.mentioned_by, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.contributor = Slot(uri=BIOLINK.contributor, name=\"contributor\", curie=BIOLINK.curie('contributor'),\n",
      "                   model_uri=DIST.contributor, domain=Agent, range=Optional[Union[Union[str, InformationContentEntityId], List[Union[str, InformationContentEntityId]]]])\n",
      "\n",
      "slots.has_contributor = Slot(uri=BIOLINK.has_contributor, name=\"has contributor\", curie=BIOLINK.curie('has_contributor'),\n",
      "                   model_uri=DIST.has_contributor, domain=InformationContentEntity, range=Optional[Union[Union[str, AgentId], List[Union[str, AgentId]]]])\n",
      "\n",
      "slots.provider = Slot(uri=BIOLINK.provider, name=\"provider\", curie=BIOLINK.curie('provider'),\n",
      "                   model_uri=DIST.provider, domain=Agent, range=Optional[Union[Union[str, InformationContentEntityId], List[Union[str, InformationContentEntityId]]]])\n",
      "\n",
      "slots.has_provider = Slot(uri=BIOLINK.has_provider, name=\"has provider\", curie=BIOLINK.curie('has_provider'),\n",
      "                   model_uri=DIST.has_provider, domain=InformationContentEntity, range=Optional[Union[Union[str, AgentId], List[Union[str, AgentId]]]])\n",
      "\n",
      "slots.publisher = Slot(uri=BIOLINK.publisher, name=\"publisher\", curie=BIOLINK.curie('publisher'),\n",
      "                   model_uri=DIST.publisher, domain=Agent, range=Optional[Union[Union[str, PublicationId], List[Union[str, PublicationId]]]])\n",
      "\n",
      "slots.has_publisher = Slot(uri=BIOLINK.has_publisher, name=\"has publisher\", curie=BIOLINK.curie('has_publisher'),\n",
      "                   model_uri=DIST.has_publisher, domain=Publication, range=Optional[Union[Union[str, AgentId], List[Union[str, AgentId]]]])\n",
      "\n",
      "slots.editor = Slot(uri=BIOLINK.editor, name=\"editor\", curie=BIOLINK.curie('editor'),\n",
      "                   model_uri=DIST.editor, domain=Agent, range=Optional[Union[Union[str, PublicationId], List[Union[str, PublicationId]]]])\n",
      "\n",
      "slots.has_editor = Slot(uri=BIOLINK.has_editor, name=\"has editor\", curie=BIOLINK.curie('has_editor'),\n",
      "                   model_uri=DIST.has_editor, domain=Publication, range=Optional[Union[Union[str, AgentId], List[Union[str, AgentId]]]])\n",
      "\n",
      "slots.author = Slot(uri=BIOLINK.author, name=\"author\", curie=BIOLINK.curie('author'),\n",
      "                   model_uri=DIST.author, domain=Agent, range=Optional[Union[Union[str, PublicationId], List[Union[str, PublicationId]]]])\n",
      "\n",
      "slots.has_author = Slot(uri=BIOLINK.has_author, name=\"has author\", curie=BIOLINK.curie('has_author'),\n",
      "                   model_uri=DIST.has_author, domain=Publication, range=Optional[Union[Union[str, AgentId], List[Union[str, AgentId]]]])\n",
      "\n",
      "slots.assesses = Slot(uri=BIOLINK.assesses, name=\"assesses\", curie=BIOLINK.curie('assesses'),\n",
      "                   model_uri=DIST.assesses, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.is_assessed_by = Slot(uri=BIOLINK.is_assessed_by, name=\"is assessed by\", curie=BIOLINK.curie('is_assessed_by'),\n",
      "                   model_uri=DIST.is_assessed_by, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.interacts_with = Slot(uri=BIOLINK.interacts_with, name=\"interacts with\", curie=BIOLINK.curie('interacts_with'),\n",
      "                   model_uri=DIST.interacts_with, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.physically_interacts_with = Slot(uri=BIOLINK.physically_interacts_with, name=\"physically interacts with\", curie=BIOLINK.curie('physically_interacts_with'),\n",
      "                   model_uri=DIST.physically_interacts_with, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.directly_physically_interacts_with = Slot(uri=BIOLINK.directly_physically_interacts_with, name=\"directly physically interacts with\", curie=BIOLINK.curie('directly_physically_interacts_with'),\n",
      "                   model_uri=DIST.directly_physically_interacts_with, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.binds = Slot(uri=BIOLINK.binds, name=\"binds\", curie=BIOLINK.curie('binds'),\n",
      "                   model_uri=DIST.binds, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.indirectly_physically_interacts_with = Slot(uri=BIOLINK.indirectly_physically_interacts_with, name=\"indirectly physically interacts with\", curie=BIOLINK.curie('indirectly_physically_interacts_with'),\n",
      "                   model_uri=DIST.indirectly_physically_interacts_with, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.genetically_interacts_with = Slot(uri=BIOLINK.genetically_interacts_with, name=\"genetically interacts with\", curie=BIOLINK.curie('genetically_interacts_with'),\n",
      "                   model_uri=DIST.genetically_interacts_with, domain=Gene, range=Optional[Union[Union[str, GeneId], List[Union[str, GeneId]]]])\n",
      "\n",
      "slots.gene_fusion_with = Slot(uri=BIOLINK.gene_fusion_with, name=\"gene_fusion_with\", curie=BIOLINK.curie('gene_fusion_with'),\n",
      "                   model_uri=DIST.gene_fusion_with, domain=Gene, range=Optional[Union[Union[str, GeneId], List[Union[str, GeneId]]]])\n",
      "\n",
      "slots.genetic_neighborhood_of = Slot(uri=BIOLINK.genetic_neighborhood_of, name=\"genetic_neighborhood_of\", curie=BIOLINK.curie('genetic_neighborhood_of'),\n",
      "                   model_uri=DIST.genetic_neighborhood_of, domain=Gene, range=Optional[Union[Union[str, GeneId], List[Union[str, GeneId]]]])\n",
      "\n",
      "slots.affects = Slot(uri=BIOLINK.affects, name=\"affects\", curie=BIOLINK.curie('affects'),\n",
      "                   model_uri=DIST.affects, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.affected_by = Slot(uri=BIOLINK.affected_by, name=\"affected by\", curie=BIOLINK.curie('affected_by'),\n",
      "                   model_uri=DIST.affected_by, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.associated_with_sensitivity_to = Slot(uri=BIOLINK.associated_with_sensitivity_to, name=\"associated with sensitivity to\", curie=BIOLINK.curie('associated_with_sensitivity_to'),\n",
      "                   model_uri=DIST.associated_with_sensitivity_to, domain=NamedThing, range=Optional[Union[Union[str, ChemicalEntityId], List[Union[str, ChemicalEntityId]]]])\n",
      "\n",
      "slots.sensitivity_associated_with = Slot(uri=BIOLINK.sensitivity_associated_with, name=\"sensitivity associated with\", curie=BIOLINK.curie('sensitivity_associated_with'),\n",
      "                   model_uri=DIST.sensitivity_associated_with, domain=ChemicalEntity, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.associated_with_resistance_to = Slot(uri=BIOLINK.associated_with_resistance_to, name=\"associated with resistance to\", curie=BIOLINK.curie('associated_with_resistance_to'),\n",
      "                   model_uri=DIST.associated_with_resistance_to, domain=NamedThing, range=Optional[Union[Union[str, ChemicalEntityId], List[Union[str, ChemicalEntityId]]]])\n",
      "\n",
      "slots.resistance_associated_with = Slot(uri=BIOLINK.resistance_associated_with, name=\"resistance associated with\", curie=BIOLINK.curie('resistance_associated_with'),\n",
      "                   model_uri=DIST.resistance_associated_with, domain=ChemicalEntity, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.diagnoses = Slot(uri=BIOLINK.diagnoses, name=\"diagnoses\", curie=BIOLINK.curie('diagnoses'),\n",
      "                   model_uri=DIST.diagnoses, domain=DiagnosticAid, range=Optional[Union[Union[str, DiseaseOrPhenotypicFeatureId], List[Union[str, DiseaseOrPhenotypicFeatureId]]]])\n",
      "\n",
      "slots.is_diagnosed_by = Slot(uri=BIOLINK.is_diagnosed_by, name=\"is diagnosed by\", curie=BIOLINK.curie('is_diagnosed_by'),\n",
      "                   model_uri=DIST.is_diagnosed_by, domain=DiseaseOrPhenotypicFeature, range=Optional[Union[Union[str, DiagnosticAidId], List[Union[str, DiagnosticAidId]]]])\n",
      "\n",
      "slots.increases_amount_or_activity_of = Slot(uri=BIOLINK.increases_amount_or_activity_of, name=\"increases amount or activity of\", curie=BIOLINK.curie('increases_amount_or_activity_of'),\n",
      "                   model_uri=DIST.increases_amount_or_activity_of, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.amount_or_activity_increased_by = Slot(uri=BIOLINK.amount_or_activity_increased_by, name=\"amount or activity increased by\", curie=BIOLINK.curie('amount_or_activity_increased_by'),\n",
      "                   model_uri=DIST.amount_or_activity_increased_by, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.decreases_amount_or_activity_of = Slot(uri=BIOLINK.decreases_amount_or_activity_of, name=\"decreases amount or activity of\", curie=BIOLINK.curie('decreases_amount_or_activity_of'),\n",
      "                   model_uri=DIST.decreases_amount_or_activity_of, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.amount_or_activity_decreased_by = Slot(uri=BIOLINK.amount_or_activity_decreased_by, name=\"amount or activity decreased by\", curie=BIOLINK.curie('amount_or_activity_decreased_by'),\n",
      "                   model_uri=DIST.amount_or_activity_decreased_by, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.chemical_role_mixin = Slot(uri=BIOLINK.chemical_role_mixin, name=\"chemical role mixin\", curie=BIOLINK.curie('chemical_role_mixin'),\n",
      "                   model_uri=DIST.chemical_role_mixin, domain=None, range=Optional[str])\n",
      "\n",
      "slots.biological_role_mixin = Slot(uri=BIOLINK.biological_role_mixin, name=\"biological role mixin\", curie=BIOLINK.curie('biological_role_mixin'),\n",
      "                   model_uri=DIST.biological_role_mixin, domain=None, range=Optional[str])\n",
      "\n",
      "slots.affects_response_to = Slot(uri=BIOLINK.affects_response_to, name=\"affects response to\", curie=BIOLINK.curie('affects_response_to'),\n",
      "                   model_uri=DIST.affects_response_to, domain=None, range=Optional[Union[Union[dict, \"ChemicalEntityOrGeneOrGeneProduct\"], List[Union[dict, \"ChemicalEntityOrGeneOrGeneProduct\"]]]])\n",
      "\n",
      "slots.response_affected_by = Slot(uri=BIOLINK.response_affected_by, name=\"response affected by\", curie=BIOLINK.curie('response_affected_by'),\n",
      "                   model_uri=DIST.response_affected_by, domain=None, range=Optional[Union[Union[dict, \"ChemicalEntityOrGeneOrGeneProduct\"], List[Union[dict, \"ChemicalEntityOrGeneOrGeneProduct\"]]]])\n",
      "\n",
      "slots.increases_response_to = Slot(uri=BIOLINK.increases_response_to, name=\"increases response to\", curie=BIOLINK.curie('increases_response_to'),\n",
      "                   model_uri=DIST.increases_response_to, domain=None, range=Optional[Union[Union[dict, \"ChemicalEntityOrGeneOrGeneProduct\"], List[Union[dict, \"ChemicalEntityOrGeneOrGeneProduct\"]]]])\n",
      "\n",
      "slots.response_increased_by = Slot(uri=BIOLINK.response_increased_by, name=\"response increased by\", curie=BIOLINK.curie('response_increased_by'),\n",
      "                   model_uri=DIST.response_increased_by, domain=None, range=Optional[Union[Union[dict, \"ChemicalEntityOrGeneOrGeneProduct\"], List[Union[dict, \"ChemicalEntityOrGeneOrGeneProduct\"]]]])\n",
      "\n",
      "slots.decreases_response_to = Slot(uri=BIOLINK.decreases_response_to, name=\"decreases response to\", curie=BIOLINK.curie('decreases_response_to'),\n",
      "                   model_uri=DIST.decreases_response_to, domain=None, range=Optional[Union[Union[dict, \"ChemicalEntityOrGeneOrGeneProduct\"], List[Union[dict, \"ChemicalEntityOrGeneOrGeneProduct\"]]]])\n",
      "\n",
      "slots.response_decreased_by = Slot(uri=BIOLINK.response_decreased_by, name=\"response decreased by\", curie=BIOLINK.curie('response_decreased_by'),\n",
      "                   model_uri=DIST.response_decreased_by, domain=None, range=Optional[Union[Union[dict, \"ChemicalEntityOrGeneOrGeneProduct\"], List[Union[dict, \"ChemicalEntityOrGeneOrGeneProduct\"]]]])\n",
      "\n",
      "slots.regulates = Slot(uri=BIOLINK.regulates, name=\"regulates\", curie=BIOLINK.curie('regulates'),\n",
      "                   model_uri=DIST.regulates, domain=None, range=Optional[Union[Union[dict, \"PhysicalEssenceOrOccurrent\"], List[Union[dict, \"PhysicalEssenceOrOccurrent\"]]]])\n",
      "\n",
      "slots.regulated_by = Slot(uri=BIOLINK.regulated_by, name=\"regulated by\", curie=BIOLINK.curie('regulated_by'),\n",
      "                   model_uri=DIST.regulated_by, domain=None, range=Optional[Union[Union[dict, \"PhysicalEssenceOrOccurrent\"], List[Union[dict, \"PhysicalEssenceOrOccurrent\"]]]])\n",
      "\n",
      "slots.disrupts = Slot(uri=BIOLINK.disrupts, name=\"disrupts\", curie=BIOLINK.curie('disrupts'),\n",
      "                   model_uri=DIST.disrupts, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.disrupted_by = Slot(uri=BIOLINK.disrupted_by, name=\"disrupted by\", curie=BIOLINK.curie('disrupted_by'),\n",
      "                   model_uri=DIST.disrupted_by, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.gene_product_of = Slot(uri=BIOLINK.gene_product_of, name=\"gene product of\", curie=BIOLINK.curie('gene_product_of'),\n",
      "                   model_uri=DIST.gene_product_of, domain=None, range=Optional[Union[Union[str, GeneId], List[Union[str, GeneId]]]])\n",
      "\n",
      "slots.has_gene_product = Slot(uri=BIOLINK.has_gene_product, name=\"has gene product\", curie=BIOLINK.curie('has_gene_product'),\n",
      "                   model_uri=DIST.has_gene_product, domain=Gene, range=Optional[Union[Union[dict, \"GeneProductMixin\"], List[Union[dict, \"GeneProductMixin\"]]]])\n",
      "\n",
      "slots.transcribed_to = Slot(uri=BIOLINK.transcribed_to, name=\"transcribed to\", curie=BIOLINK.curie('transcribed_to'),\n",
      "                   model_uri=DIST.transcribed_to, domain=Gene, range=Optional[Union[Union[str, TranscriptId], List[Union[str, TranscriptId]]]])\n",
      "\n",
      "slots.transcribed_from = Slot(uri=BIOLINK.transcribed_from, name=\"transcribed from\", curie=BIOLINK.curie('transcribed_from'),\n",
      "                   model_uri=DIST.transcribed_from, domain=Transcript, range=Optional[Union[Union[str, GeneId], List[Union[str, GeneId]]]])\n",
      "\n",
      "slots.translates_to = Slot(uri=BIOLINK.translates_to, name=\"translates to\", curie=BIOLINK.curie('translates_to'),\n",
      "                   model_uri=DIST.translates_to, domain=Transcript, range=Optional[Union[Union[str, ProteinId], List[Union[str, ProteinId]]]])\n",
      "\n",
      "slots.translation_of = Slot(uri=BIOLINK.translation_of, name=\"translation of\", curie=BIOLINK.curie('translation_of'),\n",
      "                   model_uri=DIST.translation_of, domain=Protein, range=Optional[Union[Union[str, TranscriptId], List[Union[str, TranscriptId]]]])\n",
      "\n",
      "slots.homologous_to = Slot(uri=BIOLINK.homologous_to, name=\"homologous to\", curie=BIOLINK.curie('homologous_to'),\n",
      "                   model_uri=DIST.homologous_to, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.paralogous_to = Slot(uri=BIOLINK.paralogous_to, name=\"paralogous to\", curie=BIOLINK.curie('paralogous_to'),\n",
      "                   model_uri=DIST.paralogous_to, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.orthologous_to = Slot(uri=BIOLINK.orthologous_to, name=\"orthologous to\", curie=BIOLINK.curie('orthologous_to'),\n",
      "                   model_uri=DIST.orthologous_to, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.xenologous_to = Slot(uri=BIOLINK.xenologous_to, name=\"xenologous to\", curie=BIOLINK.curie('xenologous_to'),\n",
      "                   model_uri=DIST.xenologous_to, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.coexists_with = Slot(uri=BIOLINK.coexists_with, name=\"coexists with\", curie=BIOLINK.curie('coexists_with'),\n",
      "                   model_uri=DIST.coexists_with, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.in_pathway_with = Slot(uri=BIOLINK.in_pathway_with, name=\"in pathway with\", curie=BIOLINK.curie('in_pathway_with'),\n",
      "                   model_uri=DIST.in_pathway_with, domain=None, range=Optional[Union[Union[dict, \"GeneOrGeneProduct\"], List[Union[dict, \"GeneOrGeneProduct\"]]]])\n",
      "\n",
      "slots.in_complex_with = Slot(uri=BIOLINK.in_complex_with, name=\"in complex with\", curie=BIOLINK.curie('in_complex_with'),\n",
      "                   model_uri=DIST.in_complex_with, domain=None, range=Optional[Union[Union[dict, \"GeneOrGeneProduct\"], List[Union[dict, \"GeneOrGeneProduct\"]]]])\n",
      "\n",
      "slots.in_cell_population_with = Slot(uri=BIOLINK.in_cell_population_with, name=\"in cell population with\", curie=BIOLINK.curie('in_cell_population_with'),\n",
      "                   model_uri=DIST.in_cell_population_with, domain=None, range=Optional[Union[Union[dict, \"GeneOrGeneProduct\"], List[Union[dict, \"GeneOrGeneProduct\"]]]])\n",
      "\n",
      "slots.colocalizes_with = Slot(uri=BIOLINK.colocalizes_with, name=\"colocalizes with\", curie=BIOLINK.curie('colocalizes_with'),\n",
      "                   model_uri=DIST.colocalizes_with, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.genetic_association = Slot(uri=BIOLINK.genetic_association, name=\"genetic association\", curie=BIOLINK.curie('genetic_association'),\n",
      "                   model_uri=DIST.genetic_association, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.genetically_associated_with = Slot(uri=BIOLINK.genetically_associated_with, name=\"genetically associated with\", curie=BIOLINK.curie('genetically_associated_with'),\n",
      "                   model_uri=DIST.genetically_associated_with, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.gene_associated_with_condition = Slot(uri=BIOLINK.gene_associated_with_condition, name=\"gene associated with condition\", curie=BIOLINK.curie('gene_associated_with_condition'),\n",
      "                   model_uri=DIST.gene_associated_with_condition, domain=Gene, range=Optional[Union[Union[str, DiseaseOrPhenotypicFeatureId], List[Union[str, DiseaseOrPhenotypicFeatureId]]]])\n",
      "\n",
      "slots.condition_associated_with_gene = Slot(uri=BIOLINK.condition_associated_with_gene, name=\"condition associated with gene\", curie=BIOLINK.curie('condition_associated_with_gene'),\n",
      "                   model_uri=DIST.condition_associated_with_gene, domain=DiseaseOrPhenotypicFeature, range=Optional[Union[Union[str, GeneId], List[Union[str, GeneId]]]])\n",
      "\n",
      "slots.contributes_to = Slot(uri=BIOLINK.contributes_to, name=\"contributes to\", curie=BIOLINK.curie('contributes_to'),\n",
      "                   model_uri=DIST.contributes_to, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.contribution_from = Slot(uri=BIOLINK.contribution_from, name=\"contribution from\", curie=BIOLINK.curie('contribution_from'),\n",
      "                   model_uri=DIST.contribution_from, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.causes = Slot(uri=BIOLINK.causes, name=\"causes\", curie=BIOLINK.curie('causes'),\n",
      "                   model_uri=DIST.causes, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.caused_by = Slot(uri=BIOLINK.caused_by, name=\"caused by\", curie=BIOLINK.curie('caused_by'),\n",
      "                   model_uri=DIST.caused_by, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.ameliorates_condition = Slot(uri=BIOLINK.ameliorates_condition, name=\"ameliorates condition\", curie=BIOLINK.curie('ameliorates_condition'),\n",
      "                   model_uri=DIST.ameliorates_condition, domain=None, range=Optional[Union[Union[str, DiseaseOrPhenotypicFeatureId], List[Union[str, DiseaseOrPhenotypicFeatureId]]]])\n",
      "\n",
      "slots.condition_ameliorated_by = Slot(uri=BIOLINK.condition_ameliorated_by, name=\"condition ameliorated by\", curie=BIOLINK.curie('condition_ameliorated_by'),\n",
      "                   model_uri=DIST.condition_ameliorated_by, domain=DiseaseOrPhenotypicFeature, range=Optional[Union[Union[dict, ChemicalOrDrugOrTreatment], List[Union[dict, ChemicalOrDrugOrTreatment]]]])\n",
      "\n",
      "slots.preventative_for_condition = Slot(uri=BIOLINK.preventative_for_condition, name=\"preventative for condition\", curie=BIOLINK.curie('preventative_for_condition'),\n",
      "                   model_uri=DIST.preventative_for_condition, domain=None, range=Optional[Union[Union[str, DiseaseOrPhenotypicFeatureId], List[Union[str, DiseaseOrPhenotypicFeatureId]]]])\n",
      "\n",
      "slots.condition_prevented_by = Slot(uri=BIOLINK.condition_prevented_by, name=\"condition prevented by\", curie=BIOLINK.curie('condition_prevented_by'),\n",
      "                   model_uri=DIST.condition_prevented_by, domain=DiseaseOrPhenotypicFeature, range=Optional[Union[Union[dict, ChemicalOrDrugOrTreatment], List[Union[dict, ChemicalOrDrugOrTreatment]]]])\n",
      "\n",
      "slots.promotes_condition = Slot(uri=BIOLINK.promotes_condition, name=\"promotes condition\", curie=BIOLINK.curie('promotes_condition'),\n",
      "                   model_uri=DIST.promotes_condition, domain=None, range=Optional[Union[Union[str, DiseaseOrPhenotypicFeatureId], List[Union[str, DiseaseOrPhenotypicFeatureId]]]])\n",
      "\n",
      "slots.condition_promoted_by = Slot(uri=BIOLINK.condition_promoted_by, name=\"condition promoted by\", curie=BIOLINK.curie('condition_promoted_by'),\n",
      "                   model_uri=DIST.condition_promoted_by, domain=DiseaseOrPhenotypicFeature, range=Optional[Union[Union[dict, ChemicalOrDrugOrTreatment], List[Union[dict, ChemicalOrDrugOrTreatment]]]])\n",
      "\n",
      "slots.predisposes_to_condition = Slot(uri=BIOLINK.predisposes_to_condition, name=\"predisposes to condition\", curie=BIOLINK.curie('predisposes_to_condition'),\n",
      "                   model_uri=DIST.predisposes_to_condition, domain=None, range=Optional[Union[Union[str, DiseaseOrPhenotypicFeatureId], List[Union[str, DiseaseOrPhenotypicFeatureId]]]])\n",
      "\n",
      "slots.condition_predisposed_by = Slot(uri=BIOLINK.condition_predisposed_by, name=\"condition predisposed by\", curie=BIOLINK.curie('condition_predisposed_by'),\n",
      "                   model_uri=DIST.condition_predisposed_by, domain=DiseaseOrPhenotypicFeature, range=Optional[Union[Union[dict, ChemicalOrDrugOrTreatment], List[Union[dict, ChemicalOrDrugOrTreatment]]]])\n",
      "\n",
      "slots.exacerbates_condition = Slot(uri=BIOLINK.exacerbates_condition, name=\"exacerbates condition\", curie=BIOLINK.curie('exacerbates_condition'),\n",
      "                   model_uri=DIST.exacerbates_condition, domain=None, range=Optional[Union[Union[str, DiseaseOrPhenotypicFeatureId], List[Union[str, DiseaseOrPhenotypicFeatureId]]]])\n",
      "\n",
      "slots.condition_exacerbated_by = Slot(uri=BIOLINK.condition_exacerbated_by, name=\"condition exacerbated by\", curie=BIOLINK.curie('condition_exacerbated_by'),\n",
      "                   model_uri=DIST.condition_exacerbated_by, domain=DiseaseOrPhenotypicFeature, range=Optional[Union[Union[dict, ChemicalOrDrugOrTreatment], List[Union[dict, ChemicalOrDrugOrTreatment]]]])\n",
      "\n",
      "slots.treats = Slot(uri=BIOLINK.treats, name=\"treats\", curie=BIOLINK.curie('treats'),\n",
      "                   model_uri=DIST.treats, domain=None, range=Optional[Union[Union[str, DiseaseOrPhenotypicFeatureId], List[Union[str, DiseaseOrPhenotypicFeatureId]]]])\n",
      "\n",
      "slots.treated_by = Slot(uri=BIOLINK.treated_by, name=\"treated by\", curie=BIOLINK.curie('treated_by'),\n",
      "                   model_uri=DIST.treated_by, domain=DiseaseOrPhenotypicFeature, range=Optional[Union[Union[dict, ChemicalOrDrugOrTreatment], List[Union[dict, ChemicalOrDrugOrTreatment]]]])\n",
      "\n",
      "slots.studied_to_treat = Slot(uri=BIOLINK.studied_to_treat, name=\"studied to treat\", curie=BIOLINK.curie('studied_to_treat'),\n",
      "                   model_uri=DIST.studied_to_treat, domain=None, range=Optional[Union[Union[str, DiseaseOrPhenotypicFeatureId], List[Union[str, DiseaseOrPhenotypicFeatureId]]]])\n",
      "\n",
      "slots.in_clinical_trials_for = Slot(uri=BIOLINK.in_clinical_trials_for, name=\"in clinical trials for\", curie=BIOLINK.curie('in_clinical_trials_for'),\n",
      "                   model_uri=DIST.in_clinical_trials_for, domain=None, range=Optional[Union[Union[str, DiseaseOrPhenotypicFeatureId], List[Union[str, DiseaseOrPhenotypicFeatureId]]]])\n",
      "\n",
      "slots.tested_by_clinical_trials_of = Slot(uri=BIOLINK.tested_by_clinical_trials_of, name=\"tested by clinical trials of\", curie=BIOLINK.curie('tested_by_clinical_trials_of'),\n",
      "                   model_uri=DIST.tested_by_clinical_trials_of, domain=DiseaseOrPhenotypicFeature, range=Optional[Union[Union[dict, ChemicalOrDrugOrTreatment], List[Union[dict, ChemicalOrDrugOrTreatment]]]])\n",
      "\n",
      "slots.treated_in_studies_by = Slot(uri=BIOLINK.treated_in_studies_by, name=\"treated in studies by\", curie=BIOLINK.curie('treated_in_studies_by'),\n",
      "                   model_uri=DIST.treated_in_studies_by, domain=DiseaseOrPhenotypicFeature, range=Optional[Union[Union[dict, ChemicalOrDrugOrTreatment], List[Union[dict, ChemicalOrDrugOrTreatment]]]])\n",
      "\n",
      "slots.tested_by_preclinical_trials_of = Slot(uri=BIOLINK.tested_by_preclinical_trials_of, name=\"tested by preclinical trials of\", curie=BIOLINK.curie('tested_by_preclinical_trials_of'),\n",
      "                   model_uri=DIST.tested_by_preclinical_trials_of, domain=DiseaseOrPhenotypicFeature, range=Optional[Union[Union[dict, ChemicalOrDrugOrTreatment], List[Union[dict, ChemicalOrDrugOrTreatment]]]])\n",
      "\n",
      "slots.in_preclinical_trials_for = Slot(uri=BIOLINK.in_preclinical_trials_for, name=\"in preclinical trials for\", curie=BIOLINK.curie('in_preclinical_trials_for'),\n",
      "                   model_uri=DIST.in_preclinical_trials_for, domain=None, range=Optional[Union[Union[str, DiseaseOrPhenotypicFeatureId], List[Union[str, DiseaseOrPhenotypicFeatureId]]]])\n",
      "\n",
      "slots.beneficial_in_models_for = Slot(uri=BIOLINK.beneficial_in_models_for, name=\"beneficial in models for\", curie=BIOLINK.curie('beneficial_in_models_for'),\n",
      "                   model_uri=DIST.beneficial_in_models_for, domain=None, range=Optional[Union[Union[str, DiseaseOrPhenotypicFeatureId], List[Union[str, DiseaseOrPhenotypicFeatureId]]]])\n",
      "\n",
      "slots.models_demonstrating_benefits_for = Slot(uri=BIOLINK.models_demonstrating_benefits_for, name=\"models demonstrating benefits for\", curie=BIOLINK.curie('models_demonstrating_benefits_for'),\n",
      "                   model_uri=DIST.models_demonstrating_benefits_for, domain=DiseaseOrPhenotypicFeature, range=Optional[Union[Union[dict, ChemicalOrDrugOrTreatment], List[Union[dict, ChemicalOrDrugOrTreatment]]]])\n",
      "\n",
      "slots.applied_to_treat = Slot(uri=BIOLINK.applied_to_treat, name=\"applied to treat\", curie=BIOLINK.curie('applied_to_treat'),\n",
      "                   model_uri=DIST.applied_to_treat, domain=None, range=Optional[Union[Union[str, DiseaseOrPhenotypicFeatureId], List[Union[str, DiseaseOrPhenotypicFeatureId]]]])\n",
      "\n",
      "slots.treatment_applications_from = Slot(uri=BIOLINK.treatment_applications_from, name=\"treatment applications from\", curie=BIOLINK.curie('treatment_applications_from'),\n",
      "                   model_uri=DIST.treatment_applications_from, domain=DiseaseOrPhenotypicFeature, range=Optional[Union[Union[dict, ChemicalOrDrugOrTreatment], List[Union[dict, ChemicalOrDrugOrTreatment]]]])\n",
      "\n",
      "slots.treats_or_applied_or_studied_to_treat = Slot(uri=BIOLINK.treats_or_applied_or_studied_to_treat, name=\"treats or applied or studied to treat\", curie=BIOLINK.curie('treats_or_applied_or_studied_to_treat'),\n",
      "                   model_uri=DIST.treats_or_applied_or_studied_to_treat, domain=None, range=Optional[Union[Union[str, DiseaseOrPhenotypicFeatureId], List[Union[str, DiseaseOrPhenotypicFeatureId]]]])\n",
      "\n",
      "slots.subject_of_treatment_application_or_study_for_treatment_by = Slot(uri=BIOLINK.subject_of_treatment_application_or_study_for_treatment_by, name=\"subject of treatment application or study for treatment by\", curie=BIOLINK.curie('subject_of_treatment_application_or_study_for_treatment_by'),\n",
      "                   model_uri=DIST.subject_of_treatment_application_or_study_for_treatment_by, domain=DiseaseOrPhenotypicFeature, range=Optional[Union[Union[dict, ChemicalOrDrugOrTreatment], List[Union[dict, ChemicalOrDrugOrTreatment]]]])\n",
      "\n",
      "slots.correlated_with = Slot(uri=BIOLINK.correlated_with, name=\"correlated with\", curie=BIOLINK.curie('correlated_with'),\n",
      "                   model_uri=DIST.correlated_with, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.positively_correlated_with = Slot(uri=BIOLINK.positively_correlated_with, name=\"positively correlated with\", curie=BIOLINK.curie('positively_correlated_with'),\n",
      "                   model_uri=DIST.positively_correlated_with, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.negatively_correlated_with = Slot(uri=BIOLINK.negatively_correlated_with, name=\"negatively correlated with\", curie=BIOLINK.curie('negatively_correlated_with'),\n",
      "                   model_uri=DIST.negatively_correlated_with, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.occurs_together_in_literature_with = Slot(uri=BIOLINK.occurs_together_in_literature_with, name=\"occurs together in literature with\", curie=BIOLINK.curie('occurs_together_in_literature_with'),\n",
      "                   model_uri=DIST.occurs_together_in_literature_with, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.coexpressed_with = Slot(uri=BIOLINK.coexpressed_with, name=\"coexpressed with\", curie=BIOLINK.curie('coexpressed_with'),\n",
      "                   model_uri=DIST.coexpressed_with, domain=None, range=Optional[Union[Union[dict, \"GeneOrGeneProduct\"], List[Union[dict, \"GeneOrGeneProduct\"]]]])\n",
      "\n",
      "slots.has_biomarker = Slot(uri=BIOLINK.has_biomarker, name=\"has biomarker\", curie=BIOLINK.curie('has_biomarker'),\n",
      "                   model_uri=DIST.has_biomarker, domain=DiseaseOrPhenotypicFeature, range=Optional[Union[Union[dict, \"ChemicalEntityOrGeneOrGeneProduct\"], List[Union[dict, \"ChemicalEntityOrGeneOrGeneProduct\"]]]])\n",
      "\n",
      "slots.biomarker_for = Slot(uri=BIOLINK.biomarker_for, name=\"biomarker for\", curie=BIOLINK.curie('biomarker_for'),\n",
      "                   model_uri=DIST.biomarker_for, domain=None, range=Optional[Union[Union[str, DiseaseOrPhenotypicFeatureId], List[Union[str, DiseaseOrPhenotypicFeatureId]]]])\n",
      "\n",
      "slots.expressed_in = Slot(uri=BIOLINK.expressed_in, name=\"expressed in\", curie=BIOLINK.curie('expressed_in'),\n",
      "                   model_uri=DIST.expressed_in, domain=None, range=Optional[Union[Union[str, AnatomicalEntityId], List[Union[str, AnatomicalEntityId]]]])\n",
      "\n",
      "slots.expresses = Slot(uri=BIOLINK.expresses, name=\"expresses\", curie=BIOLINK.curie('expresses'),\n",
      "                   model_uri=DIST.expresses, domain=AnatomicalEntity, range=Optional[Union[Union[dict, \"GeneOrGeneProduct\"], List[Union[dict, \"GeneOrGeneProduct\"]]]])\n",
      "\n",
      "slots.has_phenotype = Slot(uri=BIOLINK.has_phenotype, name=\"has phenotype\", curie=BIOLINK.curie('has_phenotype'),\n",
      "                   model_uri=DIST.has_phenotype, domain=BiologicalEntity, range=Optional[Union[Union[str, PhenotypicFeatureId], List[Union[str, PhenotypicFeatureId]]]])\n",
      "\n",
      "slots.phenotype_of = Slot(uri=BIOLINK.phenotype_of, name=\"phenotype of\", curie=BIOLINK.curie('phenotype_of'),\n",
      "                   model_uri=DIST.phenotype_of, domain=PhenotypicFeature, range=Optional[Union[Union[str, BiologicalEntityId], List[Union[str, BiologicalEntityId]]]])\n",
      "\n",
      "slots.occurs_in = Slot(uri=BIOLINK.occurs_in, name=\"occurs in\", curie=BIOLINK.curie('occurs_in'),\n",
      "                   model_uri=DIST.occurs_in, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.contains_process = Slot(uri=BIOLINK.contains_process, name=\"contains process\", curie=BIOLINK.curie('contains_process'),\n",
      "                   model_uri=DIST.contains_process, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.located_in = Slot(uri=BIOLINK.located_in, name=\"located in\", curie=BIOLINK.curie('located_in'),\n",
      "                   model_uri=DIST.located_in, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.location_of = Slot(uri=BIOLINK.location_of, name=\"location of\", curie=BIOLINK.curie('location_of'),\n",
      "                   model_uri=DIST.location_of, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.disease_has_location = Slot(uri=BIOLINK.disease_has_location, name=\"disease has location\", curie=BIOLINK.curie('disease_has_location'),\n",
      "                   model_uri=DIST.disease_has_location, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.location_of_disease = Slot(uri=BIOLINK.location_of_disease, name=\"location of disease\", curie=BIOLINK.curie('location_of_disease'),\n",
      "                   model_uri=DIST.location_of_disease, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.similar_to = Slot(uri=BIOLINK.similar_to, name=\"similar to\", curie=BIOLINK.curie('similar_to'),\n",
      "                   model_uri=DIST.similar_to, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.chemically_similar_to = Slot(uri=BIOLINK.chemically_similar_to, name=\"chemically similar to\", curie=BIOLINK.curie('chemically_similar_to'),\n",
      "                   model_uri=DIST.chemically_similar_to, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.has_sequence_location = Slot(uri=BIOLINK.has_sequence_location, name=\"has sequence location\", curie=BIOLINK.curie('has_sequence_location'),\n",
      "                   model_uri=DIST.has_sequence_location, domain=NucleicAcidEntity, range=Optional[Union[Union[str, NucleicAcidEntityId], List[Union[str, NucleicAcidEntityId]]]])\n",
      "\n",
      "slots.sequence_location_of = Slot(uri=BIOLINK.sequence_location_of, name=\"sequence location of\", curie=BIOLINK.curie('sequence_location_of'),\n",
      "                   model_uri=DIST.sequence_location_of, domain=NucleicAcidEntity, range=Optional[Union[Union[str, NucleicAcidEntityId], List[Union[str, NucleicAcidEntityId]]]])\n",
      "\n",
      "slots.model_of = Slot(uri=BIOLINK.model_of, name=\"model of\", curie=BIOLINK.curie('model_of'),\n",
      "                   model_uri=DIST.model_of, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.models = Slot(uri=BIOLINK.models, name=\"models\", curie=BIOLINK.curie('models'),\n",
      "                   model_uri=DIST.models, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.overlaps = Slot(uri=BIOLINK.overlaps, name=\"overlaps\", curie=BIOLINK.curie('overlaps'),\n",
      "                   model_uri=DIST.overlaps, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.has_part = Slot(uri=BIOLINK.has_part, name=\"has part\", curie=BIOLINK.curie('has_part'),\n",
      "                   model_uri=DIST.has_part, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.has_plasma_membrane_part = Slot(uri=BIOLINK.has_plasma_membrane_part, name=\"has plasma membrane part\", curie=BIOLINK.curie('has_plasma_membrane_part'),\n",
      "                   model_uri=DIST.has_plasma_membrane_part, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.composed_primarily_of = Slot(uri=BIOLINK.composed_primarily_of, name=\"composed primarily of\", curie=BIOLINK.curie('composed_primarily_of'),\n",
      "                   model_uri=DIST.composed_primarily_of, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.primarily_composed_of = Slot(uri=BIOLINK.primarily_composed_of, name=\"primarily composed of\", curie=BIOLINK.curie('primarily_composed_of'),\n",
      "                   model_uri=DIST.primarily_composed_of, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.plasma_membrane_part_of = Slot(uri=BIOLINK.plasma_membrane_part_of, name=\"plasma membrane part of\", curie=BIOLINK.curie('plasma_membrane_part_of'),\n",
      "                   model_uri=DIST.plasma_membrane_part_of, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.part_of = Slot(uri=BIOLINK.part_of, name=\"part of\", curie=BIOLINK.curie('part_of'),\n",
      "                   model_uri=DIST.part_of, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.has_input = Slot(uri=BIOLINK.has_input, name=\"has input\", curie=BIOLINK.curie('has_input'),\n",
      "                   model_uri=DIST.has_input, domain=BiologicalProcessOrActivity, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.is_input_of = Slot(uri=BIOLINK.is_input_of, name=\"is input of\", curie=BIOLINK.curie('is_input_of'),\n",
      "                   model_uri=DIST.is_input_of, domain=NamedThing, range=Optional[Union[Union[str, BiologicalProcessOrActivityId], List[Union[str, BiologicalProcessOrActivityId]]]])\n",
      "\n",
      "slots.has_output = Slot(uri=BIOLINK.has_output, name=\"has output\", curie=BIOLINK.curie('has_output'),\n",
      "                   model_uri=DIST.has_output, domain=BiologicalProcessOrActivity, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.is_output_of = Slot(uri=BIOLINK.is_output_of, name=\"is output of\", curie=BIOLINK.curie('is_output_of'),\n",
      "                   model_uri=DIST.is_output_of, domain=NamedThing, range=Optional[Union[Union[str, BiologicalProcessOrActivityId], List[Union[str, BiologicalProcessOrActivityId]]]])\n",
      "\n",
      "slots.has_participant = Slot(uri=BIOLINK.has_participant, name=\"has participant\", curie=BIOLINK.curie('has_participant'),\n",
      "                   model_uri=DIST.has_participant, domain=BiologicalProcessOrActivity, range=Optional[Union[Union[dict, Occurrent], List[Union[dict, Occurrent]]]])\n",
      "\n",
      "slots.catalyzes = Slot(uri=BIOLINK.catalyzes, name=\"catalyzes\", curie=BIOLINK.curie('catalyzes'),\n",
      "                   model_uri=DIST.catalyzes, domain=None, range=Optional[Union[Union[str, BiologicalProcessOrActivityId], List[Union[str, BiologicalProcessOrActivityId]]]])\n",
      "\n",
      "slots.has_catalyst = Slot(uri=BIOLINK.has_catalyst, name=\"has catalyst\", curie=BIOLINK.curie('has_catalyst'),\n",
      "                   model_uri=DIST.has_catalyst, domain=BiologicalProcessOrActivity, range=Optional[Union[Union[dict, Occurrent], List[Union[dict, Occurrent]]]])\n",
      "\n",
      "slots.has_substrate = Slot(uri=BIOLINK.has_substrate, name=\"has substrate\", curie=BIOLINK.curie('has_substrate'),\n",
      "                   model_uri=DIST.has_substrate, domain=None, range=Optional[Union[Union[dict, \"ChemicalEntityOrGeneOrGeneProduct\"], List[Union[dict, \"ChemicalEntityOrGeneOrGeneProduct\"]]]])\n",
      "\n",
      "slots.is_substrate_of = Slot(uri=BIOLINK.is_substrate_of, name=\"is substrate of\", curie=BIOLINK.curie('is_substrate_of'),\n",
      "                   model_uri=DIST.is_substrate_of, domain=None, range=Optional[Union[Union[dict, \"ChemicalEntityOrGeneOrGeneProduct\"], List[Union[dict, \"ChemicalEntityOrGeneOrGeneProduct\"]]]])\n",
      "\n",
      "slots.participates_in = Slot(uri=BIOLINK.participates_in, name=\"participates in\", curie=BIOLINK.curie('participates_in'),\n",
      "                   model_uri=DIST.participates_in, domain=None, range=Optional[Union[Union[str, BiologicalProcessOrActivityId], List[Union[str, BiologicalProcessOrActivityId]]]])\n",
      "\n",
      "slots.actively_involved_in = Slot(uri=BIOLINK.actively_involved_in, name=\"actively involved in\", curie=BIOLINK.curie('actively_involved_in'),\n",
      "                   model_uri=DIST.actively_involved_in, domain=NamedThing, range=Optional[Union[Union[str, BiologicalProcessOrActivityId], List[Union[str, BiologicalProcessOrActivityId]]]])\n",
      "\n",
      "slots.actively_involves = Slot(uri=BIOLINK.actively_involves, name=\"actively involves\", curie=BIOLINK.curie('actively_involves'),\n",
      "                   model_uri=DIST.actively_involves, domain=BiologicalProcessOrActivity, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.capable_of = Slot(uri=BIOLINK.capable_of, name=\"capable of\", curie=BIOLINK.curie('capable_of'),\n",
      "                   model_uri=DIST.capable_of, domain=NamedThing, range=Optional[Union[Union[dict, \"Occurrent\"], List[Union[dict, \"Occurrent\"]]]])\n",
      "\n",
      "slots.can_be_carried_out_by = Slot(uri=BIOLINK.can_be_carried_out_by, name=\"can be carried out by\", curie=BIOLINK.curie('can_be_carried_out_by'),\n",
      "                   model_uri=DIST.can_be_carried_out_by, domain=None, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.enables = Slot(uri=BIOLINK.enables, name=\"enables\", curie=BIOLINK.curie('enables'),\n",
      "                   model_uri=DIST.enables, domain=PhysicalEntity, range=Optional[Union[Union[str, BiologicalProcessOrActivityId], List[Union[str, BiologicalProcessOrActivityId]]]])\n",
      "\n",
      "slots.enabled_by = Slot(uri=BIOLINK.enabled_by, name=\"enabled by\", curie=BIOLINK.curie('enabled_by'),\n",
      "                   model_uri=DIST.enabled_by, domain=BiologicalProcessOrActivity, range=Optional[Union[Union[str, PhysicalEntityId], List[Union[str, PhysicalEntityId]]]])\n",
      "\n",
      "slots.derives_into = Slot(uri=BIOLINK.derives_into, name=\"derives into\", curie=BIOLINK.curie('derives_into'),\n",
      "                   model_uri=DIST.derives_into, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.derives_from = Slot(uri=BIOLINK.derives_from, name=\"derives from\", curie=BIOLINK.curie('derives_from'),\n",
      "                   model_uri=DIST.derives_from, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.is_metabolite_of = Slot(uri=BIOLINK.is_metabolite_of, name=\"is metabolite of\", curie=BIOLINK.curie('is_metabolite_of'),\n",
      "                   model_uri=DIST.is_metabolite_of, domain=MolecularEntity, range=Optional[Union[Union[str, MolecularEntityId], List[Union[str, MolecularEntityId]]]])\n",
      "\n",
      "slots.has_metabolite = Slot(uri=BIOLINK.has_metabolite, name=\"has metabolite\", curie=BIOLINK.curie('has_metabolite'),\n",
      "                   model_uri=DIST.has_metabolite, domain=MolecularEntity, range=Optional[Union[Union[str, MolecularEntityId], List[Union[str, MolecularEntityId]]]])\n",
      "\n",
      "slots.food_component_of = Slot(uri=BIOLINK.food_component_of, name=\"food component of\", curie=BIOLINK.curie('food_component_of'),\n",
      "                   model_uri=DIST.food_component_of, domain=ChemicalEntity, range=Optional[Union[Union[str, ChemicalEntityId], List[Union[str, ChemicalEntityId]]]])\n",
      "\n",
      "slots.has_food_component = Slot(uri=BIOLINK.has_food_component, name=\"has food component\", curie=BIOLINK.curie('has_food_component'),\n",
      "                   model_uri=DIST.has_food_component, domain=ChemicalEntity, range=Optional[Union[Union[str, ChemicalEntityId], List[Union[str, ChemicalEntityId]]]])\n",
      "\n",
      "slots.nutrient_of = Slot(uri=BIOLINK.nutrient_of, name=\"nutrient of\", curie=BIOLINK.curie('nutrient_of'),\n",
      "                   model_uri=DIST.nutrient_of, domain=ChemicalEntity, range=Optional[Union[Union[str, ChemicalEntityId], List[Union[str, ChemicalEntityId]]]])\n",
      "\n",
      "slots.has_nutrient = Slot(uri=BIOLINK.has_nutrient, name=\"has nutrient\", curie=BIOLINK.curie('has_nutrient'),\n",
      "                   model_uri=DIST.has_nutrient, domain=ChemicalEntity, range=Optional[Union[Union[str, ChemicalEntityId], List[Union[str, ChemicalEntityId]]]])\n",
      "\n",
      "slots.is_active_ingredient_of = Slot(uri=BIOLINK.is_active_ingredient_of, name=\"is active ingredient of\", curie=BIOLINK.curie('is_active_ingredient_of'),\n",
      "                   model_uri=DIST.is_active_ingredient_of, domain=MolecularEntity, range=Optional[Union[Union[str, DrugId], List[Union[str, DrugId]]]], mappings = [RO[\"0002249\"]])\n",
      "\n",
      "slots.has_active_ingredient = Slot(uri=BIOLINK.has_active_ingredient, name=\"has active ingredient\", curie=BIOLINK.curie('has_active_ingredient'),\n",
      "                   model_uri=DIST.has_active_ingredient, domain=Drug, range=Optional[Union[Union[str, MolecularEntityId], List[Union[str, MolecularEntityId]]]], mappings = [RO[\"0002248\"]])\n",
      "\n",
      "slots.is_excipient_of = Slot(uri=BIOLINK.is_excipient_of, name=\"is excipient of\", curie=BIOLINK.curie('is_excipient_of'),\n",
      "                   model_uri=DIST.is_excipient_of, domain=MolecularEntity, range=Optional[Union[Union[str, DrugId], List[Union[str, DrugId]]]], mappings = [WIKIDATA[\"Q902638\"]])\n",
      "\n",
      "slots.has_excipient = Slot(uri=BIOLINK.has_excipient, name=\"has excipient\", curie=BIOLINK.curie('has_excipient'),\n",
      "                   model_uri=DIST.has_excipient, domain=Drug, range=Optional[Union[Union[str, MolecularEntityId], List[Union[str, MolecularEntityId]]]], mappings = [WIKIDATA[\"Q902638\"]])\n",
      "\n",
      "slots.manifestation_of = Slot(uri=BIOLINK.manifestation_of, name=\"manifestation of\", curie=BIOLINK.curie('manifestation_of'),\n",
      "                   model_uri=DIST.manifestation_of, domain=NamedThing, range=Optional[Union[Union[str, DiseaseId], List[Union[str, DiseaseId]]]])\n",
      "\n",
      "slots.has_manifestation = Slot(uri=BIOLINK.has_manifestation, name=\"has manifestation\", curie=BIOLINK.curie('has_manifestation'),\n",
      "                   model_uri=DIST.has_manifestation, domain=Disease, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.mode_of_inheritance_of = Slot(uri=BIOLINK.mode_of_inheritance_of, name=\"mode of inheritance of\", curie=BIOLINK.curie('mode_of_inheritance_of'),\n",
      "                   model_uri=DIST.mode_of_inheritance_of, domain=GeneticInheritance, range=Optional[Union[Union[str, DiseaseOrPhenotypicFeatureId], List[Union[str, DiseaseOrPhenotypicFeatureId]]]])\n",
      "\n",
      "slots.has_mode_of_inheritance = Slot(uri=BIOLINK.has_mode_of_inheritance, name=\"has mode of inheritance\", curie=BIOLINK.curie('has_mode_of_inheritance'),\n",
      "                   model_uri=DIST.has_mode_of_inheritance, domain=DiseaseOrPhenotypicFeature, range=Optional[Union[Union[str, GeneticInheritanceId], List[Union[str, GeneticInheritanceId]]]])\n",
      "\n",
      "slots.produces = Slot(uri=BIOLINK.produces, name=\"produces\", curie=BIOLINK.curie('produces'),\n",
      "                   model_uri=DIST.produces, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.produced_by = Slot(uri=BIOLINK.produced_by, name=\"produced by\", curie=BIOLINK.curie('produced_by'),\n",
      "                   model_uri=DIST.produced_by, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.consumes = Slot(uri=BIOLINK.consumes, name=\"consumes\", curie=BIOLINK.curie('consumes'),\n",
      "                   model_uri=DIST.consumes, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.consumed_by = Slot(uri=BIOLINK.consumed_by, name=\"consumed by\", curie=BIOLINK.curie('consumed_by'),\n",
      "                   model_uri=DIST.consumed_by, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.temporally_related_to = Slot(uri=BIOLINK.temporally_related_to, name=\"temporally related to\", curie=BIOLINK.curie('temporally_related_to'),\n",
      "                   model_uri=DIST.temporally_related_to, domain=None, range=Optional[Union[Union[dict, \"Occurrent\"], List[Union[dict, \"Occurrent\"]]]])\n",
      "\n",
      "slots.precedes = Slot(uri=BIOLINK.precedes, name=\"precedes\", curie=BIOLINK.curie('precedes'),\n",
      "                   model_uri=DIST.precedes, domain=None, range=Optional[Union[Union[dict, \"Occurrent\"], List[Union[dict, \"Occurrent\"]]]])\n",
      "\n",
      "slots.preceded_by = Slot(uri=BIOLINK.preceded_by, name=\"preceded by\", curie=BIOLINK.curie('preceded_by'),\n",
      "                   model_uri=DIST.preceded_by, domain=None, range=Optional[Union[Union[dict, \"Occurrent\"], List[Union[dict, \"Occurrent\"]]]])\n",
      "\n",
      "slots.has_variant_part = Slot(uri=BIOLINK.has_variant_part, name=\"has variant part\", curie=BIOLINK.curie('has_variant_part'),\n",
      "                   model_uri=DIST.has_variant_part, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.variant_part_of = Slot(uri=BIOLINK.variant_part_of, name=\"variant part of\", curie=BIOLINK.curie('variant_part_of'),\n",
      "                   model_uri=DIST.variant_part_of, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.related_condition = Slot(uri=BIOLINK.related_condition, name=\"related condition\", curie=BIOLINK.curie('related_condition'),\n",
      "                   model_uri=DIST.related_condition, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.is_sequence_variant_of = Slot(uri=BIOLINK.is_sequence_variant_of, name=\"is sequence variant of\", curie=BIOLINK.curie('is_sequence_variant_of'),\n",
      "                   model_uri=DIST.is_sequence_variant_of, domain=SequenceVariant, range=Optional[Union[Union[dict, GenomicEntity], List[Union[dict, GenomicEntity]]]])\n",
      "\n",
      "slots.has_sequence_variant = Slot(uri=BIOLINK.has_sequence_variant, name=\"has sequence variant\", curie=BIOLINK.curie('has_sequence_variant'),\n",
      "                   model_uri=DIST.has_sequence_variant, domain=None, range=Optional[Union[Union[str, SequenceVariantId], List[Union[str, SequenceVariantId]]]])\n",
      "\n",
      "slots.is_missense_variant_of = Slot(uri=BIOLINK.is_missense_variant_of, name=\"is missense variant of\", curie=BIOLINK.curie('is_missense_variant_of'),\n",
      "                   model_uri=DIST.is_missense_variant_of, domain=SequenceVariant, range=Optional[Union[Union[dict, GenomicEntity], List[Union[dict, GenomicEntity]]]])\n",
      "\n",
      "slots.has_missense_variant = Slot(uri=BIOLINK.has_missense_variant, name=\"has missense variant\", curie=BIOLINK.curie('has_missense_variant'),\n",
      "                   model_uri=DIST.has_missense_variant, domain=None, range=Optional[Union[Union[str, SequenceVariantId], List[Union[str, SequenceVariantId]]]])\n",
      "\n",
      "slots.is_synonymous_variant_of = Slot(uri=BIOLINK.is_synonymous_variant_of, name=\"is synonymous variant of\", curie=BIOLINK.curie('is_synonymous_variant_of'),\n",
      "                   model_uri=DIST.is_synonymous_variant_of, domain=SequenceVariant, range=Optional[Union[Union[dict, GenomicEntity], List[Union[dict, GenomicEntity]]]])\n",
      "\n",
      "slots.has_synonymous_variant = Slot(uri=BIOLINK.has_synonymous_variant, name=\"has synonymous variant\", curie=BIOLINK.curie('has_synonymous_variant'),\n",
      "                   model_uri=DIST.has_synonymous_variant, domain=None, range=Optional[Union[Union[str, SequenceVariantId], List[Union[str, SequenceVariantId]]]])\n",
      "\n",
      "slots.is_nonsense_variant_of = Slot(uri=BIOLINK.is_nonsense_variant_of, name=\"is nonsense variant of\", curie=BIOLINK.curie('is_nonsense_variant_of'),\n",
      "                   model_uri=DIST.is_nonsense_variant_of, domain=SequenceVariant, range=Optional[Union[Union[dict, GenomicEntity], List[Union[dict, GenomicEntity]]]])\n",
      "\n",
      "slots.has_nonsense_variant = Slot(uri=BIOLINK.has_nonsense_variant, name=\"has nonsense variant\", curie=BIOLINK.curie('has_nonsense_variant'),\n",
      "                   model_uri=DIST.has_nonsense_variant, domain=None, range=Optional[Union[Union[str, SequenceVariantId], List[Union[str, SequenceVariantId]]]])\n",
      "\n",
      "slots.is_frameshift_variant_of = Slot(uri=BIOLINK.is_frameshift_variant_of, name=\"is frameshift variant of\", curie=BIOLINK.curie('is_frameshift_variant_of'),\n",
      "                   model_uri=DIST.is_frameshift_variant_of, domain=SequenceVariant, range=Optional[Union[Union[dict, GenomicEntity], List[Union[dict, GenomicEntity]]]])\n",
      "\n",
      "slots.has_frameshift_variant = Slot(uri=BIOLINK.has_frameshift_variant, name=\"has frameshift variant\", curie=BIOLINK.curie('has_frameshift_variant'),\n",
      "                   model_uri=DIST.has_frameshift_variant, domain=None, range=Optional[Union[Union[str, SequenceVariantId], List[Union[str, SequenceVariantId]]]])\n",
      "\n",
      "slots.is_splice_site_variant_of = Slot(uri=BIOLINK.is_splice_site_variant_of, name=\"is splice site variant of\", curie=BIOLINK.curie('is_splice_site_variant_of'),\n",
      "                   model_uri=DIST.is_splice_site_variant_of, domain=SequenceVariant, range=Optional[Union[Union[dict, GenomicEntity], List[Union[dict, GenomicEntity]]]])\n",
      "\n",
      "slots.has_splice_site_variant = Slot(uri=BIOLINK.has_splice_site_variant, name=\"has splice site variant\", curie=BIOLINK.curie('has_splice_site_variant'),\n",
      "                   model_uri=DIST.has_splice_site_variant, domain=None, range=Optional[Union[Union[str, SequenceVariantId], List[Union[str, SequenceVariantId]]]])\n",
      "\n",
      "slots.is_nearby_variant_of = Slot(uri=BIOLINK.is_nearby_variant_of, name=\"is nearby variant of\", curie=BIOLINK.curie('is_nearby_variant_of'),\n",
      "                   model_uri=DIST.is_nearby_variant_of, domain=SequenceVariant, range=Optional[Union[Union[dict, GenomicEntity], List[Union[dict, GenomicEntity]]]])\n",
      "\n",
      "slots.has_nearby_variant = Slot(uri=BIOLINK.has_nearby_variant, name=\"has nearby variant\", curie=BIOLINK.curie('has_nearby_variant'),\n",
      "                   model_uri=DIST.has_nearby_variant, domain=None, range=Optional[Union[Union[str, SequenceVariantId], List[Union[str, SequenceVariantId]]]])\n",
      "\n",
      "slots.is_non_coding_variant_of = Slot(uri=BIOLINK.is_non_coding_variant_of, name=\"is non coding variant of\", curie=BIOLINK.curie('is_non_coding_variant_of'),\n",
      "                   model_uri=DIST.is_non_coding_variant_of, domain=SequenceVariant, range=Optional[Union[Union[dict, GenomicEntity], List[Union[dict, GenomicEntity]]]])\n",
      "\n",
      "slots.has_non_coding_variant = Slot(uri=BIOLINK.has_non_coding_variant, name=\"has non coding variant\", curie=BIOLINK.curie('has_non_coding_variant'),\n",
      "                   model_uri=DIST.has_non_coding_variant, domain=None, range=Optional[Union[Union[str, SequenceVariantId], List[Union[str, SequenceVariantId]]]])\n",
      "\n",
      "slots.disease_has_basis_in = Slot(uri=BIOLINK.disease_has_basis_in, name=\"disease has basis in\", curie=BIOLINK.curie('disease_has_basis_in'),\n",
      "                   model_uri=DIST.disease_has_basis_in, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.occurs_in_disease = Slot(uri=BIOLINK.occurs_in_disease, name=\"occurs in disease\", curie=BIOLINK.curie('occurs_in_disease'),\n",
      "                   model_uri=DIST.occurs_in_disease, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.has_adverse_event = Slot(uri=BIOLINK.has_adverse_event, name=\"has adverse event\", curie=BIOLINK.curie('has_adverse_event'),\n",
      "                   model_uri=DIST.has_adverse_event, domain=None, range=Optional[Union[Union[str, DiseaseOrPhenotypicFeatureId], List[Union[str, DiseaseOrPhenotypicFeatureId]]]])\n",
      "\n",
      "slots.adverse_event_of = Slot(uri=BIOLINK.adverse_event_of, name=\"adverse event of\", curie=BIOLINK.curie('adverse_event_of'),\n",
      "                   model_uri=DIST.adverse_event_of, domain=DiseaseOrPhenotypicFeature, range=Optional[Union[Union[dict, ChemicalOrDrugOrTreatment], List[Union[dict, ChemicalOrDrugOrTreatment]]]])\n",
      "\n",
      "slots.has_side_effect = Slot(uri=BIOLINK.has_side_effect, name=\"has side effect\", curie=BIOLINK.curie('has_side_effect'),\n",
      "                   model_uri=DIST.has_side_effect, domain=None, range=Optional[Union[Union[str, DiseaseOrPhenotypicFeatureId], List[Union[str, DiseaseOrPhenotypicFeatureId]]]])\n",
      "\n",
      "slots.is_side_effect_of = Slot(uri=BIOLINK.is_side_effect_of, name=\"is side effect of\", curie=BIOLINK.curie('is_side_effect_of'),\n",
      "                   model_uri=DIST.is_side_effect_of, domain=DiseaseOrPhenotypicFeature, range=Optional[Union[Union[dict, ChemicalOrDrugOrTreatment], List[Union[dict, ChemicalOrDrugOrTreatment]]]])\n",
      "\n",
      "slots.contraindicated_in = Slot(uri=BIOLINK.contraindicated_in, name=\"contraindicated in\", curie=BIOLINK.curie('contraindicated_in'),\n",
      "                   model_uri=DIST.contraindicated_in, domain=None, range=Optional[Union[Union[str, BiologicalEntityId], List[Union[str, BiologicalEntityId]]]])\n",
      "\n",
      "slots.chemical_entity_or_drug_or_treatment = Slot(uri=BIOLINK.chemical_entity_or_drug_or_treatment, name=\"chemical entity or drug or treatment\", curie=BIOLINK.curie('chemical_entity_or_drug_or_treatment'),\n",
      "                   model_uri=DIST.chemical_entity_or_drug_or_treatment, domain=None, range=Optional[str])\n",
      "\n",
      "slots.has_contraindication = Slot(uri=BIOLINK.has_contraindication, name=\"has contraindication\", curie=BIOLINK.curie('has_contraindication'),\n",
      "                   model_uri=DIST.has_contraindication, domain=BiologicalEntity, range=Optional[Union[Union[dict, ChemicalOrDrugOrTreatment], List[Union[dict, ChemicalOrDrugOrTreatment]]]])\n",
      "\n",
      "slots.has_not_completed = Slot(uri=BIOLINK.has_not_completed, name=\"has not completed\", curie=BIOLINK.curie('has_not_completed'),\n",
      "                   model_uri=DIST.has_not_completed, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.not_completed_by = Slot(uri=BIOLINK.not_completed_by, name=\"not completed by\", curie=BIOLINK.curie('not_completed_by'),\n",
      "                   model_uri=DIST.not_completed_by, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.has_completed = Slot(uri=BIOLINK.has_completed, name=\"has completed\", curie=BIOLINK.curie('has_completed'),\n",
      "                   model_uri=DIST.has_completed, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.completed_by = Slot(uri=BIOLINK.completed_by, name=\"completed by\", curie=BIOLINK.curie('completed_by'),\n",
      "                   model_uri=DIST.completed_by, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.in_linkage_disequilibrium_with = Slot(uri=BIOLINK.in_linkage_disequilibrium_with, name=\"in linkage disequilibrium with\", curie=BIOLINK.curie('in_linkage_disequilibrium_with'),\n",
      "                   model_uri=DIST.in_linkage_disequilibrium_with, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.has_increased_amount = Slot(uri=BIOLINK.has_increased_amount, name=\"has increased amount\", curie=BIOLINK.curie('has_increased_amount'),\n",
      "                   model_uri=DIST.has_increased_amount, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.increased_amount_of = Slot(uri=BIOLINK.increased_amount_of, name=\"increased amount of\", curie=BIOLINK.curie('increased_amount_of'),\n",
      "                   model_uri=DIST.increased_amount_of, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.has_decreased_amount = Slot(uri=BIOLINK.has_decreased_amount, name=\"has decreased amount\", curie=BIOLINK.curie('has_decreased_amount'),\n",
      "                   model_uri=DIST.has_decreased_amount, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.decreased_amount_in = Slot(uri=BIOLINK.decreased_amount_in, name=\"decreased amount in\", curie=BIOLINK.curie('decreased_amount_in'),\n",
      "                   model_uri=DIST.decreased_amount_in, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.lacks_part = Slot(uri=BIOLINK.lacks_part, name=\"lacks part\", curie=BIOLINK.curie('lacks_part'),\n",
      "                   model_uri=DIST.lacks_part, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.missing_from = Slot(uri=BIOLINK.missing_from, name=\"missing from\", curie=BIOLINK.curie('missing_from'),\n",
      "                   model_uri=DIST.missing_from, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.develops_from = Slot(uri=BIOLINK.develops_from, name=\"develops from\", curie=BIOLINK.curie('develops_from'),\n",
      "                   model_uri=DIST.develops_from, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.develops_into = Slot(uri=BIOLINK.develops_into, name=\"develops into\", curie=BIOLINK.curie('develops_into'),\n",
      "                   model_uri=DIST.develops_into, domain=NamedThing, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.in_taxon = Slot(uri=BIOLINK.in_taxon, name=\"in taxon\", curie=BIOLINK.curie('in_taxon'),\n",
      "                   model_uri=DIST.in_taxon, domain=None, range=Optional[Union[Union[str, OrganismTaxonId], List[Union[str, OrganismTaxonId]]]])\n",
      "\n",
      "slots.in_taxon_label = Slot(uri=BIOLINK.in_taxon_label, name=\"in taxon label\", curie=BIOLINK.curie('in_taxon_label'),\n",
      "                   model_uri=DIST.in_taxon_label, domain=None, range=Optional[Union[str, LabelType]])\n",
      "\n",
      "slots.taxon_of = Slot(uri=BIOLINK.taxon_of, name=\"taxon of\", curie=BIOLINK.curie('taxon_of'),\n",
      "                   model_uri=DIST.taxon_of, domain=OrganismTaxon, range=Optional[Union[Union[dict, \"ThingWithTaxon\"], List[Union[dict, \"ThingWithTaxon\"]]]])\n",
      "\n",
      "slots.has_molecular_consequence = Slot(uri=BIOLINK.has_molecular_consequence, name=\"has molecular consequence\", curie=BIOLINK.curie('has_molecular_consequence'),\n",
      "                   model_uri=DIST.has_molecular_consequence, domain=NamedThing, range=Optional[Union[Union[str, OntologyClassId], List[Union[str, OntologyClassId]]]])\n",
      "\n",
      "slots.is_molecular_consequence_of = Slot(uri=BIOLINK.is_molecular_consequence_of, name=\"is molecular consequence of\", curie=BIOLINK.curie('is_molecular_consequence_of'),\n",
      "                   model_uri=DIST.is_molecular_consequence_of, domain=None, range=Optional[Union[Union[str, NamedThingId], List[Union[str, NamedThingId]]]])\n",
      "\n",
      "slots.association_slot = Slot(uri=BIOLINK.association_slot, name=\"association slot\", curie=BIOLINK.curie('association_slot'),\n",
      "                   model_uri=DIST.association_slot, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.qualifier = Slot(uri=BIOLINK.qualifier, name=\"qualifier\", curie=BIOLINK.curie('qualifier'),\n",
      "                   model_uri=DIST.qualifier, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.original_subject = Slot(uri=BIOLINK.original_subject, name=\"original subject\", curie=BIOLINK.curie('original_subject'),\n",
      "                   model_uri=DIST.original_subject, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.original_object = Slot(uri=BIOLINK.original_object, name=\"original object\", curie=BIOLINK.curie('original_object'),\n",
      "                   model_uri=DIST.original_object, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.original_predicate = Slot(uri=BIOLINK.original_predicate, name=\"original predicate\", curie=BIOLINK.curie('original_predicate'),\n",
      "                   model_uri=DIST.original_predicate, domain=Association, range=Optional[Union[str, URIorCURIE]])\n",
      "\n",
      "slots.subject_closure = Slot(uri=BIOLINK.subject_closure, name=\"subject closure\", curie=BIOLINK.curie('subject_closure'),\n",
      "                   model_uri=DIST.subject_closure, domain=Association, range=Optional[Union[str, List[str]]])\n",
      "\n",
      "slots.object_closure = Slot(uri=BIOLINK.object_closure, name=\"object closure\", curie=BIOLINK.curie('object_closure'),\n",
      "                   model_uri=DIST.object_closure, domain=Association, range=Optional[Union[str, List[str]]])\n",
      "\n",
      "slots.subject_category = Slot(uri=BIOLINK.subject_category, name=\"subject category\", curie=BIOLINK.curie('subject_category'),\n",
      "                   model_uri=DIST.subject_category, domain=Association, range=Optional[Union[str, OntologyClassId]])\n",
      "\n",
      "slots.object_category = Slot(uri=BIOLINK.object_category, name=\"object category\", curie=BIOLINK.curie('object_category'),\n",
      "                   model_uri=DIST.object_category, domain=Association, range=Optional[Union[str, OntologyClassId]])\n",
      "\n",
      "slots.subject_category_closure = Slot(uri=BIOLINK.subject_category_closure, name=\"subject category closure\", curie=BIOLINK.curie('subject_category_closure'),\n",
      "                   model_uri=DIST.subject_category_closure, domain=Association, range=Optional[Union[Union[str, OntologyClassId], List[Union[str, OntologyClassId]]]])\n",
      "\n",
      "slots.object_category_closure = Slot(uri=BIOLINK.object_category_closure, name=\"object category closure\", curie=BIOLINK.curie('object_category_closure'),\n",
      "                   model_uri=DIST.object_category_closure, domain=Association, range=Optional[Union[Union[str, OntologyClassId], List[Union[str, OntologyClassId]]]])\n",
      "\n",
      "slots.subject_label_closure = Slot(uri=BIOLINK.subject_label_closure, name=\"subject label closure\", curie=BIOLINK.curie('subject_label_closure'),\n",
      "                   model_uri=DIST.subject_label_closure, domain=Association, range=Optional[Union[str, List[str]]])\n",
      "\n",
      "slots.object_label_closure = Slot(uri=BIOLINK.object_label_closure, name=\"object label closure\", curie=BIOLINK.curie('object_label_closure'),\n",
      "                   model_uri=DIST.object_label_closure, domain=Association, range=Optional[Union[str, List[str]]])\n",
      "\n",
      "slots.subject_namespace = Slot(uri=BIOLINK.subject_namespace, name=\"subject namespace\", curie=BIOLINK.curie('subject_namespace'),\n",
      "                   model_uri=DIST.subject_namespace, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.object_namespace = Slot(uri=BIOLINK.object_namespace, name=\"object namespace\", curie=BIOLINK.curie('object_namespace'),\n",
      "                   model_uri=DIST.object_namespace, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.subject = Slot(uri=RDF.subject, name=\"subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.subject, domain=Association, range=Union[str, NamedThingId])\n",
      "\n",
      "slots.object = Slot(uri=RDF.object, name=\"object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.object, domain=Association, range=Union[str, NamedThingId])\n",
      "\n",
      "slots.predicate = Slot(uri=RDF.predicate, name=\"predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.predicate, domain=Association, range=Union[str, PredicateType])\n",
      "\n",
      "slots.logical_interpretation = Slot(uri=BIOLINK.logical_interpretation, name=\"logical interpretation\", curie=BIOLINK.curie('logical_interpretation'),\n",
      "                   model_uri=DIST.logical_interpretation, domain=Association, range=Optional[Union[str, \"LogicalInterpretationEnum\"]])\n",
      "\n",
      "slots.relation = Slot(uri=BIOLINK.relation, name=\"relation\", curie=BIOLINK.curie('relation'),\n",
      "                   model_uri=DIST.relation, domain=None, range=Optional[str])\n",
      "\n",
      "slots.negated = Slot(uri=BIOLINK.negated, name=\"negated\", curie=BIOLINK.curie('negated'),\n",
      "                   model_uri=DIST.negated, domain=Association, range=Optional[Union[bool, Bool]])\n",
      "\n",
      "slots.has_confidence_level = Slot(uri=BIOLINK.has_confidence_level, name=\"has confidence level\", curie=BIOLINK.curie('has_confidence_level'),\n",
      "                   model_uri=DIST.has_confidence_level, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.has_evidence = Slot(uri=BIOLINK.has_evidence, name=\"has evidence\", curie=BIOLINK.curie('has_evidence'),\n",
      "                   model_uri=DIST.has_evidence, domain=Association, range=Optional[Union[Union[str, EvidenceTypeId], List[Union[str, EvidenceTypeId]]]])\n",
      "\n",
      "slots.has_supporting_study_result = Slot(uri=BIOLINK.has_supporting_study_result, name=\"has supporting study result\", curie=BIOLINK.curie('has_supporting_study_result'),\n",
      "                   model_uri=DIST.has_supporting_study_result, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.log_odds_ratio = Slot(uri=BIOLINK.log_odds_ratio, name=\"log odds ratio\", curie=BIOLINK.curie('log_odds_ratio'),\n",
      "                   model_uri=DIST.log_odds_ratio, domain=Association, range=Optional[float])\n",
      "\n",
      "slots.log_odds_ratio_95_ci = Slot(uri=BIOLINK.log_odds_ratio_95_ci, name=\"log odds ratio 95 ci\", curie=BIOLINK.curie('log_odds_ratio_95_ci'),\n",
      "                   model_uri=DIST.log_odds_ratio_95_ci, domain=Association, range=Optional[float])\n",
      "\n",
      "slots.total_sample_size = Slot(uri=BIOLINK.total_sample_size, name=\"total sample size\", curie=BIOLINK.curie('total_sample_size'),\n",
      "                   model_uri=DIST.total_sample_size, domain=Association, range=Optional[int])\n",
      "\n",
      "slots.mechanism_of_action = Slot(uri=BIOLINK.mechanism_of_action, name=\"mechanism of action\", curie=BIOLINK.curie('mechanism_of_action'),\n",
      "                   model_uri=DIST.mechanism_of_action, domain=Association, range=Optional[Union[bool, Bool]])\n",
      "\n",
      "slots.knowledge_source = Slot(uri=BIOLINK.knowledge_source, name=\"knowledge source\", curie=BIOLINK.curie('knowledge_source'),\n",
      "                   model_uri=DIST.knowledge_source, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.provided_by = Slot(uri=BIOLINK.provided_by, name=\"provided by\", curie=BIOLINK.curie('provided_by'),\n",
      "                   model_uri=DIST.provided_by, domain=NamedThing, range=Optional[Union[str, List[str]]])\n",
      "\n",
      "slots.primary_knowledge_source = Slot(uri=BIOLINK.primary_knowledge_source, name=\"primary knowledge source\", curie=BIOLINK.curie('primary_knowledge_source'),\n",
      "                   model_uri=DIST.primary_knowledge_source, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.aggregator_knowledge_source = Slot(uri=BIOLINK.aggregator_knowledge_source, name=\"aggregator knowledge source\", curie=BIOLINK.curie('aggregator_knowledge_source'),\n",
      "                   model_uri=DIST.aggregator_knowledge_source, domain=Association, range=Optional[Union[str, List[str]]])\n",
      "\n",
      "slots.supporting_data_source = Slot(uri=BIOLINK.supporting_data_source, name=\"supporting data source\", curie=BIOLINK.curie('supporting_data_source'),\n",
      "                   model_uri=DIST.supporting_data_source, domain=Association, range=Optional[Union[str, List[str]]])\n",
      "\n",
      "slots.supporting_data_set = Slot(uri=BIOLINK.supporting_data_set, name=\"supporting data set\", curie=BIOLINK.curie('supporting_data_set'),\n",
      "                   model_uri=DIST.supporting_data_set, domain=Association, range=Optional[Union[str, List[str]]])\n",
      "\n",
      "slots.chi_squared_statistic = Slot(uri=BIOLINK.chi_squared_statistic, name=\"chi squared statistic\", curie=BIOLINK.curie('chi_squared_statistic'),\n",
      "                   model_uri=DIST.chi_squared_statistic, domain=Association, range=Optional[float])\n",
      "\n",
      "slots.p_value = Slot(uri=BIOLINK.p_value, name=\"p value\", curie=BIOLINK.curie('p_value'),\n",
      "                   model_uri=DIST.p_value, domain=Association, range=Optional[float])\n",
      "\n",
      "slots.evidence_count = Slot(uri=BIOLINK.evidence_count, name=\"evidence count\", curie=BIOLINK.curie('evidence_count'),\n",
      "                   model_uri=DIST.evidence_count, domain=Association, range=Optional[int])\n",
      "\n",
      "slots.dataset_count = Slot(uri=BIOLINK.dataset_count, name=\"dataset count\", curie=BIOLINK.curie('dataset_count'),\n",
      "                   model_uri=DIST.dataset_count, domain=Association, range=Optional[int])\n",
      "\n",
      "slots.concept_count_subject = Slot(uri=BIOLINK.concept_count_subject, name=\"concept count subject\", curie=BIOLINK.curie('concept_count_subject'),\n",
      "                   model_uri=DIST.concept_count_subject, domain=Association, range=Optional[int])\n",
      "\n",
      "slots.concept_count_object = Slot(uri=BIOLINK.concept_count_object, name=\"concept count object\", curie=BIOLINK.curie('concept_count_object'),\n",
      "                   model_uri=DIST.concept_count_object, domain=Association, range=Optional[int])\n",
      "\n",
      "slots.concept_pair_count = Slot(uri=BIOLINK.concept_pair_count, name=\"concept pair count\", curie=BIOLINK.curie('concept_pair_count'),\n",
      "                   model_uri=DIST.concept_pair_count, domain=Association, range=Optional[int])\n",
      "\n",
      "slots.expected_count = Slot(uri=BIOLINK.expected_count, name=\"expected count\", curie=BIOLINK.curie('expected_count'),\n",
      "                   model_uri=DIST.expected_count, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.relative_frequency_subject = Slot(uri=BIOLINK.relative_frequency_subject, name=\"relative frequency subject\", curie=BIOLINK.curie('relative_frequency_subject'),\n",
      "                   model_uri=DIST.relative_frequency_subject, domain=Association, range=Optional[float])\n",
      "\n",
      "slots.relative_frequency_object = Slot(uri=BIOLINK.relative_frequency_object, name=\"relative frequency object\", curie=BIOLINK.curie('relative_frequency_object'),\n",
      "                   model_uri=DIST.relative_frequency_object, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.relative_frequency_subject_confidence_interval = Slot(uri=BIOLINK.relative_frequency_subject_confidence_interval, name=\"relative frequency subject confidence interval\", curie=BIOLINK.curie('relative_frequency_subject_confidence_interval'),\n",
      "                   model_uri=DIST.relative_frequency_subject_confidence_interval, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.relative_frequency_object_confidence_interval = Slot(uri=BIOLINK.relative_frequency_object_confidence_interval, name=\"relative frequency object confidence interval\", curie=BIOLINK.curie('relative_frequency_object_confidence_interval'),\n",
      "                   model_uri=DIST.relative_frequency_object_confidence_interval, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.adjusted_p_value = Slot(uri=BIOLINK.adjusted_p_value, name=\"adjusted p value\", curie=BIOLINK.curie('adjusted_p_value'),\n",
      "                   model_uri=DIST.adjusted_p_value, domain=Association, range=Optional[float])\n",
      "\n",
      "slots.bonferonni_adjusted_p_value = Slot(uri=BIOLINK.bonferonni_adjusted_p_value, name=\"bonferonni adjusted p value\", curie=BIOLINK.curie('bonferonni_adjusted_p_value'),\n",
      "                   model_uri=DIST.bonferonni_adjusted_p_value, domain=Association, range=Optional[float])\n",
      "\n",
      "slots.supporting_text = Slot(uri=BIOLINK.supporting_text, name=\"supporting text\", curie=BIOLINK.curie('supporting_text'),\n",
      "                   model_uri=DIST.supporting_text, domain=Association, range=Optional[Union[str, List[str]]])\n",
      "\n",
      "slots.supporting_documents = Slot(uri=BIOLINK.supporting_documents, name=\"supporting documents\", curie=BIOLINK.curie('supporting_documents'),\n",
      "                   model_uri=DIST.supporting_documents, domain=Association, range=Optional[Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]]])\n",
      "\n",
      "slots.subject_location_in_text = Slot(uri=BIOLINK.subject_location_in_text, name=\"subject location in text\", curie=BIOLINK.curie('subject_location_in_text'),\n",
      "                   model_uri=DIST.subject_location_in_text, domain=Association, range=Optional[Union[int, List[int]]])\n",
      "\n",
      "slots.object_location_in_text = Slot(uri=BIOLINK.object_location_in_text, name=\"object location in text\", curie=BIOLINK.curie('object_location_in_text'),\n",
      "                   model_uri=DIST.object_location_in_text, domain=Association, range=Optional[Union[int, List[int]]])\n",
      "\n",
      "slots.extraction_confidence_score = Slot(uri=BIOLINK.extraction_confidence_score, name=\"extraction confidence score\", curie=BIOLINK.curie('extraction_confidence_score'),\n",
      "                   model_uri=DIST.extraction_confidence_score, domain=Association, range=Optional[int])\n",
      "\n",
      "slots.supporting_document_type = Slot(uri=BIOLINK.supporting_document_type, name=\"supporting document type\", curie=BIOLINK.curie('supporting_document_type'),\n",
      "                   model_uri=DIST.supporting_document_type, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.supporting_document_year = Slot(uri=BIOLINK.supporting_document_year, name=\"supporting document year\", curie=BIOLINK.curie('supporting_document_year'),\n",
      "                   model_uri=DIST.supporting_document_year, domain=Association, range=Optional[int])\n",
      "\n",
      "slots.supporting_text_section_type = Slot(uri=BIOLINK.supporting_text_section_type, name=\"supporting text section type\", curie=BIOLINK.curie('supporting_text_section_type'),\n",
      "                   model_uri=DIST.supporting_text_section_type, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.ln_ratio = Slot(uri=BIOLINK.ln_ratio, name=\"ln ratio\", curie=BIOLINK.curie('ln_ratio'),\n",
      "                   model_uri=DIST.ln_ratio, domain=Association, range=Optional[float])\n",
      "\n",
      "slots.ln_ratio_confidence_interval = Slot(uri=BIOLINK.ln_ratio_confidence_interval, name=\"ln ratio confidence interval\", curie=BIOLINK.curie('ln_ratio_confidence_interval'),\n",
      "                   model_uri=DIST.ln_ratio_confidence_interval, domain=Association, range=Optional[float])\n",
      "\n",
      "slots.interacting_molecules_category = Slot(uri=BIOLINK.interacting_molecules_category, name=\"interacting molecules category\", curie=BIOLINK.curie('interacting_molecules_category'),\n",
      "                   model_uri=DIST.interacting_molecules_category, domain=Association, range=Optional[Union[str, OntologyClassId]])\n",
      "\n",
      "slots.expression_site = Slot(uri=BIOLINK.expression_site, name=\"expression site\", curie=BIOLINK.curie('expression_site'),\n",
      "                   model_uri=DIST.expression_site, domain=Association, range=Optional[Union[str, AnatomicalEntityId]])\n",
      "\n",
      "slots.phenotypic_state = Slot(uri=BIOLINK.phenotypic_state, name=\"phenotypic state\", curie=BIOLINK.curie('phenotypic_state'),\n",
      "                   model_uri=DIST.phenotypic_state, domain=Association, range=Optional[Union[str, DiseaseOrPhenotypicFeatureId]])\n",
      "\n",
      "slots.publications = Slot(uri=BIOLINK.publications, name=\"publications\", curie=BIOLINK.curie('publications'),\n",
      "                   model_uri=DIST.publications, domain=Association, range=Optional[Union[Union[str, PublicationId], List[Union[str, PublicationId]]]])\n",
      "\n",
      "slots.associated_environmental_context = Slot(uri=BIOLINK.associated_environmental_context, name=\"associated environmental context\", curie=BIOLINK.curie('associated_environmental_context'),\n",
      "                   model_uri=DIST.associated_environmental_context, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.sequence_localization_attribute = Slot(uri=BIOLINK.sequence_localization_attribute, name=\"sequence localization attribute\", curie=BIOLINK.curie('sequence_localization_attribute'),\n",
      "                   model_uri=DIST.sequence_localization_attribute, domain=GenomicSequenceLocalization, range=Optional[str])\n",
      "\n",
      "slots.interbase_coordinate = Slot(uri=BIOLINK.interbase_coordinate, name=\"interbase coordinate\", curie=BIOLINK.curie('interbase_coordinate'),\n",
      "                   model_uri=DIST.interbase_coordinate, domain=GenomicSequenceLocalization, range=Optional[int])\n",
      "\n",
      "slots.start_interbase_coordinate = Slot(uri=BIOLINK.start_interbase_coordinate, name=\"start interbase coordinate\", curie=BIOLINK.curie('start_interbase_coordinate'),\n",
      "                   model_uri=DIST.start_interbase_coordinate, domain=GenomicSequenceLocalization, range=Optional[int])\n",
      "\n",
      "slots.end_interbase_coordinate = Slot(uri=BIOLINK.end_interbase_coordinate, name=\"end interbase coordinate\", curie=BIOLINK.curie('end_interbase_coordinate'),\n",
      "                   model_uri=DIST.end_interbase_coordinate, domain=GenomicSequenceLocalization, range=Optional[int])\n",
      "\n",
      "slots.start_coordinate = Slot(uri=BIOLINK.start_coordinate, name=\"start coordinate\", curie=BIOLINK.curie('start_coordinate'),\n",
      "                   model_uri=DIST.start_coordinate, domain=GenomicSequenceLocalization, range=Optional[int])\n",
      "\n",
      "slots.end_coordinate = Slot(uri=BIOLINK.end_coordinate, name=\"end coordinate\", curie=BIOLINK.curie('end_coordinate'),\n",
      "                   model_uri=DIST.end_coordinate, domain=GenomicSequenceLocalization, range=Optional[int])\n",
      "\n",
      "slots.genome_build = Slot(uri=BIOLINK.genome_build, name=\"genome build\", curie=BIOLINK.curie('genome_build'),\n",
      "                   model_uri=DIST.genome_build, domain=GenomicSequenceLocalization, range=Optional[Union[str, \"StrandEnum\"]])\n",
      "\n",
      "slots.strand = Slot(uri=BIOLINK.strand, name=\"strand\", curie=BIOLINK.curie('strand'),\n",
      "                   model_uri=DIST.strand, domain=GenomicSequenceLocalization, range=Optional[Union[str, \"StrandEnum\"]])\n",
      "\n",
      "slots.phase = Slot(uri=BIOLINK.phase, name=\"phase\", curie=BIOLINK.curie('phase'),\n",
      "                   model_uri=DIST.phase, domain=CodingSequence, range=Optional[Union[str, \"PhaseEnum\"]])\n",
      "\n",
      "slots.clinical_approval_status = Slot(uri=BIOLINK.clinical_approval_status, name=\"clinical approval status\", curie=BIOLINK.curie('clinical_approval_status'),\n",
      "                   model_uri=DIST.clinical_approval_status, domain=Association, range=Optional[Union[str, \"ClinicalApprovalStatusEnum\"]])\n",
      "\n",
      "slots.max_research_phase = Slot(uri=BIOLINK.max_research_phase, name=\"max research phase\", curie=BIOLINK.curie('max_research_phase'),\n",
      "                   model_uri=DIST.max_research_phase, domain=Association, range=Optional[Union[str, \"ResearchPhaseEnum\"]])\n",
      "\n",
      "slots.has_supporting_studies = Slot(uri=BIOLINK.has_supporting_studies, name=\"has supporting studies\", curie=BIOLINK.curie('has_supporting_studies'),\n",
      "                   model_uri=DIST.has_supporting_studies, domain=Association, range=Optional[Union[Union[str, StudyId], List[Union[str, StudyId]]]])\n",
      "\n",
      "slots.supporting_study_metadata = Slot(uri=BIOLINK.supporting_study_metadata, name=\"supporting study metadata\", curie=BIOLINK.curie('supporting_study_metadata'),\n",
      "                   model_uri=DIST.supporting_study_metadata, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.supporting_study_method_type = Slot(uri=BIOLINK.supporting_study_method_type, name=\"supporting study method type\", curie=BIOLINK.curie('supporting_study_method_type'),\n",
      "                   model_uri=DIST.supporting_study_method_type, domain=Association, range=Optional[Union[str, URIorCURIE]])\n",
      "\n",
      "slots.supporting_study_method_description = Slot(uri=BIOLINK.supporting_study_method_description, name=\"supporting study method description\", curie=BIOLINK.curie('supporting_study_method_description'),\n",
      "                   model_uri=DIST.supporting_study_method_description, domain=Association, range=Optional[Union[str, URIorCURIE]])\n",
      "\n",
      "slots.supporting_study_size = Slot(uri=BIOLINK.supporting_study_size, name=\"supporting study size\", curie=BIOLINK.curie('supporting_study_size'),\n",
      "                   model_uri=DIST.supporting_study_size, domain=Association, range=Optional[int])\n",
      "\n",
      "slots.supporting_study_cohort = Slot(uri=BIOLINK.supporting_study_cohort, name=\"supporting study cohort\", curie=BIOLINK.curie('supporting_study_cohort'),\n",
      "                   model_uri=DIST.supporting_study_cohort, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.supporting_study_date_range = Slot(uri=BIOLINK.supporting_study_date_range, name=\"supporting study date range\", curie=BIOLINK.curie('supporting_study_date_range'),\n",
      "                   model_uri=DIST.supporting_study_date_range, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.supporting_study_context = Slot(uri=BIOLINK.supporting_study_context, name=\"supporting study context\", curie=BIOLINK.curie('supporting_study_context'),\n",
      "                   model_uri=DIST.supporting_study_context, domain=Association, range=Optional[str])\n",
      "\n",
      "slots.knowledge_level = Slot(uri=BIOLINK.knowledge_level, name=\"knowledge level\", curie=BIOLINK.curie('knowledge_level'),\n",
      "                   model_uri=DIST.knowledge_level, domain=Association, range=Union[str, \"KnowledgeLevelEnum\"])\n",
      "\n",
      "slots.agent_type = Slot(uri=BIOLINK.agent_type, name=\"agent type\", curie=BIOLINK.curie('agent_type'),\n",
      "                   model_uri=DIST.agent_type, domain=Association, range=Union[str, \"AgentTypeEnum\"])\n",
      "\n",
      "slots.attribute_name = Slot(uri=RDFS.label, name=\"attribute_name\", curie=RDFS.curie('label'),\n",
      "                   model_uri=DIST.attribute_name, domain=Attribute, range=Optional[Union[str, LabelType]])\n",
      "\n",
      "slots.named_thing_category = Slot(uri=BIOLINK.category, name=\"named thing_category\", curie=BIOLINK.curie('category'),\n",
      "                   model_uri=DIST.named_thing_category, domain=NamedThing, range=Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]])\n",
      "\n",
      "slots.organism_taxon_has_taxonomic_rank = Slot(uri=BIOLINK.has_taxonomic_rank, name=\"organism taxon_has taxonomic rank\", curie=BIOLINK.curie('has_taxonomic_rank'),\n",
      "                   model_uri=DIST.organism_taxon_has_taxonomic_rank, domain=OrganismTaxon, range=Optional[Union[str, TaxonomicRankId]], mappings = [WIKIDATA[\"P105\"]])\n",
      "\n",
      "slots.agent_id = Slot(uri=BIOLINK.id, name=\"agent_id\", curie=BIOLINK.curie('id'),\n",
      "                   model_uri=DIST.agent_id, domain=Agent, range=Union[str, AgentId])\n",
      "\n",
      "slots.agent_name = Slot(uri=RDFS.label, name=\"agent_name\", curie=RDFS.curie('label'),\n",
      "                   model_uri=DIST.agent_name, domain=Agent, range=Optional[Union[str, LabelType]])\n",
      "\n",
      "slots.publication_id = Slot(uri=BIOLINK.id, name=\"publication_id\", curie=BIOLINK.curie('id'),\n",
      "                   model_uri=DIST.publication_id, domain=Publication, range=Union[str, PublicationId])\n",
      "\n",
      "slots.publication_name = Slot(uri=RDFS.label, name=\"publication_name\", curie=RDFS.curie('label'),\n",
      "                   model_uri=DIST.publication_name, domain=Publication, range=Optional[Union[str, LabelType]])\n",
      "\n",
      "slots.publication_publication_type = Slot(uri=DCT.type, name=\"publication_publication type\", curie=DCT.curie('type'),\n",
      "                   model_uri=DIST.publication_publication_type, domain=Publication, range=Union[str, List[str]])\n",
      "\n",
      "slots.publication_pages = Slot(uri=BIOLINK.pages, name=\"publication_pages\", curie=BIOLINK.curie('pages'),\n",
      "                   model_uri=DIST.publication_pages, domain=Publication, range=Optional[Union[str, List[str]]])\n",
      "\n",
      "slots.book_id = Slot(uri=BIOLINK.id, name=\"book_id\", curie=BIOLINK.curie('id'),\n",
      "                   model_uri=DIST.book_id, domain=Book, range=Union[str, BookId])\n",
      "\n",
      "slots.book_type = Slot(uri=RDF.type, name=\"book_type\", curie=RDF.curie('type'),\n",
      "                   model_uri=DIST.book_type, domain=Book, range=Optional[Union[str, List[str]]])\n",
      "\n",
      "slots.book_chapter_published_in = Slot(uri=BIOLINK.published_in, name=\"book chapter_published in\", curie=BIOLINK.curie('published_in'),\n",
      "                   model_uri=DIST.book_chapter_published_in, domain=BookChapter, range=Union[str, URIorCURIE])\n",
      "\n",
      "slots.serial_id = Slot(uri=BIOLINK.id, name=\"serial_id\", curie=BIOLINK.curie('id'),\n",
      "                   model_uri=DIST.serial_id, domain=Serial, range=Union[str, SerialId])\n",
      "\n",
      "slots.serial_type = Slot(uri=RDF.type, name=\"serial_type\", curie=RDF.curie('type'),\n",
      "                   model_uri=DIST.serial_type, domain=Serial, range=Optional[Union[str, List[str]]])\n",
      "\n",
      "slots.article_published_in = Slot(uri=BIOLINK.published_in, name=\"article_published in\", curie=BIOLINK.curie('published_in'),\n",
      "                   model_uri=DIST.article_published_in, domain=Article, range=Union[str, URIorCURIE])\n",
      "\n",
      "slots.article_iso_abbreviation = Slot(uri=BIOLINK.iso_abbreviation, name=\"article_iso abbreviation\", curie=BIOLINK.curie('iso_abbreviation'),\n",
      "                   model_uri=DIST.article_iso_abbreviation, domain=Article, range=Optional[str])\n",
      "\n",
      "slots.retrieval_source_resource_id = Slot(uri=BIOLINK.resource_id, name=\"retrieval source_resource id\", curie=BIOLINK.curie('resource_id'),\n",
      "                   model_uri=DIST.retrieval_source_resource_id, domain=RetrievalSource, range=Union[str, URIorCURIE])\n",
      "\n",
      "slots.retrieval_source_resource_role = Slot(uri=BIOLINK.resource_role, name=\"retrieval source_resource role\", curie=BIOLINK.curie('resource_role'),\n",
      "                   model_uri=DIST.retrieval_source_resource_role, domain=RetrievalSource, range=Union[str, \"ResourceRoleEnum\"])\n",
      "\n",
      "slots.retrieval_source_upstream_resource_ids = Slot(uri=BIOLINK.upstream_resource_ids, name=\"retrieval source_upstream resource ids\", curie=BIOLINK.curie('upstream_resource_ids'),\n",
      "                   model_uri=DIST.retrieval_source_upstream_resource_ids, domain=RetrievalSource, range=Optional[Union[str, URIorCURIE]])\n",
      "\n",
      "slots.molecular_activity_has_input = Slot(uri=BIOLINK.has_input, name=\"molecular activity_has input\", curie=BIOLINK.curie('has_input'),\n",
      "                   model_uri=DIST.molecular_activity_has_input, domain=MolecularActivity, range=Optional[Union[Union[str, MolecularEntityId], List[Union[str, MolecularEntityId]]]])\n",
      "\n",
      "slots.molecular_activity_has_output = Slot(uri=BIOLINK.has_output, name=\"molecular activity_has output\", curie=BIOLINK.curie('has_output'),\n",
      "                   model_uri=DIST.molecular_activity_has_output, domain=MolecularActivity, range=Optional[Union[Union[str, MolecularEntityId], List[Union[str, MolecularEntityId]]]])\n",
      "\n",
      "slots.molecular_activity_enabled_by = Slot(uri=BIOLINK.enabled_by, name=\"molecular activity_enabled by\", curie=BIOLINK.curie('enabled_by'),\n",
      "                   model_uri=DIST.molecular_activity_enabled_by, domain=MolecularActivity, range=Optional[Union[Union[dict, \"MacromolecularMachineMixin\"], List[Union[dict, \"MacromolecularMachineMixin\"]]]])\n",
      "\n",
      "slots.organismal_entity_has_attribute = Slot(uri=BIOLINK.has_attribute, name=\"organismal entity_has attribute\", curie=BIOLINK.curie('has_attribute'),\n",
      "                   model_uri=DIST.organismal_entity_has_attribute, domain=OrganismalEntity, range=Optional[Union[Union[str, AttributeId], List[Union[str, AttributeId]]]])\n",
      "\n",
      "slots.macromolecular_machine_mixin_name = Slot(uri=RDFS.label, name=\"macromolecular machine mixin_name\", curie=RDFS.curie('label'),\n",
      "                   model_uri=DIST.macromolecular_machine_mixin_name, domain=None, range=Optional[Union[str, SymbolType]])\n",
      "\n",
      "slots.sequence_variant_has_gene = Slot(uri=BIOLINK.has_gene, name=\"sequence variant_has gene\", curie=BIOLINK.curie('has_gene'),\n",
      "                   model_uri=DIST.sequence_variant_has_gene, domain=SequenceVariant, range=Optional[Union[Union[str, GeneId], List[Union[str, GeneId]]]])\n",
      "\n",
      "slots.sequence_variant_has_biological_sequence = Slot(uri=BIOLINK.has_biological_sequence, name=\"sequence variant_has biological sequence\", curie=BIOLINK.curie('has_biological_sequence'),\n",
      "                   model_uri=DIST.sequence_variant_has_biological_sequence, domain=SequenceVariant, range=Optional[Union[str, BiologicalSequence]])\n",
      "\n",
      "slots.sequence_variant_id = Slot(uri=BIOLINK.id, name=\"sequence variant_id\", curie=BIOLINK.curie('id'),\n",
      "                   model_uri=DIST.sequence_variant_id, domain=SequenceVariant, range=Union[str, SequenceVariantId])\n",
      "\n",
      "slots.clinical_measurement_has_attribute_type = Slot(uri=BIOLINK.has_attribute_type, name=\"clinical measurement_has attribute type\", curie=BIOLINK.curie('has_attribute_type'),\n",
      "                   model_uri=DIST.clinical_measurement_has_attribute_type, domain=ClinicalMeasurement, range=Union[str, OntologyClassId])\n",
      "\n",
      "slots.clinical_finding_has_attribute = Slot(uri=BIOLINK.has_attribute, name=\"clinical finding_has attribute\", curie=BIOLINK.curie('has_attribute'),\n",
      "                   model_uri=DIST.clinical_finding_has_attribute, domain=ClinicalFinding, range=Optional[Union[Union[str, ClinicalAttributeId], List[Union[str, ClinicalAttributeId]]]])\n",
      "\n",
      "slots.socioeconomic_exposure_has_attribute = Slot(uri=BIOLINK.has_attribute, name=\"socioeconomic exposure_has attribute\", curie=BIOLINK.curie('has_attribute'),\n",
      "                   model_uri=DIST.socioeconomic_exposure_has_attribute, domain=SocioeconomicExposure, range=Union[Union[str, SocioeconomicAttributeId], List[Union[str, SocioeconomicAttributeId]]])\n",
      "\n",
      "slots.association_type = Slot(uri=RDF.type, name=\"association_type\", curie=RDF.curie('type'),\n",
      "                   model_uri=DIST.association_type, domain=Association, range=Optional[Union[str, List[str]]])\n",
      "\n",
      "slots.association_category = Slot(uri=BIOLINK.category, name=\"association_category\", curie=BIOLINK.curie('category'),\n",
      "                   model_uri=DIST.association_category, domain=Association, range=Optional[Union[Union[str, URIorCURIE], List[Union[str, URIorCURIE]]]])\n",
      "\n",
      "slots.chemical_entity_assesses_named_thing_association_subject = Slot(uri=RDF.subject, name=\"chemical entity assesses named thing association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.chemical_entity_assesses_named_thing_association_subject, domain=ChemicalEntityAssessesNamedThingAssociation, range=Union[str, ChemicalEntityId])\n",
      "\n",
      "slots.chemical_entity_assesses_named_thing_association_object = Slot(uri=RDF.object, name=\"chemical entity assesses named thing association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.chemical_entity_assesses_named_thing_association_object, domain=ChemicalEntityAssessesNamedThingAssociation, range=Union[str, NamedThingId])\n",
      "\n",
      "slots.chemical_entity_assesses_named_thing_association_predicate = Slot(uri=RDF.predicate, name=\"chemical entity assesses named thing association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.chemical_entity_assesses_named_thing_association_predicate, domain=ChemicalEntityAssessesNamedThingAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.contributor_association_subject = Slot(uri=RDF.subject, name=\"contributor association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.contributor_association_subject, domain=ContributorAssociation, range=Union[str, InformationContentEntityId])\n",
      "\n",
      "slots.contributor_association_predicate = Slot(uri=RDF.predicate, name=\"contributor association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.contributor_association_predicate, domain=ContributorAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.contributor_association_object = Slot(uri=RDF.object, name=\"contributor association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.contributor_association_object, domain=ContributorAssociation, range=Union[str, AgentId])\n",
      "\n",
      "slots.contributor_association_qualifiers = Slot(uri=BIOLINK.qualifiers, name=\"contributor association_qualifiers\", curie=BIOLINK.curie('qualifiers'),\n",
      "                   model_uri=DIST.contributor_association_qualifiers, domain=ContributorAssociation, range=Optional[Union[Union[str, OntologyClassId], List[Union[str, OntologyClassId]]]])\n",
      "\n",
      "slots.genotype_to_genotype_part_association_predicate = Slot(uri=RDF.predicate, name=\"genotype to genotype part association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.genotype_to_genotype_part_association_predicate, domain=GenotypeToGenotypePartAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.genotype_to_genotype_part_association_subject = Slot(uri=RDF.subject, name=\"genotype to genotype part association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.genotype_to_genotype_part_association_subject, domain=GenotypeToGenotypePartAssociation, range=Union[str, GenotypeId])\n",
      "\n",
      "slots.genotype_to_genotype_part_association_object = Slot(uri=RDF.object, name=\"genotype to genotype part association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.genotype_to_genotype_part_association_object, domain=GenotypeToGenotypePartAssociation, range=Union[str, GenotypeId])\n",
      "\n",
      "slots.genotype_to_gene_association_predicate = Slot(uri=RDF.predicate, name=\"genotype to gene association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.genotype_to_gene_association_predicate, domain=GenotypeToGeneAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.genotype_to_gene_association_subject = Slot(uri=RDF.subject, name=\"genotype to gene association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.genotype_to_gene_association_subject, domain=GenotypeToGeneAssociation, range=Union[str, GenotypeId])\n",
      "\n",
      "slots.genotype_to_gene_association_object = Slot(uri=RDF.object, name=\"genotype to gene association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.genotype_to_gene_association_object, domain=GenotypeToGeneAssociation, range=Union[str, GeneId])\n",
      "\n",
      "slots.genotype_to_variant_association_predicate = Slot(uri=RDF.predicate, name=\"genotype to variant association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.genotype_to_variant_association_predicate, domain=GenotypeToVariantAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.genotype_to_variant_association_subject = Slot(uri=RDF.subject, name=\"genotype to variant association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.genotype_to_variant_association_subject, domain=GenotypeToVariantAssociation, range=Union[str, GenotypeId])\n",
      "\n",
      "slots.genotype_to_variant_association_object = Slot(uri=RDF.object, name=\"genotype to variant association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.genotype_to_variant_association_object, domain=GenotypeToVariantAssociation, range=Union[str, SequenceVariantId])\n",
      "\n",
      "slots.gene_to_gene_association_subject = Slot(uri=RDF.subject, name=\"gene to gene association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.gene_to_gene_association_subject, domain=GeneToGeneAssociation, range=Union[dict, GeneOrGeneProduct])\n",
      "\n",
      "slots.gene_to_gene_association_object = Slot(uri=RDF.object, name=\"gene to gene association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.gene_to_gene_association_object, domain=GeneToGeneAssociation, range=Union[dict, GeneOrGeneProduct])\n",
      "\n",
      "slots.gene_to_gene_homology_association_subject = Slot(uri=RDF.subject, name=\"gene to gene homology association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.gene_to_gene_homology_association_subject, domain=GeneToGeneHomologyAssociation, range=Union[dict, GeneOrGeneProduct])\n",
      "\n",
      "slots.gene_to_gene_homology_association_predicate = Slot(uri=RDF.predicate, name=\"gene to gene homology association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.gene_to_gene_homology_association_predicate, domain=GeneToGeneHomologyAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.gene_to_gene_homology_association_object = Slot(uri=RDF.object, name=\"gene to gene homology association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.gene_to_gene_homology_association_object, domain=GeneToGeneHomologyAssociation, range=Union[dict, GeneOrGeneProduct])\n",
      "\n",
      "slots.gene_to_gene_family_association_subject = Slot(uri=RDF.subject, name=\"gene to gene family association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.gene_to_gene_family_association_subject, domain=GeneToGeneFamilyAssociation, range=Union[str, GeneId])\n",
      "\n",
      "slots.gene_to_gene_family_association_object = Slot(uri=RDF.object, name=\"gene to gene family association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.gene_to_gene_family_association_object, domain=GeneToGeneFamilyAssociation, range=Union[str, GeneFamilyId])\n",
      "\n",
      "slots.gene_to_gene_family_association_predicate = Slot(uri=RDF.predicate, name=\"gene to gene family association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.gene_to_gene_family_association_predicate, domain=GeneToGeneFamilyAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.gene_expression_mixin_quantifier_qualifier = Slot(uri=BIOLINK.quantifier_qualifier, name=\"gene expression mixin_quantifier qualifier\", curie=BIOLINK.curie('quantifier_qualifier'),\n",
      "                   model_uri=DIST.gene_expression_mixin_quantifier_qualifier, domain=None, range=Optional[Union[str, OntologyClassId]])\n",
      "\n",
      "slots.gene_to_gene_coexpression_association_predicate = Slot(uri=RDF.predicate, name=\"gene to gene coexpression association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.gene_to_gene_coexpression_association_predicate, domain=GeneToGeneCoexpressionAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.pairwise_gene_to_gene_interaction_predicate = Slot(uri=RDF.predicate, name=\"pairwise gene to gene interaction_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.pairwise_gene_to_gene_interaction_predicate, domain=PairwiseGeneToGeneInteraction, range=Union[str, PredicateType])\n",
      "\n",
      "slots.pairwise_molecular_interaction_subject = Slot(uri=RDF.subject, name=\"pairwise molecular interaction_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.pairwise_molecular_interaction_subject, domain=PairwiseMolecularInteraction, range=Union[str, MolecularEntityId])\n",
      "\n",
      "slots.pairwise_molecular_interaction_id = Slot(uri=BIOLINK.id, name=\"pairwise molecular interaction_id\", curie=BIOLINK.curie('id'),\n",
      "                   model_uri=DIST.pairwise_molecular_interaction_id, domain=PairwiseMolecularInteraction, range=Union[str, PairwiseMolecularInteractionId])\n",
      "\n",
      "slots.pairwise_molecular_interaction_predicate = Slot(uri=RDF.predicate, name=\"pairwise molecular interaction_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.pairwise_molecular_interaction_predicate, domain=PairwiseMolecularInteraction, range=Union[str, PredicateType])\n",
      "\n",
      "slots.pairwise_molecular_interaction_object = Slot(uri=RDF.object, name=\"pairwise molecular interaction_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.pairwise_molecular_interaction_object, domain=PairwiseMolecularInteraction, range=Union[str, MolecularEntityId])\n",
      "\n",
      "slots.cell_line_to_entity_association_mixin_subject = Slot(uri=RDF.subject, name=\"cell line to entity association mixin_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.cell_line_to_entity_association_mixin_subject, domain=None, range=Union[str, CellLineId])\n",
      "\n",
      "slots.cell_line_to_disease_or_phenotypic_feature_association_subject = Slot(uri=RDF.subject, name=\"cell line to disease or phenotypic feature association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.cell_line_to_disease_or_phenotypic_feature_association_subject, domain=CellLineToDiseaseOrPhenotypicFeatureAssociation, range=Union[str, DiseaseOrPhenotypicFeatureId])\n",
      "\n",
      "slots.chemical_entity_to_entity_association_mixin_subject = Slot(uri=RDF.subject, name=\"chemical entity to entity association mixin_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.chemical_entity_to_entity_association_mixin_subject, domain=None, range=Union[dict, ChemicalEntityOrGeneOrGeneProduct])\n",
      "\n",
      "slots.drug_to_entity_association_mixin_subject = Slot(uri=RDF.subject, name=\"drug to entity association mixin_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.drug_to_entity_association_mixin_subject, domain=None, range=Union[str, DrugId])\n",
      "\n",
      "slots.chemical_to_entity_association_mixin_subject = Slot(uri=RDF.subject, name=\"chemical to entity association mixin_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.chemical_to_entity_association_mixin_subject, domain=None, range=Union[dict, ChemicalEntityOrGeneOrGeneProduct])\n",
      "\n",
      "slots.case_to_entity_association_mixin_subject = Slot(uri=RDF.subject, name=\"case to entity association mixin_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.case_to_entity_association_mixin_subject, domain=None, range=Union[str, CaseId])\n",
      "\n",
      "slots.chemical_to_chemical_association_object = Slot(uri=RDF.object, name=\"chemical to chemical association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.chemical_to_chemical_association_object, domain=ChemicalToChemicalAssociation, range=Union[str, ChemicalEntityId])\n",
      "\n",
      "slots.reaction_to_participant_association_subject = Slot(uri=RDF.subject, name=\"reaction to participant association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.reaction_to_participant_association_subject, domain=ReactionToParticipantAssociation, range=Union[str, MolecularEntityId])\n",
      "\n",
      "slots.reaction_to_catalyst_association_object = Slot(uri=RDF.object, name=\"reaction to catalyst association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.reaction_to_catalyst_association_object, domain=ReactionToCatalystAssociation, range=Union[dict, GeneOrGeneProduct])\n",
      "\n",
      "slots.chemical_to_chemical_derivation_association_subject = Slot(uri=RDF.subject, name=\"chemical to chemical derivation association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.chemical_to_chemical_derivation_association_subject, domain=ChemicalToChemicalDerivationAssociation, range=Union[str, ChemicalEntityId])\n",
      "\n",
      "slots.chemical_to_chemical_derivation_association_object = Slot(uri=RDF.object, name=\"chemical to chemical derivation association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.chemical_to_chemical_derivation_association_object, domain=ChemicalToChemicalDerivationAssociation, range=Union[str, ChemicalEntityId])\n",
      "\n",
      "slots.chemical_to_chemical_derivation_association_predicate = Slot(uri=RDF.predicate, name=\"chemical to chemical derivation association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.chemical_to_chemical_derivation_association_predicate, domain=ChemicalToChemicalDerivationAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.chemical_to_chemical_derivation_association_catalyst_qualifier = Slot(uri=BIOLINK.catalyst_qualifier, name=\"chemical to chemical derivation association_catalyst qualifier\", curie=BIOLINK.curie('catalyst_qualifier'),\n",
      "                   model_uri=DIST.chemical_to_chemical_derivation_association_catalyst_qualifier, domain=ChemicalToChemicalDerivationAssociation, range=Optional[Union[Union[dict, MacromolecularMachineMixin], List[Union[dict, MacromolecularMachineMixin]]]])\n",
      "\n",
      "slots.chemical_to_disease_or_phenotypic_feature_association_object = Slot(uri=RDF.object, name=\"chemical to disease or phenotypic feature association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.chemical_to_disease_or_phenotypic_feature_association_object, domain=ChemicalToDiseaseOrPhenotypicFeatureAssociation, range=Union[str, DiseaseOrPhenotypicFeatureId])\n",
      "\n",
      "slots.chemical_or_drug_or_treatment_to_disease_or_phenotypic_feature_association_predicate = Slot(uri=RDF.predicate, name=\"chemical or drug or treatment to disease or phenotypic feature association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.chemical_or_drug_or_treatment_to_disease_or_phenotypic_feature_association_predicate, domain=ChemicalOrDrugOrTreatmentToDiseaseOrPhenotypicFeatureAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.chemical_or_drug_or_treatment_side_effect_disease_or_phenotypic_feature_association_predicate = Slot(uri=RDF.predicate, name=\"chemical or drug or treatment side effect disease or phenotypic feature association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.chemical_or_drug_or_treatment_side_effect_disease_or_phenotypic_feature_association_predicate, domain=ChemicalOrDrugOrTreatmentSideEffectDiseaseOrPhenotypicFeatureAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.gene_to_pathway_association_subject = Slot(uri=RDF.subject, name=\"gene to pathway association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.gene_to_pathway_association_subject, domain=GeneToPathwayAssociation, range=Union[dict, GeneOrGeneProduct])\n",
      "\n",
      "slots.gene_to_pathway_association_object = Slot(uri=RDF.object, name=\"gene to pathway association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.gene_to_pathway_association_object, domain=GeneToPathwayAssociation, range=Union[str, PathwayId])\n",
      "\n",
      "slots.molecular_activity_to_pathway_association_subject = Slot(uri=RDF.subject, name=\"molecular activity to pathway association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.molecular_activity_to_pathway_association_subject, domain=MolecularActivityToPathwayAssociation, range=Union[str, MolecularActivityId])\n",
      "\n",
      "slots.molecular_activity_to_pathway_association_object = Slot(uri=RDF.object, name=\"molecular activity to pathway association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.molecular_activity_to_pathway_association_object, domain=MolecularActivityToPathwayAssociation, range=Union[str, PathwayId])\n",
      "\n",
      "slots.molecular_activity_to_pathway_association_predicate = Slot(uri=RDF.predicate, name=\"molecular activity to pathway association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.molecular_activity_to_pathway_association_predicate, domain=MolecularActivityToPathwayAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.chemical_to_pathway_association_subject = Slot(uri=RDF.subject, name=\"chemical to pathway association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.chemical_to_pathway_association_subject, domain=ChemicalToPathwayAssociation, range=Union[str, ChemicalEntityId])\n",
      "\n",
      "slots.chemical_to_pathway_association_object = Slot(uri=RDF.object, name=\"chemical to pathway association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.chemical_to_pathway_association_object, domain=ChemicalToPathwayAssociation, range=Union[str, PathwayId])\n",
      "\n",
      "slots.named_thing_associated_with_likelihood_of_named_thing_association_predicate = Slot(uri=RDF.predicate, name=\"named thing associated with likelihood of named thing association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.named_thing_associated_with_likelihood_of_named_thing_association_predicate, domain=NamedThingAssociatedWithLikelihoodOfNamedThingAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.named_thing_associated_with_likelihood_of_named_thing_association_subject_aspect_qualifier = Slot(uri=BIOLINK.subject_aspect_qualifier, name=\"named thing associated with likelihood of named thing association_subject aspect qualifier\", curie=BIOLINK.curie('subject_aspect_qualifier'),\n",
      "                   model_uri=DIST.named_thing_associated_with_likelihood_of_named_thing_association_subject_aspect_qualifier, domain=NamedThingAssociatedWithLikelihoodOfNamedThingAssociation, range=Optional[str])\n",
      "\n",
      "slots.named_thing_associated_with_likelihood_of_named_thing_association_subject_context_qualifier = Slot(uri=BIOLINK.subject_context_qualifier, name=\"named thing associated with likelihood of named thing association_subject context qualifier\", curie=BIOLINK.curie('subject_context_qualifier'),\n",
      "                   model_uri=DIST.named_thing_associated_with_likelihood_of_named_thing_association_subject_context_qualifier, domain=NamedThingAssociatedWithLikelihoodOfNamedThingAssociation, range=Optional[Union[str, OntologyClassId]])\n",
      "\n",
      "slots.named_thing_associated_with_likelihood_of_named_thing_association_object_aspect_qualifier = Slot(uri=BIOLINK.object_aspect_qualifier, name=\"named thing associated with likelihood of named thing association_object aspect qualifier\", curie=BIOLINK.curie('object_aspect_qualifier'),\n",
      "                   model_uri=DIST.named_thing_associated_with_likelihood_of_named_thing_association_object_aspect_qualifier, domain=NamedThingAssociatedWithLikelihoodOfNamedThingAssociation, range=Optional[str])\n",
      "\n",
      "slots.named_thing_associated_with_likelihood_of_named_thing_association_object_context_qualifier = Slot(uri=BIOLINK.object_context_qualifier, name=\"named thing associated with likelihood of named thing association_object context qualifier\", curie=BIOLINK.curie('object_context_qualifier'),\n",
      "                   model_uri=DIST.named_thing_associated_with_likelihood_of_named_thing_association_object_context_qualifier, domain=NamedThingAssociatedWithLikelihoodOfNamedThingAssociation, range=Optional[Union[str, OntologyClassId]])\n",
      "\n",
      "slots.chemical_gene_interaction_association_subject = Slot(uri=RDF.subject, name=\"chemical gene interaction association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.chemical_gene_interaction_association_subject, domain=ChemicalGeneInteractionAssociation, range=Union[str, ChemicalEntityId])\n",
      "\n",
      "slots.chemical_gene_interaction_association_object = Slot(uri=RDF.object, name=\"chemical gene interaction association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.chemical_gene_interaction_association_object, domain=ChemicalGeneInteractionAssociation, range=Union[dict, GeneOrGeneProduct])\n",
      "\n",
      "slots.chemical_gene_interaction_association_predicate = Slot(uri=RDF.predicate, name=\"chemical gene interaction association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.chemical_gene_interaction_association_predicate, domain=ChemicalGeneInteractionAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.chemical_gene_interaction_association_subject_form_or_variant_qualifier = Slot(uri=BIOLINK.subject_form_or_variant_qualifier, name=\"chemical gene interaction association_subject form or variant qualifier\", curie=BIOLINK.curie('subject_form_or_variant_qualifier'),\n",
      "                   model_uri=DIST.chemical_gene_interaction_association_subject_form_or_variant_qualifier, domain=ChemicalGeneInteractionAssociation, range=Optional[Union[str, \"ChemicalOrGeneOrGeneProductFormOrVariantEnum\"]])\n",
      "\n",
      "slots.chemical_gene_interaction_association_subject_part_qualifier = Slot(uri=BIOLINK.subject_part_qualifier, name=\"chemical gene interaction association_subject part qualifier\", curie=BIOLINK.curie('subject_part_qualifier'),\n",
      "                   model_uri=DIST.chemical_gene_interaction_association_subject_part_qualifier, domain=ChemicalGeneInteractionAssociation, range=Optional[Union[str, \"GeneOrGeneProductOrChemicalPartQualifierEnum\"]])\n",
      "\n",
      "slots.chemical_gene_interaction_association_subject_derivative_qualifier = Slot(uri=BIOLINK.subject_derivative_qualifier, name=\"chemical gene interaction association_subject derivative qualifier\", curie=BIOLINK.curie('subject_derivative_qualifier'),\n",
      "                   model_uri=DIST.chemical_gene_interaction_association_subject_derivative_qualifier, domain=ChemicalGeneInteractionAssociation, range=Optional[Union[str, \"ChemicalEntityDerivativeEnum\"]])\n",
      "\n",
      "slots.chemical_gene_interaction_association_subject_context_qualifier = Slot(uri=BIOLINK.subject_context_qualifier, name=\"chemical gene interaction association_subject context qualifier\", curie=BIOLINK.curie('subject_context_qualifier'),\n",
      "                   model_uri=DIST.chemical_gene_interaction_association_subject_context_qualifier, domain=ChemicalGeneInteractionAssociation, range=Optional[Union[str, AnatomicalEntityId]])\n",
      "\n",
      "slots.chemical_gene_interaction_association_object_form_or_variant_qualifier = Slot(uri=BIOLINK.object_form_or_variant_qualifier, name=\"chemical gene interaction association_object form or variant qualifier\", curie=BIOLINK.curie('object_form_or_variant_qualifier'),\n",
      "                   model_uri=DIST.chemical_gene_interaction_association_object_form_or_variant_qualifier, domain=ChemicalGeneInteractionAssociation, range=Optional[Union[str, \"ChemicalOrGeneOrGeneProductFormOrVariantEnum\"]])\n",
      "\n",
      "slots.chemical_gene_interaction_association_object_part_qualifier = Slot(uri=BIOLINK.object_part_qualifier, name=\"chemical gene interaction association_object part qualifier\", curie=BIOLINK.curie('object_part_qualifier'),\n",
      "                   model_uri=DIST.chemical_gene_interaction_association_object_part_qualifier, domain=ChemicalGeneInteractionAssociation, range=Optional[Union[str, \"GeneOrGeneProductOrChemicalPartQualifierEnum\"]])\n",
      "\n",
      "slots.chemical_gene_interaction_association_object_context_qualifier = Slot(uri=BIOLINK.object_context_qualifier, name=\"chemical gene interaction association_object context qualifier\", curie=BIOLINK.curie('object_context_qualifier'),\n",
      "                   model_uri=DIST.chemical_gene_interaction_association_object_context_qualifier, domain=ChemicalGeneInteractionAssociation, range=Optional[Union[str, AnatomicalEntityId]])\n",
      "\n",
      "slots.chemical_gene_interaction_association_anatomical_context_qualifier = Slot(uri=BIOLINK.anatomical_context_qualifier, name=\"chemical gene interaction association_anatomical context qualifier\", curie=BIOLINK.curie('anatomical_context_qualifier'),\n",
      "                   model_uri=DIST.chemical_gene_interaction_association_anatomical_context_qualifier, domain=ChemicalGeneInteractionAssociation, range=Optional[Union[str, AnatomicalEntityId]])\n",
      "\n",
      "slots.gene_regulates_gene_association_subject = Slot(uri=RDF.subject, name=\"gene regulates gene association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.gene_regulates_gene_association_subject, domain=GeneRegulatesGeneAssociation, range=Union[dict, GeneOrGeneProduct])\n",
      "\n",
      "slots.gene_regulates_gene_association_predicate = Slot(uri=RDF.predicate, name=\"gene regulates gene association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.gene_regulates_gene_association_predicate, domain=GeneRegulatesGeneAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.gene_regulates_gene_association_object = Slot(uri=RDF.object, name=\"gene regulates gene association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.gene_regulates_gene_association_object, domain=GeneRegulatesGeneAssociation, range=Union[dict, GeneOrGeneProduct])\n",
      "\n",
      "slots.gene_regulates_gene_association_object_aspect_qualifier = Slot(uri=BIOLINK.object_aspect_qualifier, name=\"gene regulates gene association_object aspect qualifier\", curie=BIOLINK.curie('object_aspect_qualifier'),\n",
      "                   model_uri=DIST.gene_regulates_gene_association_object_aspect_qualifier, domain=GeneRegulatesGeneAssociation, range=Union[str, \"GeneOrGeneProductOrChemicalEntityAspectEnum\"])\n",
      "\n",
      "slots.gene_regulates_gene_association_object_direction_qualifier = Slot(uri=BIOLINK.object_direction_qualifier, name=\"gene regulates gene association_object direction qualifier\", curie=BIOLINK.curie('object_direction_qualifier'),\n",
      "                   model_uri=DIST.gene_regulates_gene_association_object_direction_qualifier, domain=GeneRegulatesGeneAssociation, range=Union[str, \"DirectionQualifierEnum\"])\n",
      "\n",
      "slots.gene_regulates_gene_association_qualified_predicate = Slot(uri=BIOLINK.qualified_predicate, name=\"gene regulates gene association_qualified predicate\", curie=BIOLINK.curie('qualified_predicate'),\n",
      "                   model_uri=DIST.gene_regulates_gene_association_qualified_predicate, domain=GeneRegulatesGeneAssociation, range=str)\n",
      "\n",
      "slots.process_regulates_process_association_subject = Slot(uri=RDF.subject, name=\"process regulates process association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.process_regulates_process_association_subject, domain=ProcessRegulatesProcessAssociation, range=Union[str, BiologicalProcessId])\n",
      "\n",
      "slots.process_regulates_process_association_predicate = Slot(uri=RDF.predicate, name=\"process regulates process association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.process_regulates_process_association_predicate, domain=ProcessRegulatesProcessAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.process_regulates_process_association_object = Slot(uri=RDF.object, name=\"process regulates process association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.process_regulates_process_association_object, domain=ProcessRegulatesProcessAssociation, range=Union[str, BiologicalProcessId])\n",
      "\n",
      "slots.chemical_affects_gene_association_subject = Slot(uri=RDF.subject, name=\"chemical affects gene association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.chemical_affects_gene_association_subject, domain=ChemicalAffectsGeneAssociation, range=Union[str, ChemicalEntityId])\n",
      "\n",
      "slots.chemical_affects_gene_association_subject_form_or_variant_qualifier = Slot(uri=BIOLINK.subject_form_or_variant_qualifier, name=\"chemical affects gene association_subject form or variant qualifier\", curie=BIOLINK.curie('subject_form_or_variant_qualifier'),\n",
      "                   model_uri=DIST.chemical_affects_gene_association_subject_form_or_variant_qualifier, domain=ChemicalAffectsGeneAssociation, range=Optional[Union[str, \"ChemicalOrGeneOrGeneProductFormOrVariantEnum\"]])\n",
      "\n",
      "slots.chemical_affects_gene_association_subject_part_qualifier = Slot(uri=BIOLINK.subject_part_qualifier, name=\"chemical affects gene association_subject part qualifier\", curie=BIOLINK.curie('subject_part_qualifier'),\n",
      "                   model_uri=DIST.chemical_affects_gene_association_subject_part_qualifier, domain=ChemicalAffectsGeneAssociation, range=Optional[Union[str, \"GeneOrGeneProductOrChemicalPartQualifierEnum\"]])\n",
      "\n",
      "slots.chemical_affects_gene_association_subject_derivative_qualifier = Slot(uri=BIOLINK.subject_derivative_qualifier, name=\"chemical affects gene association_subject derivative qualifier\", curie=BIOLINK.curie('subject_derivative_qualifier'),\n",
      "                   model_uri=DIST.chemical_affects_gene_association_subject_derivative_qualifier, domain=ChemicalAffectsGeneAssociation, range=Optional[Union[str, \"ChemicalEntityDerivativeEnum\"]])\n",
      "\n",
      "slots.chemical_affects_gene_association_subject_aspect_qualifier = Slot(uri=BIOLINK.subject_aspect_qualifier, name=\"chemical affects gene association_subject aspect qualifier\", curie=BIOLINK.curie('subject_aspect_qualifier'),\n",
      "                   model_uri=DIST.chemical_affects_gene_association_subject_aspect_qualifier, domain=ChemicalAffectsGeneAssociation, range=Optional[Union[str, \"GeneOrGeneProductOrChemicalEntityAspectEnum\"]])\n",
      "\n",
      "slots.chemical_affects_gene_association_subject_context_qualifier = Slot(uri=BIOLINK.subject_context_qualifier, name=\"chemical affects gene association_subject context qualifier\", curie=BIOLINK.curie('subject_context_qualifier'),\n",
      "                   model_uri=DIST.chemical_affects_gene_association_subject_context_qualifier, domain=ChemicalAffectsGeneAssociation, range=Optional[Union[str, AnatomicalEntityId]])\n",
      "\n",
      "slots.chemical_affects_gene_association_subject_direction_qualifier = Slot(uri=BIOLINK.subject_direction_qualifier, name=\"chemical affects gene association_subject direction qualifier\", curie=BIOLINK.curie('subject_direction_qualifier'),\n",
      "                   model_uri=DIST.chemical_affects_gene_association_subject_direction_qualifier, domain=ChemicalAffectsGeneAssociation, range=Optional[Union[str, \"DirectionQualifierEnum\"]])\n",
      "\n",
      "slots.chemical_affects_gene_association_predicate = Slot(uri=RDF.predicate, name=\"chemical affects gene association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.chemical_affects_gene_association_predicate, domain=ChemicalAffectsGeneAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.chemical_affects_gene_association_qualified_predicate = Slot(uri=BIOLINK.qualified_predicate, name=\"chemical affects gene association_qualified predicate\", curie=BIOLINK.curie('qualified_predicate'),\n",
      "                   model_uri=DIST.chemical_affects_gene_association_qualified_predicate, domain=ChemicalAffectsGeneAssociation, range=Optional[str])\n",
      "\n",
      "slots.chemical_affects_gene_association_object = Slot(uri=RDF.object, name=\"chemical affects gene association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.chemical_affects_gene_association_object, domain=ChemicalAffectsGeneAssociation, range=Union[dict, GeneOrGeneProduct])\n",
      "\n",
      "slots.chemical_affects_gene_association_object_form_or_variant_qualifier = Slot(uri=BIOLINK.object_form_or_variant_qualifier, name=\"chemical affects gene association_object form or variant qualifier\", curie=BIOLINK.curie('object_form_or_variant_qualifier'),\n",
      "                   model_uri=DIST.chemical_affects_gene_association_object_form_or_variant_qualifier, domain=ChemicalAffectsGeneAssociation, range=Optional[Union[str, \"ChemicalOrGeneOrGeneProductFormOrVariantEnum\"]])\n",
      "\n",
      "slots.chemical_affects_gene_association_object_part_qualifier = Slot(uri=BIOLINK.object_part_qualifier, name=\"chemical affects gene association_object part qualifier\", curie=BIOLINK.curie('object_part_qualifier'),\n",
      "                   model_uri=DIST.chemical_affects_gene_association_object_part_qualifier, domain=ChemicalAffectsGeneAssociation, range=Optional[Union[str, \"GeneOrGeneProductOrChemicalPartQualifierEnum\"]])\n",
      "\n",
      "slots.chemical_affects_gene_association_object_aspect_qualifier = Slot(uri=BIOLINK.object_aspect_qualifier, name=\"chemical affects gene association_object aspect qualifier\", curie=BIOLINK.curie('object_aspect_qualifier'),\n",
      "                   model_uri=DIST.chemical_affects_gene_association_object_aspect_qualifier, domain=ChemicalAffectsGeneAssociation, range=Optional[Union[str, \"GeneOrGeneProductOrChemicalEntityAspectEnum\"]])\n",
      "\n",
      "slots.chemical_affects_gene_association_object_context_qualifier = Slot(uri=BIOLINK.object_context_qualifier, name=\"chemical affects gene association_object context qualifier\", curie=BIOLINK.curie('object_context_qualifier'),\n",
      "                   model_uri=DIST.chemical_affects_gene_association_object_context_qualifier, domain=ChemicalAffectsGeneAssociation, range=Optional[Union[str, AnatomicalEntityId]])\n",
      "\n",
      "slots.chemical_affects_gene_association_object_direction_qualifier = Slot(uri=BIOLINK.object_direction_qualifier, name=\"chemical affects gene association_object direction qualifier\", curie=BIOLINK.curie('object_direction_qualifier'),\n",
      "                   model_uri=DIST.chemical_affects_gene_association_object_direction_qualifier, domain=ChemicalAffectsGeneAssociation, range=Optional[Union[str, \"DirectionQualifierEnum\"]])\n",
      "\n",
      "slots.chemical_affects_gene_association_causal_mechanism_qualifier = Slot(uri=BIOLINK.causal_mechanism_qualifier, name=\"chemical affects gene association_causal mechanism qualifier\", curie=BIOLINK.curie('causal_mechanism_qualifier'),\n",
      "                   model_uri=DIST.chemical_affects_gene_association_causal_mechanism_qualifier, domain=ChemicalAffectsGeneAssociation, range=Optional[Union[str, \"CausalMechanismQualifierEnum\"]])\n",
      "\n",
      "slots.chemical_affects_gene_association_anatomical_context_qualifier = Slot(uri=BIOLINK.anatomical_context_qualifier, name=\"chemical affects gene association_anatomical context qualifier\", curie=BIOLINK.curie('anatomical_context_qualifier'),\n",
      "                   model_uri=DIST.chemical_affects_gene_association_anatomical_context_qualifier, domain=ChemicalAffectsGeneAssociation, range=Optional[Union[str, AnatomicalEntityId]])\n",
      "\n",
      "slots.chemical_affects_gene_association_species_context_qualifier = Slot(uri=BIOLINK.species_context_qualifier, name=\"chemical affects gene association_species context qualifier\", curie=BIOLINK.curie('species_context_qualifier'),\n",
      "                   model_uri=DIST.chemical_affects_gene_association_species_context_qualifier, domain=ChemicalAffectsGeneAssociation, range=Optional[Union[str, OrganismTaxonId]])\n",
      "\n",
      "slots.gene_affects_chemical_association_subject = Slot(uri=RDF.subject, name=\"gene affects chemical association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.gene_affects_chemical_association_subject, domain=GeneAffectsChemicalAssociation, range=Union[dict, GeneOrGeneProduct])\n",
      "\n",
      "slots.gene_affects_chemical_association_subject_form_or_variant_qualifier = Slot(uri=BIOLINK.subject_form_or_variant_qualifier, name=\"gene affects chemical association_subject form or variant qualifier\", curie=BIOLINK.curie('subject_form_or_variant_qualifier'),\n",
      "                   model_uri=DIST.gene_affects_chemical_association_subject_form_or_variant_qualifier, domain=GeneAffectsChemicalAssociation, range=Optional[Union[str, \"ChemicalOrGeneOrGeneProductFormOrVariantEnum\"]])\n",
      "\n",
      "slots.gene_affects_chemical_association_subject_part_qualifier = Slot(uri=BIOLINK.subject_part_qualifier, name=\"gene affects chemical association_subject part qualifier\", curie=BIOLINK.curie('subject_part_qualifier'),\n",
      "                   model_uri=DIST.gene_affects_chemical_association_subject_part_qualifier, domain=GeneAffectsChemicalAssociation, range=Optional[Union[str, \"GeneOrGeneProductOrChemicalPartQualifierEnum\"]])\n",
      "\n",
      "slots.gene_affects_chemical_association_subject_aspect_qualifier = Slot(uri=BIOLINK.subject_aspect_qualifier, name=\"gene affects chemical association_subject aspect qualifier\", curie=BIOLINK.curie('subject_aspect_qualifier'),\n",
      "                   model_uri=DIST.gene_affects_chemical_association_subject_aspect_qualifier, domain=GeneAffectsChemicalAssociation, range=Optional[Union[str, \"GeneOrGeneProductOrChemicalEntityAspectEnum\"]])\n",
      "\n",
      "slots.gene_affects_chemical_association_subject_context_qualifier = Slot(uri=BIOLINK.subject_context_qualifier, name=\"gene affects chemical association_subject context qualifier\", curie=BIOLINK.curie('subject_context_qualifier'),\n",
      "                   model_uri=DIST.gene_affects_chemical_association_subject_context_qualifier, domain=GeneAffectsChemicalAssociation, range=Optional[Union[str, AnatomicalEntityId]])\n",
      "\n",
      "slots.gene_affects_chemical_association_subject_direction_qualifier = Slot(uri=BIOLINK.subject_direction_qualifier, name=\"gene affects chemical association_subject direction qualifier\", curie=BIOLINK.curie('subject_direction_qualifier'),\n",
      "                   model_uri=DIST.gene_affects_chemical_association_subject_direction_qualifier, domain=GeneAffectsChemicalAssociation, range=Optional[Union[str, \"DirectionQualifierEnum\"]])\n",
      "\n",
      "slots.gene_affects_chemical_association_predicate = Slot(uri=RDF.predicate, name=\"gene affects chemical association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.gene_affects_chemical_association_predicate, domain=GeneAffectsChemicalAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.gene_affects_chemical_association_qualified_predicate = Slot(uri=BIOLINK.qualified_predicate, name=\"gene affects chemical association_qualified predicate\", curie=BIOLINK.curie('qualified_predicate'),\n",
      "                   model_uri=DIST.gene_affects_chemical_association_qualified_predicate, domain=GeneAffectsChemicalAssociation, range=Optional[str])\n",
      "\n",
      "slots.gene_affects_chemical_association_object = Slot(uri=RDF.object, name=\"gene affects chemical association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.gene_affects_chemical_association_object, domain=GeneAffectsChemicalAssociation, range=Union[str, ChemicalEntityId])\n",
      "\n",
      "slots.gene_affects_chemical_association_object_form_or_variant_qualifier = Slot(uri=BIOLINK.object_form_or_variant_qualifier, name=\"gene affects chemical association_object form or variant qualifier\", curie=BIOLINK.curie('object_form_or_variant_qualifier'),\n",
      "                   model_uri=DIST.gene_affects_chemical_association_object_form_or_variant_qualifier, domain=GeneAffectsChemicalAssociation, range=Optional[Union[str, \"ChemicalOrGeneOrGeneProductFormOrVariantEnum\"]])\n",
      "\n",
      "slots.gene_affects_chemical_association_object_part_qualifier = Slot(uri=BIOLINK.object_part_qualifier, name=\"gene affects chemical association_object part qualifier\", curie=BIOLINK.curie('object_part_qualifier'),\n",
      "                   model_uri=DIST.gene_affects_chemical_association_object_part_qualifier, domain=GeneAffectsChemicalAssociation, range=Optional[Union[str, \"GeneOrGeneProductOrChemicalPartQualifierEnum\"]])\n",
      "\n",
      "slots.gene_affects_chemical_association_object_derivative_qualifier = Slot(uri=BIOLINK.object_derivative_qualifier, name=\"gene affects chemical association_object derivative qualifier\", curie=BIOLINK.curie('object_derivative_qualifier'),\n",
      "                   model_uri=DIST.gene_affects_chemical_association_object_derivative_qualifier, domain=GeneAffectsChemicalAssociation, range=Optional[Union[str, \"ChemicalEntityDerivativeEnum\"]])\n",
      "\n",
      "slots.gene_affects_chemical_association_object_aspect_qualifier = Slot(uri=BIOLINK.object_aspect_qualifier, name=\"gene affects chemical association_object aspect qualifier\", curie=BIOLINK.curie('object_aspect_qualifier'),\n",
      "                   model_uri=DIST.gene_affects_chemical_association_object_aspect_qualifier, domain=GeneAffectsChemicalAssociation, range=Optional[Union[str, \"GeneOrGeneProductOrChemicalEntityAspectEnum\"]])\n",
      "\n",
      "slots.gene_affects_chemical_association_object_context_qualifier = Slot(uri=BIOLINK.object_context_qualifier, name=\"gene affects chemical association_object context qualifier\", curie=BIOLINK.curie('object_context_qualifier'),\n",
      "                   model_uri=DIST.gene_affects_chemical_association_object_context_qualifier, domain=GeneAffectsChemicalAssociation, range=Optional[Union[str, AnatomicalEntityId]])\n",
      "\n",
      "slots.gene_affects_chemical_association_object_direction_qualifier = Slot(uri=BIOLINK.object_direction_qualifier, name=\"gene affects chemical association_object direction qualifier\", curie=BIOLINK.curie('object_direction_qualifier'),\n",
      "                   model_uri=DIST.gene_affects_chemical_association_object_direction_qualifier, domain=GeneAffectsChemicalAssociation, range=Optional[Union[str, \"DirectionQualifierEnum\"]])\n",
      "\n",
      "slots.gene_affects_chemical_association_causal_mechanism_qualifier = Slot(uri=BIOLINK.causal_mechanism_qualifier, name=\"gene affects chemical association_causal mechanism qualifier\", curie=BIOLINK.curie('causal_mechanism_qualifier'),\n",
      "                   model_uri=DIST.gene_affects_chemical_association_causal_mechanism_qualifier, domain=GeneAffectsChemicalAssociation, range=Optional[Union[str, \"CausalMechanismQualifierEnum\"]])\n",
      "\n",
      "slots.gene_affects_chemical_association_anatomical_context_qualifier = Slot(uri=BIOLINK.anatomical_context_qualifier, name=\"gene affects chemical association_anatomical context qualifier\", curie=BIOLINK.curie('anatomical_context_qualifier'),\n",
      "                   model_uri=DIST.gene_affects_chemical_association_anatomical_context_qualifier, domain=GeneAffectsChemicalAssociation, range=Optional[Union[str, AnatomicalEntityId]])\n",
      "\n",
      "slots.gene_affects_chemical_association_species_context_qualifier = Slot(uri=BIOLINK.species_context_qualifier, name=\"gene affects chemical association_species context qualifier\", curie=BIOLINK.curie('species_context_qualifier'),\n",
      "                   model_uri=DIST.gene_affects_chemical_association_species_context_qualifier, domain=GeneAffectsChemicalAssociation, range=Optional[Union[str, OrganismTaxonId]])\n",
      "\n",
      "slots.drug_to_gene_association_object = Slot(uri=RDF.object, name=\"drug to gene association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.drug_to_gene_association_object, domain=DrugToGeneAssociation, range=Union[dict, GeneOrGeneProduct])\n",
      "\n",
      "slots.material_sample_to_entity_association_mixin_subject = Slot(uri=RDF.subject, name=\"material sample to entity association mixin_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.material_sample_to_entity_association_mixin_subject, domain=None, range=Union[str, MaterialSampleId])\n",
      "\n",
      "slots.material_sample_derivation_association_subject = Slot(uri=RDF.subject, name=\"material sample derivation association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.material_sample_derivation_association_subject, domain=MaterialSampleDerivationAssociation, range=Union[str, MaterialSampleId])\n",
      "\n",
      "slots.material_sample_derivation_association_object = Slot(uri=RDF.object, name=\"material sample derivation association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.material_sample_derivation_association_object, domain=MaterialSampleDerivationAssociation, range=Union[str, NamedThingId])\n",
      "\n",
      "slots.material_sample_derivation_association_predicate = Slot(uri=RDF.predicate, name=\"material sample derivation association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.material_sample_derivation_association_predicate, domain=MaterialSampleDerivationAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.disease_to_entity_association_mixin_subject = Slot(uri=RDF.subject, name=\"disease to entity association mixin_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.disease_to_entity_association_mixin_subject, domain=None, range=Union[str, DiseaseId])\n",
      "\n",
      "slots.entity_to_exposure_event_association_mixin_object = Slot(uri=RDF.object, name=\"entity to exposure event association mixin_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.entity_to_exposure_event_association_mixin_object, domain=None, range=Union[str, ExposureEventId])\n",
      "\n",
      "slots.entity_to_outcome_association_mixin_object = Slot(uri=RDF.object, name=\"entity to outcome association mixin_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.entity_to_outcome_association_mixin_object, domain=None, range=Union[dict, Outcome])\n",
      "\n",
      "slots.entity_to_phenotypic_feature_association_mixin_object = Slot(uri=RDF.object, name=\"entity to phenotypic feature association mixin_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.entity_to_phenotypic_feature_association_mixin_object, domain=None, range=Union[str, PhenotypicFeatureId])\n",
      "\n",
      "slots.phenotypic_feature_to_entity_association_mixin_subject = Slot(uri=RDF.subject, name=\"phenotypic feature to entity association mixin_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.phenotypic_feature_to_entity_association_mixin_subject, domain=None, range=Union[str, PhenotypicFeatureId])\n",
      "\n",
      "slots.information_content_entity_to_named_thing_association_subject = Slot(uri=RDF.subject, name=\"information content entity to named thing association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.information_content_entity_to_named_thing_association_subject, domain=InformationContentEntityToNamedThingAssociation, range=Union[str, NamedThingId])\n",
      "\n",
      "slots.information_content_entity_to_named_thing_association_object = Slot(uri=RDF.object, name=\"information content entity to named thing association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.information_content_entity_to_named_thing_association_object, domain=InformationContentEntityToNamedThingAssociation, range=Union[str, NamedThingId])\n",
      "\n",
      "slots.information_content_entity_to_named_thing_association_predicate = Slot(uri=RDF.predicate, name=\"information content entity to named thing association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.information_content_entity_to_named_thing_association_predicate, domain=InformationContentEntityToNamedThingAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.entity_to_disease_association_mixin_object = Slot(uri=RDF.object, name=\"entity to disease association mixin_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.entity_to_disease_association_mixin_object, domain=None, range=Union[str, DiseaseId])\n",
      "\n",
      "slots.disease_or_phenotypic_feature_to_entity_association_mixin_subject = Slot(uri=RDF.subject, name=\"disease or phenotypic feature to entity association mixin_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.disease_or_phenotypic_feature_to_entity_association_mixin_subject, domain=None, range=Union[str, DiseaseOrPhenotypicFeatureId])\n",
      "\n",
      "slots.disease_or_phenotypic_feature_to_location_association_object = Slot(uri=RDF.object, name=\"disease or phenotypic feature to location association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.disease_or_phenotypic_feature_to_location_association_object, domain=DiseaseOrPhenotypicFeatureToLocationAssociation, range=Union[str, AnatomicalEntityId])\n",
      "\n",
      "slots.disease_or_phenotypic_feature_to_genetic_inheritance_association_predicate = Slot(uri=RDF.predicate, name=\"disease or phenotypic feature to genetic inheritance association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.disease_or_phenotypic_feature_to_genetic_inheritance_association_predicate, domain=DiseaseOrPhenotypicFeatureToGeneticInheritanceAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.disease_or_phenotypic_feature_to_genetic_inheritance_association_object = Slot(uri=RDF.object, name=\"disease or phenotypic feature to genetic inheritance association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.disease_or_phenotypic_feature_to_genetic_inheritance_association_object, domain=DiseaseOrPhenotypicFeatureToGeneticInheritanceAssociation, range=Union[str, GeneticInheritanceId])\n",
      "\n",
      "slots.entity_to_disease_or_phenotypic_feature_association_mixin_object = Slot(uri=RDF.object, name=\"entity to disease or phenotypic feature association mixin_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.entity_to_disease_or_phenotypic_feature_association_mixin_object, domain=None, range=Union[str, DiseaseOrPhenotypicFeatureId])\n",
      "\n",
      "slots.genotype_to_entity_association_mixin_subject = Slot(uri=RDF.subject, name=\"genotype to entity association mixin_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.genotype_to_entity_association_mixin_subject, domain=None, range=Union[str, GenotypeId])\n",
      "\n",
      "slots.genotype_to_phenotypic_feature_association_predicate = Slot(uri=RDF.predicate, name=\"genotype to phenotypic feature association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.genotype_to_phenotypic_feature_association_predicate, domain=GenotypeToPhenotypicFeatureAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.genotype_to_phenotypic_feature_association_subject = Slot(uri=RDF.subject, name=\"genotype to phenotypic feature association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.genotype_to_phenotypic_feature_association_subject, domain=GenotypeToPhenotypicFeatureAssociation, range=Union[str, GenotypeId])\n",
      "\n",
      "slots.exposure_event_to_phenotypic_feature_association_subject = Slot(uri=RDF.subject, name=\"exposure event to phenotypic feature association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.exposure_event_to_phenotypic_feature_association_subject, domain=ExposureEventToPhenotypicFeatureAssociation, range=Union[str, ExposureEventId])\n",
      "\n",
      "slots.disease_to_phenotypic_feature_association_subject = Slot(uri=RDF.subject, name=\"disease to phenotypic feature association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.disease_to_phenotypic_feature_association_subject, domain=DiseaseToPhenotypicFeatureAssociation, range=Union[str, DiseaseId])\n",
      "\n",
      "slots.disease_to_phenotypic_feature_association_object = Slot(uri=RDF.object, name=\"disease to phenotypic feature association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.disease_to_phenotypic_feature_association_object, domain=DiseaseToPhenotypicFeatureAssociation, range=Union[str, PhenotypicFeatureId])\n",
      "\n",
      "slots.behavior_to_behavioral_feature_association_subject = Slot(uri=RDF.subject, name=\"behavior to behavioral feature association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.behavior_to_behavioral_feature_association_subject, domain=BehaviorToBehavioralFeatureAssociation, range=Union[str, BehaviorId])\n",
      "\n",
      "slots.behavior_to_behavioral_feature_association_object = Slot(uri=RDF.object, name=\"behavior to behavioral feature association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.behavior_to_behavioral_feature_association_object, domain=BehaviorToBehavioralFeatureAssociation, range=Union[str, BehavioralFeatureId])\n",
      "\n",
      "slots.gene_to_entity_association_mixin_subject = Slot(uri=RDF.subject, name=\"gene to entity association mixin_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.gene_to_entity_association_mixin_subject, domain=None, range=Union[dict, GeneOrGeneProduct])\n",
      "\n",
      "slots.variant_to_entity_association_mixin_subject = Slot(uri=RDF.subject, name=\"variant to entity association mixin_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.variant_to_entity_association_mixin_subject, domain=None, range=Union[str, SequenceVariantId])\n",
      "\n",
      "slots.gene_to_disease_or_phenotypic_feature_association_subject = Slot(uri=RDF.subject, name=\"gene to disease or phenotypic feature association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.gene_to_disease_or_phenotypic_feature_association_subject, domain=GeneToDiseaseOrPhenotypicFeatureAssociation, range=Union[dict, GeneOrGeneProduct])\n",
      "\n",
      "slots.gene_to_disease_or_phenotypic_feature_association_subject_aspect_qualifier = Slot(uri=BIOLINK.subject_aspect_qualifier, name=\"gene to disease or phenotypic feature association_subject aspect qualifier\", curie=BIOLINK.curie('subject_aspect_qualifier'),\n",
      "                   model_uri=DIST.gene_to_disease_or_phenotypic_feature_association_subject_aspect_qualifier, domain=GeneToDiseaseOrPhenotypicFeatureAssociation, range=Optional[Union[str, \"GeneOrGeneProductOrChemicalEntityAspectEnum\"]])\n",
      "\n",
      "slots.gene_to_disease_or_phenotypic_feature_association_object = Slot(uri=RDF.object, name=\"gene to disease or phenotypic feature association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.gene_to_disease_or_phenotypic_feature_association_object, domain=GeneToDiseaseOrPhenotypicFeatureAssociation, range=Union[str, DiseaseOrPhenotypicFeatureId])\n",
      "\n",
      "slots.gene_to_disease_or_phenotypic_feature_association_object_direction_qualifier = Slot(uri=BIOLINK.object_direction_qualifier, name=\"gene to disease or phenotypic feature association_object direction qualifier\", curie=BIOLINK.curie('object_direction_qualifier'),\n",
      "                   model_uri=DIST.gene_to_disease_or_phenotypic_feature_association_object_direction_qualifier, domain=GeneToDiseaseOrPhenotypicFeatureAssociation, range=Optional[Union[str, \"DirectionQualifierEnum\"]])\n",
      "\n",
      "slots.gene_to_disease_or_phenotypic_feature_association_predicate = Slot(uri=RDF.predicate, name=\"gene to disease or phenotypic feature association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.gene_to_disease_or_phenotypic_feature_association_predicate, domain=GeneToDiseaseOrPhenotypicFeatureAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.gene_to_phenotypic_feature_association_subject = Slot(uri=RDF.subject, name=\"gene to phenotypic feature association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.gene_to_phenotypic_feature_association_subject, domain=GeneToPhenotypicFeatureAssociation, range=Union[dict, GeneOrGeneProduct])\n",
      "\n",
      "slots.gene_to_phenotypic_feature_association_object = Slot(uri=RDF.object, name=\"gene to phenotypic feature association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.gene_to_phenotypic_feature_association_object, domain=GeneToPhenotypicFeatureAssociation, range=Union[str, PhenotypicFeatureId])\n",
      "\n",
      "slots.gene_to_disease_association_subject = Slot(uri=RDF.subject, name=\"gene to disease association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.gene_to_disease_association_subject, domain=GeneToDiseaseAssociation, range=Union[dict, GeneOrGeneProduct])\n",
      "\n",
      "slots.gene_to_disease_association_object = Slot(uri=RDF.object, name=\"gene to disease association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.gene_to_disease_association_object, domain=GeneToDiseaseAssociation, range=Union[str, DiseaseId])\n",
      "\n",
      "slots.causal_gene_to_disease_association_subject = Slot(uri=RDF.subject, name=\"causal gene to disease association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.causal_gene_to_disease_association_subject, domain=CausalGeneToDiseaseAssociation, range=Union[dict, GeneOrGeneProduct])\n",
      "\n",
      "slots.causal_gene_to_disease_association_object = Slot(uri=RDF.object, name=\"causal gene to disease association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.causal_gene_to_disease_association_object, domain=CausalGeneToDiseaseAssociation, range=Union[str, DiseaseId])\n",
      "\n",
      "slots.correlated_gene_to_disease_association_subject = Slot(uri=RDF.subject, name=\"correlated gene to disease association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.correlated_gene_to_disease_association_subject, domain=CorrelatedGeneToDiseaseAssociation, range=Union[dict, GeneOrGeneProduct])\n",
      "\n",
      "slots.correlated_gene_to_disease_association_object = Slot(uri=RDF.object, name=\"correlated gene to disease association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.correlated_gene_to_disease_association_object, domain=CorrelatedGeneToDiseaseAssociation, range=Union[str, DiseaseId])\n",
      "\n",
      "slots.druggable_gene_to_disease_association_subject = Slot(uri=RDF.subject, name=\"druggable gene to disease association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.druggable_gene_to_disease_association_subject, domain=DruggableGeneToDiseaseAssociation, range=Union[dict, GeneOrGeneProduct])\n",
      "\n",
      "slots.druggable_gene_to_disease_association_predicate = Slot(uri=RDF.predicate, name=\"druggable gene to disease association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.druggable_gene_to_disease_association_predicate, domain=DruggableGeneToDiseaseAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.druggable_gene_to_disease_association_has_evidence = Slot(uri=BIOLINK.has_evidence, name=\"druggable gene to disease association_has evidence\", curie=BIOLINK.curie('has_evidence'),\n",
      "                   model_uri=DIST.druggable_gene_to_disease_association_has_evidence, domain=DruggableGeneToDiseaseAssociation, range=Optional[Union[Union[str, \"DruggableGeneCategoryEnum\"], List[Union[str, \"DruggableGeneCategoryEnum\"]]]])\n",
      "\n",
      "slots.phenotypic_feature_to_disease_association_predicate = Slot(uri=RDF.predicate, name=\"phenotypic feature to disease association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.phenotypic_feature_to_disease_association_predicate, domain=PhenotypicFeatureToDiseaseAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.variant_to_gene_association_object = Slot(uri=RDF.object, name=\"variant to gene association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.variant_to_gene_association_object, domain=VariantToGeneAssociation, range=Union[str, GeneId])\n",
      "\n",
      "slots.variant_to_gene_association_predicate = Slot(uri=RDF.predicate, name=\"variant to gene association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.variant_to_gene_association_predicate, domain=VariantToGeneAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.variant_to_gene_expression_association_predicate = Slot(uri=RDF.predicate, name=\"variant to gene expression association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.variant_to_gene_expression_association_predicate, domain=VariantToGeneExpressionAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.variant_to_population_association_subject = Slot(uri=RDF.subject, name=\"variant to population association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.variant_to_population_association_subject, domain=VariantToPopulationAssociation, range=Union[str, SequenceVariantId])\n",
      "\n",
      "slots.variant_to_population_association_object = Slot(uri=RDF.object, name=\"variant to population association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.variant_to_population_association_object, domain=VariantToPopulationAssociation, range=Union[str, PopulationOfIndividualOrganismsId])\n",
      "\n",
      "slots.variant_to_population_association_has_quotient = Slot(uri=BIOLINK.has_quotient, name=\"variant to population association_has quotient\", curie=BIOLINK.curie('has_quotient'),\n",
      "                   model_uri=DIST.variant_to_population_association_has_quotient, domain=VariantToPopulationAssociation, range=Optional[float])\n",
      "\n",
      "slots.variant_to_population_association_has_count = Slot(uri=BIOLINK.has_count, name=\"variant to population association_has count\", curie=BIOLINK.curie('has_count'),\n",
      "                   model_uri=DIST.variant_to_population_association_has_count, domain=VariantToPopulationAssociation, range=Optional[int])\n",
      "\n",
      "slots.variant_to_population_association_has_total = Slot(uri=BIOLINK.has_total, name=\"variant to population association_has total\", curie=BIOLINK.curie('has_total'),\n",
      "                   model_uri=DIST.variant_to_population_association_has_total, domain=VariantToPopulationAssociation, range=Optional[int])\n",
      "\n",
      "slots.population_to_population_association_subject = Slot(uri=RDF.subject, name=\"population to population association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.population_to_population_association_subject, domain=PopulationToPopulationAssociation, range=Union[str, PopulationOfIndividualOrganismsId])\n",
      "\n",
      "slots.population_to_population_association_object = Slot(uri=RDF.object, name=\"population to population association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.population_to_population_association_object, domain=PopulationToPopulationAssociation, range=Union[str, PopulationOfIndividualOrganismsId])\n",
      "\n",
      "slots.population_to_population_association_predicate = Slot(uri=RDF.predicate, name=\"population to population association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.population_to_population_association_predicate, domain=PopulationToPopulationAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.variant_to_phenotypic_feature_association_subject = Slot(uri=RDF.subject, name=\"variant to phenotypic feature association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.variant_to_phenotypic_feature_association_subject, domain=VariantToPhenotypicFeatureAssociation, range=Union[str, SequenceVariantId])\n",
      "\n",
      "slots.variant_to_disease_association_subject = Slot(uri=RDF.subject, name=\"variant to disease association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.variant_to_disease_association_subject, domain=VariantToDiseaseAssociation, range=Union[str, NamedThingId])\n",
      "\n",
      "slots.variant_to_disease_association_predicate = Slot(uri=RDF.predicate, name=\"variant to disease association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.variant_to_disease_association_predicate, domain=VariantToDiseaseAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.variant_to_disease_association_object = Slot(uri=RDF.object, name=\"variant to disease association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.variant_to_disease_association_object, domain=VariantToDiseaseAssociation, range=Union[str, NamedThingId])\n",
      "\n",
      "slots.genotype_to_disease_association_subject = Slot(uri=RDF.subject, name=\"genotype to disease association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.genotype_to_disease_association_subject, domain=GenotypeToDiseaseAssociation, range=Union[str, NamedThingId])\n",
      "\n",
      "slots.genotype_to_disease_association_predicate = Slot(uri=RDF.predicate, name=\"genotype to disease association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.genotype_to_disease_association_predicate, domain=GenotypeToDiseaseAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.genotype_to_disease_association_object = Slot(uri=RDF.object, name=\"genotype to disease association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.genotype_to_disease_association_object, domain=GenotypeToDiseaseAssociation, range=Union[str, NamedThingId])\n",
      "\n",
      "slots.model_to_disease_association_mixin_subject = Slot(uri=RDF.subject, name=\"model to disease association mixin_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.model_to_disease_association_mixin_subject, domain=None, range=Union[str, NamedThingId])\n",
      "\n",
      "slots.model_to_disease_association_mixin_predicate = Slot(uri=RDF.predicate, name=\"model to disease association mixin_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.model_to_disease_association_mixin_predicate, domain=None, range=Union[str, PredicateType])\n",
      "\n",
      "slots.gene_as_a_model_of_disease_association_subject = Slot(uri=RDF.subject, name=\"gene as a model of disease association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.gene_as_a_model_of_disease_association_subject, domain=GeneAsAModelOfDiseaseAssociation, range=Union[dict, GeneOrGeneProduct])\n",
      "\n",
      "slots.variant_as_a_model_of_disease_association_subject = Slot(uri=RDF.subject, name=\"variant as a model of disease association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.variant_as_a_model_of_disease_association_subject, domain=VariantAsAModelOfDiseaseAssociation, range=Union[str, SequenceVariantId])\n",
      "\n",
      "slots.genotype_as_a_model_of_disease_association_subject = Slot(uri=RDF.subject, name=\"genotype as a model of disease association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.genotype_as_a_model_of_disease_association_subject, domain=GenotypeAsAModelOfDiseaseAssociation, range=Union[str, GenotypeId])\n",
      "\n",
      "slots.cell_line_as_a_model_of_disease_association_subject = Slot(uri=RDF.subject, name=\"cell line as a model of disease association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.cell_line_as_a_model_of_disease_association_subject, domain=CellLineAsAModelOfDiseaseAssociation, range=Union[str, CellLineId])\n",
      "\n",
      "slots.organismal_entity_as_a_model_of_disease_association_subject = Slot(uri=RDF.subject, name=\"organismal entity as a model of disease association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.organismal_entity_as_a_model_of_disease_association_subject, domain=OrganismalEntityAsAModelOfDiseaseAssociation, range=Union[str, OrganismalEntityId])\n",
      "\n",
      "slots.organism_to_organism_association_subject = Slot(uri=RDF.subject, name=\"organism to organism association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.organism_to_organism_association_subject, domain=OrganismToOrganismAssociation, range=Union[str, IndividualOrganismId])\n",
      "\n",
      "slots.organism_to_organism_association_object = Slot(uri=RDF.object, name=\"organism to organism association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.organism_to_organism_association_object, domain=OrganismToOrganismAssociation, range=Union[str, IndividualOrganismId])\n",
      "\n",
      "slots.taxon_to_taxon_association_subject = Slot(uri=RDF.subject, name=\"taxon to taxon association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.taxon_to_taxon_association_subject, domain=TaxonToTaxonAssociation, range=Union[str, OrganismTaxonId])\n",
      "\n",
      "slots.taxon_to_taxon_association_object = Slot(uri=RDF.object, name=\"taxon to taxon association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.taxon_to_taxon_association_object, domain=TaxonToTaxonAssociation, range=Union[str, OrganismTaxonId])\n",
      "\n",
      "slots.gene_has_variant_that_contributes_to_disease_association_subject = Slot(uri=RDF.subject, name=\"gene has variant that contributes to disease association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.gene_has_variant_that_contributes_to_disease_association_subject, domain=GeneHasVariantThatContributesToDiseaseAssociation, range=Union[dict, GeneOrGeneProduct])\n",
      "\n",
      "slots.gene_has_variant_that_contributes_to_disease_association_object = Slot(uri=RDF.object, name=\"gene has variant that contributes to disease association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.gene_has_variant_that_contributes_to_disease_association_object, domain=GeneHasVariantThatContributesToDiseaseAssociation, range=Union[str, DiseaseId])\n",
      "\n",
      "slots.gene_has_variant_that_contributes_to_disease_association_predicate = Slot(uri=RDF.predicate, name=\"gene has variant that contributes to disease association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.gene_has_variant_that_contributes_to_disease_association_predicate, domain=GeneHasVariantThatContributesToDiseaseAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.gene_to_expression_site_association_subject = Slot(uri=RDF.subject, name=\"gene to expression site association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.gene_to_expression_site_association_subject, domain=GeneToExpressionSiteAssociation, range=Union[dict, GeneOrGeneProduct])\n",
      "\n",
      "slots.gene_to_expression_site_association_object = Slot(uri=RDF.object, name=\"gene to expression site association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.gene_to_expression_site_association_object, domain=GeneToExpressionSiteAssociation, range=Union[str, AnatomicalEntityId])\n",
      "\n",
      "slots.gene_to_expression_site_association_predicate = Slot(uri=RDF.predicate, name=\"gene to expression site association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.gene_to_expression_site_association_predicate, domain=GeneToExpressionSiteAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.gene_to_expression_site_association_stage_qualifier = Slot(uri=BIOLINK.stage_qualifier, name=\"gene to expression site association_stage qualifier\", curie=BIOLINK.curie('stage_qualifier'),\n",
      "                   model_uri=DIST.gene_to_expression_site_association_stage_qualifier, domain=GeneToExpressionSiteAssociation, range=Optional[Union[str, LifeStageId]])\n",
      "\n",
      "slots.gene_to_expression_site_association_quantifier_qualifier = Slot(uri=BIOLINK.quantifier_qualifier, name=\"gene to expression site association_quantifier qualifier\", curie=BIOLINK.curie('quantifier_qualifier'),\n",
      "                   model_uri=DIST.gene_to_expression_site_association_quantifier_qualifier, domain=GeneToExpressionSiteAssociation, range=Optional[Union[str, OntologyClassId]])\n",
      "\n",
      "slots.sequence_variant_modulates_treatment_association_subject = Slot(uri=RDF.subject, name=\"sequence variant modulates treatment association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.sequence_variant_modulates_treatment_association_subject, domain=SequenceVariantModulatesTreatmentAssociation, range=Union[str, SequenceVariantId])\n",
      "\n",
      "slots.sequence_variant_modulates_treatment_association_object = Slot(uri=RDF.object, name=\"sequence variant modulates treatment association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.sequence_variant_modulates_treatment_association_object, domain=SequenceVariantModulatesTreatmentAssociation, range=Union[str, TreatmentId])\n",
      "\n",
      "slots.functional_association_subject = Slot(uri=RDF.subject, name=\"functional association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.functional_association_subject, domain=FunctionalAssociation, range=Union[dict, MacromolecularMachineMixin])\n",
      "\n",
      "slots.functional_association_object = Slot(uri=RDF.object, name=\"functional association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.functional_association_object, domain=FunctionalAssociation, range=Union[str, OntologyClassId])\n",
      "\n",
      "slots.macromolecular_machine_to_entity_association_mixin_subject = Slot(uri=RDF.subject, name=\"macromolecular machine to entity association mixin_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.macromolecular_machine_to_entity_association_mixin_subject, domain=None, range=Union[str, NamedThingId])\n",
      "\n",
      "slots.macromolecular_machine_to_molecular_activity_association_object = Slot(uri=RDF.object, name=\"macromolecular machine to molecular activity association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.macromolecular_machine_to_molecular_activity_association_object, domain=MacromolecularMachineToMolecularActivityAssociation, range=Union[str, MolecularActivityId])\n",
      "\n",
      "slots.macromolecular_machine_to_biological_process_association_object = Slot(uri=RDF.object, name=\"macromolecular machine to biological process association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.macromolecular_machine_to_biological_process_association_object, domain=MacromolecularMachineToBiologicalProcessAssociation, range=Union[str, BiologicalProcessId])\n",
      "\n",
      "slots.macromolecular_machine_to_cellular_component_association_object = Slot(uri=RDF.object, name=\"macromolecular machine to cellular component association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.macromolecular_machine_to_cellular_component_association_object, domain=MacromolecularMachineToCellularComponentAssociation, range=Union[str, CellularComponentId])\n",
      "\n",
      "slots.molecular_activity_to_chemical_entity_association_subject = Slot(uri=RDF.subject, name=\"molecular activity to chemical entity association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.molecular_activity_to_chemical_entity_association_subject, domain=MolecularActivityToChemicalEntityAssociation, range=Union[str, MolecularActivityId])\n",
      "\n",
      "slots.molecular_activity_to_chemical_entity_association_object = Slot(uri=RDF.object, name=\"molecular activity to chemical entity association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.molecular_activity_to_chemical_entity_association_object, domain=MolecularActivityToChemicalEntityAssociation, range=Union[str, ChemicalEntityId])\n",
      "\n",
      "slots.molecular_activity_to_molecular_activity_association_subject = Slot(uri=RDF.subject, name=\"molecular activity to molecular activity association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.molecular_activity_to_molecular_activity_association_subject, domain=MolecularActivityToMolecularActivityAssociation, range=Union[str, MolecularActivityId])\n",
      "\n",
      "slots.molecular_activity_to_molecular_activity_association_object = Slot(uri=RDF.object, name=\"molecular activity to molecular activity association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.molecular_activity_to_molecular_activity_association_object, domain=MolecularActivityToMolecularActivityAssociation, range=Union[str, MolecularActivityId])\n",
      "\n",
      "slots.gene_to_go_term_association_subject = Slot(uri=RDF.subject, name=\"gene to go term association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.gene_to_go_term_association_subject, domain=GeneToGoTermAssociation, range=Union[str, GeneId])\n",
      "\n",
      "slots.gene_to_go_term_association_object = Slot(uri=RDF.object, name=\"gene to go term association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.gene_to_go_term_association_object, domain=GeneToGoTermAssociation, range=Union[str, OntologyClassId])\n",
      "\n",
      "slots.genomic_sequence_localization_subject = Slot(uri=RDF.subject, name=\"genomic sequence localization_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.genomic_sequence_localization_subject, domain=GenomicSequenceLocalization, range=Union[str, NucleicAcidEntityId])\n",
      "\n",
      "slots.genomic_sequence_localization_object = Slot(uri=RDF.object, name=\"genomic sequence localization_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.genomic_sequence_localization_object, domain=GenomicSequenceLocalization, range=Union[str, NucleicAcidEntityId])\n",
      "\n",
      "slots.genomic_sequence_localization_predicate = Slot(uri=RDF.predicate, name=\"genomic sequence localization_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.genomic_sequence_localization_predicate, domain=GenomicSequenceLocalization, range=Union[str, PredicateType])\n",
      "\n",
      "slots.sequence_feature_relationship_subject = Slot(uri=RDF.subject, name=\"sequence feature relationship_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.sequence_feature_relationship_subject, domain=SequenceFeatureRelationship, range=Union[str, NucleicAcidEntityId])\n",
      "\n",
      "slots.sequence_feature_relationship_object = Slot(uri=RDF.object, name=\"sequence feature relationship_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.sequence_feature_relationship_object, domain=SequenceFeatureRelationship, range=Union[str, NucleicAcidEntityId])\n",
      "\n",
      "slots.transcript_to_gene_relationship_subject = Slot(uri=RDF.subject, name=\"transcript to gene relationship_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.transcript_to_gene_relationship_subject, domain=TranscriptToGeneRelationship, range=Union[str, TranscriptId])\n",
      "\n",
      "slots.transcript_to_gene_relationship_object = Slot(uri=RDF.object, name=\"transcript to gene relationship_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.transcript_to_gene_relationship_object, domain=TranscriptToGeneRelationship, range=Union[str, GeneId])\n",
      "\n",
      "slots.gene_to_gene_product_relationship_subject = Slot(uri=RDF.subject, name=\"gene to gene product relationship_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.gene_to_gene_product_relationship_subject, domain=GeneToGeneProductRelationship, range=Union[str, GeneId])\n",
      "\n",
      "slots.gene_to_gene_product_relationship_object = Slot(uri=RDF.object, name=\"gene to gene product relationship_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.gene_to_gene_product_relationship_object, domain=GeneToGeneProductRelationship, range=Union[dict, GeneProductMixin])\n",
      "\n",
      "slots.gene_to_gene_product_relationship_predicate = Slot(uri=RDF.predicate, name=\"gene to gene product relationship_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.gene_to_gene_product_relationship_predicate, domain=GeneToGeneProductRelationship, range=Union[str, PredicateType])\n",
      "\n",
      "slots.exon_to_transcript_relationship_subject = Slot(uri=RDF.subject, name=\"exon to transcript relationship_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.exon_to_transcript_relationship_subject, domain=ExonToTranscriptRelationship, range=Union[str, ExonId])\n",
      "\n",
      "slots.exon_to_transcript_relationship_object = Slot(uri=RDF.object, name=\"exon to transcript relationship_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.exon_to_transcript_relationship_object, domain=ExonToTranscriptRelationship, range=Union[str, TranscriptId])\n",
      "\n",
      "slots.chemical_entity_or_gene_or_gene_product_regulates_gene_association_predicate = Slot(uri=RDF.predicate, name=\"chemical entity or gene or gene product regulates gene association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.chemical_entity_or_gene_or_gene_product_regulates_gene_association_predicate, domain=ChemicalEntityOrGeneOrGeneProductRegulatesGeneAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.chemical_entity_or_gene_or_gene_product_regulates_gene_association_subject = Slot(uri=RDF.subject, name=\"chemical entity or gene or gene product regulates gene association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.chemical_entity_or_gene_or_gene_product_regulates_gene_association_subject, domain=ChemicalEntityOrGeneOrGeneProductRegulatesGeneAssociation, range=Union[dict, ChemicalEntityOrGeneOrGeneProduct])\n",
      "\n",
      "slots.chemical_entity_or_gene_or_gene_product_regulates_gene_association_object = Slot(uri=RDF.object, name=\"chemical entity or gene or gene product regulates gene association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.chemical_entity_or_gene_or_gene_product_regulates_gene_association_object, domain=ChemicalEntityOrGeneOrGeneProductRegulatesGeneAssociation, range=Union[dict, GeneOrGeneProduct])\n",
      "\n",
      "slots.chemical_entity_or_gene_or_gene_product_regulates_gene_association_object_direction_qualifier = Slot(uri=BIOLINK.object_direction_qualifier, name=\"chemical entity or gene or gene product regulates gene association_object direction qualifier\", curie=BIOLINK.curie('object_direction_qualifier'),\n",
      "                   model_uri=DIST.chemical_entity_or_gene_or_gene_product_regulates_gene_association_object_direction_qualifier, domain=ChemicalEntityOrGeneOrGeneProductRegulatesGeneAssociation, range=Optional[Union[str, \"DirectionQualifierEnum\"]])\n",
      "\n",
      "slots.anatomical_entity_to_anatomical_entity_association_subject = Slot(uri=RDF.subject, name=\"anatomical entity to anatomical entity association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.anatomical_entity_to_anatomical_entity_association_subject, domain=AnatomicalEntityToAnatomicalEntityAssociation, range=Union[str, AnatomicalEntityId])\n",
      "\n",
      "slots.anatomical_entity_to_anatomical_entity_association_object = Slot(uri=RDF.object, name=\"anatomical entity to anatomical entity association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.anatomical_entity_to_anatomical_entity_association_object, domain=AnatomicalEntityToAnatomicalEntityAssociation, range=Union[str, AnatomicalEntityId])\n",
      "\n",
      "slots.anatomical_entity_to_anatomical_entity_part_of_association_subject = Slot(uri=RDF.subject, name=\"anatomical entity to anatomical entity part of association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.anatomical_entity_to_anatomical_entity_part_of_association_subject, domain=AnatomicalEntityToAnatomicalEntityPartOfAssociation, range=Union[str, AnatomicalEntityId])\n",
      "\n",
      "slots.anatomical_entity_to_anatomical_entity_part_of_association_object = Slot(uri=RDF.object, name=\"anatomical entity to anatomical entity part of association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.anatomical_entity_to_anatomical_entity_part_of_association_object, domain=AnatomicalEntityToAnatomicalEntityPartOfAssociation, range=Union[str, AnatomicalEntityId])\n",
      "\n",
      "slots.anatomical_entity_to_anatomical_entity_part_of_association_predicate = Slot(uri=RDF.predicate, name=\"anatomical entity to anatomical entity part of association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.anatomical_entity_to_anatomical_entity_part_of_association_predicate, domain=AnatomicalEntityToAnatomicalEntityPartOfAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.anatomical_entity_to_anatomical_entity_ontogenic_association_subject = Slot(uri=RDF.subject, name=\"anatomical entity to anatomical entity ontogenic association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.anatomical_entity_to_anatomical_entity_ontogenic_association_subject, domain=AnatomicalEntityToAnatomicalEntityOntogenicAssociation, range=Union[str, AnatomicalEntityId])\n",
      "\n",
      "slots.anatomical_entity_to_anatomical_entity_ontogenic_association_object = Slot(uri=RDF.object, name=\"anatomical entity to anatomical entity ontogenic association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.anatomical_entity_to_anatomical_entity_ontogenic_association_object, domain=AnatomicalEntityToAnatomicalEntityOntogenicAssociation, range=Union[str, AnatomicalEntityId])\n",
      "\n",
      "slots.anatomical_entity_to_anatomical_entity_ontogenic_association_predicate = Slot(uri=RDF.predicate, name=\"anatomical entity to anatomical entity ontogenic association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.anatomical_entity_to_anatomical_entity_ontogenic_association_predicate, domain=AnatomicalEntityToAnatomicalEntityOntogenicAssociation, range=Union[str, PredicateType])\n",
      "\n",
      "slots.organism_taxon_to_entity_association_subject = Slot(uri=RDF.subject, name=\"organism taxon to entity association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.organism_taxon_to_entity_association_subject, domain=None, range=Union[str, OrganismTaxonId])\n",
      "\n",
      "slots.organism_taxon_to_organism_taxon_association_subject = Slot(uri=RDF.subject, name=\"organism taxon to organism taxon association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.organism_taxon_to_organism_taxon_association_subject, domain=OrganismTaxonToOrganismTaxonAssociation, range=Union[str, OrganismTaxonId])\n",
      "\n",
      "slots.organism_taxon_to_organism_taxon_association_object = Slot(uri=RDF.object, name=\"organism taxon to organism taxon association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.organism_taxon_to_organism_taxon_association_object, domain=OrganismTaxonToOrganismTaxonAssociation, range=Union[str, OrganismTaxonId])\n",
      "\n",
      "slots.organism_taxon_to_organism_taxon_specialization_subject = Slot(uri=RDF.subject, name=\"organism taxon to organism taxon specialization_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.organism_taxon_to_organism_taxon_specialization_subject, domain=OrganismTaxonToOrganismTaxonSpecialization, range=Union[str, OrganismTaxonId])\n",
      "\n",
      "slots.organism_taxon_to_organism_taxon_specialization_object = Slot(uri=RDF.object, name=\"organism taxon to organism taxon specialization_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.organism_taxon_to_organism_taxon_specialization_object, domain=OrganismTaxonToOrganismTaxonSpecialization, range=Union[str, OrganismTaxonId])\n",
      "\n",
      "slots.organism_taxon_to_organism_taxon_specialization_predicate = Slot(uri=RDF.predicate, name=\"organism taxon to organism taxon specialization_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.organism_taxon_to_organism_taxon_specialization_predicate, domain=OrganismTaxonToOrganismTaxonSpecialization, range=Union[str, PredicateType])\n",
      "\n",
      "slots.organism_taxon_to_organism_taxon_interaction_subject = Slot(uri=RDF.subject, name=\"organism taxon to organism taxon interaction_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.organism_taxon_to_organism_taxon_interaction_subject, domain=OrganismTaxonToOrganismTaxonInteraction, range=Union[str, OrganismTaxonId])\n",
      "\n",
      "slots.organism_taxon_to_organism_taxon_interaction_object = Slot(uri=RDF.object, name=\"organism taxon to organism taxon interaction_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.organism_taxon_to_organism_taxon_interaction_object, domain=OrganismTaxonToOrganismTaxonInteraction, range=Union[str, OrganismTaxonId])\n",
      "\n",
      "slots.organism_taxon_to_organism_taxon_interaction_predicate = Slot(uri=RDF.predicate, name=\"organism taxon to organism taxon interaction_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.organism_taxon_to_organism_taxon_interaction_predicate, domain=OrganismTaxonToOrganismTaxonInteraction, range=Union[str, PredicateType])\n",
      "\n",
      "slots.organism_taxon_to_organism_taxon_interaction_associated_environmental_context = Slot(uri=BIOLINK.associated_environmental_context, name=\"organism taxon to organism taxon interaction_associated environmental context\", curie=BIOLINK.curie('associated_environmental_context'),\n",
      "                   model_uri=DIST.organism_taxon_to_organism_taxon_interaction_associated_environmental_context, domain=OrganismTaxonToOrganismTaxonInteraction, range=Optional[str])\n",
      "\n",
      "slots.organism_taxon_to_environment_association_subject = Slot(uri=RDF.subject, name=\"organism taxon to environment association_subject\", curie=RDF.curie('subject'),\n",
      "                   model_uri=DIST.organism_taxon_to_environment_association_subject, domain=OrganismTaxonToEnvironmentAssociation, range=Union[str, OrganismTaxonId])\n",
      "\n",
      "slots.organism_taxon_to_environment_association_object = Slot(uri=RDF.object, name=\"organism taxon to environment association_object\", curie=RDF.curie('object'),\n",
      "                   model_uri=DIST.organism_taxon_to_environment_association_object, domain=OrganismTaxonToEnvironmentAssociation, range=Union[str, NamedThingId])\n",
      "\n",
      "slots.organism_taxon_to_environment_association_predicate = Slot(uri=RDF.predicate, name=\"organism taxon to environment association_predicate\", curie=RDF.curie('predicate'),\n",
      "                   model_uri=DIST.organism_taxon_to_environment_association_predicate, domain=OrganismTaxonToEnvironmentAssociation, range=Union[str, PredicateType])\n"
     ]
    }
   ],
   "source": [
    "import yaml\n",
    "from linkml.generators.pythongen import PythonGenerator\n",
    "from logging import ERROR\n",
    "\n",
    "model_str = yaml.dump(model)\n",
    "\n",
    "# Note: Jupyter appears to generate output even if the log_level is set.\n",
    "gen = PythonGenerator(model_str, log_level=ERROR)\n",
    "print(gen.serialize())"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
